{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# NGS pipeline for Alzheimer's Disease RNA-seq data\n",
    "\n",
    "Pipeline to analyze RNA-seq data of an Alzheimer experiment. I have used 5 control samples and 5 Alzheimer samples."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import csv\n",
    "from IPython.display import FileLink\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Initialize and setup some global variables:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# The git directory\n",
    "os.environ['DIR'] = '/home/mees/NGS_Alzheimer/'\n",
    "# Working directory to the sequencing data (I use an external disk, since I dont have a lot of space left on the pc)\n",
    "os.environ['WORKDIR'] = '/media/mees/Elements/ngs_data/'\n",
    "# The directory to the downloaded genome\n",
    "os.environ['GENOMEDIR'] = '/media/mees/Elements/ngs_data/genomes/'\n",
    "# Location to save the indexed reference genome\n",
    "os.environ['REF_GENOME'] = '/media/mees/Elements/ngs_data/genomes/ref_genome/'\n",
    "# FastQC perl file (without extension)\n",
    "os.environ['FASTQC'] = '/home/mees/NGS/FastQC/fastqc'\n",
    "# Hisat2-build file (without extension) or command\n",
    "os.environ['HISAT_BUILD'] = '/home/mees/NGS_Alzheimer/hisat2-2.2.0/hisat2-build'\n",
    "# Hisat2 file (without extension) or command\n",
    "os.environ['HISAT'] = '/home/mees/NGS_Alzheimer/hisat2-2.2.0/hisat2'\n",
    "# Samtools file (without extension) or command\n",
    "os.environ['SAMTOOLS'] = 'samtools'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download Data\n",
    "\n",
    "First download the human genome for reference and save it as genome.fa in $GENOMEDIR.\n",
    "\n",
    "Second, download the human gene transfer file for feature counting as genome.gtf in $GENOMEDIR.\n",
    "\n",
    "Third, download the sample data via SRA tools. This process is found in download_data.sh. This data is download in $WORKDIR."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!wget ftp://ftp.ensembl.org/pub/release-100/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\n",
    "!gzip -d Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz\n",
    "!mv Homo_sapiens.GRCh38.dna.primary_assembly.fa $GENOMEDIR/genome.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2020-06-25 12:35:33--  ftp://ftp.ensembl.org/pub/release-100/gtf/homo_sapiens/Homo_sapiens.GRCh38.100.gtf.gz\n",
      "           => ‘Homo_sapiens.GRCh38.100.gtf.gz’\n",
      "Resolving ftp.ensembl.org (ftp.ensembl.org)... 193.62.193.139\n",
      "Connecting to ftp.ensembl.org (ftp.ensembl.org)|193.62.193.139|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /pub/release-100/gtf/homo_sapiens ... done.\n",
      "==> SIZE Homo_sapiens.GRCh38.100.gtf.gz ... 46973686\n",
      "==> PASV ... done.    ==> RETR Homo_sapiens.GRCh38.100.gtf.gz ... done.\n",
      "Length: 46973686 (45M) (unauthoritative)\n",
      "\n",
      "Homo_sapiens.GRCh38 100%[===================>]  44,80M  13,6MB/s    in 3,3s    \n",
      "\n",
      "2020-06-25 12:35:36 (13,4 MB/s) - ‘Homo_sapiens.GRCh38.100.gtf.gz’ saved [46973686]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget ftp://ftp.ensembl.org/pub/release-100/gtf/homo_sapiens/Homo_sapiens.GRCh38.100.gtf.gz\n",
    "!gzip -d Homo_sapiens.GRCh38.100.gtf.gz\n",
    "!mv Homo_sapiens.GRCh38.100.gtf $GENOMEDIR/genome.gtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "2020-06-29T10:11:16 prefetch.2.10.7 int: buffer insufficient while reading uri within cloud module - cannot Get Cloud Location\n",
      "2020-06-29T10:11:17 prefetch.2.10.7: 1) 'SRR6145423' is found locally\n",
      "2020-06-29T10:11:17 prefetch.2.10.7: 'SRR6145423' has 0 unresolved dependencies\n",
      "\n",
      "2020-06-29T10:11:17 prefetch.2.10.7 int: buffer insufficient while reading uri within cloud module - cannot Get Cloud Location\n",
      "2020-06-29T10:11:18 prefetch.2.10.7: 2) 'SRR6145424' is found locally\n",
      "2020-06-29T10:11:18 prefetch.2.10.7: 'SRR6145424' has 0 unresolved dependencies\n",
      "\n",
      "2020-06-29T10:11:18 prefetch.2.10.7 int: buffer insufficient while reading uri within cloud module - cannot Get Cloud Location\n",
      "2020-06-29T10:11:19 prefetch.2.10.7: 3) 'SRR6145425' is found locally\n",
      "2020-06-29T10:11:19 prefetch.2.10.7: 'SRR6145425' has 0 unresolved dependencies\n",
      "\n",
      "2020-06-29T10:11:19 prefetch.2.10.7 int: buffer insufficient while reading uri within cloud module - cannot Get Cloud Location\n",
      "2020-06-29T10:11:20 prefetch.2.10.7: 4) 'SRR6145426' is found locally\n",
      "2020-06-29T10:11:20 prefetch.2.10.7: 'SRR6145426' has 0 unresolved dependencies\n",
      "\n",
      "2020-06-29T10:11:20 prefetch.2.10.7 int: buffer insufficient while reading uri within cloud module - cannot Get Cloud Location\n",
      "2020-06-29T10:11:21 prefetch.2.10.7: 5) 'SRR6145427' is found locally\n",
      "2020-06-29T10:11:21 prefetch.2.10.7: 'SRR6145427' has 0 unresolved dependencies\n",
      "\n",
      "2020-06-29T10:11:21 prefetch.2.10.7 int: buffer insufficient while reading uri within cloud module - cannot Get Cloud Location\n",
      "2020-06-29T10:11:22 prefetch.2.10.7: 6) 'SRR6145433' is found locally\n",
      "2020-06-29T10:11:22 prefetch.2.10.7: 'SRR6145433' has 0 unresolved dependencies\n",
      "\n",
      "2020-06-29T10:11:22 prefetch.2.10.7 int: buffer insufficient while reading uri within cloud module - cannot Get Cloud Location\n",
      "2020-06-29T10:11:23 prefetch.2.10.7: 7) 'SRR6145434' is found locally\n",
      "2020-06-29T10:11:23 prefetch.2.10.7: 'SRR6145434' has 0 unresolved dependencies\n",
      "\n",
      "2020-06-29T10:11:23 prefetch.2.10.7 int: buffer insufficient while reading uri within cloud module - cannot Get Cloud Location\n",
      "2020-06-29T10:11:23 prefetch.2.10.7: 8) 'SRR6145435' is found locally\n",
      "2020-06-29T10:11:24 prefetch.2.10.7: 'SRR6145435' has 0 unresolved dependencies\n",
      "\n",
      "2020-06-29T10:11:24 prefetch.2.10.7 int: buffer insufficient while reading uri within cloud module - cannot Get Cloud Location\n",
      "2020-06-29T10:11:24 prefetch.2.10.7: 9) 'SRR6145436' is found locally\n",
      "2020-06-29T10:11:24 prefetch.2.10.7: 'SRR6145436' has 0 unresolved dependencies\n",
      "\n",
      "2020-06-29T10:11:24 prefetch.2.10.7 int: buffer insufficient while reading uri within cloud module - cannot Get Cloud Location\n",
      "2020-06-29T10:11:25 prefetch.2.10.7: 10) 'SRR6145437' is found locally\n",
      "2020-06-29T10:11:25 prefetch.2.10.7: 'SRR6145437' has 0 unresolved dependencies\n",
      "Succesfully dowloaded the SRA data.\n",
      "Converting SRR6145423.sra to fastq.\n",
      "2020-06-29T10:11:25 fasterq-dump.2.10.7 err: fasterq-dump.c fastdump_csra() checking ouput-file 'SRR6145423/SRR6145423.sra.fastq' -> RC(rcExe,rcFile,rcPacking,rcName,rcExists)\n",
      "fasterq-dump quit with error code 3\n",
      "Error while converting SRR6145423.sra.\n",
      "Converting SRR6145424.sra to fastq.\n",
      "2020-06-29T10:11:25 fasterq-dump.2.10.7 err: fasterq-dump.c fastdump_csra() checking ouput-file 'SRR6145424/SRR6145424.sra.fastq' -> RC(rcExe,rcFile,rcPacking,rcName,rcExists)\n",
      "fasterq-dump quit with error code 3\n",
      "Error while converting SRR6145424.sra.\n",
      "Converting SRR6145425.sra to fastq.\n",
      "2020-06-29T10:11:25 fasterq-dump.2.10.7 err: fasterq-dump.c fastdump_csra() checking ouput-file 'SRR6145425/SRR6145425.sra.fastq' -> RC(rcExe,rcFile,rcPacking,rcName,rcExists)\n",
      "fasterq-dump quit with error code 3\n",
      "Error while converting SRR6145425.sra.\n",
      "Converting SRR6145426.sra to fastq.\n",
      "2020-06-29T10:11:25 fasterq-dump.2.10.7 err: fasterq-dump.c fastdump_csra() checking ouput-file 'SRR6145426/SRR6145426.sra.fastq' -> RC(rcExe,rcFile,rcPacking,rcName,rcExists)\n",
      "fasterq-dump quit with error code 3\n",
      "Error while converting SRR6145426.sra.\n",
      "Converting SRR6145427.sra to fastq.\n",
      "2020-06-29T10:11:25 fasterq-dump.2.10.7 err: fasterq-dump.c fastdump_csra() checking ouput-file 'SRR6145427/SRR6145427.sra.fastq' -> RC(rcExe,rcFile,rcPacking,rcName,rcExists)\n",
      "fasterq-dump quit with error code 3\n",
      "Error while converting SRR6145427.sra.\n",
      "Converting SRR6145433.sra to fastq.\n",
      "2020-06-29T10:11:25 fasterq-dump.2.10.7 err: fasterq-dump.c fastdump_csra() checking ouput-file 'SRR6145433/SRR6145433.sra.fastq' -> RC(rcExe,rcFile,rcPacking,rcName,rcExists)\n",
      "fasterq-dump quit with error code 3\n",
      "Error while converting SRR6145433.sra.\n",
      "Converting SRR6145434.sra to fastq.\n",
      "2020-06-29T10:11:26 fasterq-dump.2.10.7 err: fasterq-dump.c fastdump_csra() checking ouput-file 'SRR6145434/SRR6145434.sra.fastq' -> RC(rcExe,rcFile,rcPacking,rcName,rcExists)\n",
      "fasterq-dump quit with error code 3\n",
      "Error while converting SRR6145434.sra.\n",
      "Converting SRR6145435.sra to fastq.\n",
      "2020-06-29T10:11:26 fasterq-dump.2.10.7 err: fasterq-dump.c fastdump_csra() checking ouput-file 'SRR6145435/SRR6145435.sra.fastq' -> RC(rcExe,rcFile,rcPacking,rcName,rcExists)\n",
      "fasterq-dump quit with error code 3\n",
      "Error while converting SRR6145435.sra.\n",
      "Converting SRR6145436.sra to fastq.\n",
      "2020-06-29T10:11:26 fasterq-dump.2.10.7 err: fasterq-dump.c fastdump_csra() checking ouput-file 'SRR6145436/SRR6145436.sra.fastq' -> RC(rcExe,rcFile,rcPacking,rcName,rcExists)\n",
      "fasterq-dump quit with error code 3\n",
      "Error while converting SRR6145436.sra.\n",
      "Converting SRR6145437.sra to fastq.\n",
      "2020-06-29T10:11:26 fasterq-dump.2.10.7 err: fasterq-dump.c fastdump_csra() checking ouput-file 'SRR6145437/SRR6145437.sra.fastq' -> RC(rcExe,rcFile,rcPacking,rcName,rcExists)\n",
      "fasterq-dump quit with error code 3\n",
      "Error while converting SRR6145437.sra.\n"
     ]
    }
   ],
   "source": [
    "!mkdir -p $WORKDIR\n",
    "!./src/download_data.sh -d $WORKDIR"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Quality Control\n",
    "\n",
    "The quality of the RNA-seq data is determined using FastQC, which creates HTML reports. The script that contains calls to FastQC is fastqc.sh.\n",
    "\n",
    "U can specifiy -c to copy the final reports to another directory, for instance, I copy it to the git repository."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir -p data/FastQC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started analysis of SRR6145423.sra.fastq\n",
      "Approx 5% complete for SRR6145423.sra.fastq\n",
      "Approx 10% complete for SRR6145423.sra.fastq\n",
      "Approx 15% complete for SRR6145423.sra.fastq\n",
      "^C\n"
     ]
    }
   ],
   "source": [
    "!./src/fastqc.sh -d $WORKDIR -l $FASTQC -c $DIR/data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<a href='data/FastQC/SRR6145434.sra_fastqc.html' target='_blank'>data/FastQC/SRR6145434.sra_fastqc.html</a><br>"
      ],
      "text/plain": [
       "/home/mees/NGS_Alzheimer/data/FastQC/SRR6145434.sra_fastqc.html"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<a href='data/FastQC/SRR6145433.sra_fastqc.html' target='_blank'>data/FastQC/SRR6145433.sra_fastqc.html</a><br>"
      ],
      "text/plain": [
       "/home/mees/NGS_Alzheimer/data/FastQC/SRR6145433.sra_fastqc.html"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<a href='data/FastQC/SRR6145425.sra_fastqc.html' target='_blank'>data/FastQC/SRR6145425.sra_fastqc.html</a><br>"
      ],
      "text/plain": [
       "/home/mees/NGS_Alzheimer/data/FastQC/SRR6145425.sra_fastqc.html"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<a href='data/FastQC/SRR6145427.sra_fastqc.html' target='_blank'>data/FastQC/SRR6145427.sra_fastqc.html</a><br>"
      ],
      "text/plain": [
       "/home/mees/NGS_Alzheimer/data/FastQC/SRR6145427.sra_fastqc.html"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<a href='data/FastQC/SRR6145426.sra_fastqc.html' target='_blank'>data/FastQC/SRR6145426.sra_fastqc.html</a><br>"
      ],
      "text/plain": [
       "/home/mees/NGS_Alzheimer/data/FastQC/SRR6145426.sra_fastqc.html"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<a href='data/FastQC/SRR6145424.sra_fastqc.html' target='_blank'>data/FastQC/SRR6145424.sra_fastqc.html</a><br>"
      ],
      "text/plain": [
       "/home/mees/NGS_Alzheimer/data/FastQC/SRR6145424.sra_fastqc.html"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<a href='data/FastQC/SRR6145435.sra_fastqc.html' target='_blank'>data/FastQC/SRR6145435.sra_fastqc.html</a><br>"
      ],
      "text/plain": [
       "/home/mees/NGS_Alzheimer/data/FastQC/SRR6145435.sra_fastqc.html"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<a href='data/FastQC/SRR6145436.sra_fastqc.html' target='_blank'>data/FastQC/SRR6145436.sra_fastqc.html</a><br>"
      ],
      "text/plain": [
       "/home/mees/NGS_Alzheimer/data/FastQC/SRR6145436.sra_fastqc.html"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<a href='data/FastQC/SRR6145423.sra_fastqc.html' target='_blank'>data/FastQC/SRR6145423.sra_fastqc.html</a><br>"
      ],
      "text/plain": [
       "/home/mees/NGS_Alzheimer/data/FastQC/SRR6145423.sra_fastqc.html"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<a href='data/FastQC/SRR6145437.sra_fastqc.html' target='_blank'>data/FastQC/SRR6145437.sra_fastqc.html</a><br>"
      ],
      "text/plain": [
       "/home/mees/NGS_Alzheimer/data/FastQC/SRR6145437.sra_fastqc.html"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for file in os.listdir('data/FastQC'):\n",
    "    if file.endswith(\".html\"):\n",
    "        html_file = FileLink(os.path.join('data/FastQC', file))\n",
    "        display(html_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "FastQC shows that there are Truseq adapter index 7 and NNNNNNN sequences.\n",
    "We want to remove these sequences using cutadapt, however, this is still in progress at the bottom of this notebook."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Alignment with Hisat2\n",
    "\n",
    "First, hisat2 needs to build the reference genome using hisat2-build, which can be found in build.sh.\n",
    "\n",
    "Then, the samples can be aligned using hisat2, which can be found in align.sh."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HISAT2 version 2.2.0 by Daehwan Kim (infphilo@gmail.com, http://www.ccb.jhu.edu/people/infphilo)\r\n",
      "Usage: hisat2-build [options]* <reference_in> <ht2_index_base>\r\n",
      "    reference_in            comma-separated list of files with ref sequences\r\n",
      "    hisat2_index_base       write ht2 data to files with this dir/basename\r\n",
      "Options:\r\n",
      "    -c                      reference sequences given on cmd line (as\r\n",
      "                            <reference_in>)\r\n",
      "    --large-index           force generated index to be 'large', even if ref\r\n",
      "                            has fewer than 4 billion nucleotides\r\n",
      "    -a/--noauto             disable automatic -p/--bmax/--dcv memory-fitting\r\n",
      "    -p <int>                number of threads\r\n",
      "    --bmax <int>            max bucket sz for blockwise suffix-array builder\r\n",
      "    --bmaxdivn <int>        max bucket sz as divisor of ref len (default: 4)\r\n",
      "    --dcv <int>             diff-cover period for blockwise (default: 1024)\r\n",
      "    --nodc                  disable diff-cover (algorithm becomes quadratic)\r\n",
      "    -r/--noref              don't build .3/.4.ht2 (packed reference) portion\r\n",
      "    -3/--justref            just build .3/.4.ht2 (packed reference) portion\r\n",
      "    -o/--offrate <int>      SA is sampled every 2^offRate BWT chars (default: 5)\r\n",
      "    -t/--ftabchars <int>    # of chars consumed in initial lookup (default: 10)\r\n",
      "    --localoffrate <int>    SA (local) is sampled every 2^offRate BWT chars (default: 3)\r\n",
      "    --localftabchars <int>  # of chars consumed in initial lookup in a local index (default: 6)\r\n",
      "    --snp <path>            SNP file name\r\n",
      "    --haplotype <path>      haplotype file name\r\n",
      "    --ss <path>             Splice site file name\r\n",
      "    --exon <path>           Exon file name\r\n",
      "    --repeat-ref <path>     Repeat reference file name\r\n",
      "    --repeat-info <path>    Repeat information file name\r\n",
      "    --repeat-snp <path>     Repeat snp file name\r\n",
      "    --repeat-haplotype <path>   Repeat haplotype file name\r\n",
      "    --seed <int>            seed for random number generator\r\n",
      "    -q/--quiet              disable verbose output (for debugging)\r\n",
      "    -h/--help               print detailed description of tool and its options\r\n",
      "    --usage                 print this usage message\r\n",
      "    --version               print version information and quit\r\n"
     ]
    }
   ],
   "source": [
    "# Commented so I don't accidently run it again\n",
    "#!./src/build.sh -g $GENOMEDIR/genome.fa -o $REF_GENOME -l $HISAT_BUILD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "37552625 reads; of these:\n",
      "  37552625 (100.00%) were unpaired; of these:\n",
      "    2360539 (6.29%) aligned 0 times\n",
      "    31238345 (83.19%) aligned exactly 1 time\n",
      "    3953741 (10.53%) aligned >1 times\n",
      "93.71% overall alignment rate\n",
      "Hisat2 is aligning SRR6145423 to the reference genome.\n",
      "38052606 reads; of these:\n",
      "  38052606 (100.00%) were unpaired; of these:\n",
      "    3423266 (9.00%) aligned 0 times\n",
      "    28619217 (75.21%) aligned exactly 1 time\n",
      "    6010123 (15.79%) aligned >1 times\n",
      "91.00% overall alignment rate\n",
      "Hisat2 is aligning SRR6145424 to the reference genome.\n",
      "36212177 reads; of these:\n",
      "  36212177 (100.00%) were unpaired; of these:\n",
      "    2478081 (6.84%) aligned 0 times\n",
      "    30107607 (83.14%) aligned exactly 1 time\n",
      "    3626489 (10.01%) aligned >1 times\n",
      "93.16% overall alignment rate\n",
      "Hisat2 is aligning SRR6145425 to the reference genome.\n",
      "42174836 reads; of these:\n",
      "  42174836 (100.00%) were unpaired; of these:\n",
      "    2849120 (6.76%) aligned 0 times\n",
      "    34036585 (80.70%) aligned exactly 1 time\n",
      "    5289131 (12.54%) aligned >1 times\n",
      "93.24% overall alignment rate\n",
      "Hisat2 is aligning SRR6145426 to the reference genome.\n",
      "38610276 reads; of these:\n",
      "  38610276 (100.00%) were unpaired; of these:\n",
      "    2369133 (6.14%) aligned 0 times\n",
      "    30757903 (79.66%) aligned exactly 1 time\n",
      "    5483240 (14.20%) aligned >1 times\n",
      "93.86% overall alignment rate\n",
      "Hisat2 is aligning SRR6145427 to the reference genome.\n",
      "40299174 reads; of these:\n",
      "  40299174 (100.00%) were unpaired; of these:\n",
      "    2232074 (5.54%) aligned 0 times\n",
      "    27833152 (69.07%) aligned exactly 1 time\n",
      "    10233948 (25.39%) aligned >1 times\n",
      "94.46% overall alignment rate\n",
      "Hisat2 is aligning SRR6145433 to the reference genome.\n",
      "39739040 reads; of these:\n",
      "  39739040 (100.00%) were unpaired; of these:\n",
      "    2352643 (5.92%) aligned 0 times\n",
      "    27289549 (68.67%) aligned exactly 1 time\n",
      "    10096848 (25.41%) aligned >1 times\n",
      "94.08% overall alignment rate\n",
      "Hisat2 is aligning SRR6145434 to the reference genome.\n",
      "39459711 reads; of these:\n",
      "  39459711 (100.00%) were unpaired; of these:\n",
      "    2175603 (5.51%) aligned 0 times\n",
      "    32962979 (83.54%) aligned exactly 1 time\n",
      "    4321129 (10.95%) aligned >1 times\n",
      "94.49% overall alignment rate\n",
      "Hisat2 is aligning SRR6145435 to the reference genome.\n",
      "38019285 reads; of these:\n",
      "  38019285 (100.00%) were unpaired; of these:\n",
      "    2550704 (6.71%) aligned 0 times\n",
      "    31004952 (81.55%) aligned exactly 1 time\n",
      "    4463629 (11.74%) aligned >1 times\n",
      "93.29% overall alignment rate\n",
      "Hisat2 is aligning SRR6145436 to the reference genome.\n",
      "40847873 reads; of these:\n",
      "  40847873 (100.00%) were unpaired; of these:\n",
      "    2442791 (5.98%) aligned 0 times\n",
      "    33282504 (81.48%) aligned exactly 1 time\n",
      "    5122578 (12.54%) aligned >1 times\n",
      "94.02% overall alignment rate\n",
      "Hisat2 is aligning SRR6145437 to the reference genome.\n"
     ]
    }
   ],
   "source": [
    "!./src/align.sh -g $REF_GENOME -f $WORKDIR/ -l $HISAT"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Converting .sam to .bam\n",
    "In order to view the alignment in IGV and run HTSeq, the created sam files should processed, which is performed with samtools in samtobam.sh.\n",
    "\n",
    "1. First, the files must be converted to bam files.\n",
    "2. Second, the bam files must be sorted.\n",
    "3. Third, the sorted bam files must be indexed."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Program: samtools (Tools for alignments in the SAM format)\r\n",
      "Version: 1.10 (using htslib 1.10)\r\n",
      "\r\n",
      "Usage:   samtools <command> [options]\r\n",
      "\r\n",
      "Commands:\r\n",
      "  -- Indexing\r\n",
      "     dict           create a sequence dictionary file\r\n",
      "     faidx          index/extract FASTA\r\n",
      "     fqidx          index/extract FASTQ\r\n",
      "     index          index alignment\r\n",
      "\r\n",
      "  -- Editing\r\n",
      "     calmd          recalculate MD/NM tags and '=' bases\r\n",
      "     fixmate        fix mate information\r\n",
      "     reheader       replace BAM header\r\n",
      "     targetcut      cut fosmid regions (for fosmid pool only)\r\n",
      "     addreplacerg   adds or replaces RG tags\r\n",
      "     markdup        mark duplicates\r\n",
      "\r\n",
      "  -- File operations\r\n",
      "     collate        shuffle and group alignments by name\r\n",
      "     cat            concatenate BAMs\r\n",
      "     merge          merge sorted alignments\r\n",
      "     mpileup        multi-way pileup\r\n",
      "     sort           sort alignment file\r\n",
      "     split          splits a file by read group\r\n",
      "     quickcheck     quickly check if SAM/BAM/CRAM file appears intact\r\n",
      "     fastq          converts a BAM to a FASTQ\r\n",
      "     fasta          converts a BAM to a FASTA\r\n",
      "\r\n",
      "  -- Statistics\r\n",
      "     bedcov         read depth per BED region\r\n",
      "     coverage       alignment depth and percent coverage\r\n",
      "     depth          compute the depth\r\n",
      "     flagstat       simple stats\r\n",
      "     idxstats       BAM index stats\r\n",
      "     phase          phase heterozygotes\r\n",
      "     stats          generate stats (former bamcheck)\r\n",
      "\r\n",
      "  -- Viewing\r\n",
      "     flags          explain BAM flags\r\n",
      "     tview          text alignment viewer\r\n",
      "     view           SAM<->BAM<->CRAM conversion\r\n",
      "     depad          convert padded BAM to unpadded BAM\r\n",
      "\r\n",
      "Samtools is converting SRR6145423.sam to a bam file.\r\n",
      "Samtools is sorting SRR6145423.bam\r\n",
      "Samtools is indexing SRR6145423.sorted.bam\r\n",
      "\r\n",
      "Program: samtools (Tools for alignments in the SAM format)\r\n",
      "Version: 1.10 (using htslib 1.10)\r\n",
      "\r\n",
      "Usage:   samtools <command> [options]\r\n",
      "\r\n",
      "Commands:\r\n",
      "  -- Indexing\r\n",
      "     dict           create a sequence dictionary file\r\n",
      "     faidx          index/extract FASTA\r\n",
      "     fqidx          index/extract FASTQ\r\n",
      "     index          index alignment\r\n",
      "\r\n",
      "  -- Editing\r\n",
      "     calmd          recalculate MD/NM tags and '=' bases\r\n",
      "     fixmate        fix mate information\r\n",
      "     reheader       replace BAM header\r\n",
      "     targetcut      cut fosmid regions (for fosmid pool only)\r\n",
      "     addreplacerg   adds or replaces RG tags\r\n",
      "     markdup        mark duplicates\r\n",
      "\r\n",
      "  -- File operations\r\n",
      "     collate        shuffle and group alignments by name\r\n",
      "     cat            concatenate BAMs\r\n",
      "     merge          merge sorted alignments\r\n",
      "     mpileup        multi-way pileup\r\n",
      "     sort           sort alignment file\r\n",
      "     split          splits a file by read group\r\n",
      "     quickcheck     quickly check if SAM/BAM/CRAM file appears intact\r\n",
      "     fastq          converts a BAM to a FASTQ\r\n",
      "     fasta          converts a BAM to a FASTA\r\n",
      "\r\n",
      "  -- Statistics\r\n",
      "     bedcov         read depth per BED region\r\n",
      "     coverage       alignment depth and percent coverage\r\n",
      "     depth          compute the depth\r\n",
      "     flagstat       simple stats\r\n",
      "     idxstats       BAM index stats\r\n",
      "     phase          phase heterozygotes\r\n",
      "     stats          generate stats (former bamcheck)\r\n",
      "\r\n",
      "  -- Viewing\r\n",
      "     flags          explain BAM flags\r\n",
      "     tview          text alignment viewer\r\n",
      "     view           SAM<->BAM<->CRAM conversion\r\n",
      "     depad          convert padded BAM to unpadded BAM\r\n",
      "\r\n",
      "Samtools is converting SRR6145424.sam to a bam file.\r\n",
      "Samtools is sorting SRR6145424.bam\r\n",
      "Samtools is indexing SRR6145424.sorted.bam\r\n",
      "\r\n",
      "Program: samtools (Tools for alignments in the SAM format)\r\n",
      "Version: 1.10 (using htslib 1.10)\r\n",
      "\r\n",
      "Usage:   samtools <command> [options]\r\n",
      "\r\n",
      "Commands:\r\n",
      "  -- Indexing\r\n",
      "     dict           create a sequence dictionary file\r\n",
      "     faidx          index/extract FASTA\r\n",
      "     fqidx          index/extract FASTQ\r\n",
      "     index          index alignment\r\n",
      "\r\n",
      "  -- Editing\r\n",
      "     calmd          recalculate MD/NM tags and '=' bases\r\n",
      "     fixmate        fix mate information\r\n",
      "     reheader       replace BAM header\r\n",
      "     targetcut      cut fosmid regions (for fosmid pool only)\r\n",
      "     addreplacerg   adds or replaces RG tags\r\n",
      "     markdup        mark duplicates\r\n",
      "\r\n",
      "  -- File operations\r\n",
      "     collate        shuffle and group alignments by name\r\n",
      "     cat            concatenate BAMs\r\n",
      "     merge          merge sorted alignments\r\n",
      "     mpileup        multi-way pileup\r\n",
      "     sort           sort alignment file\r\n",
      "     split          splits a file by read group\r\n",
      "     quickcheck     quickly check if SAM/BAM/CRAM file appears intact\r\n",
      "     fastq          converts a BAM to a FASTQ\r\n",
      "     fasta          converts a BAM to a FASTA\r\n",
      "\r\n",
      "  -- Statistics\r\n",
      "     bedcov         read depth per BED region\r\n",
      "     coverage       alignment depth and percent coverage\r\n",
      "     depth          compute the depth\r\n",
      "     flagstat       simple stats\r\n",
      "     idxstats       BAM index stats\r\n",
      "     phase          phase heterozygotes\r\n",
      "     stats          generate stats (former bamcheck)\r\n",
      "\r\n",
      "  -- Viewing\r\n",
      "     flags          explain BAM flags\r\n",
      "     tview          text alignment viewer\r\n",
      "     view           SAM<->BAM<->CRAM conversion\r\n",
      "     depad          convert padded BAM to unpadded BAM\r\n",
      "\r\n",
      "Samtools is converting SRR6145425.sam to a bam file.\r\n",
      "Samtools is sorting SRR6145425.bam\r\n",
      "Samtools is indexing SRR6145425.sorted.bam\r\n",
      "\r\n",
      "Program: samtools (Tools for alignments in the SAM format)\r\n",
      "Version: 1.10 (using htslib 1.10)\r\n",
      "\r\n",
      "Usage:   samtools <command> [options]\r\n",
      "\r\n",
      "Commands:\r\n",
      "  -- Indexing\r\n",
      "     dict           create a sequence dictionary file\r\n",
      "     faidx          index/extract FASTA\r\n",
      "     fqidx          index/extract FASTQ\r\n",
      "     index          index alignment\r\n",
      "\r\n",
      "  -- Editing\r\n",
      "     calmd          recalculate MD/NM tags and '=' bases\r\n",
      "     fixmate        fix mate information\r\n",
      "     reheader       replace BAM header\r\n",
      "     targetcut      cut fosmid regions (for fosmid pool only)\r\n",
      "     addreplacerg   adds or replaces RG tags\r\n",
      "     markdup        mark duplicates\r\n",
      "\r\n",
      "  -- File operations\r\n",
      "     collate        shuffle and group alignments by name\r\n",
      "     cat            concatenate BAMs\r\n",
      "     merge          merge sorted alignments\r\n",
      "     mpileup        multi-way pileup\r\n",
      "     sort           sort alignment file\r\n",
      "     split          splits a file by read group\r\n",
      "     quickcheck     quickly check if SAM/BAM/CRAM file appears intact\r\n",
      "     fastq          converts a BAM to a FASTQ\r\n",
      "     fasta          converts a BAM to a FASTA\r\n",
      "\r\n",
      "  -- Statistics\r\n",
      "     bedcov         read depth per BED region\r\n",
      "     coverage       alignment depth and percent coverage\r\n",
      "     depth          compute the depth\r\n",
      "     flagstat       simple stats\r\n",
      "     idxstats       BAM index stats\r\n",
      "     phase          phase heterozygotes\r\n",
      "     stats          generate stats (former bamcheck)\r\n",
      "\r\n",
      "  -- Viewing\r\n",
      "     flags          explain BAM flags\r\n",
      "     tview          text alignment viewer\r\n",
      "     view           SAM<->BAM<->CRAM conversion\r\n",
      "     depad          convert padded BAM to unpadded BAM\r\n",
      "\r\n",
      "Samtools is converting SRR6145426.sam to a bam file.\r\n",
      "Samtools is sorting SRR6145426.bam\r\n",
      "Samtools is indexing SRR6145426.sorted.bam\r\n",
      "\r\n",
      "Program: samtools (Tools for alignments in the SAM format)\r\n",
      "Version: 1.10 (using htslib 1.10)\r\n",
      "\r\n",
      "Usage:   samtools <command> [options]\r\n",
      "\r\n",
      "Commands:\r\n",
      "  -- Indexing\r\n",
      "     dict           create a sequence dictionary file\r\n",
      "     faidx          index/extract FASTA\r\n",
      "     fqidx          index/extract FASTQ\r\n",
      "     index          index alignment\r\n",
      "\r\n",
      "  -- Editing\r\n",
      "     calmd          recalculate MD/NM tags and '=' bases\r\n",
      "     fixmate        fix mate information\r\n",
      "     reheader       replace BAM header\r\n",
      "     targetcut      cut fosmid regions (for fosmid pool only)\r\n",
      "     addreplacerg   adds or replaces RG tags\r\n",
      "     markdup        mark duplicates\r\n",
      "\r\n",
      "  -- File operations\r\n",
      "     collate        shuffle and group alignments by name\r\n",
      "     cat            concatenate BAMs\r\n",
      "     merge          merge sorted alignments\r\n",
      "     mpileup        multi-way pileup\r\n",
      "     sort           sort alignment file\r\n",
      "     split          splits a file by read group\r\n",
      "     quickcheck     quickly check if SAM/BAM/CRAM file appears intact\r\n",
      "     fastq          converts a BAM to a FASTQ\r\n",
      "     fasta          converts a BAM to a FASTA\r\n",
      "\r\n",
      "  -- Statistics\r\n",
      "     bedcov         read depth per BED region\r\n",
      "     coverage       alignment depth and percent coverage\r\n",
      "     depth          compute the depth\r\n",
      "     flagstat       simple stats\r\n",
      "     idxstats       BAM index stats\r\n",
      "     phase          phase heterozygotes\r\n",
      "     stats          generate stats (former bamcheck)\r\n",
      "\r\n",
      "  -- Viewing\r\n",
      "     flags          explain BAM flags\r\n",
      "     tview          text alignment viewer\r\n",
      "     view           SAM<->BAM<->CRAM conversion\r\n",
      "     depad          convert padded BAM to unpadded BAM\r\n",
      "\r\n",
      "Samtools is converting SRR6145427.sam to a bam file.\r\n",
      "Samtools is sorting SRR6145427.bam\r\n",
      "Samtools is indexing SRR6145427.sorted.bam\r\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Program: samtools (Tools for alignments in the SAM format)\r\n",
      "Version: 1.10 (using htslib 1.10)\r\n",
      "\r\n",
      "Usage:   samtools <command> [options]\r\n",
      "\r\n",
      "Commands:\r\n",
      "  -- Indexing\r\n",
      "     dict           create a sequence dictionary file\r\n",
      "     faidx          index/extract FASTA\r\n",
      "     fqidx          index/extract FASTQ\r\n",
      "     index          index alignment\r\n",
      "\r\n",
      "  -- Editing\r\n",
      "     calmd          recalculate MD/NM tags and '=' bases\r\n",
      "     fixmate        fix mate information\r\n",
      "     reheader       replace BAM header\r\n",
      "     targetcut      cut fosmid regions (for fosmid pool only)\r\n",
      "     addreplacerg   adds or replaces RG tags\r\n",
      "     markdup        mark duplicates\r\n",
      "\r\n",
      "  -- File operations\r\n",
      "     collate        shuffle and group alignments by name\r\n",
      "     cat            concatenate BAMs\r\n",
      "     merge          merge sorted alignments\r\n",
      "     mpileup        multi-way pileup\r\n",
      "     sort           sort alignment file\r\n",
      "     split          splits a file by read group\r\n",
      "     quickcheck     quickly check if SAM/BAM/CRAM file appears intact\r\n",
      "     fastq          converts a BAM to a FASTQ\r\n",
      "     fasta          converts a BAM to a FASTA\r\n",
      "\r\n",
      "  -- Statistics\r\n",
      "     bedcov         read depth per BED region\r\n",
      "     coverage       alignment depth and percent coverage\r\n",
      "     depth          compute the depth\r\n",
      "     flagstat       simple stats\r\n",
      "     idxstats       BAM index stats\r\n",
      "     phase          phase heterozygotes\r\n",
      "     stats          generate stats (former bamcheck)\r\n",
      "\r\n",
      "  -- Viewing\r\n",
      "     flags          explain BAM flags\r\n",
      "     tview          text alignment viewer\r\n",
      "     view           SAM<->BAM<->CRAM conversion\r\n",
      "     depad          convert padded BAM to unpadded BAM\r\n",
      "\r\n",
      "Samtools is converting SRR6145433.sam to a bam file.\r\n",
      "Samtools is sorting SRR6145433.bam\r\n",
      "Samtools is indexing SRR6145433.sorted.bam\r\n",
      "\r\n",
      "Program: samtools (Tools for alignments in the SAM format)\r\n",
      "Version: 1.10 (using htslib 1.10)\r\n",
      "\r\n",
      "Usage:   samtools <command> [options]\r\n",
      "\r\n",
      "Commands:\r\n",
      "  -- Indexing\r\n",
      "     dict           create a sequence dictionary file\r\n",
      "     faidx          index/extract FASTA\r\n",
      "     fqidx          index/extract FASTQ\r\n",
      "     index          index alignment\r\n",
      "\r\n",
      "  -- Editing\r\n",
      "     calmd          recalculate MD/NM tags and '=' bases\r\n",
      "     fixmate        fix mate information\r\n",
      "     reheader       replace BAM header\r\n",
      "     targetcut      cut fosmid regions (for fosmid pool only)\r\n",
      "     addreplacerg   adds or replaces RG tags\r\n",
      "     markdup        mark duplicates\r\n",
      "\r\n",
      "  -- File operations\r\n",
      "     collate        shuffle and group alignments by name\r\n",
      "     cat            concatenate BAMs\r\n",
      "     merge          merge sorted alignments\r\n",
      "     mpileup        multi-way pileup\r\n",
      "     sort           sort alignment file\r\n",
      "     split          splits a file by read group\r\n",
      "     quickcheck     quickly check if SAM/BAM/CRAM file appears intact\r\n",
      "     fastq          converts a BAM to a FASTQ\r\n",
      "     fasta          converts a BAM to a FASTA\r\n",
      "\r\n",
      "  -- Statistics\r\n",
      "     bedcov         read depth per BED region\r\n",
      "     coverage       alignment depth and percent coverage\r\n",
      "     depth          compute the depth\r\n",
      "     flagstat       simple stats\r\n",
      "     idxstats       BAM index stats\r\n",
      "     phase          phase heterozygotes\r\n",
      "     stats          generate stats (former bamcheck)\r\n",
      "\r\n",
      "  -- Viewing\r\n",
      "     flags          explain BAM flags\r\n",
      "     tview          text alignment viewer\r\n",
      "     view           SAM<->BAM<->CRAM conversion\r\n",
      "     depad          convert padded BAM to unpadded BAM\r\n",
      "\r\n",
      "Samtools is converting SRR6145434.sam to a bam file.\r\n",
      "Samtools is sorting SRR6145434.bam\r\n",
      "Samtools is indexing SRR6145434.sorted.bam\r\n",
      "\r\n",
      "Program: samtools (Tools for alignments in the SAM format)\r\n",
      "Version: 1.10 (using htslib 1.10)\r\n",
      "\r\n",
      "Usage:   samtools <command> [options]\r\n",
      "\r\n",
      "Commands:\r\n",
      "  -- Indexing\r\n",
      "     dict           create a sequence dictionary file\r\n",
      "     faidx          index/extract FASTA\r\n",
      "     fqidx          index/extract FASTQ\r\n",
      "     index          index alignment\r\n",
      "\r\n",
      "  -- Editing\r\n",
      "     calmd          recalculate MD/NM tags and '=' bases\r\n",
      "     fixmate        fix mate information\r\n",
      "     reheader       replace BAM header\r\n",
      "     targetcut      cut fosmid regions (for fosmid pool only)\r\n",
      "     addreplacerg   adds or replaces RG tags\r\n",
      "     markdup        mark duplicates\r\n",
      "\r\n",
      "  -- File operations\r\n",
      "     collate        shuffle and group alignments by name\r\n",
      "     cat            concatenate BAMs\r\n",
      "     merge          merge sorted alignments\r\n",
      "     mpileup        multi-way pileup\r\n",
      "     sort           sort alignment file\r\n",
      "     split          splits a file by read group\r\n",
      "     quickcheck     quickly check if SAM/BAM/CRAM file appears intact\r\n",
      "     fastq          converts a BAM to a FASTQ\r\n",
      "     fasta          converts a BAM to a FASTA\r\n",
      "\r\n",
      "  -- Statistics\r\n",
      "     bedcov         read depth per BED region\r\n",
      "     coverage       alignment depth and percent coverage\r\n",
      "     depth          compute the depth\r\n",
      "     flagstat       simple stats\r\n",
      "     idxstats       BAM index stats\r\n",
      "     phase          phase heterozygotes\r\n",
      "     stats          generate stats (former bamcheck)\r\n",
      "\r\n",
      "  -- Viewing\r\n",
      "     flags          explain BAM flags\r\n",
      "     tview          text alignment viewer\r\n",
      "     view           SAM<->BAM<->CRAM conversion\r\n",
      "     depad          convert padded BAM to unpadded BAM\r\n",
      "\r\n",
      "Samtools is converting SRR6145435.sam to a bam file.\r\n",
      "Samtools is sorting SRR6145435.bam\r\n",
      "Samtools is indexing SRR6145435.sorted.bam\r\n",
      "\r\n",
      "Program: samtools (Tools for alignments in the SAM format)\r\n",
      "Version: 1.10 (using htslib 1.10)\r\n",
      "\r\n",
      "Usage:   samtools <command> [options]\r\n",
      "\r\n",
      "Commands:\r\n",
      "  -- Indexing\r\n",
      "     dict           create a sequence dictionary file\r\n",
      "     faidx          index/extract FASTA\r\n",
      "     fqidx          index/extract FASTQ\r\n",
      "     index          index alignment\r\n",
      "\r\n",
      "  -- Editing\r\n",
      "     calmd          recalculate MD/NM tags and '=' bases\r\n",
      "     fixmate        fix mate information\r\n",
      "     reheader       replace BAM header\r\n",
      "     targetcut      cut fosmid regions (for fosmid pool only)\r\n",
      "     addreplacerg   adds or replaces RG tags\r\n",
      "     markdup        mark duplicates\r\n",
      "\r\n",
      "  -- File operations\r\n",
      "     collate        shuffle and group alignments by name\r\n",
      "     cat            concatenate BAMs\r\n",
      "     merge          merge sorted alignments\r\n",
      "     mpileup        multi-way pileup\r\n",
      "     sort           sort alignment file\r\n",
      "     split          splits a file by read group\r\n",
      "     quickcheck     quickly check if SAM/BAM/CRAM file appears intact\r\n",
      "     fastq          converts a BAM to a FASTQ\r\n",
      "     fasta          converts a BAM to a FASTA\r\n",
      "\r\n",
      "  -- Statistics\r\n",
      "     bedcov         read depth per BED region\r\n",
      "     coverage       alignment depth and percent coverage\r\n",
      "     depth          compute the depth\r\n",
      "     flagstat       simple stats\r\n",
      "     idxstats       BAM index stats\r\n",
      "     phase          phase heterozygotes\r\n",
      "     stats          generate stats (former bamcheck)\r\n",
      "\r\n",
      "  -- Viewing\r\n",
      "     flags          explain BAM flags\r\n",
      "     tview          text alignment viewer\r\n",
      "     view           SAM<->BAM<->CRAM conversion\r\n",
      "     depad          convert padded BAM to unpadded BAM\r\n",
      "\r\n",
      "Samtools is converting SRR6145436.sam to a bam file.\r\n",
      "Samtools is sorting SRR6145436.bam\r\n",
      "Samtools is indexing SRR6145436.sorted.bam\r\n",
      "\r\n",
      "Program: samtools (Tools for alignments in the SAM format)\r\n",
      "Version: 1.10 (using htslib 1.10)\r\n",
      "\r\n",
      "Usage:   samtools <command> [options]\r\n",
      "\r\n",
      "Commands:\r\n",
      "  -- Indexing\r\n",
      "     dict           create a sequence dictionary file\r\n",
      "     faidx          index/extract FASTA\r\n",
      "     fqidx          index/extract FASTQ\r\n",
      "     index          index alignment\r\n",
      "\r\n",
      "  -- Editing\r\n",
      "     calmd          recalculate MD/NM tags and '=' bases\r\n",
      "     fixmate        fix mate information\r\n",
      "     reheader       replace BAM header\r\n",
      "     targetcut      cut fosmid regions (for fosmid pool only)\r\n",
      "     addreplacerg   adds or replaces RG tags\r\n",
      "     markdup        mark duplicates\r\n",
      "\r\n",
      "  -- File operations\r\n",
      "     collate        shuffle and group alignments by name\r\n",
      "     cat            concatenate BAMs\r\n",
      "     merge          merge sorted alignments\r\n",
      "     mpileup        multi-way pileup\r\n",
      "     sort           sort alignment file\r\n",
      "     split          splits a file by read group\r\n",
      "     quickcheck     quickly check if SAM/BAM/CRAM file appears intact\r\n",
      "     fastq          converts a BAM to a FASTQ\r\n",
      "     fasta          converts a BAM to a FASTA\r\n",
      "\r\n",
      "  -- Statistics\r\n",
      "     bedcov         read depth per BED region\r\n",
      "     coverage       alignment depth and percent coverage\r\n",
      "     depth          compute the depth\r\n",
      "     flagstat       simple stats\r\n",
      "     idxstats       BAM index stats\r\n",
      "     phase          phase heterozygotes\r\n",
      "     stats          generate stats (former bamcheck)\r\n",
      "\r\n",
      "  -- Viewing\r\n",
      "     flags          explain BAM flags\r\n",
      "     tview          text alignment viewer\r\n",
      "     view           SAM<->BAM<->CRAM conversion\r\n",
      "     depad          convert padded BAM to unpadded BAM\r\n",
      "\r\n",
      "Samtools is converting SRR6145437.sam to a bam file.\r\n",
      "Samtools is sorting SRR6145437.bam\r\n",
      "Samtools is indexing SRR6145437.sorted.bam\r\n"
     ]
    }
   ],
   "source": [
    "!./src/samtobam.sh -d $WORKDIR -l $SAMTOOLS"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Counting Gene Features\n",
    "\n",
    "Now that the alignment is finished, we count reads that are aligned to genes using HTSeq-count, which can be found in htseq.sh. In this script, it is specified that the alignments are ordered on position and that it is unstranded."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HTSeq is counting SRR6145423.sorted.bam.\n",
      "100000 GFF lines processed.\n",
      "200000 GFF lines processed.\n",
      "300000 GFF lines processed.\n",
      "400000 GFF lines processed.\n",
      "500000 GFF lines processed.\n",
      "600000 GFF lines processed.\n",
      "700000 GFF lines processed.\n",
      "800000 GFF lines processed.\n",
      "900000 GFF lines processed.\n",
      "1000000 GFF lines processed.\n",
      "1100000 GFF lines processed.\n",
      "1200000 GFF lines processed.\n",
      "1300000 GFF lines processed.\n",
      "1400000 GFF lines processed.\n",
      "1500000 GFF lines processed.\n",
      "1600000 GFF lines processed.\n",
      "1700000 GFF lines processed.\n",
      "1800000 GFF lines processed.\n",
      "1900000 GFF lines processed.\n",
      "2000000 GFF lines processed.\n",
      "2100000 GFF lines processed.\n",
      "2200000 GFF lines processed.\n",
      "2300000 GFF lines processed.\n",
      "2400000 GFF lines processed.\n",
      "2500000 GFF lines processed.\n",
      "2600000 GFF lines processed.\n",
      "2700000 GFF lines processed.\n",
      "2800000 GFF lines processed.\n",
      "2900000 GFF lines processed.\n",
      "2908224 GFF lines processed.\n",
      "100000 SAM alignment records processed.\n",
      "200000 SAM alignment records processed.\n",
      "300000 SAM alignment records processed.\n",
      "400000 SAM alignment records processed.\n",
      "500000 SAM alignment records processed.\n",
      "600000 SAM alignment records processed.\n",
      "700000 SAM alignment records processed.\n",
      "800000 SAM alignment records processed.\n",
      "900000 SAM alignment records processed.\n",
      "1000000 SAM alignment records processed.\n",
      "1100000 SAM alignment records processed.\n",
      "1200000 SAM alignment records processed.\n",
      "1300000 SAM alignment records processed.\n",
      "1400000 SAM alignment records processed.\n",
      "1500000 SAM alignment records processed.\n",
      "1600000 SAM alignment records processed.\n",
      "1700000 SAM alignment records processed.\n",
      "1800000 SAM alignment records processed.\n",
      "1900000 SAM alignment records processed.\n",
      "2000000 SAM alignment records processed.\n",
      "2100000 SAM alignment records processed.\n",
      "2200000 SAM alignment records processed.\n",
      "2300000 SAM alignment records processed.\n",
      "2400000 SAM alignment records processed.\n",
      "2500000 SAM alignment records processed.\n",
      "2600000 SAM alignment records processed.\n",
      "2700000 SAM alignment records processed.\n",
      "2800000 SAM alignment records processed.\n",
      "2900000 SAM alignment records processed.\n",
      "3000000 SAM alignment records processed.\n",
      "3100000 SAM alignment records processed.\n",
      "3200000 SAM alignment records processed.\n",
      "3300000 SAM alignment records processed.\n",
      "3400000 SAM alignment records processed.\n",
      "3500000 SAM alignment records processed.\n",
      "3600000 SAM alignment records processed.\n",
      "3700000 SAM alignment records processed.\n",
      "3800000 SAM alignment records processed.\n",
      "3900000 SAM alignment records processed.\n",
      "4000000 SAM alignment records processed.\n",
      "4100000 SAM alignment records processed.\n",
      "4200000 SAM alignment records processed.\n",
      "4300000 SAM alignment records processed.\n",
      "4400000 SAM alignment records processed.\n",
      "4500000 SAM alignment records processed.\n",
      "4600000 SAM alignment records processed.\n",
      "4700000 SAM alignment records processed.\n",
      "4800000 SAM alignment records processed.\n",
      "4900000 SAM alignment records processed.\n",
      "5000000 SAM alignment records processed.\n",
      "5100000 SAM alignment records processed.\n",
      "5200000 SAM alignment records processed.\n",
      "5300000 SAM alignment records processed.\n",
      "5400000 SAM alignment records processed.\n",
      "5500000 SAM alignment records processed.\n",
      "5600000 SAM alignment records processed.\n",
      "5700000 SAM alignment records processed.\n",
      "5800000 SAM alignment records processed.\n",
      "5900000 SAM alignment records processed.\n",
      "6000000 SAM alignment records processed.\n",
      "6100000 SAM alignment records processed.\n",
      "6200000 SAM alignment records processed.\n",
      "6300000 SAM alignment records processed.\n",
      "6400000 SAM alignment records processed.\n",
      "6500000 SAM alignment records processed.\n",
      "6600000 SAM alignment records processed.\n",
      "6700000 SAM alignment records processed.\n",
      "6800000 SAM alignment records processed.\n",
      "6900000 SAM alignment records processed.\n",
      "7000000 SAM alignment records processed.\n",
      "7100000 SAM alignment records processed.\n",
      "7200000 SAM alignment records processed.\n",
      "7300000 SAM alignment records processed.\n",
      "7400000 SAM alignment records processed.\n",
      "7500000 SAM alignment records processed.\n",
      "7600000 SAM alignment records processed.\n",
      "7700000 SAM alignment records processed.\n",
      "7800000 SAM alignment records processed.\n",
      "7900000 SAM alignment records processed.\n",
      "8000000 SAM alignment records processed.\n",
      "8100000 SAM alignment records processed.\n",
      "8200000 SAM alignment records processed.\n",
      "8300000 SAM alignment records processed.\n",
      "8400000 SAM alignment records processed.\n",
      "8500000 SAM alignment records processed.\n",
      "8600000 SAM alignment records processed.\n",
      "8700000 SAM alignment records processed.\n",
      "8800000 SAM alignment records processed.\n",
      "8900000 SAM alignment records processed.\n",
      "9000000 SAM alignment records processed.\n",
      "9100000 SAM alignment records processed.\n",
      "9200000 SAM alignment records processed.\n",
      "9300000 SAM alignment records processed.\n",
      "9400000 SAM alignment records processed.\n",
      "9500000 SAM alignment records processed.\n",
      "9600000 SAM alignment records processed.\n",
      "9700000 SAM alignment records processed.\n",
      "9800000 SAM alignment records processed.\n",
      "9900000 SAM alignment records processed.\n",
      "10000000 SAM alignment records processed.\n",
      "10100000 SAM alignment records processed.\n",
      "10200000 SAM alignment records processed.\n",
      "10300000 SAM alignment records processed.\n",
      "10400000 SAM alignment records processed.\n",
      "10500000 SAM alignment records processed.\n",
      "10600000 SAM alignment records processed.\n",
      "10700000 SAM alignment records processed.\n",
      "10800000 SAM alignment records processed.\n",
      "10900000 SAM alignment records processed.\n",
      "11000000 SAM alignment records processed.\n",
      "11100000 SAM alignment records processed.\n",
      "11200000 SAM alignment records processed.\n",
      "11300000 SAM alignment records processed.\n",
      "11400000 SAM alignment records processed.\n",
      "11500000 SAM alignment records processed.\n",
      "11600000 SAM alignment records processed.\n",
      "11700000 SAM alignment records processed.\n",
      "11800000 SAM alignment records processed.\n",
      "11900000 SAM alignment records processed.\n",
      "12000000 SAM alignment records processed.\n",
      "12100000 SAM alignment records processed.\n",
      "12200000 SAM alignment records processed.\n",
      "12300000 SAM alignment records processed.\n",
      "12400000 SAM alignment records processed.\n",
      "12500000 SAM alignment records processed.\n",
      "12600000 SAM alignment records processed.\n",
      "12700000 SAM alignment records processed.\n",
      "12800000 SAM alignment records processed.\n",
      "12900000 SAM alignment records processed.\n",
      "13000000 SAM alignment records processed.\n",
      "13100000 SAM alignment records processed.\n",
      "13200000 SAM alignment records processed.\n",
      "13300000 SAM alignment records processed.\n",
      "13400000 SAM alignment records processed.\n",
      "13500000 SAM alignment records processed.\n",
      "13600000 SAM alignment records processed.\n",
      "13700000 SAM alignment records processed.\n",
      "13800000 SAM alignment records processed.\n",
      "13900000 SAM alignment records processed.\n",
      "14000000 SAM alignment records processed.\n",
      "14100000 SAM alignment records processed.\n",
      "14200000 SAM alignment records processed.\n",
      "14300000 SAM alignment records processed.\n",
      "14400000 SAM alignment records processed.\n",
      "14500000 SAM alignment records processed.\n",
      "14600000 SAM alignment records processed.\n",
      "14700000 SAM alignment records processed.\n",
      "14800000 SAM alignment records processed.\n",
      "14900000 SAM alignment records processed.\n",
      "15000000 SAM alignment records processed.\n",
      "15100000 SAM alignment records processed.\n",
      "15200000 SAM alignment records processed.\n",
      "15300000 SAM alignment records processed.\n",
      "15400000 SAM alignment records processed.\n",
      "15500000 SAM alignment records processed.\n",
      "15600000 SAM alignment records processed.\n",
      "15700000 SAM alignment records processed.\n",
      "15800000 SAM alignment records processed.\n",
      "15900000 SAM alignment records processed.\n",
      "16000000 SAM alignment records processed.\n",
      "16100000 SAM alignment records processed.\n",
      "16200000 SAM alignment records processed.\n",
      "16300000 SAM alignment records processed.\n",
      "16400000 SAM alignment records processed.\n",
      "16500000 SAM alignment records processed.\n",
      "16600000 SAM alignment records processed.\n",
      "16700000 SAM alignment records processed.\n",
      "16800000 SAM alignment records processed.\n",
      "16900000 SAM alignment records processed.\n",
      "17000000 SAM alignment records processed.\n",
      "17100000 SAM alignment records processed.\n",
      "17200000 SAM alignment records processed.\n",
      "17300000 SAM alignment records processed.\n",
      "17400000 SAM alignment records processed.\n",
      "17500000 SAM alignment records processed.\n",
      "17600000 SAM alignment records processed.\n",
      "17700000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17800000 SAM alignment records processed.\n",
      "17900000 SAM alignment records processed.\n",
      "18000000 SAM alignment records processed.\n",
      "18100000 SAM alignment records processed.\n",
      "18200000 SAM alignment records processed.\n",
      "18300000 SAM alignment records processed.\n",
      "18400000 SAM alignment records processed.\n",
      "18500000 SAM alignment records processed.\n",
      "18600000 SAM alignment records processed.\n",
      "18700000 SAM alignment records processed.\n",
      "18800000 SAM alignment records processed.\n",
      "18900000 SAM alignment records processed.\n",
      "19000000 SAM alignment records processed.\n",
      "19100000 SAM alignment records processed.\n",
      "19200000 SAM alignment records processed.\n",
      "19300000 SAM alignment records processed.\n",
      "19400000 SAM alignment records processed.\n",
      "19500000 SAM alignment records processed.\n",
      "19600000 SAM alignment records processed.\n",
      "19700000 SAM alignment records processed.\n",
      "19800000 SAM alignment records processed.\n",
      "19900000 SAM alignment records processed.\n",
      "20000000 SAM alignment records processed.\n",
      "20100000 SAM alignment records processed.\n",
      "20200000 SAM alignment records processed.\n",
      "20300000 SAM alignment records processed.\n",
      "20400000 SAM alignment records processed.\n",
      "20500000 SAM alignment records processed.\n",
      "20600000 SAM alignment records processed.\n",
      "20700000 SAM alignment records processed.\n",
      "20800000 SAM alignment records processed.\n",
      "20900000 SAM alignment records processed.\n",
      "21000000 SAM alignment records processed.\n",
      "21100000 SAM alignment records processed.\n",
      "21200000 SAM alignment records processed.\n",
      "21300000 SAM alignment records processed.\n",
      "21400000 SAM alignment records processed.\n",
      "21500000 SAM alignment records processed.\n",
      "21600000 SAM alignment records processed.\n",
      "21700000 SAM alignment records processed.\n",
      "21800000 SAM alignment records processed.\n",
      "21900000 SAM alignment records processed.\n",
      "22000000 SAM alignment records processed.\n",
      "22100000 SAM alignment records processed.\n",
      "22200000 SAM alignment records processed.\n",
      "22300000 SAM alignment records processed.\n",
      "22400000 SAM alignment records processed.\n",
      "22500000 SAM alignment records processed.\n",
      "22600000 SAM alignment records processed.\n",
      "22700000 SAM alignment records processed.\n",
      "22800000 SAM alignment records processed.\n",
      "22900000 SAM alignment records processed.\n",
      "23000000 SAM alignment records processed.\n",
      "23100000 SAM alignment records processed.\n",
      "23200000 SAM alignment records processed.\n",
      "23300000 SAM alignment records processed.\n",
      "23400000 SAM alignment records processed.\n",
      "23500000 SAM alignment records processed.\n",
      "23600000 SAM alignment records processed.\n",
      "23700000 SAM alignment records processed.\n",
      "23800000 SAM alignment records processed.\n",
      "23900000 SAM alignment records processed.\n",
      "24000000 SAM alignment records processed.\n",
      "24100000 SAM alignment records processed.\n",
      "24200000 SAM alignment records processed.\n",
      "24300000 SAM alignment records processed.\n",
      "24400000 SAM alignment records processed.\n",
      "24500000 SAM alignment records processed.\n",
      "24600000 SAM alignment records processed.\n",
      "24700000 SAM alignment records processed.\n",
      "24800000 SAM alignment records processed.\n",
      "24900000 SAM alignment records processed.\n",
      "25000000 SAM alignment records processed.\n",
      "25100000 SAM alignment records processed.\n",
      "25200000 SAM alignment records processed.\n",
      "25300000 SAM alignment records processed.\n",
      "25400000 SAM alignment records processed.\n",
      "25500000 SAM alignment records processed.\n",
      "25600000 SAM alignment records processed.\n",
      "25700000 SAM alignment records processed.\n",
      "25800000 SAM alignment records processed.\n",
      "25900000 SAM alignment records processed.\n",
      "26000000 SAM alignment records processed.\n",
      "26100000 SAM alignment records processed.\n",
      "26200000 SAM alignment records processed.\n",
      "26300000 SAM alignment records processed.\n",
      "26400000 SAM alignment records processed.\n",
      "26500000 SAM alignment records processed.\n",
      "26600000 SAM alignment records processed.\n",
      "26700000 SAM alignment records processed.\n",
      "26800000 SAM alignment records processed.\n",
      "26900000 SAM alignment records processed.\n",
      "27000000 SAM alignment records processed.\n",
      "27100000 SAM alignment records processed.\n",
      "27200000 SAM alignment records processed.\n",
      "27300000 SAM alignment records processed.\n",
      "27400000 SAM alignment records processed.\n",
      "27500000 SAM alignment records processed.\n",
      "27600000 SAM alignment records processed.\n",
      "27700000 SAM alignment records processed.\n",
      "27800000 SAM alignment records processed.\n",
      "27900000 SAM alignment records processed.\n",
      "28000000 SAM alignment records processed.\n",
      "28100000 SAM alignment records processed.\n",
      "28200000 SAM alignment records processed.\n",
      "28300000 SAM alignment records processed.\n",
      "28400000 SAM alignment records processed.\n",
      "28500000 SAM alignment records processed.\n",
      "28600000 SAM alignment records processed.\n",
      "28700000 SAM alignment records processed.\n",
      "28800000 SAM alignment records processed.\n",
      "28900000 SAM alignment records processed.\n",
      "29000000 SAM alignment records processed.\n",
      "29100000 SAM alignment records processed.\n",
      "29200000 SAM alignment records processed.\n",
      "29300000 SAM alignment records processed.\n",
      "29400000 SAM alignment records processed.\n",
      "29500000 SAM alignment records processed.\n",
      "29600000 SAM alignment records processed.\n",
      "29700000 SAM alignment records processed.\n",
      "29800000 SAM alignment records processed.\n",
      "29900000 SAM alignment records processed.\n",
      "30000000 SAM alignment records processed.\n",
      "30100000 SAM alignment records processed.\n",
      "30200000 SAM alignment records processed.\n",
      "30300000 SAM alignment records processed.\n",
      "30400000 SAM alignment records processed.\n",
      "30500000 SAM alignment records processed.\n",
      "30600000 SAM alignment records processed.\n",
      "30700000 SAM alignment records processed.\n",
      "30800000 SAM alignment records processed.\n",
      "30900000 SAM alignment records processed.\n",
      "31000000 SAM alignment records processed.\n",
      "31100000 SAM alignment records processed.\n",
      "31200000 SAM alignment records processed.\n",
      "31300000 SAM alignment records processed.\n",
      "31400000 SAM alignment records processed.\n",
      "31500000 SAM alignment records processed.\n",
      "31600000 SAM alignment records processed.\n",
      "31700000 SAM alignment records processed.\n",
      "31800000 SAM alignment records processed.\n",
      "31900000 SAM alignment records processed.\n",
      "32000000 SAM alignment records processed.\n",
      "32100000 SAM alignment records processed.\n",
      "32200000 SAM alignment records processed.\n",
      "32300000 SAM alignment records processed.\n",
      "32400000 SAM alignment records processed.\n",
      "32500000 SAM alignment records processed.\n",
      "32600000 SAM alignment records processed.\n",
      "32700000 SAM alignment records processed.\n",
      "32800000 SAM alignment records processed.\n",
      "32900000 SAM alignment records processed.\n",
      "33000000 SAM alignment records processed.\n",
      "33100000 SAM alignment records processed.\n",
      "33200000 SAM alignment records processed.\n",
      "33300000 SAM alignment records processed.\n",
      "33400000 SAM alignment records processed.\n",
      "33500000 SAM alignment records processed.\n",
      "33600000 SAM alignment records processed.\n",
      "33700000 SAM alignment records processed.\n",
      "33800000 SAM alignment records processed.\n",
      "33900000 SAM alignment records processed.\n",
      "34000000 SAM alignment records processed.\n",
      "34100000 SAM alignment records processed.\n",
      "34200000 SAM alignment records processed.\n",
      "34300000 SAM alignment records processed.\n",
      "34400000 SAM alignment records processed.\n",
      "34500000 SAM alignment records processed.\n",
      "34600000 SAM alignment records processed.\n",
      "34700000 SAM alignment records processed.\n",
      "34800000 SAM alignment records processed.\n",
      "34900000 SAM alignment records processed.\n",
      "35000000 SAM alignment records processed.\n",
      "35100000 SAM alignment records processed.\n",
      "35200000 SAM alignment records processed.\n",
      "35300000 SAM alignment records processed.\n",
      "35400000 SAM alignment records processed.\n",
      "35500000 SAM alignment records processed.\n",
      "35600000 SAM alignment records processed.\n",
      "35700000 SAM alignment records processed.\n",
      "35800000 SAM alignment records processed.\n",
      "35900000 SAM alignment records processed.\n",
      "36000000 SAM alignment records processed.\n",
      "36100000 SAM alignment records processed.\n",
      "36200000 SAM alignment records processed.\n",
      "36300000 SAM alignment records processed.\n",
      "36400000 SAM alignment records processed.\n",
      "36500000 SAM alignment records processed.\n",
      "36600000 SAM alignment records processed.\n",
      "36700000 SAM alignment records processed.\n",
      "36800000 SAM alignment records processed.\n",
      "36900000 SAM alignment records processed.\n",
      "37000000 SAM alignment records processed.\n",
      "37100000 SAM alignment records processed.\n",
      "37200000 SAM alignment records processed.\n",
      "37300000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "37400000 SAM alignment records processed.\n",
      "37500000 SAM alignment records processed.\n",
      "37600000 SAM alignment records processed.\n",
      "37700000 SAM alignment records processed.\n",
      "37800000 SAM alignment records processed.\n",
      "37900000 SAM alignment records processed.\n",
      "38000000 SAM alignment records processed.\n",
      "38100000 SAM alignment records processed.\n",
      "38200000 SAM alignment records processed.\n",
      "38300000 SAM alignment records processed.\n",
      "38400000 SAM alignment records processed.\n",
      "38500000 SAM alignment records processed.\n",
      "38600000 SAM alignment records processed.\n",
      "38700000 SAM alignment records processed.\n",
      "38800000 SAM alignment records processed.\n",
      "38900000 SAM alignment records processed.\n",
      "39000000 SAM alignment records processed.\n",
      "39100000 SAM alignment records processed.\n",
      "39200000 SAM alignment records processed.\n",
      "39300000 SAM alignment records processed.\n",
      "39400000 SAM alignment records processed.\n",
      "39500000 SAM alignment records processed.\n",
      "39600000 SAM alignment records processed.\n",
      "39700000 SAM alignment records processed.\n",
      "39800000 SAM alignment records processed.\n",
      "39900000 SAM alignment records processed.\n",
      "40000000 SAM alignment records processed.\n",
      "40100000 SAM alignment records processed.\n",
      "40200000 SAM alignment records processed.\n",
      "40300000 SAM alignment records processed.\n",
      "40400000 SAM alignment records processed.\n",
      "40500000 SAM alignment records processed.\n",
      "40600000 SAM alignment records processed.\n",
      "40700000 SAM alignment records processed.\n",
      "40800000 SAM alignment records processed.\n",
      "40900000 SAM alignment records processed.\n",
      "41000000 SAM alignment records processed.\n",
      "41100000 SAM alignment records processed.\n",
      "41200000 SAM alignment records processed.\n",
      "41300000 SAM alignment records processed.\n",
      "41400000 SAM alignment records processed.\n",
      "41500000 SAM alignment records processed.\n",
      "41600000 SAM alignment records processed.\n",
      "41700000 SAM alignment records processed.\n",
      "41800000 SAM alignment records processed.\n",
      "41900000 SAM alignment records processed.\n",
      "42000000 SAM alignment records processed.\n",
      "42100000 SAM alignment records processed.\n",
      "42200000 SAM alignment records processed.\n",
      "42300000 SAM alignment records processed.\n",
      "42400000 SAM alignment records processed.\n",
      "42500000 SAM alignment records processed.\n",
      "42600000 SAM alignment records processed.\n",
      "42700000 SAM alignment records processed.\n",
      "42800000 SAM alignment records processed.\n",
      "42900000 SAM alignment records processed.\n",
      "43000000 SAM alignment records processed.\n",
      "43100000 SAM alignment records processed.\n",
      "43200000 SAM alignment records processed.\n",
      "43300000 SAM alignment records processed.\n",
      "43400000 SAM alignment records processed.\n",
      "43500000 SAM alignment records processed.\n",
      "43600000 SAM alignment records processed.\n",
      "43700000 SAM alignment records processed.\n",
      "43800000 SAM alignment records processed.\n",
      "43900000 SAM alignment records processed.\n",
      "44000000 SAM alignment records processed.\n",
      "44100000 SAM alignment records processed.\n",
      "44200000 SAM alignment records processed.\n",
      "44300000 SAM alignment records processed.\n",
      "44400000 SAM alignment records processed.\n",
      "44500000 SAM alignment records processed.\n",
      "44600000 SAM alignment records processed.\n",
      "44700000 SAM alignment records processed.\n",
      "44800000 SAM alignment records processed.\n",
      "44900000 SAM alignment records processed.\n",
      "45000000 SAM alignment records processed.\n",
      "45100000 SAM alignment records processed.\n",
      "45200000 SAM alignment records processed.\n",
      "45300000 SAM alignment records processed.\n",
      "45400000 SAM alignment records processed.\n",
      "45500000 SAM alignment records processed.\n",
      "45600000 SAM alignment records processed.\n",
      "45700000 SAM alignment records processed.\n",
      "45800000 SAM alignment records processed.\n",
      "45900000 SAM alignment records processed.\n",
      "46000000 SAM alignment records processed.\n",
      "46100000 SAM alignment records processed.\n",
      "46200000 SAM alignment records processed.\n",
      "46300000 SAM alignment records processed.\n",
      "46400000 SAM alignment records processed.\n",
      "46500000 SAM alignment records processed.\n",
      "46600000 SAM alignment records processed.\n",
      "46700000 SAM alignment records processed.\n",
      "46800000 SAM alignment records processed.\n",
      "46900000 SAM alignment records processed.\n",
      "47000000 SAM alignment records processed.\n",
      "47100000 SAM alignment records processed.\n",
      "47200000 SAM alignment records processed.\n",
      "47300000 SAM alignment records processed.\n",
      "47400000 SAM alignment records processed.\n",
      "47427180 BAM alignments  processed.\n",
      "HTSeq is counting SRR6145424.sorted.bam.\n",
      "100000 GFF lines processed.\n",
      "200000 GFF lines processed.\n",
      "300000 GFF lines processed.\n",
      "400000 GFF lines processed.\n",
      "500000 GFF lines processed.\n",
      "600000 GFF lines processed.\n",
      "700000 GFF lines processed.\n",
      "800000 GFF lines processed.\n",
      "900000 GFF lines processed.\n",
      "1000000 GFF lines processed.\n",
      "1100000 GFF lines processed.\n",
      "1200000 GFF lines processed.\n",
      "1300000 GFF lines processed.\n",
      "1400000 GFF lines processed.\n",
      "1500000 GFF lines processed.\n",
      "1600000 GFF lines processed.\n",
      "1700000 GFF lines processed.\n",
      "1800000 GFF lines processed.\n",
      "1900000 GFF lines processed.\n",
      "2000000 GFF lines processed.\n",
      "2100000 GFF lines processed.\n",
      "2200000 GFF lines processed.\n",
      "2300000 GFF lines processed.\n",
      "2400000 GFF lines processed.\n",
      "2500000 GFF lines processed.\n",
      "2600000 GFF lines processed.\n",
      "2700000 GFF lines processed.\n",
      "2800000 GFF lines processed.\n",
      "2900000 GFF lines processed.\n",
      "2908224 GFF lines processed.\n",
      "100000 SAM alignment records processed.\n",
      "200000 SAM alignment records processed.\n",
      "300000 SAM alignment records processed.\n",
      "400000 SAM alignment records processed.\n",
      "500000 SAM alignment records processed.\n",
      "600000 SAM alignment records processed.\n",
      "700000 SAM alignment records processed.\n",
      "800000 SAM alignment records processed.\n",
      "900000 SAM alignment records processed.\n",
      "1000000 SAM alignment records processed.\n",
      "1100000 SAM alignment records processed.\n",
      "1200000 SAM alignment records processed.\n",
      "1300000 SAM alignment records processed.\n",
      "1400000 SAM alignment records processed.\n",
      "1500000 SAM alignment records processed.\n",
      "1600000 SAM alignment records processed.\n",
      "1700000 SAM alignment records processed.\n",
      "1800000 SAM alignment records processed.\n",
      "1900000 SAM alignment records processed.\n",
      "2000000 SAM alignment records processed.\n",
      "2100000 SAM alignment records processed.\n",
      "2200000 SAM alignment records processed.\n",
      "2300000 SAM alignment records processed.\n",
      "2400000 SAM alignment records processed.\n",
      "2500000 SAM alignment records processed.\n",
      "2600000 SAM alignment records processed.\n",
      "2700000 SAM alignment records processed.\n",
      "2800000 SAM alignment records processed.\n",
      "2900000 SAM alignment records processed.\n",
      "3000000 SAM alignment records processed.\n",
      "3100000 SAM alignment records processed.\n",
      "3200000 SAM alignment records processed.\n",
      "3300000 SAM alignment records processed.\n",
      "3400000 SAM alignment records processed.\n",
      "3500000 SAM alignment records processed.\n",
      "3600000 SAM alignment records processed.\n",
      "3700000 SAM alignment records processed.\n",
      "3800000 SAM alignment records processed.\n",
      "3900000 SAM alignment records processed.\n",
      "4000000 SAM alignment records processed.\n",
      "4100000 SAM alignment records processed.\n",
      "4200000 SAM alignment records processed.\n",
      "4300000 SAM alignment records processed.\n",
      "4400000 SAM alignment records processed.\n",
      "4500000 SAM alignment records processed.\n",
      "4600000 SAM alignment records processed.\n",
      "4700000 SAM alignment records processed.\n",
      "4800000 SAM alignment records processed.\n",
      "4900000 SAM alignment records processed.\n",
      "5000000 SAM alignment records processed.\n",
      "5100000 SAM alignment records processed.\n",
      "5200000 SAM alignment records processed.\n",
      "5300000 SAM alignment records processed.\n",
      "5400000 SAM alignment records processed.\n",
      "5500000 SAM alignment records processed.\n",
      "5600000 SAM alignment records processed.\n",
      "5700000 SAM alignment records processed.\n",
      "5800000 SAM alignment records processed.\n",
      "5900000 SAM alignment records processed.\n",
      "6000000 SAM alignment records processed.\n",
      "6100000 SAM alignment records processed.\n",
      "6200000 SAM alignment records processed.\n",
      "6300000 SAM alignment records processed.\n",
      "6400000 SAM alignment records processed.\n",
      "6500000 SAM alignment records processed.\n",
      "6600000 SAM alignment records processed.\n",
      "6700000 SAM alignment records processed.\n",
      "6800000 SAM alignment records processed.\n",
      "6900000 SAM alignment records processed.\n",
      "7000000 SAM alignment records processed.\n",
      "7100000 SAM alignment records processed.\n",
      "7200000 SAM alignment records processed.\n",
      "7300000 SAM alignment records processed.\n",
      "7400000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7500000 SAM alignment records processed.\n",
      "7600000 SAM alignment records processed.\n",
      "7700000 SAM alignment records processed.\n",
      "7800000 SAM alignment records processed.\n",
      "7900000 SAM alignment records processed.\n",
      "8000000 SAM alignment records processed.\n",
      "8100000 SAM alignment records processed.\n",
      "8200000 SAM alignment records processed.\n",
      "8300000 SAM alignment records processed.\n",
      "8400000 SAM alignment records processed.\n",
      "8500000 SAM alignment records processed.\n",
      "8600000 SAM alignment records processed.\n",
      "8700000 SAM alignment records processed.\n",
      "8800000 SAM alignment records processed.\n",
      "8900000 SAM alignment records processed.\n",
      "9000000 SAM alignment records processed.\n",
      "9100000 SAM alignment records processed.\n",
      "9200000 SAM alignment records processed.\n",
      "9300000 SAM alignment records processed.\n",
      "9400000 SAM alignment records processed.\n",
      "9500000 SAM alignment records processed.\n",
      "9600000 SAM alignment records processed.\n",
      "9700000 SAM alignment records processed.\n",
      "9800000 SAM alignment records processed.\n",
      "9900000 SAM alignment records processed.\n",
      "10000000 SAM alignment records processed.\n",
      "10100000 SAM alignment records processed.\n",
      "10200000 SAM alignment records processed.\n",
      "10300000 SAM alignment records processed.\n",
      "10400000 SAM alignment records processed.\n",
      "10500000 SAM alignment records processed.\n",
      "10600000 SAM alignment records processed.\n",
      "10700000 SAM alignment records processed.\n",
      "10800000 SAM alignment records processed.\n",
      "10900000 SAM alignment records processed.\n",
      "11000000 SAM alignment records processed.\n",
      "11100000 SAM alignment records processed.\n",
      "11200000 SAM alignment records processed.\n",
      "11300000 SAM alignment records processed.\n",
      "11400000 SAM alignment records processed.\n",
      "11500000 SAM alignment records processed.\n",
      "11600000 SAM alignment records processed.\n",
      "11700000 SAM alignment records processed.\n",
      "11800000 SAM alignment records processed.\n",
      "11900000 SAM alignment records processed.\n",
      "12000000 SAM alignment records processed.\n",
      "12100000 SAM alignment records processed.\n",
      "12200000 SAM alignment records processed.\n",
      "12300000 SAM alignment records processed.\n",
      "12400000 SAM alignment records processed.\n",
      "12500000 SAM alignment records processed.\n",
      "12600000 SAM alignment records processed.\n",
      "12700000 SAM alignment records processed.\n",
      "12800000 SAM alignment records processed.\n",
      "12900000 SAM alignment records processed.\n",
      "13000000 SAM alignment records processed.\n",
      "13100000 SAM alignment records processed.\n",
      "13200000 SAM alignment records processed.\n",
      "13300000 SAM alignment records processed.\n",
      "13400000 SAM alignment records processed.\n",
      "13500000 SAM alignment records processed.\n",
      "13600000 SAM alignment records processed.\n",
      "13700000 SAM alignment records processed.\n",
      "13800000 SAM alignment records processed.\n",
      "13900000 SAM alignment records processed.\n",
      "14000000 SAM alignment records processed.\n",
      "14100000 SAM alignment records processed.\n",
      "14200000 SAM alignment records processed.\n",
      "14300000 SAM alignment records processed.\n",
      "14400000 SAM alignment records processed.\n",
      "14500000 SAM alignment records processed.\n",
      "14600000 SAM alignment records processed.\n",
      "14700000 SAM alignment records processed.\n",
      "14800000 SAM alignment records processed.\n",
      "14900000 SAM alignment records processed.\n",
      "15000000 SAM alignment records processed.\n",
      "15100000 SAM alignment records processed.\n",
      "15200000 SAM alignment records processed.\n",
      "15300000 SAM alignment records processed.\n",
      "15400000 SAM alignment records processed.\n",
      "15500000 SAM alignment records processed.\n",
      "15600000 SAM alignment records processed.\n",
      "15700000 SAM alignment records processed.\n",
      "15800000 SAM alignment records processed.\n",
      "15900000 SAM alignment records processed.\n",
      "16000000 SAM alignment records processed.\n",
      "16100000 SAM alignment records processed.\n",
      "16200000 SAM alignment records processed.\n",
      "16300000 SAM alignment records processed.\n",
      "16400000 SAM alignment records processed.\n",
      "16500000 SAM alignment records processed.\n",
      "16600000 SAM alignment records processed.\n",
      "16700000 SAM alignment records processed.\n",
      "16800000 SAM alignment records processed.\n",
      "16900000 SAM alignment records processed.\n",
      "17000000 SAM alignment records processed.\n",
      "17100000 SAM alignment records processed.\n",
      "17200000 SAM alignment records processed.\n",
      "17300000 SAM alignment records processed.\n",
      "17400000 SAM alignment records processed.\n",
      "17500000 SAM alignment records processed.\n",
      "17600000 SAM alignment records processed.\n",
      "17700000 SAM alignment records processed.\n",
      "17800000 SAM alignment records processed.\n",
      "17900000 SAM alignment records processed.\n",
      "18000000 SAM alignment records processed.\n",
      "18100000 SAM alignment records processed.\n",
      "18200000 SAM alignment records processed.\n",
      "18300000 SAM alignment records processed.\n",
      "18400000 SAM alignment records processed.\n",
      "18500000 SAM alignment records processed.\n",
      "18600000 SAM alignment records processed.\n",
      "18700000 SAM alignment records processed.\n",
      "18800000 SAM alignment records processed.\n",
      "18900000 SAM alignment records processed.\n",
      "19000000 SAM alignment records processed.\n",
      "19100000 SAM alignment records processed.\n",
      "19200000 SAM alignment records processed.\n",
      "19300000 SAM alignment records processed.\n",
      "19400000 SAM alignment records processed.\n",
      "19500000 SAM alignment records processed.\n",
      "19600000 SAM alignment records processed.\n",
      "19700000 SAM alignment records processed.\n",
      "19800000 SAM alignment records processed.\n",
      "19900000 SAM alignment records processed.\n",
      "20000000 SAM alignment records processed.\n",
      "20100000 SAM alignment records processed.\n",
      "20200000 SAM alignment records processed.\n",
      "20300000 SAM alignment records processed.\n",
      "20400000 SAM alignment records processed.\n",
      "20500000 SAM alignment records processed.\n",
      "20600000 SAM alignment records processed.\n",
      "20700000 SAM alignment records processed.\n",
      "20800000 SAM alignment records processed.\n",
      "20900000 SAM alignment records processed.\n",
      "21000000 SAM alignment records processed.\n",
      "21100000 SAM alignment records processed.\n",
      "21200000 SAM alignment records processed.\n",
      "21300000 SAM alignment records processed.\n",
      "21400000 SAM alignment records processed.\n",
      "21500000 SAM alignment records processed.\n",
      "21600000 SAM alignment records processed.\n",
      "21700000 SAM alignment records processed.\n",
      "21800000 SAM alignment records processed.\n",
      "21900000 SAM alignment records processed.\n",
      "22000000 SAM alignment records processed.\n",
      "22100000 SAM alignment records processed.\n",
      "22200000 SAM alignment records processed.\n",
      "22300000 SAM alignment records processed.\n",
      "22400000 SAM alignment records processed.\n",
      "22500000 SAM alignment records processed.\n",
      "22600000 SAM alignment records processed.\n",
      "22700000 SAM alignment records processed.\n",
      "22800000 SAM alignment records processed.\n",
      "22900000 SAM alignment records processed.\n",
      "23000000 SAM alignment records processed.\n",
      "23100000 SAM alignment records processed.\n",
      "23200000 SAM alignment records processed.\n",
      "23300000 SAM alignment records processed.\n",
      "23400000 SAM alignment records processed.\n",
      "23500000 SAM alignment records processed.\n",
      "23600000 SAM alignment records processed.\n",
      "23700000 SAM alignment records processed.\n",
      "23800000 SAM alignment records processed.\n",
      "23900000 SAM alignment records processed.\n",
      "24000000 SAM alignment records processed.\n",
      "24100000 SAM alignment records processed.\n",
      "24200000 SAM alignment records processed.\n",
      "24300000 SAM alignment records processed.\n",
      "24400000 SAM alignment records processed.\n",
      "24500000 SAM alignment records processed.\n",
      "24600000 SAM alignment records processed.\n",
      "24700000 SAM alignment records processed.\n",
      "24800000 SAM alignment records processed.\n",
      "24900000 SAM alignment records processed.\n",
      "25000000 SAM alignment records processed.\n",
      "25100000 SAM alignment records processed.\n",
      "25200000 SAM alignment records processed.\n",
      "25300000 SAM alignment records processed.\n",
      "25400000 SAM alignment records processed.\n",
      "25500000 SAM alignment records processed.\n",
      "25600000 SAM alignment records processed.\n",
      "25700000 SAM alignment records processed.\n",
      "25800000 SAM alignment records processed.\n",
      "25900000 SAM alignment records processed.\n",
      "26000000 SAM alignment records processed.\n",
      "26100000 SAM alignment records processed.\n",
      "26200000 SAM alignment records processed.\n",
      "26300000 SAM alignment records processed.\n",
      "26400000 SAM alignment records processed.\n",
      "26500000 SAM alignment records processed.\n",
      "26600000 SAM alignment records processed.\n",
      "26700000 SAM alignment records processed.\n",
      "26800000 SAM alignment records processed.\n",
      "26900000 SAM alignment records processed.\n",
      "27000000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27100000 SAM alignment records processed.\n",
      "27200000 SAM alignment records processed.\n",
      "27300000 SAM alignment records processed.\n",
      "27400000 SAM alignment records processed.\n",
      "27500000 SAM alignment records processed.\n",
      "27600000 SAM alignment records processed.\n",
      "27700000 SAM alignment records processed.\n",
      "27800000 SAM alignment records processed.\n",
      "27900000 SAM alignment records processed.\n",
      "28000000 SAM alignment records processed.\n",
      "28100000 SAM alignment records processed.\n",
      "28200000 SAM alignment records processed.\n",
      "28300000 SAM alignment records processed.\n",
      "28400000 SAM alignment records processed.\n",
      "28500000 SAM alignment records processed.\n",
      "28600000 SAM alignment records processed.\n",
      "28700000 SAM alignment records processed.\n",
      "28800000 SAM alignment records processed.\n",
      "28900000 SAM alignment records processed.\n",
      "29000000 SAM alignment records processed.\n",
      "29100000 SAM alignment records processed.\n",
      "29200000 SAM alignment records processed.\n",
      "29300000 SAM alignment records processed.\n",
      "29400000 SAM alignment records processed.\n",
      "29500000 SAM alignment records processed.\n",
      "29600000 SAM alignment records processed.\n",
      "29700000 SAM alignment records processed.\n",
      "29800000 SAM alignment records processed.\n",
      "29900000 SAM alignment records processed.\n",
      "30000000 SAM alignment records processed.\n",
      "30100000 SAM alignment records processed.\n",
      "30200000 SAM alignment records processed.\n",
      "30300000 SAM alignment records processed.\n",
      "30400000 SAM alignment records processed.\n",
      "30500000 SAM alignment records processed.\n",
      "30600000 SAM alignment records processed.\n",
      "30700000 SAM alignment records processed.\n",
      "30800000 SAM alignment records processed.\n",
      "30900000 SAM alignment records processed.\n",
      "31000000 SAM alignment records processed.\n",
      "31100000 SAM alignment records processed.\n",
      "31200000 SAM alignment records processed.\n",
      "31300000 SAM alignment records processed.\n",
      "31400000 SAM alignment records processed.\n",
      "31500000 SAM alignment records processed.\n",
      "31600000 SAM alignment records processed.\n",
      "31700000 SAM alignment records processed.\n",
      "31800000 SAM alignment records processed.\n",
      "31900000 SAM alignment records processed.\n",
      "32000000 SAM alignment records processed.\n",
      "32100000 SAM alignment records processed.\n",
      "32200000 SAM alignment records processed.\n",
      "32300000 SAM alignment records processed.\n",
      "32400000 SAM alignment records processed.\n",
      "32500000 SAM alignment records processed.\n",
      "32600000 SAM alignment records processed.\n",
      "32700000 SAM alignment records processed.\n",
      "32800000 SAM alignment records processed.\n",
      "32900000 SAM alignment records processed.\n",
      "33000000 SAM alignment records processed.\n",
      "33100000 SAM alignment records processed.\n",
      "33200000 SAM alignment records processed.\n",
      "33300000 SAM alignment records processed.\n",
      "33400000 SAM alignment records processed.\n",
      "33500000 SAM alignment records processed.\n",
      "33600000 SAM alignment records processed.\n",
      "33700000 SAM alignment records processed.\n",
      "33800000 SAM alignment records processed.\n",
      "33900000 SAM alignment records processed.\n",
      "34000000 SAM alignment records processed.\n",
      "34100000 SAM alignment records processed.\n",
      "34200000 SAM alignment records processed.\n",
      "34300000 SAM alignment records processed.\n",
      "34400000 SAM alignment records processed.\n",
      "34500000 SAM alignment records processed.\n",
      "34600000 SAM alignment records processed.\n",
      "34700000 SAM alignment records processed.\n",
      "34800000 SAM alignment records processed.\n",
      "34900000 SAM alignment records processed.\n",
      "35000000 SAM alignment records processed.\n",
      "35100000 SAM alignment records processed.\n",
      "35200000 SAM alignment records processed.\n",
      "35300000 SAM alignment records processed.\n",
      "35400000 SAM alignment records processed.\n",
      "35500000 SAM alignment records processed.\n",
      "35600000 SAM alignment records processed.\n",
      "35700000 SAM alignment records processed.\n",
      "35800000 SAM alignment records processed.\n",
      "35900000 SAM alignment records processed.\n",
      "36000000 SAM alignment records processed.\n",
      "36100000 SAM alignment records processed.\n",
      "36200000 SAM alignment records processed.\n",
      "36300000 SAM alignment records processed.\n",
      "36400000 SAM alignment records processed.\n",
      "36500000 SAM alignment records processed.\n",
      "36600000 SAM alignment records processed.\n",
      "36700000 SAM alignment records processed.\n",
      "36800000 SAM alignment records processed.\n",
      "36900000 SAM alignment records processed.\n",
      "37000000 SAM alignment records processed.\n",
      "37100000 SAM alignment records processed.\n",
      "37200000 SAM alignment records processed.\n",
      "37300000 SAM alignment records processed.\n",
      "37400000 SAM alignment records processed.\n",
      "37500000 SAM alignment records processed.\n",
      "37600000 SAM alignment records processed.\n",
      "37700000 SAM alignment records processed.\n",
      "37800000 SAM alignment records processed.\n",
      "37900000 SAM alignment records processed.\n",
      "38000000 SAM alignment records processed.\n",
      "38100000 SAM alignment records processed.\n",
      "38200000 SAM alignment records processed.\n",
      "38300000 SAM alignment records processed.\n",
      "38400000 SAM alignment records processed.\n",
      "38500000 SAM alignment records processed.\n",
      "38600000 SAM alignment records processed.\n",
      "38700000 SAM alignment records processed.\n",
      "38800000 SAM alignment records processed.\n",
      "38900000 SAM alignment records processed.\n",
      "39000000 SAM alignment records processed.\n",
      "39100000 SAM alignment records processed.\n",
      "39200000 SAM alignment records processed.\n",
      "39300000 SAM alignment records processed.\n",
      "39400000 SAM alignment records processed.\n",
      "39500000 SAM alignment records processed.\n",
      "39600000 SAM alignment records processed.\n",
      "39700000 SAM alignment records processed.\n",
      "39800000 SAM alignment records processed.\n",
      "39900000 SAM alignment records processed.\n",
      "40000000 SAM alignment records processed.\n",
      "40100000 SAM alignment records processed.\n",
      "40200000 SAM alignment records processed.\n",
      "40300000 SAM alignment records processed.\n",
      "40400000 SAM alignment records processed.\n",
      "40500000 SAM alignment records processed.\n",
      "40600000 SAM alignment records processed.\n",
      "40700000 SAM alignment records processed.\n",
      "40800000 SAM alignment records processed.\n",
      "40900000 SAM alignment records processed.\n",
      "41000000 SAM alignment records processed.\n",
      "41100000 SAM alignment records processed.\n",
      "41200000 SAM alignment records processed.\n",
      "41300000 SAM alignment records processed.\n",
      "41400000 SAM alignment records processed.\n",
      "41500000 SAM alignment records processed.\n",
      "41600000 SAM alignment records processed.\n",
      "41700000 SAM alignment records processed.\n",
      "41800000 SAM alignment records processed.\n",
      "41900000 SAM alignment records processed.\n",
      "42000000 SAM alignment records processed.\n",
      "42100000 SAM alignment records processed.\n",
      "42200000 SAM alignment records processed.\n",
      "42300000 SAM alignment records processed.\n",
      "42400000 SAM alignment records processed.\n",
      "42500000 SAM alignment records processed.\n",
      "42600000 SAM alignment records processed.\n",
      "42700000 SAM alignment records processed.\n",
      "42800000 SAM alignment records processed.\n",
      "42900000 SAM alignment records processed.\n",
      "43000000 SAM alignment records processed.\n",
      "43100000 SAM alignment records processed.\n",
      "43200000 SAM alignment records processed.\n",
      "43300000 SAM alignment records processed.\n",
      "43400000 SAM alignment records processed.\n",
      "43500000 SAM alignment records processed.\n",
      "43600000 SAM alignment records processed.\n",
      "43700000 SAM alignment records processed.\n",
      "43800000 SAM alignment records processed.\n",
      "43900000 SAM alignment records processed.\n",
      "44000000 SAM alignment records processed.\n",
      "44100000 SAM alignment records processed.\n",
      "44200000 SAM alignment records processed.\n",
      "44300000 SAM alignment records processed.\n",
      "44400000 SAM alignment records processed.\n",
      "44500000 SAM alignment records processed.\n",
      "44600000 SAM alignment records processed.\n",
      "44700000 SAM alignment records processed.\n",
      "44800000 SAM alignment records processed.\n",
      "44900000 SAM alignment records processed.\n",
      "45000000 SAM alignment records processed.\n",
      "45100000 SAM alignment records processed.\n",
      "45200000 SAM alignment records processed.\n",
      "45300000 SAM alignment records processed.\n",
      "45400000 SAM alignment records processed.\n",
      "45500000 SAM alignment records processed.\n",
      "45600000 SAM alignment records processed.\n",
      "45700000 SAM alignment records processed.\n",
      "45800000 SAM alignment records processed.\n",
      "45900000 SAM alignment records processed.\n",
      "46000000 SAM alignment records processed.\n",
      "46100000 SAM alignment records processed.\n",
      "46200000 SAM alignment records processed.\n",
      "46300000 SAM alignment records processed.\n",
      "46400000 SAM alignment records processed.\n",
      "46500000 SAM alignment records processed.\n",
      "46600000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "46700000 SAM alignment records processed.\n",
      "46800000 SAM alignment records processed.\n",
      "46900000 SAM alignment records processed.\n",
      "47000000 SAM alignment records processed.\n",
      "47100000 SAM alignment records processed.\n",
      "47200000 SAM alignment records processed.\n",
      "47300000 SAM alignment records processed.\n",
      "47400000 SAM alignment records processed.\n",
      "47500000 SAM alignment records processed.\n",
      "47600000 SAM alignment records processed.\n",
      "47700000 SAM alignment records processed.\n",
      "47800000 SAM alignment records processed.\n",
      "47900000 SAM alignment records processed.\n",
      "48000000 SAM alignment records processed.\n",
      "48100000 SAM alignment records processed.\n",
      "48200000 SAM alignment records processed.\n",
      "48300000 SAM alignment records processed.\n",
      "48400000 SAM alignment records processed.\n",
      "48500000 SAM alignment records processed.\n",
      "48600000 SAM alignment records processed.\n",
      "48700000 SAM alignment records processed.\n",
      "48800000 SAM alignment records processed.\n",
      "48900000 SAM alignment records processed.\n",
      "49000000 SAM alignment records processed.\n",
      "49100000 SAM alignment records processed.\n",
      "49200000 SAM alignment records processed.\n",
      "49300000 SAM alignment records processed.\n",
      "49400000 SAM alignment records processed.\n",
      "49500000 SAM alignment records processed.\n",
      "49600000 SAM alignment records processed.\n",
      "49700000 SAM alignment records processed.\n",
      "49800000 SAM alignment records processed.\n",
      "49900000 SAM alignment records processed.\n",
      "50000000 SAM alignment records processed.\n",
      "50100000 SAM alignment records processed.\n",
      "50200000 SAM alignment records processed.\n",
      "50300000 SAM alignment records processed.\n",
      "50400000 SAM alignment records processed.\n",
      "50500000 SAM alignment records processed.\n",
      "50600000 SAM alignment records processed.\n",
      "50700000 SAM alignment records processed.\n",
      "50800000 SAM alignment records processed.\n",
      "50900000 SAM alignment records processed.\n",
      "51000000 SAM alignment records processed.\n",
      "51100000 SAM alignment records processed.\n",
      "51200000 SAM alignment records processed.\n",
      "51300000 SAM alignment records processed.\n",
      "51400000 SAM alignment records processed.\n",
      "51500000 SAM alignment records processed.\n",
      "51600000 SAM alignment records processed.\n",
      "51700000 SAM alignment records processed.\n",
      "51800000 SAM alignment records processed.\n",
      "51900000 SAM alignment records processed.\n",
      "52000000 SAM alignment records processed.\n",
      "52100000 SAM alignment records processed.\n",
      "52200000 SAM alignment records processed.\n",
      "52300000 SAM alignment records processed.\n",
      "52400000 SAM alignment records processed.\n",
      "52500000 SAM alignment records processed.\n",
      "52600000 SAM alignment records processed.\n",
      "52700000 SAM alignment records processed.\n",
      "52800000 SAM alignment records processed.\n",
      "52900000 SAM alignment records processed.\n",
      "53000000 SAM alignment records processed.\n",
      "53100000 SAM alignment records processed.\n",
      "53200000 SAM alignment records processed.\n",
      "53300000 SAM alignment records processed.\n",
      "53400000 SAM alignment records processed.\n",
      "53500000 SAM alignment records processed.\n",
      "53600000 SAM alignment records processed.\n",
      "53700000 SAM alignment records processed.\n",
      "53800000 SAM alignment records processed.\n",
      "53900000 SAM alignment records processed.\n",
      "54000000 SAM alignment records processed.\n",
      "54100000 SAM alignment records processed.\n",
      "54200000 SAM alignment records processed.\n",
      "54300000 SAM alignment records processed.\n",
      "54400000 SAM alignment records processed.\n",
      "54500000 SAM alignment records processed.\n",
      "54600000 SAM alignment records processed.\n",
      "54700000 SAM alignment records processed.\n",
      "54800000 SAM alignment records processed.\n",
      "54900000 SAM alignment records processed.\n",
      "55000000 SAM alignment records processed.\n",
      "55100000 SAM alignment records processed.\n",
      "55200000 SAM alignment records processed.\n",
      "55300000 SAM alignment records processed.\n",
      "55400000 SAM alignment records processed.\n",
      "55500000 SAM alignment records processed.\n",
      "55589955 BAM alignments  processed.\n",
      "HTSeq is counting SRR6145425.sorted.bam.\n",
      "100000 GFF lines processed.\n",
      "200000 GFF lines processed.\n",
      "300000 GFF lines processed.\n",
      "400000 GFF lines processed.\n",
      "500000 GFF lines processed.\n",
      "600000 GFF lines processed.\n",
      "700000 GFF lines processed.\n",
      "800000 GFF lines processed.\n",
      "900000 GFF lines processed.\n",
      "1000000 GFF lines processed.\n",
      "1100000 GFF lines processed.\n",
      "1200000 GFF lines processed.\n",
      "1300000 GFF lines processed.\n",
      "1400000 GFF lines processed.\n",
      "1500000 GFF lines processed.\n",
      "1600000 GFF lines processed.\n",
      "1700000 GFF lines processed.\n",
      "1800000 GFF lines processed.\n",
      "1900000 GFF lines processed.\n",
      "2000000 GFF lines processed.\n",
      "2100000 GFF lines processed.\n",
      "2200000 GFF lines processed.\n",
      "2300000 GFF lines processed.\n",
      "2400000 GFF lines processed.\n",
      "2500000 GFF lines processed.\n",
      "2600000 GFF lines processed.\n",
      "2700000 GFF lines processed.\n",
      "2800000 GFF lines processed.\n",
      "2900000 GFF lines processed.\n",
      "2908224 GFF lines processed.\n",
      "100000 SAM alignment records processed.\n",
      "200000 SAM alignment records processed.\n",
      "300000 SAM alignment records processed.\n",
      "400000 SAM alignment records processed.\n",
      "500000 SAM alignment records processed.\n",
      "600000 SAM alignment records processed.\n",
      "700000 SAM alignment records processed.\n",
      "800000 SAM alignment records processed.\n",
      "900000 SAM alignment records processed.\n",
      "1000000 SAM alignment records processed.\n",
      "1100000 SAM alignment records processed.\n",
      "1200000 SAM alignment records processed.\n",
      "1300000 SAM alignment records processed.\n",
      "1400000 SAM alignment records processed.\n",
      "1500000 SAM alignment records processed.\n",
      "1600000 SAM alignment records processed.\n",
      "1700000 SAM alignment records processed.\n",
      "1800000 SAM alignment records processed.\n",
      "1900000 SAM alignment records processed.\n",
      "2000000 SAM alignment records processed.\n",
      "2100000 SAM alignment records processed.\n",
      "2200000 SAM alignment records processed.\n",
      "2300000 SAM alignment records processed.\n",
      "2400000 SAM alignment records processed.\n",
      "2500000 SAM alignment records processed.\n",
      "2600000 SAM alignment records processed.\n",
      "2700000 SAM alignment records processed.\n",
      "2800000 SAM alignment records processed.\n",
      "2900000 SAM alignment records processed.\n",
      "3000000 SAM alignment records processed.\n",
      "3100000 SAM alignment records processed.\n",
      "3200000 SAM alignment records processed.\n",
      "3300000 SAM alignment records processed.\n",
      "3400000 SAM alignment records processed.\n",
      "3500000 SAM alignment records processed.\n",
      "3600000 SAM alignment records processed.\n",
      "3700000 SAM alignment records processed.\n",
      "3800000 SAM alignment records processed.\n",
      "3900000 SAM alignment records processed.\n",
      "4000000 SAM alignment records processed.\n",
      "4100000 SAM alignment records processed.\n",
      "4200000 SAM alignment records processed.\n",
      "4300000 SAM alignment records processed.\n",
      "4400000 SAM alignment records processed.\n",
      "4500000 SAM alignment records processed.\n",
      "4600000 SAM alignment records processed.\n",
      "4700000 SAM alignment records processed.\n",
      "4800000 SAM alignment records processed.\n",
      "4900000 SAM alignment records processed.\n",
      "5000000 SAM alignment records processed.\n",
      "5100000 SAM alignment records processed.\n",
      "5200000 SAM alignment records processed.\n",
      "5300000 SAM alignment records processed.\n",
      "5400000 SAM alignment records processed.\n",
      "5500000 SAM alignment records processed.\n",
      "5600000 SAM alignment records processed.\n",
      "5700000 SAM alignment records processed.\n",
      "5800000 SAM alignment records processed.\n",
      "5900000 SAM alignment records processed.\n",
      "6000000 SAM alignment records processed.\n",
      "6100000 SAM alignment records processed.\n",
      "6200000 SAM alignment records processed.\n",
      "6300000 SAM alignment records processed.\n",
      "6400000 SAM alignment records processed.\n",
      "6500000 SAM alignment records processed.\n",
      "6600000 SAM alignment records processed.\n",
      "6700000 SAM alignment records processed.\n",
      "6800000 SAM alignment records processed.\n",
      "6900000 SAM alignment records processed.\n",
      "7000000 SAM alignment records processed.\n",
      "7100000 SAM alignment records processed.\n",
      "7200000 SAM alignment records processed.\n",
      "7300000 SAM alignment records processed.\n",
      "7400000 SAM alignment records processed.\n",
      "7500000 SAM alignment records processed.\n",
      "7600000 SAM alignment records processed.\n",
      "7700000 SAM alignment records processed.\n",
      "7800000 SAM alignment records processed.\n",
      "7900000 SAM alignment records processed.\n",
      "8000000 SAM alignment records processed.\n",
      "8100000 SAM alignment records processed.\n",
      "8200000 SAM alignment records processed.\n",
      "8300000 SAM alignment records processed.\n",
      "8400000 SAM alignment records processed.\n",
      "8500000 SAM alignment records processed.\n",
      "8600000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8700000 SAM alignment records processed.\n",
      "8800000 SAM alignment records processed.\n",
      "8900000 SAM alignment records processed.\n",
      "9000000 SAM alignment records processed.\n",
      "9100000 SAM alignment records processed.\n",
      "9200000 SAM alignment records processed.\n",
      "9300000 SAM alignment records processed.\n",
      "9400000 SAM alignment records processed.\n",
      "9500000 SAM alignment records processed.\n",
      "9600000 SAM alignment records processed.\n",
      "9700000 SAM alignment records processed.\n",
      "9800000 SAM alignment records processed.\n",
      "9900000 SAM alignment records processed.\n",
      "10000000 SAM alignment records processed.\n",
      "10100000 SAM alignment records processed.\n",
      "10200000 SAM alignment records processed.\n",
      "10300000 SAM alignment records processed.\n",
      "10400000 SAM alignment records processed.\n",
      "10500000 SAM alignment records processed.\n",
      "10600000 SAM alignment records processed.\n",
      "10700000 SAM alignment records processed.\n",
      "10800000 SAM alignment records processed.\n",
      "10900000 SAM alignment records processed.\n",
      "11000000 SAM alignment records processed.\n",
      "11100000 SAM alignment records processed.\n",
      "11200000 SAM alignment records processed.\n",
      "11300000 SAM alignment records processed.\n",
      "11400000 SAM alignment records processed.\n",
      "11500000 SAM alignment records processed.\n",
      "11600000 SAM alignment records processed.\n",
      "11700000 SAM alignment records processed.\n",
      "11800000 SAM alignment records processed.\n",
      "11900000 SAM alignment records processed.\n",
      "12000000 SAM alignment records processed.\n",
      "12100000 SAM alignment records processed.\n",
      "12200000 SAM alignment records processed.\n",
      "12300000 SAM alignment records processed.\n",
      "12400000 SAM alignment records processed.\n",
      "12500000 SAM alignment records processed.\n",
      "12600000 SAM alignment records processed.\n",
      "12700000 SAM alignment records processed.\n",
      "12800000 SAM alignment records processed.\n",
      "12900000 SAM alignment records processed.\n",
      "13000000 SAM alignment records processed.\n",
      "13100000 SAM alignment records processed.\n",
      "13200000 SAM alignment records processed.\n",
      "13300000 SAM alignment records processed.\n",
      "13400000 SAM alignment records processed.\n",
      "13500000 SAM alignment records processed.\n",
      "13600000 SAM alignment records processed.\n",
      "13700000 SAM alignment records processed.\n",
      "13800000 SAM alignment records processed.\n",
      "13900000 SAM alignment records processed.\n",
      "14000000 SAM alignment records processed.\n",
      "14100000 SAM alignment records processed.\n",
      "14200000 SAM alignment records processed.\n",
      "14300000 SAM alignment records processed.\n",
      "14400000 SAM alignment records processed.\n",
      "14500000 SAM alignment records processed.\n",
      "14600000 SAM alignment records processed.\n",
      "14700000 SAM alignment records processed.\n",
      "14800000 SAM alignment records processed.\n",
      "14900000 SAM alignment records processed.\n",
      "15000000 SAM alignment records processed.\n",
      "15100000 SAM alignment records processed.\n",
      "15200000 SAM alignment records processed.\n",
      "15300000 SAM alignment records processed.\n",
      "15400000 SAM alignment records processed.\n",
      "15500000 SAM alignment records processed.\n",
      "15600000 SAM alignment records processed.\n",
      "15700000 SAM alignment records processed.\n",
      "15800000 SAM alignment records processed.\n",
      "15900000 SAM alignment records processed.\n",
      "16000000 SAM alignment records processed.\n",
      "16100000 SAM alignment records processed.\n",
      "16200000 SAM alignment records processed.\n",
      "16300000 SAM alignment records processed.\n",
      "16400000 SAM alignment records processed.\n",
      "16500000 SAM alignment records processed.\n",
      "16600000 SAM alignment records processed.\n",
      "16700000 SAM alignment records processed.\n",
      "16800000 SAM alignment records processed.\n",
      "16900000 SAM alignment records processed.\n",
      "17000000 SAM alignment records processed.\n",
      "17100000 SAM alignment records processed.\n",
      "17200000 SAM alignment records processed.\n",
      "17300000 SAM alignment records processed.\n",
      "17400000 SAM alignment records processed.\n",
      "17500000 SAM alignment records processed.\n",
      "17600000 SAM alignment records processed.\n",
      "17700000 SAM alignment records processed.\n",
      "17800000 SAM alignment records processed.\n",
      "17900000 SAM alignment records processed.\n",
      "18000000 SAM alignment records processed.\n",
      "18100000 SAM alignment records processed.\n",
      "18200000 SAM alignment records processed.\n",
      "18300000 SAM alignment records processed.\n",
      "18400000 SAM alignment records processed.\n",
      "18500000 SAM alignment records processed.\n",
      "18600000 SAM alignment records processed.\n",
      "18700000 SAM alignment records processed.\n",
      "18800000 SAM alignment records processed.\n",
      "18900000 SAM alignment records processed.\n",
      "19000000 SAM alignment records processed.\n",
      "19100000 SAM alignment records processed.\n",
      "19200000 SAM alignment records processed.\n",
      "19300000 SAM alignment records processed.\n",
      "19400000 SAM alignment records processed.\n",
      "19500000 SAM alignment records processed.\n",
      "19600000 SAM alignment records processed.\n",
      "19700000 SAM alignment records processed.\n",
      "19800000 SAM alignment records processed.\n",
      "19900000 SAM alignment records processed.\n",
      "20000000 SAM alignment records processed.\n",
      "20100000 SAM alignment records processed.\n",
      "20200000 SAM alignment records processed.\n",
      "20300000 SAM alignment records processed.\n",
      "20400000 SAM alignment records processed.\n",
      "20500000 SAM alignment records processed.\n",
      "20600000 SAM alignment records processed.\n",
      "20700000 SAM alignment records processed.\n",
      "20800000 SAM alignment records processed.\n",
      "20900000 SAM alignment records processed.\n",
      "21000000 SAM alignment records processed.\n",
      "21100000 SAM alignment records processed.\n",
      "21200000 SAM alignment records processed.\n",
      "21300000 SAM alignment records processed.\n",
      "21400000 SAM alignment records processed.\n",
      "21500000 SAM alignment records processed.\n",
      "21600000 SAM alignment records processed.\n",
      "21700000 SAM alignment records processed.\n",
      "21800000 SAM alignment records processed.\n",
      "21900000 SAM alignment records processed.\n",
      "22000000 SAM alignment records processed.\n",
      "22100000 SAM alignment records processed.\n",
      "22200000 SAM alignment records processed.\n",
      "22300000 SAM alignment records processed.\n",
      "22400000 SAM alignment records processed.\n",
      "22500000 SAM alignment records processed.\n",
      "22600000 SAM alignment records processed.\n",
      "22700000 SAM alignment records processed.\n",
      "22800000 SAM alignment records processed.\n",
      "22900000 SAM alignment records processed.\n",
      "23000000 SAM alignment records processed.\n",
      "23100000 SAM alignment records processed.\n",
      "23200000 SAM alignment records processed.\n",
      "23300000 SAM alignment records processed.\n",
      "23400000 SAM alignment records processed.\n",
      "23500000 SAM alignment records processed.\n",
      "23600000 SAM alignment records processed.\n",
      "23700000 SAM alignment records processed.\n",
      "23800000 SAM alignment records processed.\n",
      "23900000 SAM alignment records processed.\n",
      "24000000 SAM alignment records processed.\n",
      "24100000 SAM alignment records processed.\n",
      "24200000 SAM alignment records processed.\n",
      "24300000 SAM alignment records processed.\n",
      "24400000 SAM alignment records processed.\n",
      "24500000 SAM alignment records processed.\n",
      "24600000 SAM alignment records processed.\n",
      "24700000 SAM alignment records processed.\n",
      "24800000 SAM alignment records processed.\n",
      "24900000 SAM alignment records processed.\n",
      "25000000 SAM alignment records processed.\n",
      "25100000 SAM alignment records processed.\n",
      "25200000 SAM alignment records processed.\n",
      "25300000 SAM alignment records processed.\n",
      "25400000 SAM alignment records processed.\n",
      "25500000 SAM alignment records processed.\n",
      "25600000 SAM alignment records processed.\n",
      "25700000 SAM alignment records processed.\n",
      "25800000 SAM alignment records processed.\n",
      "25900000 SAM alignment records processed.\n",
      "26000000 SAM alignment records processed.\n",
      "26100000 SAM alignment records processed.\n",
      "26200000 SAM alignment records processed.\n",
      "26300000 SAM alignment records processed.\n",
      "26400000 SAM alignment records processed.\n",
      "26500000 SAM alignment records processed.\n",
      "26600000 SAM alignment records processed.\n",
      "26700000 SAM alignment records processed.\n",
      "26800000 SAM alignment records processed.\n",
      "26900000 SAM alignment records processed.\n",
      "27000000 SAM alignment records processed.\n",
      "27100000 SAM alignment records processed.\n",
      "27200000 SAM alignment records processed.\n",
      "27300000 SAM alignment records processed.\n",
      "27400000 SAM alignment records processed.\n",
      "27500000 SAM alignment records processed.\n",
      "27600000 SAM alignment records processed.\n",
      "27700000 SAM alignment records processed.\n",
      "27800000 SAM alignment records processed.\n",
      "27900000 SAM alignment records processed.\n",
      "28000000 SAM alignment records processed.\n",
      "28100000 SAM alignment records processed.\n",
      "28200000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28300000 SAM alignment records processed.\n",
      "28400000 SAM alignment records processed.\n",
      "28500000 SAM alignment records processed.\n",
      "28600000 SAM alignment records processed.\n",
      "28700000 SAM alignment records processed.\n",
      "28800000 SAM alignment records processed.\n",
      "28900000 SAM alignment records processed.\n",
      "29000000 SAM alignment records processed.\n",
      "29100000 SAM alignment records processed.\n",
      "29200000 SAM alignment records processed.\n",
      "29300000 SAM alignment records processed.\n",
      "29400000 SAM alignment records processed.\n",
      "29500000 SAM alignment records processed.\n",
      "29600000 SAM alignment records processed.\n",
      "29700000 SAM alignment records processed.\n",
      "29800000 SAM alignment records processed.\n",
      "29900000 SAM alignment records processed.\n",
      "30000000 SAM alignment records processed.\n",
      "30100000 SAM alignment records processed.\n",
      "30200000 SAM alignment records processed.\n",
      "30300000 SAM alignment records processed.\n",
      "30400000 SAM alignment records processed.\n",
      "30500000 SAM alignment records processed.\n",
      "30600000 SAM alignment records processed.\n",
      "30700000 SAM alignment records processed.\n",
      "30800000 SAM alignment records processed.\n",
      "30900000 SAM alignment records processed.\n",
      "31000000 SAM alignment records processed.\n",
      "31100000 SAM alignment records processed.\n",
      "31200000 SAM alignment records processed.\n",
      "31300000 SAM alignment records processed.\n",
      "31400000 SAM alignment records processed.\n",
      "31500000 SAM alignment records processed.\n",
      "31600000 SAM alignment records processed.\n",
      "31700000 SAM alignment records processed.\n",
      "31800000 SAM alignment records processed.\n",
      "31900000 SAM alignment records processed.\n",
      "32000000 SAM alignment records processed.\n",
      "32100000 SAM alignment records processed.\n",
      "32200000 SAM alignment records processed.\n",
      "32300000 SAM alignment records processed.\n",
      "32400000 SAM alignment records processed.\n",
      "32500000 SAM alignment records processed.\n",
      "32600000 SAM alignment records processed.\n",
      "32700000 SAM alignment records processed.\n",
      "32800000 SAM alignment records processed.\n",
      "32900000 SAM alignment records processed.\n",
      "33000000 SAM alignment records processed.\n",
      "33100000 SAM alignment records processed.\n",
      "33200000 SAM alignment records processed.\n",
      "33300000 SAM alignment records processed.\n",
      "33400000 SAM alignment records processed.\n",
      "33500000 SAM alignment records processed.\n",
      "33600000 SAM alignment records processed.\n",
      "33700000 SAM alignment records processed.\n",
      "33800000 SAM alignment records processed.\n",
      "33900000 SAM alignment records processed.\n",
      "34000000 SAM alignment records processed.\n",
      "34100000 SAM alignment records processed.\n",
      "34200000 SAM alignment records processed.\n",
      "34300000 SAM alignment records processed.\n",
      "34400000 SAM alignment records processed.\n",
      "34500000 SAM alignment records processed.\n",
      "34600000 SAM alignment records processed.\n",
      "34700000 SAM alignment records processed.\n",
      "34800000 SAM alignment records processed.\n",
      "34900000 SAM alignment records processed.\n",
      "35000000 SAM alignment records processed.\n",
      "35100000 SAM alignment records processed.\n",
      "35200000 SAM alignment records processed.\n",
      "35300000 SAM alignment records processed.\n",
      "35400000 SAM alignment records processed.\n",
      "35500000 SAM alignment records processed.\n",
      "35600000 SAM alignment records processed.\n",
      "35700000 SAM alignment records processed.\n",
      "35800000 SAM alignment records processed.\n",
      "35900000 SAM alignment records processed.\n",
      "36000000 SAM alignment records processed.\n",
      "36100000 SAM alignment records processed.\n",
      "36200000 SAM alignment records processed.\n",
      "36300000 SAM alignment records processed.\n",
      "36400000 SAM alignment records processed.\n",
      "36500000 SAM alignment records processed.\n",
      "36600000 SAM alignment records processed.\n",
      "36700000 SAM alignment records processed.\n",
      "36800000 SAM alignment records processed.\n",
      "36900000 SAM alignment records processed.\n",
      "37000000 SAM alignment records processed.\n",
      "37100000 SAM alignment records processed.\n",
      "37200000 SAM alignment records processed.\n",
      "37300000 SAM alignment records processed.\n",
      "37400000 SAM alignment records processed.\n",
      "37500000 SAM alignment records processed.\n",
      "37600000 SAM alignment records processed.\n",
      "37700000 SAM alignment records processed.\n",
      "37800000 SAM alignment records processed.\n",
      "37900000 SAM alignment records processed.\n",
      "38000000 SAM alignment records processed.\n",
      "38100000 SAM alignment records processed.\n",
      "38200000 SAM alignment records processed.\n",
      "38300000 SAM alignment records processed.\n",
      "38400000 SAM alignment records processed.\n",
      "38500000 SAM alignment records processed.\n",
      "38600000 SAM alignment records processed.\n",
      "38700000 SAM alignment records processed.\n",
      "38800000 SAM alignment records processed.\n",
      "38900000 SAM alignment records processed.\n",
      "39000000 SAM alignment records processed.\n",
      "39100000 SAM alignment records processed.\n",
      "39200000 SAM alignment records processed.\n",
      "39300000 SAM alignment records processed.\n",
      "39400000 SAM alignment records processed.\n",
      "39500000 SAM alignment records processed.\n",
      "39600000 SAM alignment records processed.\n",
      "39700000 SAM alignment records processed.\n",
      "39800000 SAM alignment records processed.\n",
      "39900000 SAM alignment records processed.\n",
      "40000000 SAM alignment records processed.\n",
      "40100000 SAM alignment records processed.\n",
      "40200000 SAM alignment records processed.\n",
      "40300000 SAM alignment records processed.\n",
      "40400000 SAM alignment records processed.\n",
      "40500000 SAM alignment records processed.\n",
      "40600000 SAM alignment records processed.\n",
      "40700000 SAM alignment records processed.\n",
      "40800000 SAM alignment records processed.\n",
      "40900000 SAM alignment records processed.\n",
      "41000000 SAM alignment records processed.\n",
      "41100000 SAM alignment records processed.\n",
      "41200000 SAM alignment records processed.\n",
      "41300000 SAM alignment records processed.\n",
      "41400000 SAM alignment records processed.\n",
      "41500000 SAM alignment records processed.\n",
      "41600000 SAM alignment records processed.\n",
      "41700000 SAM alignment records processed.\n",
      "41800000 SAM alignment records processed.\n",
      "41900000 SAM alignment records processed.\n",
      "42000000 SAM alignment records processed.\n",
      "42100000 SAM alignment records processed.\n",
      "42200000 SAM alignment records processed.\n",
      "42300000 SAM alignment records processed.\n",
      "42400000 SAM alignment records processed.\n",
      "42500000 SAM alignment records processed.\n",
      "42600000 SAM alignment records processed.\n",
      "42700000 SAM alignment records processed.\n",
      "42800000 SAM alignment records processed.\n",
      "42900000 SAM alignment records processed.\n",
      "43000000 SAM alignment records processed.\n",
      "43100000 SAM alignment records processed.\n",
      "43200000 SAM alignment records processed.\n",
      "43300000 SAM alignment records processed.\n",
      "43400000 SAM alignment records processed.\n",
      "43500000 SAM alignment records processed.\n",
      "43600000 SAM alignment records processed.\n",
      "43700000 SAM alignment records processed.\n",
      "43800000 SAM alignment records processed.\n",
      "43900000 SAM alignment records processed.\n",
      "44000000 SAM alignment records processed.\n",
      "44100000 SAM alignment records processed.\n",
      "44200000 SAM alignment records processed.\n",
      "44300000 SAM alignment records processed.\n",
      "44400000 SAM alignment records processed.\n",
      "44500000 SAM alignment records processed.\n",
      "44600000 SAM alignment records processed.\n",
      "44700000 SAM alignment records processed.\n",
      "44800000 SAM alignment records processed.\n",
      "44900000 SAM alignment records processed.\n",
      "44914525 BAM alignments  processed.\n",
      "HTSeq is counting SRR6145426.sorted.bam.\n",
      "100000 GFF lines processed.\n",
      "200000 GFF lines processed.\n",
      "300000 GFF lines processed.\n",
      "400000 GFF lines processed.\n",
      "500000 GFF lines processed.\n",
      "600000 GFF lines processed.\n",
      "700000 GFF lines processed.\n",
      "800000 GFF lines processed.\n",
      "900000 GFF lines processed.\n",
      "1000000 GFF lines processed.\n",
      "1100000 GFF lines processed.\n",
      "1200000 GFF lines processed.\n",
      "1300000 GFF lines processed.\n",
      "1400000 GFF lines processed.\n",
      "1500000 GFF lines processed.\n",
      "1600000 GFF lines processed.\n",
      "1700000 GFF lines processed.\n",
      "1800000 GFF lines processed.\n",
      "1900000 GFF lines processed.\n",
      "2000000 GFF lines processed.\n",
      "2100000 GFF lines processed.\n",
      "2200000 GFF lines processed.\n",
      "2300000 GFF lines processed.\n",
      "2400000 GFF lines processed.\n",
      "2500000 GFF lines processed.\n",
      "2600000 GFF lines processed.\n",
      "2700000 GFF lines processed.\n",
      "2800000 GFF lines processed.\n",
      "2900000 GFF lines processed.\n",
      "2908224 GFF lines processed.\n",
      "100000 SAM alignment records processed.\n",
      "200000 SAM alignment records processed.\n",
      "300000 SAM alignment records processed.\n",
      "400000 SAM alignment records processed.\n",
      "500000 SAM alignment records processed.\n",
      "600000 SAM alignment records processed.\n",
      "700000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "800000 SAM alignment records processed.\n",
      "900000 SAM alignment records processed.\n",
      "1000000 SAM alignment records processed.\n",
      "1100000 SAM alignment records processed.\n",
      "1200000 SAM alignment records processed.\n",
      "1300000 SAM alignment records processed.\n",
      "1400000 SAM alignment records processed.\n",
      "1500000 SAM alignment records processed.\n",
      "1600000 SAM alignment records processed.\n",
      "1700000 SAM alignment records processed.\n",
      "1800000 SAM alignment records processed.\n",
      "1900000 SAM alignment records processed.\n",
      "2000000 SAM alignment records processed.\n",
      "2100000 SAM alignment records processed.\n",
      "2200000 SAM alignment records processed.\n",
      "2300000 SAM alignment records processed.\n",
      "2400000 SAM alignment records processed.\n",
      "2500000 SAM alignment records processed.\n",
      "2600000 SAM alignment records processed.\n",
      "2700000 SAM alignment records processed.\n",
      "2800000 SAM alignment records processed.\n",
      "2900000 SAM alignment records processed.\n",
      "3000000 SAM alignment records processed.\n",
      "3100000 SAM alignment records processed.\n",
      "3200000 SAM alignment records processed.\n",
      "3300000 SAM alignment records processed.\n",
      "3400000 SAM alignment records processed.\n",
      "3500000 SAM alignment records processed.\n",
      "3600000 SAM alignment records processed.\n",
      "3700000 SAM alignment records processed.\n",
      "3800000 SAM alignment records processed.\n",
      "3900000 SAM alignment records processed.\n",
      "4000000 SAM alignment records processed.\n",
      "4100000 SAM alignment records processed.\n",
      "4200000 SAM alignment records processed.\n",
      "4300000 SAM alignment records processed.\n",
      "4400000 SAM alignment records processed.\n",
      "4500000 SAM alignment records processed.\n",
      "4600000 SAM alignment records processed.\n",
      "4700000 SAM alignment records processed.\n",
      "4800000 SAM alignment records processed.\n",
      "4900000 SAM alignment records processed.\n",
      "5000000 SAM alignment records processed.\n",
      "5100000 SAM alignment records processed.\n",
      "5200000 SAM alignment records processed.\n",
      "5300000 SAM alignment records processed.\n",
      "5400000 SAM alignment records processed.\n",
      "5500000 SAM alignment records processed.\n",
      "5600000 SAM alignment records processed.\n",
      "5700000 SAM alignment records processed.\n",
      "5800000 SAM alignment records processed.\n",
      "5900000 SAM alignment records processed.\n",
      "6000000 SAM alignment records processed.\n",
      "6100000 SAM alignment records processed.\n",
      "6200000 SAM alignment records processed.\n",
      "6300000 SAM alignment records processed.\n",
      "6400000 SAM alignment records processed.\n",
      "6500000 SAM alignment records processed.\n",
      "6600000 SAM alignment records processed.\n",
      "6700000 SAM alignment records processed.\n",
      "6800000 SAM alignment records processed.\n",
      "6900000 SAM alignment records processed.\n",
      "7000000 SAM alignment records processed.\n",
      "7100000 SAM alignment records processed.\n",
      "7200000 SAM alignment records processed.\n",
      "7300000 SAM alignment records processed.\n",
      "7400000 SAM alignment records processed.\n",
      "7500000 SAM alignment records processed.\n",
      "7600000 SAM alignment records processed.\n",
      "7700000 SAM alignment records processed.\n",
      "7800000 SAM alignment records processed.\n",
      "7900000 SAM alignment records processed.\n",
      "8000000 SAM alignment records processed.\n",
      "8100000 SAM alignment records processed.\n",
      "8200000 SAM alignment records processed.\n",
      "8300000 SAM alignment records processed.\n",
      "8400000 SAM alignment records processed.\n",
      "8500000 SAM alignment records processed.\n",
      "8600000 SAM alignment records processed.\n",
      "8700000 SAM alignment records processed.\n",
      "8800000 SAM alignment records processed.\n",
      "8900000 SAM alignment records processed.\n",
      "9000000 SAM alignment records processed.\n",
      "9100000 SAM alignment records processed.\n",
      "9200000 SAM alignment records processed.\n",
      "9300000 SAM alignment records processed.\n",
      "9400000 SAM alignment records processed.\n",
      "9500000 SAM alignment records processed.\n",
      "9600000 SAM alignment records processed.\n",
      "9700000 SAM alignment records processed.\n",
      "9800000 SAM alignment records processed.\n",
      "9900000 SAM alignment records processed.\n",
      "10000000 SAM alignment records processed.\n",
      "10100000 SAM alignment records processed.\n",
      "10200000 SAM alignment records processed.\n",
      "10300000 SAM alignment records processed.\n",
      "10400000 SAM alignment records processed.\n",
      "10500000 SAM alignment records processed.\n",
      "10600000 SAM alignment records processed.\n",
      "10700000 SAM alignment records processed.\n",
      "10800000 SAM alignment records processed.\n",
      "10900000 SAM alignment records processed.\n",
      "11000000 SAM alignment records processed.\n",
      "11100000 SAM alignment records processed.\n",
      "11200000 SAM alignment records processed.\n",
      "11300000 SAM alignment records processed.\n",
      "11400000 SAM alignment records processed.\n",
      "11500000 SAM alignment records processed.\n",
      "11600000 SAM alignment records processed.\n",
      "11700000 SAM alignment records processed.\n",
      "11800000 SAM alignment records processed.\n",
      "11900000 SAM alignment records processed.\n",
      "12000000 SAM alignment records processed.\n",
      "12100000 SAM alignment records processed.\n",
      "12200000 SAM alignment records processed.\n",
      "12300000 SAM alignment records processed.\n",
      "12400000 SAM alignment records processed.\n",
      "12500000 SAM alignment records processed.\n",
      "12600000 SAM alignment records processed.\n",
      "12700000 SAM alignment records processed.\n",
      "12800000 SAM alignment records processed.\n",
      "12900000 SAM alignment records processed.\n",
      "13000000 SAM alignment records processed.\n",
      "13100000 SAM alignment records processed.\n",
      "13200000 SAM alignment records processed.\n",
      "13300000 SAM alignment records processed.\n",
      "13400000 SAM alignment records processed.\n",
      "13500000 SAM alignment records processed.\n",
      "13600000 SAM alignment records processed.\n",
      "13700000 SAM alignment records processed.\n",
      "13800000 SAM alignment records processed.\n",
      "13900000 SAM alignment records processed.\n",
      "14000000 SAM alignment records processed.\n",
      "14100000 SAM alignment records processed.\n",
      "14200000 SAM alignment records processed.\n",
      "14300000 SAM alignment records processed.\n",
      "14400000 SAM alignment records processed.\n",
      "14500000 SAM alignment records processed.\n",
      "14600000 SAM alignment records processed.\n",
      "14700000 SAM alignment records processed.\n",
      "14800000 SAM alignment records processed.\n",
      "14900000 SAM alignment records processed.\n",
      "15000000 SAM alignment records processed.\n",
      "15100000 SAM alignment records processed.\n",
      "15200000 SAM alignment records processed.\n",
      "15300000 SAM alignment records processed.\n",
      "15400000 SAM alignment records processed.\n",
      "15500000 SAM alignment records processed.\n",
      "15600000 SAM alignment records processed.\n",
      "15700000 SAM alignment records processed.\n",
      "15800000 SAM alignment records processed.\n",
      "15900000 SAM alignment records processed.\n",
      "16000000 SAM alignment records processed.\n",
      "16100000 SAM alignment records processed.\n",
      "16200000 SAM alignment records processed.\n",
      "16300000 SAM alignment records processed.\n",
      "16400000 SAM alignment records processed.\n",
      "16500000 SAM alignment records processed.\n",
      "16600000 SAM alignment records processed.\n",
      "16700000 SAM alignment records processed.\n",
      "16800000 SAM alignment records processed.\n",
      "16900000 SAM alignment records processed.\n",
      "17000000 SAM alignment records processed.\n",
      "17100000 SAM alignment records processed.\n",
      "17200000 SAM alignment records processed.\n",
      "17300000 SAM alignment records processed.\n",
      "17400000 SAM alignment records processed.\n",
      "17500000 SAM alignment records processed.\n",
      "17600000 SAM alignment records processed.\n",
      "17700000 SAM alignment records processed.\n",
      "17800000 SAM alignment records processed.\n",
      "17900000 SAM alignment records processed.\n",
      "18000000 SAM alignment records processed.\n",
      "18100000 SAM alignment records processed.\n",
      "18200000 SAM alignment records processed.\n",
      "18300000 SAM alignment records processed.\n",
      "18400000 SAM alignment records processed.\n",
      "18500000 SAM alignment records processed.\n",
      "18600000 SAM alignment records processed.\n",
      "18700000 SAM alignment records processed.\n",
      "18800000 SAM alignment records processed.\n",
      "18900000 SAM alignment records processed.\n",
      "19000000 SAM alignment records processed.\n",
      "19100000 SAM alignment records processed.\n",
      "19200000 SAM alignment records processed.\n",
      "19300000 SAM alignment records processed.\n",
      "19400000 SAM alignment records processed.\n",
      "19500000 SAM alignment records processed.\n",
      "19600000 SAM alignment records processed.\n",
      "19700000 SAM alignment records processed.\n",
      "19800000 SAM alignment records processed.\n",
      "19900000 SAM alignment records processed.\n",
      "20000000 SAM alignment records processed.\n",
      "20100000 SAM alignment records processed.\n",
      "20200000 SAM alignment records processed.\n",
      "20300000 SAM alignment records processed.\n",
      "20400000 SAM alignment records processed.\n",
      "20500000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20600000 SAM alignment records processed.\n",
      "20700000 SAM alignment records processed.\n",
      "20800000 SAM alignment records processed.\n",
      "20900000 SAM alignment records processed.\n",
      "21000000 SAM alignment records processed.\n",
      "21100000 SAM alignment records processed.\n",
      "21200000 SAM alignment records processed.\n",
      "21300000 SAM alignment records processed.\n",
      "21400000 SAM alignment records processed.\n",
      "21500000 SAM alignment records processed.\n",
      "21600000 SAM alignment records processed.\n",
      "21700000 SAM alignment records processed.\n",
      "21800000 SAM alignment records processed.\n",
      "21900000 SAM alignment records processed.\n",
      "22000000 SAM alignment records processed.\n",
      "22100000 SAM alignment records processed.\n",
      "22200000 SAM alignment records processed.\n",
      "22300000 SAM alignment records processed.\n",
      "22400000 SAM alignment records processed.\n",
      "22500000 SAM alignment records processed.\n",
      "22600000 SAM alignment records processed.\n",
      "22700000 SAM alignment records processed.\n",
      "22800000 SAM alignment records processed.\n",
      "22900000 SAM alignment records processed.\n",
      "23000000 SAM alignment records processed.\n",
      "23100000 SAM alignment records processed.\n",
      "23200000 SAM alignment records processed.\n",
      "23300000 SAM alignment records processed.\n",
      "23400000 SAM alignment records processed.\n",
      "23500000 SAM alignment records processed.\n",
      "23600000 SAM alignment records processed.\n",
      "23700000 SAM alignment records processed.\n",
      "23800000 SAM alignment records processed.\n",
      "23900000 SAM alignment records processed.\n",
      "24000000 SAM alignment records processed.\n",
      "24100000 SAM alignment records processed.\n",
      "24200000 SAM alignment records processed.\n",
      "24300000 SAM alignment records processed.\n",
      "24400000 SAM alignment records processed.\n",
      "24500000 SAM alignment records processed.\n",
      "24600000 SAM alignment records processed.\n",
      "24700000 SAM alignment records processed.\n",
      "24800000 SAM alignment records processed.\n",
      "24900000 SAM alignment records processed.\n",
      "25000000 SAM alignment records processed.\n",
      "25100000 SAM alignment records processed.\n",
      "25200000 SAM alignment records processed.\n",
      "25300000 SAM alignment records processed.\n",
      "25400000 SAM alignment records processed.\n",
      "25500000 SAM alignment records processed.\n",
      "25600000 SAM alignment records processed.\n",
      "25700000 SAM alignment records processed.\n",
      "25800000 SAM alignment records processed.\n",
      "25900000 SAM alignment records processed.\n",
      "26000000 SAM alignment records processed.\n",
      "26100000 SAM alignment records processed.\n",
      "26200000 SAM alignment records processed.\n",
      "26300000 SAM alignment records processed.\n",
      "26400000 SAM alignment records processed.\n",
      "26500000 SAM alignment records processed.\n",
      "26600000 SAM alignment records processed.\n",
      "26700000 SAM alignment records processed.\n",
      "26800000 SAM alignment records processed.\n",
      "26900000 SAM alignment records processed.\n",
      "27000000 SAM alignment records processed.\n",
      "27100000 SAM alignment records processed.\n",
      "27200000 SAM alignment records processed.\n",
      "27300000 SAM alignment records processed.\n",
      "27400000 SAM alignment records processed.\n",
      "27500000 SAM alignment records processed.\n",
      "27600000 SAM alignment records processed.\n",
      "27700000 SAM alignment records processed.\n",
      "27800000 SAM alignment records processed.\n",
      "27900000 SAM alignment records processed.\n",
      "28000000 SAM alignment records processed.\n",
      "28100000 SAM alignment records processed.\n",
      "28200000 SAM alignment records processed.\n",
      "28300000 SAM alignment records processed.\n",
      "28400000 SAM alignment records processed.\n",
      "28500000 SAM alignment records processed.\n",
      "28600000 SAM alignment records processed.\n",
      "28700000 SAM alignment records processed.\n",
      "28800000 SAM alignment records processed.\n",
      "28900000 SAM alignment records processed.\n",
      "29000000 SAM alignment records processed.\n",
      "29100000 SAM alignment records processed.\n",
      "29200000 SAM alignment records processed.\n",
      "29300000 SAM alignment records processed.\n",
      "29400000 SAM alignment records processed.\n",
      "29500000 SAM alignment records processed.\n",
      "29600000 SAM alignment records processed.\n",
      "29700000 SAM alignment records processed.\n",
      "29800000 SAM alignment records processed.\n",
      "29900000 SAM alignment records processed.\n",
      "30000000 SAM alignment records processed.\n",
      "30100000 SAM alignment records processed.\n",
      "30200000 SAM alignment records processed.\n",
      "30300000 SAM alignment records processed.\n",
      "30400000 SAM alignment records processed.\n",
      "30500000 SAM alignment records processed.\n",
      "30600000 SAM alignment records processed.\n",
      "30700000 SAM alignment records processed.\n",
      "30800000 SAM alignment records processed.\n",
      "30900000 SAM alignment records processed.\n",
      "31000000 SAM alignment records processed.\n",
      "31100000 SAM alignment records processed.\n",
      "31200000 SAM alignment records processed.\n",
      "31300000 SAM alignment records processed.\n",
      "31400000 SAM alignment records processed.\n",
      "31500000 SAM alignment records processed.\n",
      "31600000 SAM alignment records processed.\n",
      "31700000 SAM alignment records processed.\n",
      "31800000 SAM alignment records processed.\n",
      "31900000 SAM alignment records processed.\n",
      "32000000 SAM alignment records processed.\n",
      "32100000 SAM alignment records processed.\n",
      "32200000 SAM alignment records processed.\n",
      "32300000 SAM alignment records processed.\n",
      "32400000 SAM alignment records processed.\n",
      "32500000 SAM alignment records processed.\n",
      "32600000 SAM alignment records processed.\n",
      "32700000 SAM alignment records processed.\n",
      "32800000 SAM alignment records processed.\n",
      "32900000 SAM alignment records processed.\n",
      "33000000 SAM alignment records processed.\n",
      "33100000 SAM alignment records processed.\n",
      "33200000 SAM alignment records processed.\n",
      "33300000 SAM alignment records processed.\n",
      "33400000 SAM alignment records processed.\n",
      "33500000 SAM alignment records processed.\n",
      "33600000 SAM alignment records processed.\n",
      "33700000 SAM alignment records processed.\n",
      "33800000 SAM alignment records processed.\n",
      "33900000 SAM alignment records processed.\n",
      "34000000 SAM alignment records processed.\n",
      "34100000 SAM alignment records processed.\n",
      "34200000 SAM alignment records processed.\n",
      "34300000 SAM alignment records processed.\n",
      "34400000 SAM alignment records processed.\n",
      "34500000 SAM alignment records processed.\n",
      "34600000 SAM alignment records processed.\n",
      "34700000 SAM alignment records processed.\n",
      "34800000 SAM alignment records processed.\n",
      "34900000 SAM alignment records processed.\n",
      "35000000 SAM alignment records processed.\n",
      "35100000 SAM alignment records processed.\n",
      "35200000 SAM alignment records processed.\n",
      "35300000 SAM alignment records processed.\n",
      "35400000 SAM alignment records processed.\n",
      "35500000 SAM alignment records processed.\n",
      "35600000 SAM alignment records processed.\n",
      "35700000 SAM alignment records processed.\n",
      "35800000 SAM alignment records processed.\n",
      "35900000 SAM alignment records processed.\n",
      "36000000 SAM alignment records processed.\n",
      "36100000 SAM alignment records processed.\n",
      "36200000 SAM alignment records processed.\n",
      "36300000 SAM alignment records processed.\n",
      "36400000 SAM alignment records processed.\n",
      "36500000 SAM alignment records processed.\n",
      "36600000 SAM alignment records processed.\n",
      "36700000 SAM alignment records processed.\n",
      "36800000 SAM alignment records processed.\n",
      "36900000 SAM alignment records processed.\n",
      "37000000 SAM alignment records processed.\n",
      "37100000 SAM alignment records processed.\n",
      "37200000 SAM alignment records processed.\n",
      "37300000 SAM alignment records processed.\n",
      "37400000 SAM alignment records processed.\n",
      "37500000 SAM alignment records processed.\n",
      "37600000 SAM alignment records processed.\n",
      "37700000 SAM alignment records processed.\n",
      "37800000 SAM alignment records processed.\n",
      "37900000 SAM alignment records processed.\n",
      "38000000 SAM alignment records processed.\n",
      "38100000 SAM alignment records processed.\n",
      "38200000 SAM alignment records processed.\n",
      "38300000 SAM alignment records processed.\n",
      "38400000 SAM alignment records processed.\n",
      "38500000 SAM alignment records processed.\n",
      "38600000 SAM alignment records processed.\n",
      "38700000 SAM alignment records processed.\n",
      "38800000 SAM alignment records processed.\n",
      "38900000 SAM alignment records processed.\n",
      "39000000 SAM alignment records processed.\n",
      "39100000 SAM alignment records processed.\n",
      "39200000 SAM alignment records processed.\n",
      "39300000 SAM alignment records processed.\n",
      "39400000 SAM alignment records processed.\n",
      "39500000 SAM alignment records processed.\n",
      "39600000 SAM alignment records processed.\n",
      "39700000 SAM alignment records processed.\n",
      "39800000 SAM alignment records processed.\n",
      "39900000 SAM alignment records processed.\n",
      "40000000 SAM alignment records processed.\n",
      "40100000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "40200000 SAM alignment records processed.\n",
      "40300000 SAM alignment records processed.\n",
      "40400000 SAM alignment records processed.\n",
      "40500000 SAM alignment records processed.\n",
      "40600000 SAM alignment records processed.\n",
      "40700000 SAM alignment records processed.\n",
      "40800000 SAM alignment records processed.\n",
      "40900000 SAM alignment records processed.\n",
      "41000000 SAM alignment records processed.\n",
      "41100000 SAM alignment records processed.\n",
      "41200000 SAM alignment records processed.\n",
      "41300000 SAM alignment records processed.\n",
      "41400000 SAM alignment records processed.\n",
      "41500000 SAM alignment records processed.\n",
      "41600000 SAM alignment records processed.\n",
      "41700000 SAM alignment records processed.\n",
      "41800000 SAM alignment records processed.\n",
      "41900000 SAM alignment records processed.\n",
      "42000000 SAM alignment records processed.\n",
      "42100000 SAM alignment records processed.\n",
      "42200000 SAM alignment records processed.\n",
      "42300000 SAM alignment records processed.\n",
      "42400000 SAM alignment records processed.\n",
      "42500000 SAM alignment records processed.\n",
      "42600000 SAM alignment records processed.\n",
      "42700000 SAM alignment records processed.\n",
      "42800000 SAM alignment records processed.\n",
      "42900000 SAM alignment records processed.\n",
      "43000000 SAM alignment records processed.\n",
      "43100000 SAM alignment records processed.\n",
      "43200000 SAM alignment records processed.\n",
      "43300000 SAM alignment records processed.\n",
      "43400000 SAM alignment records processed.\n",
      "43500000 SAM alignment records processed.\n",
      "43600000 SAM alignment records processed.\n",
      "43700000 SAM alignment records processed.\n",
      "43800000 SAM alignment records processed.\n",
      "43900000 SAM alignment records processed.\n",
      "44000000 SAM alignment records processed.\n",
      "44100000 SAM alignment records processed.\n",
      "44200000 SAM alignment records processed.\n",
      "44300000 SAM alignment records processed.\n",
      "44400000 SAM alignment records processed.\n",
      "44500000 SAM alignment records processed.\n",
      "44600000 SAM alignment records processed.\n",
      "44700000 SAM alignment records processed.\n",
      "44800000 SAM alignment records processed.\n",
      "44900000 SAM alignment records processed.\n",
      "45000000 SAM alignment records processed.\n",
      "45100000 SAM alignment records processed.\n",
      "45200000 SAM alignment records processed.\n",
      "45300000 SAM alignment records processed.\n",
      "45400000 SAM alignment records processed.\n",
      "45500000 SAM alignment records processed.\n",
      "45600000 SAM alignment records processed.\n",
      "45700000 SAM alignment records processed.\n",
      "45800000 SAM alignment records processed.\n",
      "45900000 SAM alignment records processed.\n",
      "46000000 SAM alignment records processed.\n",
      "46100000 SAM alignment records processed.\n",
      "46200000 SAM alignment records processed.\n",
      "46300000 SAM alignment records processed.\n",
      "46400000 SAM alignment records processed.\n",
      "46500000 SAM alignment records processed.\n",
      "46600000 SAM alignment records processed.\n",
      "46700000 SAM alignment records processed.\n",
      "46800000 SAM alignment records processed.\n",
      "46900000 SAM alignment records processed.\n",
      "47000000 SAM alignment records processed.\n",
      "47100000 SAM alignment records processed.\n",
      "47200000 SAM alignment records processed.\n",
      "47300000 SAM alignment records processed.\n",
      "47400000 SAM alignment records processed.\n",
      "47500000 SAM alignment records processed.\n",
      "47600000 SAM alignment records processed.\n",
      "47700000 SAM alignment records processed.\n",
      "47800000 SAM alignment records processed.\n",
      "47900000 SAM alignment records processed.\n",
      "48000000 SAM alignment records processed.\n",
      "48100000 SAM alignment records processed.\n",
      "48200000 SAM alignment records processed.\n",
      "48300000 SAM alignment records processed.\n",
      "48400000 SAM alignment records processed.\n",
      "48500000 SAM alignment records processed.\n",
      "48600000 SAM alignment records processed.\n",
      "48700000 SAM alignment records processed.\n",
      "48800000 SAM alignment records processed.\n",
      "48900000 SAM alignment records processed.\n",
      "49000000 SAM alignment records processed.\n",
      "49100000 SAM alignment records processed.\n",
      "49200000 SAM alignment records processed.\n",
      "49300000 SAM alignment records processed.\n",
      "49400000 SAM alignment records processed.\n",
      "49500000 SAM alignment records processed.\n",
      "49600000 SAM alignment records processed.\n",
      "49700000 SAM alignment records processed.\n",
      "49800000 SAM alignment records processed.\n",
      "49900000 SAM alignment records processed.\n",
      "50000000 SAM alignment records processed.\n",
      "50100000 SAM alignment records processed.\n",
      "50200000 SAM alignment records processed.\n",
      "50300000 SAM alignment records processed.\n",
      "50400000 SAM alignment records processed.\n",
      "50500000 SAM alignment records processed.\n",
      "50600000 SAM alignment records processed.\n",
      "50700000 SAM alignment records processed.\n",
      "50800000 SAM alignment records processed.\n",
      "50900000 SAM alignment records processed.\n",
      "51000000 SAM alignment records processed.\n",
      "51100000 SAM alignment records processed.\n",
      "51200000 SAM alignment records processed.\n",
      "51300000 SAM alignment records processed.\n",
      "51400000 SAM alignment records processed.\n",
      "51500000 SAM alignment records processed.\n",
      "51600000 SAM alignment records processed.\n",
      "51700000 SAM alignment records processed.\n",
      "51800000 SAM alignment records processed.\n",
      "51900000 SAM alignment records processed.\n",
      "52000000 SAM alignment records processed.\n",
      "52100000 SAM alignment records processed.\n",
      "52200000 SAM alignment records processed.\n",
      "52300000 SAM alignment records processed.\n",
      "52400000 SAM alignment records processed.\n",
      "52500000 SAM alignment records processed.\n",
      "52600000 SAM alignment records processed.\n",
      "52700000 SAM alignment records processed.\n",
      "52800000 SAM alignment records processed.\n",
      "52900000 SAM alignment records processed.\n",
      "53000000 SAM alignment records processed.\n",
      "53100000 SAM alignment records processed.\n",
      "53200000 SAM alignment records processed.\n",
      "53300000 SAM alignment records processed.\n",
      "53400000 SAM alignment records processed.\n",
      "53500000 SAM alignment records processed.\n",
      "53600000 SAM alignment records processed.\n",
      "53700000 SAM alignment records processed.\n",
      "53800000 SAM alignment records processed.\n",
      "53900000 SAM alignment records processed.\n",
      "54000000 SAM alignment records processed.\n",
      "54100000 SAM alignment records processed.\n",
      "54200000 SAM alignment records processed.\n",
      "54300000 SAM alignment records processed.\n",
      "54400000 SAM alignment records processed.\n",
      "54500000 SAM alignment records processed.\n",
      "54600000 SAM alignment records processed.\n",
      "54700000 SAM alignment records processed.\n",
      "54800000 SAM alignment records processed.\n",
      "54900000 SAM alignment records processed.\n",
      "55000000 SAM alignment records processed.\n",
      "55100000 SAM alignment records processed.\n",
      "55200000 SAM alignment records processed.\n",
      "55300000 SAM alignment records processed.\n",
      "55400000 SAM alignment records processed.\n",
      "55500000 SAM alignment records processed.\n",
      "55600000 SAM alignment records processed.\n",
      "55700000 SAM alignment records processed.\n",
      "55800000 SAM alignment records processed.\n",
      "55900000 SAM alignment records processed.\n",
      "56000000 SAM alignment records processed.\n",
      "56032699 BAM alignments  processed.\n",
      "HTSeq is counting SRR6145427.sorted.bam.\n",
      "100000 GFF lines processed.\n",
      "200000 GFF lines processed.\n",
      "300000 GFF lines processed.\n",
      "400000 GFF lines processed.\n",
      "500000 GFF lines processed.\n",
      "600000 GFF lines processed.\n",
      "700000 GFF lines processed.\n",
      "800000 GFF lines processed.\n",
      "900000 GFF lines processed.\n",
      "1000000 GFF lines processed.\n",
      "1100000 GFF lines processed.\n",
      "1200000 GFF lines processed.\n",
      "1300000 GFF lines processed.\n",
      "1400000 GFF lines processed.\n",
      "1500000 GFF lines processed.\n",
      "1600000 GFF lines processed.\n",
      "1700000 GFF lines processed.\n",
      "1800000 GFF lines processed.\n",
      "1900000 GFF lines processed.\n",
      "2000000 GFF lines processed.\n",
      "2100000 GFF lines processed.\n",
      "2200000 GFF lines processed.\n",
      "2300000 GFF lines processed.\n",
      "2400000 GFF lines processed.\n",
      "2500000 GFF lines processed.\n",
      "2600000 GFF lines processed.\n",
      "2700000 GFF lines processed.\n",
      "2800000 GFF lines processed.\n",
      "2900000 GFF lines processed.\n",
      "2908224 GFF lines processed.\n",
      "100000 SAM alignment records processed.\n",
      "200000 SAM alignment records processed.\n",
      "300000 SAM alignment records processed.\n",
      "400000 SAM alignment records processed.\n",
      "500000 SAM alignment records processed.\n",
      "600000 SAM alignment records processed.\n",
      "700000 SAM alignment records processed.\n",
      "800000 SAM alignment records processed.\n",
      "900000 SAM alignment records processed.\n",
      "1000000 SAM alignment records processed.\n",
      "1100000 SAM alignment records processed.\n",
      "1200000 SAM alignment records processed.\n",
      "1300000 SAM alignment records processed.\n",
      "1400000 SAM alignment records processed.\n",
      "1500000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1600000 SAM alignment records processed.\n",
      "1700000 SAM alignment records processed.\n",
      "1800000 SAM alignment records processed.\n",
      "1900000 SAM alignment records processed.\n",
      "2000000 SAM alignment records processed.\n",
      "2100000 SAM alignment records processed.\n",
      "2200000 SAM alignment records processed.\n",
      "2300000 SAM alignment records processed.\n",
      "2400000 SAM alignment records processed.\n",
      "2500000 SAM alignment records processed.\n",
      "2600000 SAM alignment records processed.\n",
      "2700000 SAM alignment records processed.\n",
      "2800000 SAM alignment records processed.\n",
      "2900000 SAM alignment records processed.\n",
      "3000000 SAM alignment records processed.\n",
      "3100000 SAM alignment records processed.\n",
      "3200000 SAM alignment records processed.\n",
      "3300000 SAM alignment records processed.\n",
      "3400000 SAM alignment records processed.\n",
      "3500000 SAM alignment records processed.\n",
      "3600000 SAM alignment records processed.\n",
      "3700000 SAM alignment records processed.\n",
      "3800000 SAM alignment records processed.\n",
      "3900000 SAM alignment records processed.\n",
      "4000000 SAM alignment records processed.\n",
      "4100000 SAM alignment records processed.\n",
      "4200000 SAM alignment records processed.\n",
      "4300000 SAM alignment records processed.\n",
      "4400000 SAM alignment records processed.\n",
      "4500000 SAM alignment records processed.\n",
      "4600000 SAM alignment records processed.\n",
      "4700000 SAM alignment records processed.\n",
      "4800000 SAM alignment records processed.\n",
      "4900000 SAM alignment records processed.\n",
      "5000000 SAM alignment records processed.\n",
      "5100000 SAM alignment records processed.\n",
      "5200000 SAM alignment records processed.\n",
      "5300000 SAM alignment records processed.\n",
      "5400000 SAM alignment records processed.\n",
      "5500000 SAM alignment records processed.\n",
      "5600000 SAM alignment records processed.\n",
      "5700000 SAM alignment records processed.\n",
      "5800000 SAM alignment records processed.\n",
      "5900000 SAM alignment records processed.\n",
      "6000000 SAM alignment records processed.\n",
      "6100000 SAM alignment records processed.\n",
      "6200000 SAM alignment records processed.\n",
      "6300000 SAM alignment records processed.\n",
      "6400000 SAM alignment records processed.\n",
      "6500000 SAM alignment records processed.\n",
      "6600000 SAM alignment records processed.\n",
      "6700000 SAM alignment records processed.\n",
      "6800000 SAM alignment records processed.\n",
      "6900000 SAM alignment records processed.\n",
      "7000000 SAM alignment records processed.\n",
      "7100000 SAM alignment records processed.\n",
      "7200000 SAM alignment records processed.\n",
      "7300000 SAM alignment records processed.\n",
      "7400000 SAM alignment records processed.\n",
      "7500000 SAM alignment records processed.\n",
      "7600000 SAM alignment records processed.\n",
      "7700000 SAM alignment records processed.\n",
      "7800000 SAM alignment records processed.\n",
      "7900000 SAM alignment records processed.\n",
      "8000000 SAM alignment records processed.\n",
      "8100000 SAM alignment records processed.\n",
      "8200000 SAM alignment records processed.\n",
      "8300000 SAM alignment records processed.\n",
      "8400000 SAM alignment records processed.\n",
      "8500000 SAM alignment records processed.\n",
      "8600000 SAM alignment records processed.\n",
      "8700000 SAM alignment records processed.\n",
      "8800000 SAM alignment records processed.\n",
      "8900000 SAM alignment records processed.\n",
      "9000000 SAM alignment records processed.\n",
      "9100000 SAM alignment records processed.\n",
      "9200000 SAM alignment records processed.\n",
      "9300000 SAM alignment records processed.\n",
      "9400000 SAM alignment records processed.\n",
      "9500000 SAM alignment records processed.\n",
      "9600000 SAM alignment records processed.\n",
      "9700000 SAM alignment records processed.\n",
      "9800000 SAM alignment records processed.\n",
      "9900000 SAM alignment records processed.\n",
      "10000000 SAM alignment records processed.\n",
      "10100000 SAM alignment records processed.\n",
      "10200000 SAM alignment records processed.\n",
      "10300000 SAM alignment records processed.\n",
      "10400000 SAM alignment records processed.\n",
      "10500000 SAM alignment records processed.\n",
      "10600000 SAM alignment records processed.\n",
      "10700000 SAM alignment records processed.\n",
      "10800000 SAM alignment records processed.\n",
      "10900000 SAM alignment records processed.\n",
      "11000000 SAM alignment records processed.\n",
      "11100000 SAM alignment records processed.\n",
      "11200000 SAM alignment records processed.\n",
      "11300000 SAM alignment records processed.\n",
      "11400000 SAM alignment records processed.\n",
      "11500000 SAM alignment records processed.\n",
      "11600000 SAM alignment records processed.\n",
      "11700000 SAM alignment records processed.\n",
      "11800000 SAM alignment records processed.\n",
      "11900000 SAM alignment records processed.\n",
      "12000000 SAM alignment records processed.\n",
      "12100000 SAM alignment records processed.\n",
      "12200000 SAM alignment records processed.\n",
      "12300000 SAM alignment records processed.\n",
      "12400000 SAM alignment records processed.\n",
      "12500000 SAM alignment records processed.\n",
      "12600000 SAM alignment records processed.\n",
      "12700000 SAM alignment records processed.\n",
      "12800000 SAM alignment records processed.\n",
      "12900000 SAM alignment records processed.\n",
      "13000000 SAM alignment records processed.\n",
      "13100000 SAM alignment records processed.\n",
      "13200000 SAM alignment records processed.\n",
      "13300000 SAM alignment records processed.\n",
      "13400000 SAM alignment records processed.\n",
      "13500000 SAM alignment records processed.\n",
      "13600000 SAM alignment records processed.\n",
      "13700000 SAM alignment records processed.\n",
      "13800000 SAM alignment records processed.\n",
      "13900000 SAM alignment records processed.\n",
      "14000000 SAM alignment records processed.\n",
      "14100000 SAM alignment records processed.\n",
      "14200000 SAM alignment records processed.\n",
      "14300000 SAM alignment records processed.\n",
      "14400000 SAM alignment records processed.\n",
      "14500000 SAM alignment records processed.\n",
      "14600000 SAM alignment records processed.\n",
      "14700000 SAM alignment records processed.\n",
      "14800000 SAM alignment records processed.\n",
      "14900000 SAM alignment records processed.\n",
      "15000000 SAM alignment records processed.\n",
      "15100000 SAM alignment records processed.\n",
      "15200000 SAM alignment records processed.\n",
      "15300000 SAM alignment records processed.\n",
      "15400000 SAM alignment records processed.\n",
      "15500000 SAM alignment records processed.\n",
      "15600000 SAM alignment records processed.\n",
      "15700000 SAM alignment records processed.\n",
      "15800000 SAM alignment records processed.\n",
      "15900000 SAM alignment records processed.\n",
      "16000000 SAM alignment records processed.\n",
      "16100000 SAM alignment records processed.\n",
      "16200000 SAM alignment records processed.\n",
      "16300000 SAM alignment records processed.\n",
      "16400000 SAM alignment records processed.\n",
      "16500000 SAM alignment records processed.\n",
      "16600000 SAM alignment records processed.\n",
      "16700000 SAM alignment records processed.\n",
      "16800000 SAM alignment records processed.\n",
      "16900000 SAM alignment records processed.\n",
      "17000000 SAM alignment records processed.\n",
      "17100000 SAM alignment records processed.\n",
      "17200000 SAM alignment records processed.\n",
      "17300000 SAM alignment records processed.\n",
      "17400000 SAM alignment records processed.\n",
      "17500000 SAM alignment records processed.\n",
      "17600000 SAM alignment records processed.\n",
      "17700000 SAM alignment records processed.\n",
      "17800000 SAM alignment records processed.\n",
      "17900000 SAM alignment records processed.\n",
      "18000000 SAM alignment records processed.\n",
      "18100000 SAM alignment records processed.\n",
      "18200000 SAM alignment records processed.\n",
      "18300000 SAM alignment records processed.\n",
      "18400000 SAM alignment records processed.\n",
      "18500000 SAM alignment records processed.\n",
      "18600000 SAM alignment records processed.\n",
      "18700000 SAM alignment records processed.\n",
      "18800000 SAM alignment records processed.\n",
      "18900000 SAM alignment records processed.\n",
      "19000000 SAM alignment records processed.\n",
      "19100000 SAM alignment records processed.\n",
      "19200000 SAM alignment records processed.\n",
      "19300000 SAM alignment records processed.\n",
      "19400000 SAM alignment records processed.\n",
      "19500000 SAM alignment records processed.\n",
      "19600000 SAM alignment records processed.\n",
      "19700000 SAM alignment records processed.\n",
      "19800000 SAM alignment records processed.\n",
      "19900000 SAM alignment records processed.\n",
      "20000000 SAM alignment records processed.\n",
      "20100000 SAM alignment records processed.\n",
      "20200000 SAM alignment records processed.\n",
      "20300000 SAM alignment records processed.\n",
      "20400000 SAM alignment records processed.\n",
      "20500000 SAM alignment records processed.\n",
      "20600000 SAM alignment records processed.\n",
      "20700000 SAM alignment records processed.\n",
      "20800000 SAM alignment records processed.\n",
      "20900000 SAM alignment records processed.\n",
      "21000000 SAM alignment records processed.\n",
      "21100000 SAM alignment records processed.\n",
      "21200000 SAM alignment records processed.\n",
      "21300000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21400000 SAM alignment records processed.\n",
      "21500000 SAM alignment records processed.\n",
      "21600000 SAM alignment records processed.\n",
      "21700000 SAM alignment records processed.\n",
      "21800000 SAM alignment records processed.\n",
      "21900000 SAM alignment records processed.\n",
      "22000000 SAM alignment records processed.\n",
      "22100000 SAM alignment records processed.\n",
      "22200000 SAM alignment records processed.\n",
      "22300000 SAM alignment records processed.\n",
      "22400000 SAM alignment records processed.\n",
      "22500000 SAM alignment records processed.\n",
      "22600000 SAM alignment records processed.\n",
      "22700000 SAM alignment records processed.\n",
      "22800000 SAM alignment records processed.\n",
      "22900000 SAM alignment records processed.\n",
      "23000000 SAM alignment records processed.\n",
      "23100000 SAM alignment records processed.\n",
      "23200000 SAM alignment records processed.\n",
      "23300000 SAM alignment records processed.\n",
      "23400000 SAM alignment records processed.\n",
      "23500000 SAM alignment records processed.\n",
      "23600000 SAM alignment records processed.\n",
      "23700000 SAM alignment records processed.\n",
      "23800000 SAM alignment records processed.\n",
      "23900000 SAM alignment records processed.\n",
      "24000000 SAM alignment records processed.\n",
      "24100000 SAM alignment records processed.\n",
      "24200000 SAM alignment records processed.\n",
      "24300000 SAM alignment records processed.\n",
      "24400000 SAM alignment records processed.\n",
      "24500000 SAM alignment records processed.\n",
      "24600000 SAM alignment records processed.\n",
      "24700000 SAM alignment records processed.\n",
      "24800000 SAM alignment records processed.\n",
      "24900000 SAM alignment records processed.\n",
      "25000000 SAM alignment records processed.\n",
      "25100000 SAM alignment records processed.\n",
      "25200000 SAM alignment records processed.\n",
      "25300000 SAM alignment records processed.\n",
      "25400000 SAM alignment records processed.\n",
      "25500000 SAM alignment records processed.\n",
      "25600000 SAM alignment records processed.\n",
      "25700000 SAM alignment records processed.\n",
      "25800000 SAM alignment records processed.\n",
      "25900000 SAM alignment records processed.\n",
      "26000000 SAM alignment records processed.\n",
      "26100000 SAM alignment records processed.\n",
      "26200000 SAM alignment records processed.\n",
      "26300000 SAM alignment records processed.\n",
      "26400000 SAM alignment records processed.\n",
      "26500000 SAM alignment records processed.\n",
      "26600000 SAM alignment records processed.\n",
      "26700000 SAM alignment records processed.\n",
      "26800000 SAM alignment records processed.\n",
      "26900000 SAM alignment records processed.\n",
      "27000000 SAM alignment records processed.\n",
      "27100000 SAM alignment records processed.\n",
      "27200000 SAM alignment records processed.\n",
      "27300000 SAM alignment records processed.\n",
      "27400000 SAM alignment records processed.\n",
      "27500000 SAM alignment records processed.\n",
      "27600000 SAM alignment records processed.\n",
      "27700000 SAM alignment records processed.\n",
      "27800000 SAM alignment records processed.\n",
      "27900000 SAM alignment records processed.\n",
      "28000000 SAM alignment records processed.\n",
      "28100000 SAM alignment records processed.\n",
      "28200000 SAM alignment records processed.\n",
      "28300000 SAM alignment records processed.\n",
      "28400000 SAM alignment records processed.\n",
      "28500000 SAM alignment records processed.\n",
      "28600000 SAM alignment records processed.\n",
      "28700000 SAM alignment records processed.\n",
      "28800000 SAM alignment records processed.\n",
      "28900000 SAM alignment records processed.\n",
      "29000000 SAM alignment records processed.\n",
      "29100000 SAM alignment records processed.\n",
      "29200000 SAM alignment records processed.\n",
      "29300000 SAM alignment records processed.\n",
      "29400000 SAM alignment records processed.\n",
      "29500000 SAM alignment records processed.\n",
      "29600000 SAM alignment records processed.\n",
      "29700000 SAM alignment records processed.\n",
      "29800000 SAM alignment records processed.\n",
      "29900000 SAM alignment records processed.\n",
      "30000000 SAM alignment records processed.\n",
      "30100000 SAM alignment records processed.\n",
      "30200000 SAM alignment records processed.\n",
      "30300000 SAM alignment records processed.\n",
      "30400000 SAM alignment records processed.\n",
      "30500000 SAM alignment records processed.\n",
      "30600000 SAM alignment records processed.\n",
      "30700000 SAM alignment records processed.\n",
      "30800000 SAM alignment records processed.\n",
      "30900000 SAM alignment records processed.\n",
      "31000000 SAM alignment records processed.\n",
      "31100000 SAM alignment records processed.\n",
      "31200000 SAM alignment records processed.\n",
      "31300000 SAM alignment records processed.\n",
      "31400000 SAM alignment records processed.\n",
      "31500000 SAM alignment records processed.\n",
      "31600000 SAM alignment records processed.\n",
      "31700000 SAM alignment records processed.\n",
      "31800000 SAM alignment records processed.\n",
      "31900000 SAM alignment records processed.\n",
      "32000000 SAM alignment records processed.\n",
      "32100000 SAM alignment records processed.\n",
      "32200000 SAM alignment records processed.\n",
      "32300000 SAM alignment records processed.\n",
      "32400000 SAM alignment records processed.\n",
      "32500000 SAM alignment records processed.\n",
      "32600000 SAM alignment records processed.\n",
      "32700000 SAM alignment records processed.\n",
      "32800000 SAM alignment records processed.\n",
      "32900000 SAM alignment records processed.\n",
      "33000000 SAM alignment records processed.\n",
      "33100000 SAM alignment records processed.\n",
      "33200000 SAM alignment records processed.\n",
      "33300000 SAM alignment records processed.\n",
      "33400000 SAM alignment records processed.\n",
      "33500000 SAM alignment records processed.\n",
      "33600000 SAM alignment records processed.\n",
      "33700000 SAM alignment records processed.\n",
      "33800000 SAM alignment records processed.\n",
      "33900000 SAM alignment records processed.\n",
      "34000000 SAM alignment records processed.\n",
      "34100000 SAM alignment records processed.\n",
      "34200000 SAM alignment records processed.\n",
      "34300000 SAM alignment records processed.\n",
      "34400000 SAM alignment records processed.\n",
      "34500000 SAM alignment records processed.\n",
      "34600000 SAM alignment records processed.\n",
      "34700000 SAM alignment records processed.\n",
      "34800000 SAM alignment records processed.\n",
      "34900000 SAM alignment records processed.\n",
      "35000000 SAM alignment records processed.\n",
      "35100000 SAM alignment records processed.\n",
      "35200000 SAM alignment records processed.\n",
      "35300000 SAM alignment records processed.\n",
      "35400000 SAM alignment records processed.\n",
      "35500000 SAM alignment records processed.\n",
      "35600000 SAM alignment records processed.\n",
      "35700000 SAM alignment records processed.\n",
      "35800000 SAM alignment records processed.\n",
      "35900000 SAM alignment records processed.\n",
      "36000000 SAM alignment records processed.\n",
      "36100000 SAM alignment records processed.\n",
      "36200000 SAM alignment records processed.\n",
      "36300000 SAM alignment records processed.\n",
      "36400000 SAM alignment records processed.\n",
      "36500000 SAM alignment records processed.\n",
      "36600000 SAM alignment records processed.\n",
      "36700000 SAM alignment records processed.\n",
      "36800000 SAM alignment records processed.\n",
      "36900000 SAM alignment records processed.\n",
      "37000000 SAM alignment records processed.\n",
      "37100000 SAM alignment records processed.\n",
      "37200000 SAM alignment records processed.\n",
      "37300000 SAM alignment records processed.\n",
      "37400000 SAM alignment records processed.\n",
      "37500000 SAM alignment records processed.\n",
      "37600000 SAM alignment records processed.\n",
      "37700000 SAM alignment records processed.\n",
      "37800000 SAM alignment records processed.\n",
      "37900000 SAM alignment records processed.\n",
      "38000000 SAM alignment records processed.\n",
      "38100000 SAM alignment records processed.\n",
      "38200000 SAM alignment records processed.\n",
      "38300000 SAM alignment records processed.\n",
      "38400000 SAM alignment records processed.\n",
      "38500000 SAM alignment records processed.\n",
      "38600000 SAM alignment records processed.\n",
      "38700000 SAM alignment records processed.\n",
      "38800000 SAM alignment records processed.\n",
      "38900000 SAM alignment records processed.\n",
      "39000000 SAM alignment records processed.\n",
      "39100000 SAM alignment records processed.\n",
      "39200000 SAM alignment records processed.\n",
      "39300000 SAM alignment records processed.\n",
      "39400000 SAM alignment records processed.\n",
      "39500000 SAM alignment records processed.\n",
      "39600000 SAM alignment records processed.\n",
      "39700000 SAM alignment records processed.\n",
      "39800000 SAM alignment records processed.\n",
      "39900000 SAM alignment records processed.\n",
      "40000000 SAM alignment records processed.\n",
      "40100000 SAM alignment records processed.\n",
      "40200000 SAM alignment records processed.\n",
      "40300000 SAM alignment records processed.\n",
      "40400000 SAM alignment records processed.\n",
      "40500000 SAM alignment records processed.\n",
      "40600000 SAM alignment records processed.\n",
      "40700000 SAM alignment records processed.\n",
      "40800000 SAM alignment records processed.\n",
      "40900000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "41000000 SAM alignment records processed.\n",
      "41100000 SAM alignment records processed.\n",
      "41200000 SAM alignment records processed.\n",
      "41300000 SAM alignment records processed.\n",
      "41400000 SAM alignment records processed.\n",
      "41500000 SAM alignment records processed.\n",
      "41600000 SAM alignment records processed.\n",
      "41700000 SAM alignment records processed.\n",
      "41800000 SAM alignment records processed.\n",
      "41900000 SAM alignment records processed.\n",
      "42000000 SAM alignment records processed.\n",
      "42100000 SAM alignment records processed.\n",
      "42200000 SAM alignment records processed.\n",
      "42300000 SAM alignment records processed.\n",
      "42400000 SAM alignment records processed.\n",
      "42500000 SAM alignment records processed.\n",
      "42600000 SAM alignment records processed.\n",
      "42700000 SAM alignment records processed.\n",
      "42800000 SAM alignment records processed.\n",
      "42900000 SAM alignment records processed.\n",
      "43000000 SAM alignment records processed.\n",
      "43100000 SAM alignment records processed.\n",
      "43200000 SAM alignment records processed.\n",
      "43300000 SAM alignment records processed.\n",
      "43400000 SAM alignment records processed.\n",
      "43500000 SAM alignment records processed.\n",
      "43600000 SAM alignment records processed.\n",
      "43700000 SAM alignment records processed.\n",
      "43800000 SAM alignment records processed.\n",
      "43900000 SAM alignment records processed.\n",
      "44000000 SAM alignment records processed.\n",
      "44100000 SAM alignment records processed.\n",
      "44200000 SAM alignment records processed.\n",
      "44300000 SAM alignment records processed.\n",
      "44400000 SAM alignment records processed.\n",
      "44500000 SAM alignment records processed.\n",
      "44600000 SAM alignment records processed.\n",
      "44700000 SAM alignment records processed.\n",
      "44800000 SAM alignment records processed.\n",
      "44900000 SAM alignment records processed.\n",
      "45000000 SAM alignment records processed.\n",
      "45100000 SAM alignment records processed.\n",
      "45200000 SAM alignment records processed.\n",
      "45300000 SAM alignment records processed.\n",
      "45400000 SAM alignment records processed.\n",
      "45500000 SAM alignment records processed.\n",
      "45600000 SAM alignment records processed.\n",
      "45700000 SAM alignment records processed.\n",
      "45800000 SAM alignment records processed.\n",
      "45900000 SAM alignment records processed.\n",
      "46000000 SAM alignment records processed.\n",
      "46100000 SAM alignment records processed.\n",
      "46200000 SAM alignment records processed.\n",
      "46300000 SAM alignment records processed.\n",
      "46400000 SAM alignment records processed.\n",
      "46500000 SAM alignment records processed.\n",
      "46600000 SAM alignment records processed.\n",
      "46700000 SAM alignment records processed.\n",
      "46800000 SAM alignment records processed.\n",
      "46900000 SAM alignment records processed.\n",
      "47000000 SAM alignment records processed.\n",
      "47100000 SAM alignment records processed.\n",
      "47200000 SAM alignment records processed.\n",
      "47300000 SAM alignment records processed.\n",
      "47400000 SAM alignment records processed.\n",
      "47500000 SAM alignment records processed.\n",
      "47600000 SAM alignment records processed.\n",
      "47700000 SAM alignment records processed.\n",
      "47800000 SAM alignment records processed.\n",
      "47900000 SAM alignment records processed.\n",
      "48000000 SAM alignment records processed.\n",
      "48100000 SAM alignment records processed.\n",
      "48200000 SAM alignment records processed.\n",
      "48300000 SAM alignment records processed.\n",
      "48400000 SAM alignment records processed.\n",
      "48500000 SAM alignment records processed.\n",
      "48600000 SAM alignment records processed.\n",
      "48700000 SAM alignment records processed.\n",
      "48800000 SAM alignment records processed.\n",
      "48900000 SAM alignment records processed.\n",
      "49000000 SAM alignment records processed.\n",
      "49100000 SAM alignment records processed.\n",
      "49200000 SAM alignment records processed.\n",
      "49300000 SAM alignment records processed.\n",
      "49400000 SAM alignment records processed.\n",
      "49500000 SAM alignment records processed.\n",
      "49600000 SAM alignment records processed.\n",
      "49700000 SAM alignment records processed.\n",
      "49800000 SAM alignment records processed.\n",
      "49900000 SAM alignment records processed.\n",
      "50000000 SAM alignment records processed.\n",
      "50100000 SAM alignment records processed.\n",
      "50200000 SAM alignment records processed.\n",
      "50300000 SAM alignment records processed.\n",
      "50400000 SAM alignment records processed.\n",
      "50500000 SAM alignment records processed.\n",
      "50600000 SAM alignment records processed.\n",
      "50700000 SAM alignment records processed.\n",
      "50800000 SAM alignment records processed.\n",
      "50900000 SAM alignment records processed.\n",
      "51000000 SAM alignment records processed.\n",
      "51100000 SAM alignment records processed.\n",
      "51200000 SAM alignment records processed.\n",
      "51300000 SAM alignment records processed.\n",
      "51400000 SAM alignment records processed.\n",
      "51500000 SAM alignment records processed.\n",
      "51600000 SAM alignment records processed.\n",
      "51700000 SAM alignment records processed.\n",
      "51800000 SAM alignment records processed.\n",
      "51900000 SAM alignment records processed.\n",
      "52000000 SAM alignment records processed.\n",
      "52100000 SAM alignment records processed.\n",
      "52200000 SAM alignment records processed.\n",
      "52300000 SAM alignment records processed.\n",
      "52400000 SAM alignment records processed.\n",
      "52500000 SAM alignment records processed.\n",
      "52600000 SAM alignment records processed.\n",
      "52700000 SAM alignment records processed.\n",
      "52800000 SAM alignment records processed.\n",
      "52900000 SAM alignment records processed.\n",
      "52999175 BAM alignments  processed.\n",
      "HTSeq is counting SRR6145433.sorted.bam.\n",
      "100000 GFF lines processed.\n",
      "200000 GFF lines processed.\n",
      "300000 GFF lines processed.\n",
      "400000 GFF lines processed.\n",
      "500000 GFF lines processed.\n",
      "600000 GFF lines processed.\n",
      "700000 GFF lines processed.\n",
      "800000 GFF lines processed.\n",
      "900000 GFF lines processed.\n",
      "1000000 GFF lines processed.\n",
      "1100000 GFF lines processed.\n",
      "1200000 GFF lines processed.\n",
      "1300000 GFF lines processed.\n",
      "1400000 GFF lines processed.\n",
      "1500000 GFF lines processed.\n",
      "1600000 GFF lines processed.\n",
      "1700000 GFF lines processed.\n",
      "1800000 GFF lines processed.\n",
      "1900000 GFF lines processed.\n",
      "2000000 GFF lines processed.\n",
      "2100000 GFF lines processed.\n",
      "2200000 GFF lines processed.\n",
      "2300000 GFF lines processed.\n",
      "2400000 GFF lines processed.\n",
      "2500000 GFF lines processed.\n",
      "2600000 GFF lines processed.\n",
      "2700000 GFF lines processed.\n",
      "2800000 GFF lines processed.\n",
      "2900000 GFF lines processed.\n",
      "2908224 GFF lines processed.\n",
      "100000 SAM alignment records processed.\n",
      "200000 SAM alignment records processed.\n",
      "300000 SAM alignment records processed.\n",
      "400000 SAM alignment records processed.\n",
      "500000 SAM alignment records processed.\n",
      "600000 SAM alignment records processed.\n",
      "700000 SAM alignment records processed.\n",
      "800000 SAM alignment records processed.\n",
      "900000 SAM alignment records processed.\n",
      "1000000 SAM alignment records processed.\n",
      "1100000 SAM alignment records processed.\n",
      "1200000 SAM alignment records processed.\n",
      "1300000 SAM alignment records processed.\n",
      "1400000 SAM alignment records processed.\n",
      "1500000 SAM alignment records processed.\n",
      "1600000 SAM alignment records processed.\n",
      "1700000 SAM alignment records processed.\n",
      "1800000 SAM alignment records processed.\n",
      "1900000 SAM alignment records processed.\n",
      "2000000 SAM alignment records processed.\n",
      "2100000 SAM alignment records processed.\n",
      "2200000 SAM alignment records processed.\n",
      "2300000 SAM alignment records processed.\n",
      "2400000 SAM alignment records processed.\n",
      "2500000 SAM alignment records processed.\n",
      "2600000 SAM alignment records processed.\n",
      "2700000 SAM alignment records processed.\n",
      "2800000 SAM alignment records processed.\n",
      "2900000 SAM alignment records processed.\n",
      "3000000 SAM alignment records processed.\n",
      "3100000 SAM alignment records processed.\n",
      "3200000 SAM alignment records processed.\n",
      "3300000 SAM alignment records processed.\n",
      "3400000 SAM alignment records processed.\n",
      "3500000 SAM alignment records processed.\n",
      "3600000 SAM alignment records processed.\n",
      "3700000 SAM alignment records processed.\n",
      "3800000 SAM alignment records processed.\n",
      "3900000 SAM alignment records processed.\n",
      "4000000 SAM alignment records processed.\n",
      "4100000 SAM alignment records processed.\n",
      "4200000 SAM alignment records processed.\n",
      "4300000 SAM alignment records processed.\n",
      "4400000 SAM alignment records processed.\n",
      "4500000 SAM alignment records processed.\n",
      "4600000 SAM alignment records processed.\n",
      "4700000 SAM alignment records processed.\n",
      "4800000 SAM alignment records processed.\n",
      "4900000 SAM alignment records processed.\n",
      "5000000 SAM alignment records processed.\n",
      "5100000 SAM alignment records processed.\n",
      "5200000 SAM alignment records processed.\n",
      "5300000 SAM alignment records processed.\n",
      "5400000 SAM alignment records processed.\n",
      "5500000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5600000 SAM alignment records processed.\n",
      "5700000 SAM alignment records processed.\n",
      "5800000 SAM alignment records processed.\n",
      "5900000 SAM alignment records processed.\n",
      "6000000 SAM alignment records processed.\n",
      "6100000 SAM alignment records processed.\n",
      "6200000 SAM alignment records processed.\n",
      "6300000 SAM alignment records processed.\n",
      "6400000 SAM alignment records processed.\n",
      "6500000 SAM alignment records processed.\n",
      "6600000 SAM alignment records processed.\n",
      "6700000 SAM alignment records processed.\n",
      "6800000 SAM alignment records processed.\n",
      "6900000 SAM alignment records processed.\n",
      "7000000 SAM alignment records processed.\n",
      "7100000 SAM alignment records processed.\n",
      "7200000 SAM alignment records processed.\n",
      "7300000 SAM alignment records processed.\n",
      "7400000 SAM alignment records processed.\n",
      "7500000 SAM alignment records processed.\n",
      "7600000 SAM alignment records processed.\n",
      "7700000 SAM alignment records processed.\n",
      "7800000 SAM alignment records processed.\n",
      "7900000 SAM alignment records processed.\n",
      "8000000 SAM alignment records processed.\n",
      "8100000 SAM alignment records processed.\n",
      "8200000 SAM alignment records processed.\n",
      "8300000 SAM alignment records processed.\n",
      "8400000 SAM alignment records processed.\n",
      "8500000 SAM alignment records processed.\n",
      "8600000 SAM alignment records processed.\n",
      "8700000 SAM alignment records processed.\n",
      "8800000 SAM alignment records processed.\n",
      "8900000 SAM alignment records processed.\n",
      "9000000 SAM alignment records processed.\n",
      "9100000 SAM alignment records processed.\n",
      "9200000 SAM alignment records processed.\n",
      "9300000 SAM alignment records processed.\n",
      "9400000 SAM alignment records processed.\n",
      "9500000 SAM alignment records processed.\n",
      "9600000 SAM alignment records processed.\n",
      "9700000 SAM alignment records processed.\n",
      "9800000 SAM alignment records processed.\n",
      "9900000 SAM alignment records processed.\n",
      "10000000 SAM alignment records processed.\n",
      "10100000 SAM alignment records processed.\n",
      "10200000 SAM alignment records processed.\n",
      "10300000 SAM alignment records processed.\n",
      "10400000 SAM alignment records processed.\n",
      "10500000 SAM alignment records processed.\n",
      "10600000 SAM alignment records processed.\n",
      "10700000 SAM alignment records processed.\n",
      "10800000 SAM alignment records processed.\n",
      "10900000 SAM alignment records processed.\n",
      "11000000 SAM alignment records processed.\n",
      "11100000 SAM alignment records processed.\n",
      "11200000 SAM alignment records processed.\n",
      "11300000 SAM alignment records processed.\n",
      "11400000 SAM alignment records processed.\n",
      "11500000 SAM alignment records processed.\n",
      "11600000 SAM alignment records processed.\n",
      "11700000 SAM alignment records processed.\n",
      "11800000 SAM alignment records processed.\n",
      "11900000 SAM alignment records processed.\n",
      "12000000 SAM alignment records processed.\n",
      "12100000 SAM alignment records processed.\n",
      "12200000 SAM alignment records processed.\n",
      "12300000 SAM alignment records processed.\n",
      "12400000 SAM alignment records processed.\n",
      "12500000 SAM alignment records processed.\n",
      "12600000 SAM alignment records processed.\n",
      "12700000 SAM alignment records processed.\n",
      "12800000 SAM alignment records processed.\n",
      "12900000 SAM alignment records processed.\n",
      "13000000 SAM alignment records processed.\n",
      "13100000 SAM alignment records processed.\n",
      "13200000 SAM alignment records processed.\n",
      "13300000 SAM alignment records processed.\n",
      "13400000 SAM alignment records processed.\n",
      "13500000 SAM alignment records processed.\n",
      "13600000 SAM alignment records processed.\n",
      "13700000 SAM alignment records processed.\n",
      "13800000 SAM alignment records processed.\n",
      "13900000 SAM alignment records processed.\n",
      "14000000 SAM alignment records processed.\n",
      "14100000 SAM alignment records processed.\n",
      "14200000 SAM alignment records processed.\n",
      "14300000 SAM alignment records processed.\n",
      "14400000 SAM alignment records processed.\n",
      "14500000 SAM alignment records processed.\n",
      "14600000 SAM alignment records processed.\n",
      "14700000 SAM alignment records processed.\n",
      "14800000 SAM alignment records processed.\n",
      "14900000 SAM alignment records processed.\n",
      "15000000 SAM alignment records processed.\n",
      "15100000 SAM alignment records processed.\n",
      "15200000 SAM alignment records processed.\n",
      "15300000 SAM alignment records processed.\n",
      "15400000 SAM alignment records processed.\n",
      "15500000 SAM alignment records processed.\n",
      "15600000 SAM alignment records processed.\n",
      "15700000 SAM alignment records processed.\n",
      "15800000 SAM alignment records processed.\n",
      "15900000 SAM alignment records processed.\n",
      "16000000 SAM alignment records processed.\n",
      "16100000 SAM alignment records processed.\n",
      "16200000 SAM alignment records processed.\n",
      "16300000 SAM alignment records processed.\n",
      "16400000 SAM alignment records processed.\n",
      "16500000 SAM alignment records processed.\n",
      "16600000 SAM alignment records processed.\n",
      "16700000 SAM alignment records processed.\n",
      "16800000 SAM alignment records processed.\n",
      "16900000 SAM alignment records processed.\n",
      "17000000 SAM alignment records processed.\n",
      "17100000 SAM alignment records processed.\n",
      "17200000 SAM alignment records processed.\n",
      "17300000 SAM alignment records processed.\n",
      "17400000 SAM alignment records processed.\n",
      "17500000 SAM alignment records processed.\n",
      "17600000 SAM alignment records processed.\n",
      "17700000 SAM alignment records processed.\n",
      "17800000 SAM alignment records processed.\n",
      "17900000 SAM alignment records processed.\n",
      "18000000 SAM alignment records processed.\n",
      "18100000 SAM alignment records processed.\n",
      "18200000 SAM alignment records processed.\n",
      "18300000 SAM alignment records processed.\n",
      "18400000 SAM alignment records processed.\n",
      "18500000 SAM alignment records processed.\n",
      "18600000 SAM alignment records processed.\n",
      "18700000 SAM alignment records processed.\n",
      "18800000 SAM alignment records processed.\n",
      "18900000 SAM alignment records processed.\n",
      "19000000 SAM alignment records processed.\n",
      "19100000 SAM alignment records processed.\n",
      "19200000 SAM alignment records processed.\n",
      "19300000 SAM alignment records processed.\n",
      "19400000 SAM alignment records processed.\n",
      "19500000 SAM alignment records processed.\n",
      "19600000 SAM alignment records processed.\n",
      "19700000 SAM alignment records processed.\n",
      "19800000 SAM alignment records processed.\n",
      "19900000 SAM alignment records processed.\n",
      "20000000 SAM alignment records processed.\n",
      "20100000 SAM alignment records processed.\n",
      "20200000 SAM alignment records processed.\n",
      "20300000 SAM alignment records processed.\n",
      "20400000 SAM alignment records processed.\n",
      "20500000 SAM alignment records processed.\n",
      "20600000 SAM alignment records processed.\n",
      "20700000 SAM alignment records processed.\n",
      "20800000 SAM alignment records processed.\n",
      "20900000 SAM alignment records processed.\n",
      "21000000 SAM alignment records processed.\n",
      "21100000 SAM alignment records processed.\n",
      "21200000 SAM alignment records processed.\n",
      "21300000 SAM alignment records processed.\n",
      "21400000 SAM alignment records processed.\n",
      "21500000 SAM alignment records processed.\n",
      "21600000 SAM alignment records processed.\n",
      "21700000 SAM alignment records processed.\n",
      "21800000 SAM alignment records processed.\n",
      "21900000 SAM alignment records processed.\n",
      "22000000 SAM alignment records processed.\n",
      "22100000 SAM alignment records processed.\n",
      "22200000 SAM alignment records processed.\n",
      "22300000 SAM alignment records processed.\n",
      "22400000 SAM alignment records processed.\n",
      "22500000 SAM alignment records processed.\n",
      "22600000 SAM alignment records processed.\n",
      "22700000 SAM alignment records processed.\n",
      "22800000 SAM alignment records processed.\n",
      "22900000 SAM alignment records processed.\n",
      "23000000 SAM alignment records processed.\n",
      "23100000 SAM alignment records processed.\n",
      "23200000 SAM alignment records processed.\n",
      "23300000 SAM alignment records processed.\n",
      "23400000 SAM alignment records processed.\n",
      "23500000 SAM alignment records processed.\n",
      "23600000 SAM alignment records processed.\n",
      "23700000 SAM alignment records processed.\n",
      "23800000 SAM alignment records processed.\n",
      "23900000 SAM alignment records processed.\n",
      "24000000 SAM alignment records processed.\n",
      "24100000 SAM alignment records processed.\n",
      "24200000 SAM alignment records processed.\n",
      "24300000 SAM alignment records processed.\n",
      "24400000 SAM alignment records processed.\n",
      "24500000 SAM alignment records processed.\n",
      "24600000 SAM alignment records processed.\n",
      "24700000 SAM alignment records processed.\n",
      "24800000 SAM alignment records processed.\n",
      "24900000 SAM alignment records processed.\n",
      "25000000 SAM alignment records processed.\n",
      "25100000 SAM alignment records processed.\n",
      "25200000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25300000 SAM alignment records processed.\n",
      "25400000 SAM alignment records processed.\n",
      "25500000 SAM alignment records processed.\n",
      "25600000 SAM alignment records processed.\n",
      "25700000 SAM alignment records processed.\n",
      "25800000 SAM alignment records processed.\n",
      "25900000 SAM alignment records processed.\n",
      "26000000 SAM alignment records processed.\n",
      "26100000 SAM alignment records processed.\n",
      "26200000 SAM alignment records processed.\n",
      "26300000 SAM alignment records processed.\n",
      "26400000 SAM alignment records processed.\n",
      "26500000 SAM alignment records processed.\n",
      "26600000 SAM alignment records processed.\n",
      "26700000 SAM alignment records processed.\n",
      "26800000 SAM alignment records processed.\n",
      "26900000 SAM alignment records processed.\n",
      "27000000 SAM alignment records processed.\n",
      "27100000 SAM alignment records processed.\n",
      "27200000 SAM alignment records processed.\n",
      "27300000 SAM alignment records processed.\n",
      "27400000 SAM alignment records processed.\n",
      "27500000 SAM alignment records processed.\n",
      "27600000 SAM alignment records processed.\n",
      "27700000 SAM alignment records processed.\n",
      "27800000 SAM alignment records processed.\n",
      "27900000 SAM alignment records processed.\n",
      "28000000 SAM alignment records processed.\n",
      "28100000 SAM alignment records processed.\n",
      "28200000 SAM alignment records processed.\n",
      "28300000 SAM alignment records processed.\n",
      "28400000 SAM alignment records processed.\n",
      "28500000 SAM alignment records processed.\n",
      "28600000 SAM alignment records processed.\n",
      "28700000 SAM alignment records processed.\n",
      "28800000 SAM alignment records processed.\n",
      "28900000 SAM alignment records processed.\n",
      "29000000 SAM alignment records processed.\n",
      "29100000 SAM alignment records processed.\n",
      "29200000 SAM alignment records processed.\n",
      "29300000 SAM alignment records processed.\n",
      "29400000 SAM alignment records processed.\n",
      "29500000 SAM alignment records processed.\n",
      "29600000 SAM alignment records processed.\n",
      "29700000 SAM alignment records processed.\n",
      "29800000 SAM alignment records processed.\n",
      "29900000 SAM alignment records processed.\n",
      "30000000 SAM alignment records processed.\n",
      "30100000 SAM alignment records processed.\n",
      "30200000 SAM alignment records processed.\n",
      "30300000 SAM alignment records processed.\n",
      "30400000 SAM alignment records processed.\n",
      "30500000 SAM alignment records processed.\n",
      "30600000 SAM alignment records processed.\n",
      "30700000 SAM alignment records processed.\n",
      "30800000 SAM alignment records processed.\n",
      "30900000 SAM alignment records processed.\n",
      "31000000 SAM alignment records processed.\n",
      "31100000 SAM alignment records processed.\n",
      "31200000 SAM alignment records processed.\n",
      "31300000 SAM alignment records processed.\n",
      "31400000 SAM alignment records processed.\n",
      "31500000 SAM alignment records processed.\n",
      "31600000 SAM alignment records processed.\n",
      "31700000 SAM alignment records processed.\n",
      "31800000 SAM alignment records processed.\n",
      "31900000 SAM alignment records processed.\n",
      "32000000 SAM alignment records processed.\n",
      "32100000 SAM alignment records processed.\n",
      "32200000 SAM alignment records processed.\n",
      "32300000 SAM alignment records processed.\n",
      "32400000 SAM alignment records processed.\n",
      "32500000 SAM alignment records processed.\n",
      "32600000 SAM alignment records processed.\n",
      "32700000 SAM alignment records processed.\n",
      "32800000 SAM alignment records processed.\n",
      "32900000 SAM alignment records processed.\n",
      "33000000 SAM alignment records processed.\n",
      "33100000 SAM alignment records processed.\n",
      "33200000 SAM alignment records processed.\n",
      "33300000 SAM alignment records processed.\n",
      "33400000 SAM alignment records processed.\n",
      "33500000 SAM alignment records processed.\n",
      "33600000 SAM alignment records processed.\n",
      "33700000 SAM alignment records processed.\n",
      "33800000 SAM alignment records processed.\n",
      "33900000 SAM alignment records processed.\n",
      "34000000 SAM alignment records processed.\n",
      "34100000 SAM alignment records processed.\n",
      "34200000 SAM alignment records processed.\n",
      "34300000 SAM alignment records processed.\n",
      "34400000 SAM alignment records processed.\n",
      "34500000 SAM alignment records processed.\n",
      "34600000 SAM alignment records processed.\n",
      "34700000 SAM alignment records processed.\n",
      "34800000 SAM alignment records processed.\n",
      "34900000 SAM alignment records processed.\n",
      "35000000 SAM alignment records processed.\n",
      "35100000 SAM alignment records processed.\n",
      "35200000 SAM alignment records processed.\n",
      "35300000 SAM alignment records processed.\n",
      "35400000 SAM alignment records processed.\n",
      "35500000 SAM alignment records processed.\n",
      "35600000 SAM alignment records processed.\n",
      "35700000 SAM alignment records processed.\n",
      "35800000 SAM alignment records processed.\n",
      "35900000 SAM alignment records processed.\n",
      "36000000 SAM alignment records processed.\n",
      "36100000 SAM alignment records processed.\n",
      "36200000 SAM alignment records processed.\n",
      "36300000 SAM alignment records processed.\n",
      "36400000 SAM alignment records processed.\n",
      "36500000 SAM alignment records processed.\n",
      "36600000 SAM alignment records processed.\n",
      "36700000 SAM alignment records processed.\n",
      "36800000 SAM alignment records processed.\n",
      "36900000 SAM alignment records processed.\n",
      "37000000 SAM alignment records processed.\n",
      "37100000 SAM alignment records processed.\n",
      "37200000 SAM alignment records processed.\n",
      "37300000 SAM alignment records processed.\n",
      "37400000 SAM alignment records processed.\n",
      "37500000 SAM alignment records processed.\n",
      "37600000 SAM alignment records processed.\n",
      "37700000 SAM alignment records processed.\n",
      "37800000 SAM alignment records processed.\n",
      "37900000 SAM alignment records processed.\n",
      "38000000 SAM alignment records processed.\n",
      "38100000 SAM alignment records processed.\n",
      "38200000 SAM alignment records processed.\n",
      "38300000 SAM alignment records processed.\n",
      "38400000 SAM alignment records processed.\n",
      "38500000 SAM alignment records processed.\n",
      "38600000 SAM alignment records processed.\n",
      "38700000 SAM alignment records processed.\n",
      "38800000 SAM alignment records processed.\n",
      "38900000 SAM alignment records processed.\n",
      "39000000 SAM alignment records processed.\n",
      "39100000 SAM alignment records processed.\n",
      "39200000 SAM alignment records processed.\n",
      "39300000 SAM alignment records processed.\n",
      "39400000 SAM alignment records processed.\n",
      "39500000 SAM alignment records processed.\n",
      "39600000 SAM alignment records processed.\n",
      "39700000 SAM alignment records processed.\n",
      "39800000 SAM alignment records processed.\n",
      "39900000 SAM alignment records processed.\n",
      "40000000 SAM alignment records processed.\n",
      "40100000 SAM alignment records processed.\n",
      "40200000 SAM alignment records processed.\n",
      "40300000 SAM alignment records processed.\n",
      "40400000 SAM alignment records processed.\n",
      "40500000 SAM alignment records processed.\n",
      "40600000 SAM alignment records processed.\n",
      "40700000 SAM alignment records processed.\n",
      "40800000 SAM alignment records processed.\n",
      "40900000 SAM alignment records processed.\n",
      "41000000 SAM alignment records processed.\n",
      "41100000 SAM alignment records processed.\n",
      "41200000 SAM alignment records processed.\n",
      "41300000 SAM alignment records processed.\n",
      "41400000 SAM alignment records processed.\n",
      "41500000 SAM alignment records processed.\n",
      "41600000 SAM alignment records processed.\n",
      "41700000 SAM alignment records processed.\n",
      "41800000 SAM alignment records processed.\n",
      "41900000 SAM alignment records processed.\n",
      "42000000 SAM alignment records processed.\n",
      "42100000 SAM alignment records processed.\n",
      "42200000 SAM alignment records processed.\n",
      "42300000 SAM alignment records processed.\n",
      "42400000 SAM alignment records processed.\n",
      "42500000 SAM alignment records processed.\n",
      "42600000 SAM alignment records processed.\n",
      "42700000 SAM alignment records processed.\n",
      "42800000 SAM alignment records processed.\n",
      "42900000 SAM alignment records processed.\n",
      "43000000 SAM alignment records processed.\n",
      "43100000 SAM alignment records processed.\n",
      "43200000 SAM alignment records processed.\n",
      "43300000 SAM alignment records processed.\n",
      "43400000 SAM alignment records processed.\n",
      "43500000 SAM alignment records processed.\n",
      "43600000 SAM alignment records processed.\n",
      "43700000 SAM alignment records processed.\n",
      "43800000 SAM alignment records processed.\n",
      "43900000 SAM alignment records processed.\n",
      "44000000 SAM alignment records processed.\n",
      "44100000 SAM alignment records processed.\n",
      "44200000 SAM alignment records processed.\n",
      "44300000 SAM alignment records processed.\n",
      "44400000 SAM alignment records processed.\n",
      "44500000 SAM alignment records processed.\n",
      "44600000 SAM alignment records processed.\n",
      "44700000 SAM alignment records processed.\n",
      "44800000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "44900000 SAM alignment records processed.\n",
      "45000000 SAM alignment records processed.\n",
      "45100000 SAM alignment records processed.\n",
      "45200000 SAM alignment records processed.\n",
      "45300000 SAM alignment records processed.\n",
      "45400000 SAM alignment records processed.\n",
      "45500000 SAM alignment records processed.\n",
      "45600000 SAM alignment records processed.\n",
      "45700000 SAM alignment records processed.\n",
      "45800000 SAM alignment records processed.\n",
      "45900000 SAM alignment records processed.\n",
      "46000000 SAM alignment records processed.\n",
      "46100000 SAM alignment records processed.\n",
      "46200000 SAM alignment records processed.\n",
      "46300000 SAM alignment records processed.\n",
      "46400000 SAM alignment records processed.\n",
      "46500000 SAM alignment records processed.\n",
      "46600000 SAM alignment records processed.\n",
      "46700000 SAM alignment records processed.\n",
      "46800000 SAM alignment records processed.\n",
      "46900000 SAM alignment records processed.\n",
      "47000000 SAM alignment records processed.\n",
      "47100000 SAM alignment records processed.\n",
      "47200000 SAM alignment records processed.\n",
      "47300000 SAM alignment records processed.\n",
      "47400000 SAM alignment records processed.\n",
      "47500000 SAM alignment records processed.\n",
      "47600000 SAM alignment records processed.\n",
      "47700000 SAM alignment records processed.\n",
      "47800000 SAM alignment records processed.\n",
      "47900000 SAM alignment records processed.\n",
      "48000000 SAM alignment records processed.\n",
      "48100000 SAM alignment records processed.\n",
      "48200000 SAM alignment records processed.\n",
      "48300000 SAM alignment records processed.\n",
      "48400000 SAM alignment records processed.\n",
      "48500000 SAM alignment records processed.\n",
      "48600000 SAM alignment records processed.\n",
      "48700000 SAM alignment records processed.\n",
      "48800000 SAM alignment records processed.\n",
      "48900000 SAM alignment records processed.\n",
      "49000000 SAM alignment records processed.\n",
      "49100000 SAM alignment records processed.\n",
      "49200000 SAM alignment records processed.\n",
      "49300000 SAM alignment records processed.\n",
      "49400000 SAM alignment records processed.\n",
      "49500000 SAM alignment records processed.\n",
      "49600000 SAM alignment records processed.\n",
      "49700000 SAM alignment records processed.\n",
      "49800000 SAM alignment records processed.\n",
      "49900000 SAM alignment records processed.\n",
      "50000000 SAM alignment records processed.\n",
      "50100000 SAM alignment records processed.\n",
      "50200000 SAM alignment records processed.\n",
      "50300000 SAM alignment records processed.\n",
      "50400000 SAM alignment records processed.\n",
      "50500000 SAM alignment records processed.\n",
      "50600000 SAM alignment records processed.\n",
      "50700000 SAM alignment records processed.\n",
      "50800000 SAM alignment records processed.\n",
      "50900000 SAM alignment records processed.\n",
      "51000000 SAM alignment records processed.\n",
      "51100000 SAM alignment records processed.\n",
      "51200000 SAM alignment records processed.\n",
      "51300000 SAM alignment records processed.\n",
      "51400000 SAM alignment records processed.\n",
      "51500000 SAM alignment records processed.\n",
      "51600000 SAM alignment records processed.\n",
      "51700000 SAM alignment records processed.\n",
      "51800000 SAM alignment records processed.\n",
      "51900000 SAM alignment records processed.\n",
      "52000000 SAM alignment records processed.\n",
      "52100000 SAM alignment records processed.\n",
      "52200000 SAM alignment records processed.\n",
      "52300000 SAM alignment records processed.\n",
      "52400000 SAM alignment records processed.\n",
      "52500000 SAM alignment records processed.\n",
      "52600000 SAM alignment records processed.\n",
      "52700000 SAM alignment records processed.\n",
      "52800000 SAM alignment records processed.\n",
      "52900000 SAM alignment records processed.\n",
      "53000000 SAM alignment records processed.\n",
      "53100000 SAM alignment records processed.\n",
      "53200000 SAM alignment records processed.\n",
      "53300000 SAM alignment records processed.\n",
      "53400000 SAM alignment records processed.\n",
      "53500000 SAM alignment records processed.\n",
      "53600000 SAM alignment records processed.\n",
      "53700000 SAM alignment records processed.\n",
      "53800000 SAM alignment records processed.\n",
      "53900000 SAM alignment records processed.\n",
      "54000000 SAM alignment records processed.\n",
      "54100000 SAM alignment records processed.\n",
      "54200000 SAM alignment records processed.\n",
      "54300000 SAM alignment records processed.\n",
      "54400000 SAM alignment records processed.\n",
      "54500000 SAM alignment records processed.\n",
      "54600000 SAM alignment records processed.\n",
      "54700000 SAM alignment records processed.\n",
      "54800000 SAM alignment records processed.\n",
      "54900000 SAM alignment records processed.\n",
      "55000000 SAM alignment records processed.\n",
      "55100000 SAM alignment records processed.\n",
      "55200000 SAM alignment records processed.\n",
      "55300000 SAM alignment records processed.\n",
      "55400000 SAM alignment records processed.\n",
      "55500000 SAM alignment records processed.\n",
      "55600000 SAM alignment records processed.\n",
      "55700000 SAM alignment records processed.\n",
      "55800000 SAM alignment records processed.\n",
      "55900000 SAM alignment records processed.\n",
      "56000000 SAM alignment records processed.\n",
      "56100000 SAM alignment records processed.\n",
      "56200000 SAM alignment records processed.\n",
      "56300000 SAM alignment records processed.\n",
      "56400000 SAM alignment records processed.\n",
      "56500000 SAM alignment records processed.\n",
      "56600000 SAM alignment records processed.\n",
      "56700000 SAM alignment records processed.\n",
      "56800000 SAM alignment records processed.\n",
      "56900000 SAM alignment records processed.\n",
      "57000000 SAM alignment records processed.\n",
      "57100000 SAM alignment records processed.\n",
      "57200000 SAM alignment records processed.\n",
      "57300000 SAM alignment records processed.\n",
      "57400000 SAM alignment records processed.\n",
      "57500000 SAM alignment records processed.\n",
      "57600000 SAM alignment records processed.\n",
      "57700000 SAM alignment records processed.\n",
      "57800000 SAM alignment records processed.\n",
      "57900000 SAM alignment records processed.\n",
      "58000000 SAM alignment records processed.\n",
      "58100000 SAM alignment records processed.\n",
      "58200000 SAM alignment records processed.\n",
      "58300000 SAM alignment records processed.\n",
      "58400000 SAM alignment records processed.\n",
      "58500000 SAM alignment records processed.\n",
      "58600000 SAM alignment records processed.\n",
      "58700000 SAM alignment records processed.\n",
      "58800000 SAM alignment records processed.\n",
      "58900000 SAM alignment records processed.\n",
      "59000000 SAM alignment records processed.\n",
      "59100000 SAM alignment records processed.\n",
      "59200000 SAM alignment records processed.\n",
      "59300000 SAM alignment records processed.\n",
      "59400000 SAM alignment records processed.\n",
      "59500000 SAM alignment records processed.\n",
      "59600000 SAM alignment records processed.\n",
      "59700000 SAM alignment records processed.\n",
      "59800000 SAM alignment records processed.\n",
      "59900000 SAM alignment records processed.\n",
      "60000000 SAM alignment records processed.\n",
      "60100000 SAM alignment records processed.\n",
      "60200000 SAM alignment records processed.\n",
      "60300000 SAM alignment records processed.\n",
      "60400000 SAM alignment records processed.\n",
      "60500000 SAM alignment records processed.\n",
      "60600000 SAM alignment records processed.\n",
      "60700000 SAM alignment records processed.\n",
      "60800000 SAM alignment records processed.\n",
      "60900000 SAM alignment records processed.\n",
      "61000000 SAM alignment records processed.\n",
      "61100000 SAM alignment records processed.\n",
      "61200000 SAM alignment records processed.\n",
      "61300000 SAM alignment records processed.\n",
      "61400000 SAM alignment records processed.\n",
      "61500000 SAM alignment records processed.\n",
      "61600000 SAM alignment records processed.\n",
      "61700000 SAM alignment records processed.\n",
      "61800000 SAM alignment records processed.\n",
      "61900000 SAM alignment records processed.\n",
      "62000000 SAM alignment records processed.\n",
      "62100000 SAM alignment records processed.\n",
      "62200000 SAM alignment records processed.\n",
      "62300000 SAM alignment records processed.\n",
      "62400000 SAM alignment records processed.\n",
      "62500000 SAM alignment records processed.\n",
      "62600000 SAM alignment records processed.\n",
      "62700000 SAM alignment records processed.\n",
      "62800000 SAM alignment records processed.\n",
      "62900000 SAM alignment records processed.\n",
      "63000000 SAM alignment records processed.\n",
      "63100000 SAM alignment records processed.\n",
      "63200000 SAM alignment records processed.\n",
      "63300000 SAM alignment records processed.\n",
      "63400000 SAM alignment records processed.\n",
      "63500000 SAM alignment records processed.\n",
      "63600000 SAM alignment records processed.\n",
      "63700000 SAM alignment records processed.\n",
      "63800000 SAM alignment records processed.\n",
      "63900000 SAM alignment records processed.\n",
      "64000000 SAM alignment records processed.\n",
      "64100000 SAM alignment records processed.\n",
      "64200000 SAM alignment records processed.\n",
      "64300000 SAM alignment records processed.\n",
      "64400000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "64500000 SAM alignment records processed.\n",
      "64600000 SAM alignment records processed.\n",
      "64700000 SAM alignment records processed.\n",
      "64800000 SAM alignment records processed.\n",
      "64900000 SAM alignment records processed.\n",
      "65000000 SAM alignment records processed.\n",
      "65100000 SAM alignment records processed.\n",
      "65200000 SAM alignment records processed.\n",
      "65300000 SAM alignment records processed.\n",
      "65400000 SAM alignment records processed.\n",
      "65500000 SAM alignment records processed.\n",
      "65600000 SAM alignment records processed.\n",
      "65700000 SAM alignment records processed.\n",
      "65800000 SAM alignment records processed.\n",
      "65900000 SAM alignment records processed.\n",
      "66000000 SAM alignment records processed.\n",
      "66100000 SAM alignment records processed.\n",
      "66200000 SAM alignment records processed.\n",
      "66300000 SAM alignment records processed.\n",
      "66400000 SAM alignment records processed.\n",
      "66500000 SAM alignment records processed.\n",
      "66600000 SAM alignment records processed.\n",
      "66700000 SAM alignment records processed.\n",
      "66800000 SAM alignment records processed.\n",
      "66900000 SAM alignment records processed.\n",
      "67000000 SAM alignment records processed.\n",
      "67100000 SAM alignment records processed.\n",
      "67200000 SAM alignment records processed.\n",
      "67300000 SAM alignment records processed.\n",
      "67400000 SAM alignment records processed.\n",
      "67500000 SAM alignment records processed.\n",
      "67600000 SAM alignment records processed.\n",
      "67700000 SAM alignment records processed.\n",
      "67800000 SAM alignment records processed.\n",
      "67900000 SAM alignment records processed.\n",
      "68000000 SAM alignment records processed.\n",
      "68100000 SAM alignment records processed.\n",
      "68200000 SAM alignment records processed.\n",
      "68300000 SAM alignment records processed.\n",
      "68400000 SAM alignment records processed.\n",
      "68500000 SAM alignment records processed.\n",
      "68600000 SAM alignment records processed.\n",
      "68700000 SAM alignment records processed.\n",
      "68800000 SAM alignment records processed.\n",
      "68900000 SAM alignment records processed.\n",
      "69000000 SAM alignment records processed.\n",
      "69100000 SAM alignment records processed.\n",
      "69200000 SAM alignment records processed.\n",
      "69300000 SAM alignment records processed.\n",
      "69400000 SAM alignment records processed.\n",
      "69500000 SAM alignment records processed.\n",
      "69600000 SAM alignment records processed.\n",
      "69700000 SAM alignment records processed.\n",
      "69800000 SAM alignment records processed.\n",
      "69900000 SAM alignment records processed.\n",
      "70000000 SAM alignment records processed.\n",
      "70100000 SAM alignment records processed.\n",
      "70200000 SAM alignment records processed.\n",
      "70300000 SAM alignment records processed.\n",
      "70400000 SAM alignment records processed.\n",
      "70500000 SAM alignment records processed.\n",
      "70600000 SAM alignment records processed.\n",
      "70700000 SAM alignment records processed.\n",
      "70800000 SAM alignment records processed.\n",
      "70900000 SAM alignment records processed.\n",
      "71000000 SAM alignment records processed.\n",
      "71100000 SAM alignment records processed.\n",
      "71200000 SAM alignment records processed.\n",
      "71300000 SAM alignment records processed.\n",
      "71400000 SAM alignment records processed.\n",
      "71500000 SAM alignment records processed.\n",
      "71600000 SAM alignment records processed.\n",
      "71700000 SAM alignment records processed.\n",
      "71800000 SAM alignment records processed.\n",
      "71900000 SAM alignment records processed.\n",
      "72000000 SAM alignment records processed.\n",
      "72100000 SAM alignment records processed.\n",
      "72200000 SAM alignment records processed.\n",
      "72300000 SAM alignment records processed.\n",
      "72400000 SAM alignment records processed.\n",
      "72500000 SAM alignment records processed.\n",
      "72600000 SAM alignment records processed.\n",
      "72700000 SAM alignment records processed.\n",
      "72800000 SAM alignment records processed.\n",
      "72900000 SAM alignment records processed.\n",
      "73000000 SAM alignment records processed.\n",
      "73100000 SAM alignment records processed.\n",
      "73200000 SAM alignment records processed.\n",
      "73300000 SAM alignment records processed.\n",
      "73400000 SAM alignment records processed.\n",
      "73500000 SAM alignment records processed.\n",
      "73600000 SAM alignment records processed.\n",
      "73700000 SAM alignment records processed.\n",
      "73800000 SAM alignment records processed.\n",
      "73900000 SAM alignment records processed.\n",
      "73906282 BAM alignments  processed.\n",
      "HTSeq is counting SRR6145434.sorted.bam.\n",
      "100000 GFF lines processed.\n",
      "200000 GFF lines processed.\n",
      "300000 GFF lines processed.\n",
      "400000 GFF lines processed.\n",
      "500000 GFF lines processed.\n",
      "600000 GFF lines processed.\n",
      "700000 GFF lines processed.\n",
      "800000 GFF lines processed.\n",
      "900000 GFF lines processed.\n",
      "1000000 GFF lines processed.\n",
      "1100000 GFF lines processed.\n",
      "1200000 GFF lines processed.\n",
      "1300000 GFF lines processed.\n",
      "1400000 GFF lines processed.\n",
      "1500000 GFF lines processed.\n",
      "1600000 GFF lines processed.\n",
      "1700000 GFF lines processed.\n",
      "1800000 GFF lines processed.\n",
      "1900000 GFF lines processed.\n",
      "2000000 GFF lines processed.\n",
      "2100000 GFF lines processed.\n",
      "2200000 GFF lines processed.\n",
      "2300000 GFF lines processed.\n",
      "2400000 GFF lines processed.\n",
      "2500000 GFF lines processed.\n",
      "2600000 GFF lines processed.\n",
      "2700000 GFF lines processed.\n",
      "2800000 GFF lines processed.\n",
      "2900000 GFF lines processed.\n",
      "2908224 GFF lines processed.\n",
      "100000 SAM alignment records processed.\n",
      "200000 SAM alignment records processed.\n",
      "300000 SAM alignment records processed.\n",
      "400000 SAM alignment records processed.\n",
      "500000 SAM alignment records processed.\n",
      "600000 SAM alignment records processed.\n",
      "700000 SAM alignment records processed.\n",
      "800000 SAM alignment records processed.\n",
      "900000 SAM alignment records processed.\n",
      "1000000 SAM alignment records processed.\n",
      "1100000 SAM alignment records processed.\n",
      "1200000 SAM alignment records processed.\n",
      "1300000 SAM alignment records processed.\n",
      "1400000 SAM alignment records processed.\n",
      "1500000 SAM alignment records processed.\n",
      "1600000 SAM alignment records processed.\n",
      "1700000 SAM alignment records processed.\n",
      "1800000 SAM alignment records processed.\n",
      "1900000 SAM alignment records processed.\n",
      "2000000 SAM alignment records processed.\n",
      "2100000 SAM alignment records processed.\n",
      "2200000 SAM alignment records processed.\n",
      "2300000 SAM alignment records processed.\n",
      "2400000 SAM alignment records processed.\n",
      "2500000 SAM alignment records processed.\n",
      "2600000 SAM alignment records processed.\n",
      "2700000 SAM alignment records processed.\n",
      "2800000 SAM alignment records processed.\n",
      "2900000 SAM alignment records processed.\n",
      "3000000 SAM alignment records processed.\n",
      "3100000 SAM alignment records processed.\n",
      "3200000 SAM alignment records processed.\n",
      "3300000 SAM alignment records processed.\n",
      "3400000 SAM alignment records processed.\n",
      "3500000 SAM alignment records processed.\n",
      "3600000 SAM alignment records processed.\n",
      "3700000 SAM alignment records processed.\n",
      "3800000 SAM alignment records processed.\n",
      "3900000 SAM alignment records processed.\n",
      "4000000 SAM alignment records processed.\n",
      "4100000 SAM alignment records processed.\n",
      "4200000 SAM alignment records processed.\n",
      "4300000 SAM alignment records processed.\n",
      "4400000 SAM alignment records processed.\n",
      "4500000 SAM alignment records processed.\n",
      "4600000 SAM alignment records processed.\n",
      "4700000 SAM alignment records processed.\n",
      "4800000 SAM alignment records processed.\n",
      "4900000 SAM alignment records processed.\n",
      "5000000 SAM alignment records processed.\n",
      "5100000 SAM alignment records processed.\n",
      "5200000 SAM alignment records processed.\n",
      "5300000 SAM alignment records processed.\n",
      "5400000 SAM alignment records processed.\n",
      "5500000 SAM alignment records processed.\n",
      "5600000 SAM alignment records processed.\n",
      "5700000 SAM alignment records processed.\n",
      "5800000 SAM alignment records processed.\n",
      "5900000 SAM alignment records processed.\n",
      "6000000 SAM alignment records processed.\n",
      "6100000 SAM alignment records processed.\n",
      "6200000 SAM alignment records processed.\n",
      "6300000 SAM alignment records processed.\n",
      "6400000 SAM alignment records processed.\n",
      "6500000 SAM alignment records processed.\n",
      "6600000 SAM alignment records processed.\n",
      "6700000 SAM alignment records processed.\n",
      "6800000 SAM alignment records processed.\n",
      "6900000 SAM alignment records processed.\n",
      "7000000 SAM alignment records processed.\n",
      "7100000 SAM alignment records processed.\n",
      "7200000 SAM alignment records processed.\n",
      "7300000 SAM alignment records processed.\n",
      "7400000 SAM alignment records processed.\n",
      "7500000 SAM alignment records processed.\n",
      "7600000 SAM alignment records processed.\n",
      "7700000 SAM alignment records processed.\n",
      "7800000 SAM alignment records processed.\n",
      "7900000 SAM alignment records processed.\n",
      "8000000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8100000 SAM alignment records processed.\n",
      "8200000 SAM alignment records processed.\n",
      "8300000 SAM alignment records processed.\n",
      "8400000 SAM alignment records processed.\n",
      "8500000 SAM alignment records processed.\n",
      "8600000 SAM alignment records processed.\n",
      "8700000 SAM alignment records processed.\n",
      "8800000 SAM alignment records processed.\n",
      "8900000 SAM alignment records processed.\n",
      "9000000 SAM alignment records processed.\n",
      "9100000 SAM alignment records processed.\n",
      "9200000 SAM alignment records processed.\n",
      "9300000 SAM alignment records processed.\n",
      "9400000 SAM alignment records processed.\n",
      "9500000 SAM alignment records processed.\n",
      "9600000 SAM alignment records processed.\n",
      "9700000 SAM alignment records processed.\n",
      "9800000 SAM alignment records processed.\n",
      "9900000 SAM alignment records processed.\n",
      "10000000 SAM alignment records processed.\n",
      "10100000 SAM alignment records processed.\n",
      "10200000 SAM alignment records processed.\n",
      "10300000 SAM alignment records processed.\n",
      "10400000 SAM alignment records processed.\n",
      "10500000 SAM alignment records processed.\n",
      "10600000 SAM alignment records processed.\n",
      "10700000 SAM alignment records processed.\n",
      "10800000 SAM alignment records processed.\n",
      "10900000 SAM alignment records processed.\n",
      "11000000 SAM alignment records processed.\n",
      "11100000 SAM alignment records processed.\n",
      "11200000 SAM alignment records processed.\n",
      "11300000 SAM alignment records processed.\n",
      "11400000 SAM alignment records processed.\n",
      "11500000 SAM alignment records processed.\n",
      "11600000 SAM alignment records processed.\n",
      "11700000 SAM alignment records processed.\n",
      "11800000 SAM alignment records processed.\n",
      "11900000 SAM alignment records processed.\n",
      "12000000 SAM alignment records processed.\n",
      "12100000 SAM alignment records processed.\n",
      "12200000 SAM alignment records processed.\n",
      "12300000 SAM alignment records processed.\n",
      "12400000 SAM alignment records processed.\n",
      "12500000 SAM alignment records processed.\n",
      "12600000 SAM alignment records processed.\n",
      "12700000 SAM alignment records processed.\n",
      "12800000 SAM alignment records processed.\n",
      "12900000 SAM alignment records processed.\n",
      "13000000 SAM alignment records processed.\n",
      "13100000 SAM alignment records processed.\n",
      "13200000 SAM alignment records processed.\n",
      "13300000 SAM alignment records processed.\n",
      "13400000 SAM alignment records processed.\n",
      "13500000 SAM alignment records processed.\n",
      "13600000 SAM alignment records processed.\n",
      "13700000 SAM alignment records processed.\n",
      "13800000 SAM alignment records processed.\n",
      "13900000 SAM alignment records processed.\n",
      "14000000 SAM alignment records processed.\n",
      "14100000 SAM alignment records processed.\n",
      "14200000 SAM alignment records processed.\n",
      "14300000 SAM alignment records processed.\n",
      "14400000 SAM alignment records processed.\n",
      "14500000 SAM alignment records processed.\n",
      "14600000 SAM alignment records processed.\n",
      "14700000 SAM alignment records processed.\n",
      "14800000 SAM alignment records processed.\n",
      "14900000 SAM alignment records processed.\n",
      "15000000 SAM alignment records processed.\n",
      "15100000 SAM alignment records processed.\n",
      "15200000 SAM alignment records processed.\n",
      "15300000 SAM alignment records processed.\n",
      "15400000 SAM alignment records processed.\n",
      "15500000 SAM alignment records processed.\n",
      "15600000 SAM alignment records processed.\n",
      "15700000 SAM alignment records processed.\n",
      "15800000 SAM alignment records processed.\n",
      "15900000 SAM alignment records processed.\n",
      "16000000 SAM alignment records processed.\n",
      "16100000 SAM alignment records processed.\n",
      "16200000 SAM alignment records processed.\n",
      "16300000 SAM alignment records processed.\n",
      "16400000 SAM alignment records processed.\n",
      "16500000 SAM alignment records processed.\n",
      "16600000 SAM alignment records processed.\n",
      "16700000 SAM alignment records processed.\n",
      "16800000 SAM alignment records processed.\n",
      "16900000 SAM alignment records processed.\n",
      "17000000 SAM alignment records processed.\n",
      "17100000 SAM alignment records processed.\n",
      "17200000 SAM alignment records processed.\n",
      "17300000 SAM alignment records processed.\n",
      "17400000 SAM alignment records processed.\n",
      "17500000 SAM alignment records processed.\n",
      "17600000 SAM alignment records processed.\n",
      "17700000 SAM alignment records processed.\n",
      "17800000 SAM alignment records processed.\n",
      "17900000 SAM alignment records processed.\n",
      "18000000 SAM alignment records processed.\n",
      "18100000 SAM alignment records processed.\n",
      "18200000 SAM alignment records processed.\n",
      "18300000 SAM alignment records processed.\n",
      "18400000 SAM alignment records processed.\n",
      "18500000 SAM alignment records processed.\n",
      "18600000 SAM alignment records processed.\n",
      "18700000 SAM alignment records processed.\n",
      "18800000 SAM alignment records processed.\n",
      "18900000 SAM alignment records processed.\n",
      "19000000 SAM alignment records processed.\n",
      "19100000 SAM alignment records processed.\n",
      "19200000 SAM alignment records processed.\n",
      "19300000 SAM alignment records processed.\n",
      "19400000 SAM alignment records processed.\n",
      "19500000 SAM alignment records processed.\n",
      "19600000 SAM alignment records processed.\n",
      "19700000 SAM alignment records processed.\n",
      "19800000 SAM alignment records processed.\n",
      "19900000 SAM alignment records processed.\n",
      "20000000 SAM alignment records processed.\n",
      "20100000 SAM alignment records processed.\n",
      "20200000 SAM alignment records processed.\n",
      "20300000 SAM alignment records processed.\n",
      "20400000 SAM alignment records processed.\n",
      "20500000 SAM alignment records processed.\n",
      "20600000 SAM alignment records processed.\n",
      "20700000 SAM alignment records processed.\n",
      "20800000 SAM alignment records processed.\n",
      "20900000 SAM alignment records processed.\n",
      "21000000 SAM alignment records processed.\n",
      "21100000 SAM alignment records processed.\n",
      "21200000 SAM alignment records processed.\n",
      "21300000 SAM alignment records processed.\n",
      "21400000 SAM alignment records processed.\n",
      "21500000 SAM alignment records processed.\n",
      "21600000 SAM alignment records processed.\n",
      "21700000 SAM alignment records processed.\n",
      "21800000 SAM alignment records processed.\n",
      "21900000 SAM alignment records processed.\n",
      "22000000 SAM alignment records processed.\n",
      "22100000 SAM alignment records processed.\n",
      "22200000 SAM alignment records processed.\n",
      "22300000 SAM alignment records processed.\n",
      "22400000 SAM alignment records processed.\n",
      "22500000 SAM alignment records processed.\n",
      "22600000 SAM alignment records processed.\n",
      "22700000 SAM alignment records processed.\n",
      "22800000 SAM alignment records processed.\n",
      "22900000 SAM alignment records processed.\n",
      "23000000 SAM alignment records processed.\n",
      "23100000 SAM alignment records processed.\n",
      "23200000 SAM alignment records processed.\n",
      "23300000 SAM alignment records processed.\n",
      "23400000 SAM alignment records processed.\n",
      "23500000 SAM alignment records processed.\n",
      "23600000 SAM alignment records processed.\n",
      "23700000 SAM alignment records processed.\n",
      "23800000 SAM alignment records processed.\n",
      "23900000 SAM alignment records processed.\n",
      "24000000 SAM alignment records processed.\n",
      "24100000 SAM alignment records processed.\n",
      "24200000 SAM alignment records processed.\n",
      "24300000 SAM alignment records processed.\n",
      "24400000 SAM alignment records processed.\n",
      "24500000 SAM alignment records processed.\n",
      "24600000 SAM alignment records processed.\n",
      "24700000 SAM alignment records processed.\n",
      "24800000 SAM alignment records processed.\n",
      "24900000 SAM alignment records processed.\n",
      "25000000 SAM alignment records processed.\n",
      "25100000 SAM alignment records processed.\n",
      "25200000 SAM alignment records processed.\n",
      "25300000 SAM alignment records processed.\n",
      "25400000 SAM alignment records processed.\n",
      "25500000 SAM alignment records processed.\n",
      "25600000 SAM alignment records processed.\n",
      "25700000 SAM alignment records processed.\n",
      "25800000 SAM alignment records processed.\n",
      "25900000 SAM alignment records processed.\n",
      "26000000 SAM alignment records processed.\n",
      "26100000 SAM alignment records processed.\n",
      "26200000 SAM alignment records processed.\n",
      "26300000 SAM alignment records processed.\n",
      "26400000 SAM alignment records processed.\n",
      "26500000 SAM alignment records processed.\n",
      "26600000 SAM alignment records processed.\n",
      "26700000 SAM alignment records processed.\n",
      "26800000 SAM alignment records processed.\n",
      "26900000 SAM alignment records processed.\n",
      "27000000 SAM alignment records processed.\n",
      "27100000 SAM alignment records processed.\n",
      "27200000 SAM alignment records processed.\n",
      "27300000 SAM alignment records processed.\n",
      "27400000 SAM alignment records processed.\n",
      "27500000 SAM alignment records processed.\n",
      "27600000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27700000 SAM alignment records processed.\n",
      "27800000 SAM alignment records processed.\n",
      "27900000 SAM alignment records processed.\n",
      "28000000 SAM alignment records processed.\n",
      "28100000 SAM alignment records processed.\n",
      "28200000 SAM alignment records processed.\n",
      "28300000 SAM alignment records processed.\n",
      "28400000 SAM alignment records processed.\n",
      "28500000 SAM alignment records processed.\n",
      "28600000 SAM alignment records processed.\n",
      "28700000 SAM alignment records processed.\n",
      "28800000 SAM alignment records processed.\n",
      "28900000 SAM alignment records processed.\n",
      "29000000 SAM alignment records processed.\n",
      "29100000 SAM alignment records processed.\n",
      "29200000 SAM alignment records processed.\n",
      "29300000 SAM alignment records processed.\n",
      "29400000 SAM alignment records processed.\n",
      "29500000 SAM alignment records processed.\n",
      "29600000 SAM alignment records processed.\n",
      "29700000 SAM alignment records processed.\n",
      "29800000 SAM alignment records processed.\n",
      "29900000 SAM alignment records processed.\n",
      "30000000 SAM alignment records processed.\n",
      "30100000 SAM alignment records processed.\n",
      "30200000 SAM alignment records processed.\n",
      "30300000 SAM alignment records processed.\n",
      "30400000 SAM alignment records processed.\n",
      "30500000 SAM alignment records processed.\n",
      "30600000 SAM alignment records processed.\n",
      "30700000 SAM alignment records processed.\n",
      "30800000 SAM alignment records processed.\n",
      "30900000 SAM alignment records processed.\n",
      "31000000 SAM alignment records processed.\n",
      "31100000 SAM alignment records processed.\n",
      "31200000 SAM alignment records processed.\n",
      "31300000 SAM alignment records processed.\n",
      "31400000 SAM alignment records processed.\n",
      "31500000 SAM alignment records processed.\n",
      "31600000 SAM alignment records processed.\n",
      "31700000 SAM alignment records processed.\n",
      "31800000 SAM alignment records processed.\n",
      "31900000 SAM alignment records processed.\n",
      "32000000 SAM alignment records processed.\n",
      "32100000 SAM alignment records processed.\n",
      "32200000 SAM alignment records processed.\n",
      "32300000 SAM alignment records processed.\n",
      "32400000 SAM alignment records processed.\n",
      "32500000 SAM alignment records processed.\n",
      "32600000 SAM alignment records processed.\n",
      "32700000 SAM alignment records processed.\n",
      "32800000 SAM alignment records processed.\n",
      "32900000 SAM alignment records processed.\n",
      "33000000 SAM alignment records processed.\n",
      "33100000 SAM alignment records processed.\n",
      "33200000 SAM alignment records processed.\n",
      "33300000 SAM alignment records processed.\n",
      "33400000 SAM alignment records processed.\n",
      "33500000 SAM alignment records processed.\n",
      "33600000 SAM alignment records processed.\n",
      "33700000 SAM alignment records processed.\n",
      "33800000 SAM alignment records processed.\n",
      "33900000 SAM alignment records processed.\n",
      "34000000 SAM alignment records processed.\n",
      "34100000 SAM alignment records processed.\n",
      "34200000 SAM alignment records processed.\n",
      "34300000 SAM alignment records processed.\n",
      "34400000 SAM alignment records processed.\n",
      "34500000 SAM alignment records processed.\n",
      "34600000 SAM alignment records processed.\n",
      "34700000 SAM alignment records processed.\n",
      "34800000 SAM alignment records processed.\n",
      "34900000 SAM alignment records processed.\n",
      "35000000 SAM alignment records processed.\n",
      "35100000 SAM alignment records processed.\n",
      "35200000 SAM alignment records processed.\n",
      "35300000 SAM alignment records processed.\n",
      "35400000 SAM alignment records processed.\n",
      "35500000 SAM alignment records processed.\n",
      "35600000 SAM alignment records processed.\n",
      "35700000 SAM alignment records processed.\n",
      "35800000 SAM alignment records processed.\n",
      "35900000 SAM alignment records processed.\n",
      "36000000 SAM alignment records processed.\n",
      "36100000 SAM alignment records processed.\n",
      "36200000 SAM alignment records processed.\n",
      "36300000 SAM alignment records processed.\n",
      "36400000 SAM alignment records processed.\n",
      "36500000 SAM alignment records processed.\n",
      "36600000 SAM alignment records processed.\n",
      "36700000 SAM alignment records processed.\n",
      "36800000 SAM alignment records processed.\n",
      "36900000 SAM alignment records processed.\n",
      "37000000 SAM alignment records processed.\n",
      "37100000 SAM alignment records processed.\n",
      "37200000 SAM alignment records processed.\n",
      "37300000 SAM alignment records processed.\n",
      "37400000 SAM alignment records processed.\n",
      "37500000 SAM alignment records processed.\n",
      "37600000 SAM alignment records processed.\n",
      "37700000 SAM alignment records processed.\n",
      "37800000 SAM alignment records processed.\n",
      "37900000 SAM alignment records processed.\n",
      "38000000 SAM alignment records processed.\n",
      "38100000 SAM alignment records processed.\n",
      "38200000 SAM alignment records processed.\n",
      "38300000 SAM alignment records processed.\n",
      "38400000 SAM alignment records processed.\n",
      "38500000 SAM alignment records processed.\n",
      "38600000 SAM alignment records processed.\n",
      "38700000 SAM alignment records processed.\n",
      "38800000 SAM alignment records processed.\n",
      "38900000 SAM alignment records processed.\n",
      "39000000 SAM alignment records processed.\n",
      "39100000 SAM alignment records processed.\n",
      "39200000 SAM alignment records processed.\n",
      "39300000 SAM alignment records processed.\n",
      "39400000 SAM alignment records processed.\n",
      "39500000 SAM alignment records processed.\n",
      "39600000 SAM alignment records processed.\n",
      "39700000 SAM alignment records processed.\n",
      "39800000 SAM alignment records processed.\n",
      "39900000 SAM alignment records processed.\n",
      "40000000 SAM alignment records processed.\n",
      "40100000 SAM alignment records processed.\n",
      "40200000 SAM alignment records processed.\n",
      "40300000 SAM alignment records processed.\n",
      "40400000 SAM alignment records processed.\n",
      "40500000 SAM alignment records processed.\n",
      "40600000 SAM alignment records processed.\n",
      "40700000 SAM alignment records processed.\n",
      "40800000 SAM alignment records processed.\n",
      "40900000 SAM alignment records processed.\n",
      "41000000 SAM alignment records processed.\n",
      "41100000 SAM alignment records processed.\n",
      "41200000 SAM alignment records processed.\n",
      "41300000 SAM alignment records processed.\n",
      "41400000 SAM alignment records processed.\n",
      "41500000 SAM alignment records processed.\n",
      "41600000 SAM alignment records processed.\n",
      "41700000 SAM alignment records processed.\n",
      "41800000 SAM alignment records processed.\n",
      "41900000 SAM alignment records processed.\n",
      "42000000 SAM alignment records processed.\n",
      "42100000 SAM alignment records processed.\n",
      "42200000 SAM alignment records processed.\n",
      "42300000 SAM alignment records processed.\n",
      "42400000 SAM alignment records processed.\n",
      "42500000 SAM alignment records processed.\n",
      "42600000 SAM alignment records processed.\n",
      "42700000 SAM alignment records processed.\n",
      "42800000 SAM alignment records processed.\n",
      "42900000 SAM alignment records processed.\n",
      "43000000 SAM alignment records processed.\n",
      "43100000 SAM alignment records processed.\n",
      "43200000 SAM alignment records processed.\n",
      "43300000 SAM alignment records processed.\n",
      "43400000 SAM alignment records processed.\n",
      "43500000 SAM alignment records processed.\n",
      "43600000 SAM alignment records processed.\n",
      "43700000 SAM alignment records processed.\n",
      "43800000 SAM alignment records processed.\n",
      "43900000 SAM alignment records processed.\n",
      "44000000 SAM alignment records processed.\n",
      "44100000 SAM alignment records processed.\n",
      "44200000 SAM alignment records processed.\n",
      "44300000 SAM alignment records processed.\n",
      "44400000 SAM alignment records processed.\n",
      "44500000 SAM alignment records processed.\n",
      "44600000 SAM alignment records processed.\n",
      "44700000 SAM alignment records processed.\n",
      "44800000 SAM alignment records processed.\n",
      "44900000 SAM alignment records processed.\n",
      "45000000 SAM alignment records processed.\n",
      "45100000 SAM alignment records processed.\n",
      "45200000 SAM alignment records processed.\n",
      "45300000 SAM alignment records processed.\n",
      "45400000 SAM alignment records processed.\n",
      "45500000 SAM alignment records processed.\n",
      "45600000 SAM alignment records processed.\n",
      "45700000 SAM alignment records processed.\n",
      "45800000 SAM alignment records processed.\n",
      "45900000 SAM alignment records processed.\n",
      "46000000 SAM alignment records processed.\n",
      "46100000 SAM alignment records processed.\n",
      "46200000 SAM alignment records processed.\n",
      "46300000 SAM alignment records processed.\n",
      "46400000 SAM alignment records processed.\n",
      "46500000 SAM alignment records processed.\n",
      "46600000 SAM alignment records processed.\n",
      "46700000 SAM alignment records processed.\n",
      "46800000 SAM alignment records processed.\n",
      "46900000 SAM alignment records processed.\n",
      "47000000 SAM alignment records processed.\n",
      "47100000 SAM alignment records processed.\n",
      "47200000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "47300000 SAM alignment records processed.\n",
      "47400000 SAM alignment records processed.\n",
      "47500000 SAM alignment records processed.\n",
      "47600000 SAM alignment records processed.\n",
      "47700000 SAM alignment records processed.\n",
      "47800000 SAM alignment records processed.\n",
      "47900000 SAM alignment records processed.\n",
      "48000000 SAM alignment records processed.\n",
      "48100000 SAM alignment records processed.\n",
      "48200000 SAM alignment records processed.\n",
      "48300000 SAM alignment records processed.\n",
      "48400000 SAM alignment records processed.\n",
      "48500000 SAM alignment records processed.\n",
      "48600000 SAM alignment records processed.\n",
      "48700000 SAM alignment records processed.\n",
      "48800000 SAM alignment records processed.\n",
      "48900000 SAM alignment records processed.\n",
      "49000000 SAM alignment records processed.\n",
      "49100000 SAM alignment records processed.\n",
      "49200000 SAM alignment records processed.\n",
      "49300000 SAM alignment records processed.\n",
      "49400000 SAM alignment records processed.\n",
      "49500000 SAM alignment records processed.\n",
      "49600000 SAM alignment records processed.\n",
      "49700000 SAM alignment records processed.\n",
      "49800000 SAM alignment records processed.\n",
      "49900000 SAM alignment records processed.\n",
      "50000000 SAM alignment records processed.\n",
      "50100000 SAM alignment records processed.\n",
      "50200000 SAM alignment records processed.\n",
      "50300000 SAM alignment records processed.\n",
      "50400000 SAM alignment records processed.\n",
      "50500000 SAM alignment records processed.\n",
      "50600000 SAM alignment records processed.\n",
      "50700000 SAM alignment records processed.\n",
      "50800000 SAM alignment records processed.\n",
      "50900000 SAM alignment records processed.\n",
      "51000000 SAM alignment records processed.\n",
      "51100000 SAM alignment records processed.\n",
      "51200000 SAM alignment records processed.\n",
      "51300000 SAM alignment records processed.\n",
      "51400000 SAM alignment records processed.\n",
      "51500000 SAM alignment records processed.\n",
      "51600000 SAM alignment records processed.\n",
      "51700000 SAM alignment records processed.\n",
      "51800000 SAM alignment records processed.\n",
      "51900000 SAM alignment records processed.\n",
      "52000000 SAM alignment records processed.\n",
      "52100000 SAM alignment records processed.\n",
      "52200000 SAM alignment records processed.\n",
      "52300000 SAM alignment records processed.\n",
      "52400000 SAM alignment records processed.\n",
      "52500000 SAM alignment records processed.\n",
      "52600000 SAM alignment records processed.\n",
      "52700000 SAM alignment records processed.\n",
      "52800000 SAM alignment records processed.\n",
      "52900000 SAM alignment records processed.\n",
      "53000000 SAM alignment records processed.\n",
      "53100000 SAM alignment records processed.\n",
      "53200000 SAM alignment records processed.\n",
      "53300000 SAM alignment records processed.\n",
      "53400000 SAM alignment records processed.\n",
      "53500000 SAM alignment records processed.\n",
      "53600000 SAM alignment records processed.\n",
      "53700000 SAM alignment records processed.\n",
      "53800000 SAM alignment records processed.\n",
      "53900000 SAM alignment records processed.\n",
      "54000000 SAM alignment records processed.\n",
      "54100000 SAM alignment records processed.\n",
      "54200000 SAM alignment records processed.\n",
      "54300000 SAM alignment records processed.\n",
      "54400000 SAM alignment records processed.\n",
      "54500000 SAM alignment records processed.\n",
      "54600000 SAM alignment records processed.\n",
      "54700000 SAM alignment records processed.\n",
      "54800000 SAM alignment records processed.\n",
      "54900000 SAM alignment records processed.\n",
      "55000000 SAM alignment records processed.\n",
      "55100000 SAM alignment records processed.\n",
      "55200000 SAM alignment records processed.\n",
      "55300000 SAM alignment records processed.\n",
      "55400000 SAM alignment records processed.\n",
      "55500000 SAM alignment records processed.\n",
      "55600000 SAM alignment records processed.\n",
      "55700000 SAM alignment records processed.\n",
      "55800000 SAM alignment records processed.\n",
      "55900000 SAM alignment records processed.\n",
      "56000000 SAM alignment records processed.\n",
      "56100000 SAM alignment records processed.\n",
      "56200000 SAM alignment records processed.\n",
      "56300000 SAM alignment records processed.\n",
      "56400000 SAM alignment records processed.\n",
      "56500000 SAM alignment records processed.\n",
      "56600000 SAM alignment records processed.\n",
      "56700000 SAM alignment records processed.\n",
      "56800000 SAM alignment records processed.\n",
      "56900000 SAM alignment records processed.\n",
      "57000000 SAM alignment records processed.\n",
      "57100000 SAM alignment records processed.\n",
      "57200000 SAM alignment records processed.\n",
      "57300000 SAM alignment records processed.\n",
      "57400000 SAM alignment records processed.\n",
      "57500000 SAM alignment records processed.\n",
      "57600000 SAM alignment records processed.\n",
      "57700000 SAM alignment records processed.\n",
      "57800000 SAM alignment records processed.\n",
      "57900000 SAM alignment records processed.\n",
      "58000000 SAM alignment records processed.\n",
      "58100000 SAM alignment records processed.\n",
      "58200000 SAM alignment records processed.\n",
      "58300000 SAM alignment records processed.\n",
      "58400000 SAM alignment records processed.\n",
      "58500000 SAM alignment records processed.\n",
      "58600000 SAM alignment records processed.\n",
      "58700000 SAM alignment records processed.\n",
      "58800000 SAM alignment records processed.\n",
      "58900000 SAM alignment records processed.\n",
      "59000000 SAM alignment records processed.\n",
      "59100000 SAM alignment records processed.\n",
      "59200000 SAM alignment records processed.\n",
      "59300000 SAM alignment records processed.\n",
      "59400000 SAM alignment records processed.\n",
      "59500000 SAM alignment records processed.\n",
      "59600000 SAM alignment records processed.\n",
      "59700000 SAM alignment records processed.\n",
      "59800000 SAM alignment records processed.\n",
      "59900000 SAM alignment records processed.\n",
      "60000000 SAM alignment records processed.\n",
      "60100000 SAM alignment records processed.\n",
      "60200000 SAM alignment records processed.\n",
      "60300000 SAM alignment records processed.\n",
      "60400000 SAM alignment records processed.\n",
      "60500000 SAM alignment records processed.\n",
      "60600000 SAM alignment records processed.\n",
      "60700000 SAM alignment records processed.\n",
      "60800000 SAM alignment records processed.\n",
      "60900000 SAM alignment records processed.\n",
      "61000000 SAM alignment records processed.\n",
      "61100000 SAM alignment records processed.\n",
      "61200000 SAM alignment records processed.\n",
      "61300000 SAM alignment records processed.\n",
      "61400000 SAM alignment records processed.\n",
      "61500000 SAM alignment records processed.\n",
      "61600000 SAM alignment records processed.\n",
      "61700000 SAM alignment records processed.\n",
      "61800000 SAM alignment records processed.\n",
      "61900000 SAM alignment records processed.\n",
      "62000000 SAM alignment records processed.\n",
      "62100000 SAM alignment records processed.\n",
      "62200000 SAM alignment records processed.\n",
      "62300000 SAM alignment records processed.\n",
      "62400000 SAM alignment records processed.\n",
      "62500000 SAM alignment records processed.\n",
      "62600000 SAM alignment records processed.\n",
      "62700000 SAM alignment records processed.\n",
      "62800000 SAM alignment records processed.\n",
      "62900000 SAM alignment records processed.\n",
      "63000000 SAM alignment records processed.\n",
      "63100000 SAM alignment records processed.\n",
      "63200000 SAM alignment records processed.\n",
      "63300000 SAM alignment records processed.\n",
      "63400000 SAM alignment records processed.\n",
      "63500000 SAM alignment records processed.\n",
      "63600000 SAM alignment records processed.\n",
      "63700000 SAM alignment records processed.\n",
      "63800000 SAM alignment records processed.\n",
      "63900000 SAM alignment records processed.\n",
      "64000000 SAM alignment records processed.\n",
      "64100000 SAM alignment records processed.\n",
      "64200000 SAM alignment records processed.\n",
      "64300000 SAM alignment records processed.\n",
      "64400000 SAM alignment records processed.\n",
      "64500000 SAM alignment records processed.\n",
      "64600000 SAM alignment records processed.\n",
      "64700000 SAM alignment records processed.\n",
      "64800000 SAM alignment records processed.\n",
      "64900000 SAM alignment records processed.\n",
      "65000000 SAM alignment records processed.\n",
      "65100000 SAM alignment records processed.\n",
      "65200000 SAM alignment records processed.\n",
      "65300000 SAM alignment records processed.\n",
      "65400000 SAM alignment records processed.\n",
      "65500000 SAM alignment records processed.\n",
      "65600000 SAM alignment records processed.\n",
      "65700000 SAM alignment records processed.\n",
      "65800000 SAM alignment records processed.\n",
      "65900000 SAM alignment records processed.\n",
      "66000000 SAM alignment records processed.\n",
      "66100000 SAM alignment records processed.\n",
      "66200000 SAM alignment records processed.\n",
      "66300000 SAM alignment records processed.\n",
      "66400000 SAM alignment records processed.\n",
      "66500000 SAM alignment records processed.\n",
      "66600000 SAM alignment records processed.\n",
      "66700000 SAM alignment records processed.\n",
      "66800000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "66900000 SAM alignment records processed.\n",
      "67000000 SAM alignment records processed.\n",
      "67100000 SAM alignment records processed.\n",
      "67200000 SAM alignment records processed.\n",
      "67300000 SAM alignment records processed.\n",
      "67400000 SAM alignment records processed.\n",
      "67500000 SAM alignment records processed.\n",
      "67600000 SAM alignment records processed.\n",
      "67700000 SAM alignment records processed.\n",
      "67800000 SAM alignment records processed.\n",
      "67900000 SAM alignment records processed.\n",
      "68000000 SAM alignment records processed.\n",
      "68100000 SAM alignment records processed.\n",
      "68200000 SAM alignment records processed.\n",
      "68300000 SAM alignment records processed.\n",
      "68400000 SAM alignment records processed.\n",
      "68500000 SAM alignment records processed.\n",
      "68600000 SAM alignment records processed.\n",
      "68700000 SAM alignment records processed.\n",
      "68800000 SAM alignment records processed.\n",
      "68900000 SAM alignment records processed.\n",
      "69000000 SAM alignment records processed.\n",
      "69100000 SAM alignment records processed.\n",
      "69200000 SAM alignment records processed.\n",
      "69300000 SAM alignment records processed.\n",
      "69400000 SAM alignment records processed.\n",
      "69500000 SAM alignment records processed.\n",
      "69600000 SAM alignment records processed.\n",
      "69700000 SAM alignment records processed.\n",
      "69800000 SAM alignment records processed.\n",
      "69900000 SAM alignment records processed.\n",
      "70000000 SAM alignment records processed.\n",
      "70100000 SAM alignment records processed.\n",
      "70200000 SAM alignment records processed.\n",
      "70300000 SAM alignment records processed.\n",
      "70400000 SAM alignment records processed.\n",
      "70500000 SAM alignment records processed.\n",
      "70600000 SAM alignment records processed.\n",
      "70700000 SAM alignment records processed.\n",
      "70800000 SAM alignment records processed.\n",
      "70900000 SAM alignment records processed.\n",
      "71000000 SAM alignment records processed.\n",
      "71100000 SAM alignment records processed.\n",
      "71200000 SAM alignment records processed.\n",
      "71300000 SAM alignment records processed.\n",
      "71400000 SAM alignment records processed.\n",
      "71500000 SAM alignment records processed.\n",
      "71600000 SAM alignment records processed.\n",
      "71700000 SAM alignment records processed.\n",
      "71800000 SAM alignment records processed.\n",
      "71900000 SAM alignment records processed.\n",
      "72000000 SAM alignment records processed.\n",
      "72100000 SAM alignment records processed.\n",
      "72200000 SAM alignment records processed.\n",
      "72300000 SAM alignment records processed.\n",
      "72374357 BAM alignments  processed.\n",
      "HTSeq is counting SRR6145435.sorted.bam.\n",
      "100000 GFF lines processed.\n",
      "200000 GFF lines processed.\n",
      "300000 GFF lines processed.\n",
      "400000 GFF lines processed.\n",
      "500000 GFF lines processed.\n",
      "600000 GFF lines processed.\n",
      "700000 GFF lines processed.\n",
      "800000 GFF lines processed.\n",
      "900000 GFF lines processed.\n",
      "1000000 GFF lines processed.\n",
      "1100000 GFF lines processed.\n",
      "1200000 GFF lines processed.\n",
      "1300000 GFF lines processed.\n",
      "1400000 GFF lines processed.\n",
      "1500000 GFF lines processed.\n",
      "1600000 GFF lines processed.\n",
      "1700000 GFF lines processed.\n",
      "1800000 GFF lines processed.\n",
      "1900000 GFF lines processed.\n",
      "2000000 GFF lines processed.\n",
      "2100000 GFF lines processed.\n",
      "2200000 GFF lines processed.\n",
      "2300000 GFF lines processed.\n",
      "2400000 GFF lines processed.\n",
      "2500000 GFF lines processed.\n",
      "2600000 GFF lines processed.\n",
      "2700000 GFF lines processed.\n",
      "2800000 GFF lines processed.\n",
      "2900000 GFF lines processed.\n",
      "2908224 GFF lines processed.\n",
      "100000 SAM alignment records processed.\n",
      "200000 SAM alignment records processed.\n",
      "300000 SAM alignment records processed.\n",
      "400000 SAM alignment records processed.\n",
      "500000 SAM alignment records processed.\n",
      "600000 SAM alignment records processed.\n",
      "700000 SAM alignment records processed.\n",
      "800000 SAM alignment records processed.\n",
      "900000 SAM alignment records processed.\n",
      "1000000 SAM alignment records processed.\n",
      "1100000 SAM alignment records processed.\n",
      "1200000 SAM alignment records processed.\n",
      "1300000 SAM alignment records processed.\n",
      "1400000 SAM alignment records processed.\n",
      "1500000 SAM alignment records processed.\n",
      "1600000 SAM alignment records processed.\n",
      "1700000 SAM alignment records processed.\n",
      "1800000 SAM alignment records processed.\n",
      "1900000 SAM alignment records processed.\n",
      "2000000 SAM alignment records processed.\n",
      "2100000 SAM alignment records processed.\n",
      "2200000 SAM alignment records processed.\n",
      "2300000 SAM alignment records processed.\n",
      "2400000 SAM alignment records processed.\n",
      "2500000 SAM alignment records processed.\n",
      "2600000 SAM alignment records processed.\n",
      "2700000 SAM alignment records processed.\n",
      "2800000 SAM alignment records processed.\n",
      "2900000 SAM alignment records processed.\n",
      "3000000 SAM alignment records processed.\n",
      "3100000 SAM alignment records processed.\n",
      "3200000 SAM alignment records processed.\n",
      "3300000 SAM alignment records processed.\n",
      "3400000 SAM alignment records processed.\n",
      "3500000 SAM alignment records processed.\n",
      "3600000 SAM alignment records processed.\n",
      "3700000 SAM alignment records processed.\n",
      "3800000 SAM alignment records processed.\n",
      "3900000 SAM alignment records processed.\n",
      "4000000 SAM alignment records processed.\n",
      "4100000 SAM alignment records processed.\n",
      "4200000 SAM alignment records processed.\n",
      "4300000 SAM alignment records processed.\n",
      "4400000 SAM alignment records processed.\n",
      "4500000 SAM alignment records processed.\n",
      "4600000 SAM alignment records processed.\n",
      "4700000 SAM alignment records processed.\n",
      "4800000 SAM alignment records processed.\n",
      "4900000 SAM alignment records processed.\n",
      "5000000 SAM alignment records processed.\n",
      "5100000 SAM alignment records processed.\n",
      "5200000 SAM alignment records processed.\n",
      "5300000 SAM alignment records processed.\n",
      "5400000 SAM alignment records processed.\n",
      "5500000 SAM alignment records processed.\n",
      "5600000 SAM alignment records processed.\n",
      "5700000 SAM alignment records processed.\n",
      "5800000 SAM alignment records processed.\n",
      "5900000 SAM alignment records processed.\n",
      "6000000 SAM alignment records processed.\n",
      "6100000 SAM alignment records processed.\n",
      "6200000 SAM alignment records processed.\n",
      "6300000 SAM alignment records processed.\n",
      "6400000 SAM alignment records processed.\n",
      "6500000 SAM alignment records processed.\n",
      "6600000 SAM alignment records processed.\n",
      "6700000 SAM alignment records processed.\n",
      "6800000 SAM alignment records processed.\n",
      "6900000 SAM alignment records processed.\n",
      "7000000 SAM alignment records processed.\n",
      "7100000 SAM alignment records processed.\n",
      "7200000 SAM alignment records processed.\n",
      "7300000 SAM alignment records processed.\n",
      "7400000 SAM alignment records processed.\n",
      "7500000 SAM alignment records processed.\n",
      "7600000 SAM alignment records processed.\n",
      "7700000 SAM alignment records processed.\n",
      "7800000 SAM alignment records processed.\n",
      "7900000 SAM alignment records processed.\n",
      "8000000 SAM alignment records processed.\n",
      "8100000 SAM alignment records processed.\n",
      "8200000 SAM alignment records processed.\n",
      "8300000 SAM alignment records processed.\n",
      "8400000 SAM alignment records processed.\n",
      "8500000 SAM alignment records processed.\n",
      "8600000 SAM alignment records processed.\n",
      "8700000 SAM alignment records processed.\n",
      "8800000 SAM alignment records processed.\n",
      "8900000 SAM alignment records processed.\n",
      "9000000 SAM alignment records processed.\n",
      "9100000 SAM alignment records processed.\n",
      "9200000 SAM alignment records processed.\n",
      "9300000 SAM alignment records processed.\n",
      "9400000 SAM alignment records processed.\n",
      "9500000 SAM alignment records processed.\n",
      "9600000 SAM alignment records processed.\n",
      "9700000 SAM alignment records processed.\n",
      "9800000 SAM alignment records processed.\n",
      "9900000 SAM alignment records processed.\n",
      "10000000 SAM alignment records processed.\n",
      "10100000 SAM alignment records processed.\n",
      "10200000 SAM alignment records processed.\n",
      "10300000 SAM alignment records processed.\n",
      "10400000 SAM alignment records processed.\n",
      "10500000 SAM alignment records processed.\n",
      "10600000 SAM alignment records processed.\n",
      "10700000 SAM alignment records processed.\n",
      "10800000 SAM alignment records processed.\n",
      "10900000 SAM alignment records processed.\n",
      "11000000 SAM alignment records processed.\n",
      "11100000 SAM alignment records processed.\n",
      "11200000 SAM alignment records processed.\n",
      "11300000 SAM alignment records processed.\n",
      "11400000 SAM alignment records processed.\n",
      "11500000 SAM alignment records processed.\n",
      "11600000 SAM alignment records processed.\n",
      "11700000 SAM alignment records processed.\n",
      "11800000 SAM alignment records processed.\n",
      "11900000 SAM alignment records processed.\n",
      "12000000 SAM alignment records processed.\n",
      "12100000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12200000 SAM alignment records processed.\n",
      "12300000 SAM alignment records processed.\n",
      "12400000 SAM alignment records processed.\n",
      "12500000 SAM alignment records processed.\n",
      "12600000 SAM alignment records processed.\n",
      "12700000 SAM alignment records processed.\n",
      "12800000 SAM alignment records processed.\n",
      "12900000 SAM alignment records processed.\n",
      "13000000 SAM alignment records processed.\n",
      "13100000 SAM alignment records processed.\n",
      "13200000 SAM alignment records processed.\n",
      "13300000 SAM alignment records processed.\n",
      "13400000 SAM alignment records processed.\n",
      "13500000 SAM alignment records processed.\n",
      "13600000 SAM alignment records processed.\n",
      "13700000 SAM alignment records processed.\n",
      "13800000 SAM alignment records processed.\n",
      "13900000 SAM alignment records processed.\n",
      "14000000 SAM alignment records processed.\n",
      "14100000 SAM alignment records processed.\n",
      "14200000 SAM alignment records processed.\n",
      "14300000 SAM alignment records processed.\n",
      "14400000 SAM alignment records processed.\n",
      "14500000 SAM alignment records processed.\n",
      "14600000 SAM alignment records processed.\n",
      "14700000 SAM alignment records processed.\n",
      "14800000 SAM alignment records processed.\n",
      "14900000 SAM alignment records processed.\n",
      "15000000 SAM alignment records processed.\n",
      "15100000 SAM alignment records processed.\n",
      "15200000 SAM alignment records processed.\n",
      "15300000 SAM alignment records processed.\n",
      "15400000 SAM alignment records processed.\n",
      "15500000 SAM alignment records processed.\n",
      "15600000 SAM alignment records processed.\n",
      "15700000 SAM alignment records processed.\n",
      "15800000 SAM alignment records processed.\n",
      "15900000 SAM alignment records processed.\n",
      "16000000 SAM alignment records processed.\n",
      "16100000 SAM alignment records processed.\n",
      "16200000 SAM alignment records processed.\n",
      "16300000 SAM alignment records processed.\n",
      "16400000 SAM alignment records processed.\n",
      "16500000 SAM alignment records processed.\n",
      "16600000 SAM alignment records processed.\n",
      "16700000 SAM alignment records processed.\n",
      "16800000 SAM alignment records processed.\n",
      "16900000 SAM alignment records processed.\n",
      "17000000 SAM alignment records processed.\n",
      "17100000 SAM alignment records processed.\n",
      "17200000 SAM alignment records processed.\n",
      "17300000 SAM alignment records processed.\n",
      "17400000 SAM alignment records processed.\n",
      "17500000 SAM alignment records processed.\n",
      "17600000 SAM alignment records processed.\n",
      "17700000 SAM alignment records processed.\n",
      "17800000 SAM alignment records processed.\n",
      "17900000 SAM alignment records processed.\n",
      "18000000 SAM alignment records processed.\n",
      "18100000 SAM alignment records processed.\n",
      "18200000 SAM alignment records processed.\n",
      "18300000 SAM alignment records processed.\n",
      "18400000 SAM alignment records processed.\n",
      "18500000 SAM alignment records processed.\n",
      "18600000 SAM alignment records processed.\n",
      "18700000 SAM alignment records processed.\n",
      "18800000 SAM alignment records processed.\n",
      "18900000 SAM alignment records processed.\n",
      "19000000 SAM alignment records processed.\n",
      "19100000 SAM alignment records processed.\n",
      "19200000 SAM alignment records processed.\n",
      "19300000 SAM alignment records processed.\n",
      "19400000 SAM alignment records processed.\n",
      "19500000 SAM alignment records processed.\n",
      "19600000 SAM alignment records processed.\n",
      "19700000 SAM alignment records processed.\n",
      "19800000 SAM alignment records processed.\n",
      "19900000 SAM alignment records processed.\n",
      "20000000 SAM alignment records processed.\n",
      "20100000 SAM alignment records processed.\n",
      "20200000 SAM alignment records processed.\n",
      "20300000 SAM alignment records processed.\n",
      "20400000 SAM alignment records processed.\n",
      "20500000 SAM alignment records processed.\n",
      "20600000 SAM alignment records processed.\n",
      "20700000 SAM alignment records processed.\n",
      "20800000 SAM alignment records processed.\n",
      "20900000 SAM alignment records processed.\n",
      "21000000 SAM alignment records processed.\n",
      "21100000 SAM alignment records processed.\n",
      "21200000 SAM alignment records processed.\n",
      "21300000 SAM alignment records processed.\n",
      "21400000 SAM alignment records processed.\n",
      "21500000 SAM alignment records processed.\n",
      "21600000 SAM alignment records processed.\n",
      "21700000 SAM alignment records processed.\n",
      "21800000 SAM alignment records processed.\n",
      "21900000 SAM alignment records processed.\n",
      "22000000 SAM alignment records processed.\n",
      "22100000 SAM alignment records processed.\n",
      "22200000 SAM alignment records processed.\n",
      "22300000 SAM alignment records processed.\n",
      "22400000 SAM alignment records processed.\n",
      "22500000 SAM alignment records processed.\n",
      "22600000 SAM alignment records processed.\n",
      "22700000 SAM alignment records processed.\n",
      "22800000 SAM alignment records processed.\n",
      "22900000 SAM alignment records processed.\n",
      "23000000 SAM alignment records processed.\n",
      "23100000 SAM alignment records processed.\n",
      "23200000 SAM alignment records processed.\n",
      "23300000 SAM alignment records processed.\n",
      "23400000 SAM alignment records processed.\n",
      "23500000 SAM alignment records processed.\n",
      "23600000 SAM alignment records processed.\n",
      "23700000 SAM alignment records processed.\n",
      "23800000 SAM alignment records processed.\n",
      "23900000 SAM alignment records processed.\n",
      "24000000 SAM alignment records processed.\n",
      "24100000 SAM alignment records processed.\n",
      "24200000 SAM alignment records processed.\n",
      "24300000 SAM alignment records processed.\n",
      "24400000 SAM alignment records processed.\n",
      "24500000 SAM alignment records processed.\n",
      "24600000 SAM alignment records processed.\n",
      "24700000 SAM alignment records processed.\n",
      "24800000 SAM alignment records processed.\n",
      "24900000 SAM alignment records processed.\n",
      "25000000 SAM alignment records processed.\n",
      "25100000 SAM alignment records processed.\n",
      "25200000 SAM alignment records processed.\n",
      "25300000 SAM alignment records processed.\n",
      "25400000 SAM alignment records processed.\n",
      "25500000 SAM alignment records processed.\n",
      "25600000 SAM alignment records processed.\n",
      "25700000 SAM alignment records processed.\n",
      "25800000 SAM alignment records processed.\n",
      "25900000 SAM alignment records processed.\n",
      "26000000 SAM alignment records processed.\n",
      "26100000 SAM alignment records processed.\n",
      "26200000 SAM alignment records processed.\n",
      "26300000 SAM alignment records processed.\n",
      "26400000 SAM alignment records processed.\n",
      "26500000 SAM alignment records processed.\n",
      "26600000 SAM alignment records processed.\n",
      "26700000 SAM alignment records processed.\n",
      "26800000 SAM alignment records processed.\n",
      "26900000 SAM alignment records processed.\n",
      "27000000 SAM alignment records processed.\n",
      "27100000 SAM alignment records processed.\n",
      "27200000 SAM alignment records processed.\n",
      "27300000 SAM alignment records processed.\n",
      "27400000 SAM alignment records processed.\n",
      "27500000 SAM alignment records processed.\n",
      "27600000 SAM alignment records processed.\n",
      "27700000 SAM alignment records processed.\n",
      "27800000 SAM alignment records processed.\n",
      "27900000 SAM alignment records processed.\n",
      "28000000 SAM alignment records processed.\n",
      "28100000 SAM alignment records processed.\n",
      "28200000 SAM alignment records processed.\n",
      "28300000 SAM alignment records processed.\n",
      "28400000 SAM alignment records processed.\n",
      "28500000 SAM alignment records processed.\n",
      "28600000 SAM alignment records processed.\n",
      "28700000 SAM alignment records processed.\n",
      "28800000 SAM alignment records processed.\n",
      "28900000 SAM alignment records processed.\n",
      "29000000 SAM alignment records processed.\n",
      "29100000 SAM alignment records processed.\n",
      "29200000 SAM alignment records processed.\n",
      "29300000 SAM alignment records processed.\n",
      "29400000 SAM alignment records processed.\n",
      "29500000 SAM alignment records processed.\n",
      "29600000 SAM alignment records processed.\n",
      "29700000 SAM alignment records processed.\n",
      "29800000 SAM alignment records processed.\n",
      "29900000 SAM alignment records processed.\n",
      "30000000 SAM alignment records processed.\n",
      "30100000 SAM alignment records processed.\n",
      "30200000 SAM alignment records processed.\n",
      "30300000 SAM alignment records processed.\n",
      "30400000 SAM alignment records processed.\n",
      "30500000 SAM alignment records processed.\n",
      "30600000 SAM alignment records processed.\n",
      "30700000 SAM alignment records processed.\n",
      "30800000 SAM alignment records processed.\n",
      "30900000 SAM alignment records processed.\n",
      "31000000 SAM alignment records processed.\n",
      "31100000 SAM alignment records processed.\n",
      "31200000 SAM alignment records processed.\n",
      "31300000 SAM alignment records processed.\n",
      "31400000 SAM alignment records processed.\n",
      "31500000 SAM alignment records processed.\n",
      "31600000 SAM alignment records processed.\n",
      "31700000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31800000 SAM alignment records processed.\n",
      "31900000 SAM alignment records processed.\n",
      "32000000 SAM alignment records processed.\n",
      "32100000 SAM alignment records processed.\n",
      "32200000 SAM alignment records processed.\n",
      "32300000 SAM alignment records processed.\n",
      "32400000 SAM alignment records processed.\n",
      "32500000 SAM alignment records processed.\n",
      "32600000 SAM alignment records processed.\n",
      "32700000 SAM alignment records processed.\n",
      "32800000 SAM alignment records processed.\n",
      "32900000 SAM alignment records processed.\n",
      "33000000 SAM alignment records processed.\n",
      "33100000 SAM alignment records processed.\n",
      "33200000 SAM alignment records processed.\n",
      "33300000 SAM alignment records processed.\n",
      "33400000 SAM alignment records processed.\n",
      "33500000 SAM alignment records processed.\n",
      "33600000 SAM alignment records processed.\n",
      "33700000 SAM alignment records processed.\n",
      "33800000 SAM alignment records processed.\n",
      "33900000 SAM alignment records processed.\n",
      "34000000 SAM alignment records processed.\n",
      "34100000 SAM alignment records processed.\n",
      "34200000 SAM alignment records processed.\n",
      "34300000 SAM alignment records processed.\n",
      "34400000 SAM alignment records processed.\n",
      "34500000 SAM alignment records processed.\n",
      "34600000 SAM alignment records processed.\n",
      "34700000 SAM alignment records processed.\n",
      "34800000 SAM alignment records processed.\n",
      "34900000 SAM alignment records processed.\n",
      "35000000 SAM alignment records processed.\n",
      "35100000 SAM alignment records processed.\n",
      "35200000 SAM alignment records processed.\n",
      "35300000 SAM alignment records processed.\n",
      "35400000 SAM alignment records processed.\n",
      "35500000 SAM alignment records processed.\n",
      "35600000 SAM alignment records processed.\n",
      "35700000 SAM alignment records processed.\n",
      "35800000 SAM alignment records processed.\n",
      "35900000 SAM alignment records processed.\n",
      "36000000 SAM alignment records processed.\n",
      "36100000 SAM alignment records processed.\n",
      "36200000 SAM alignment records processed.\n",
      "36300000 SAM alignment records processed.\n",
      "36400000 SAM alignment records processed.\n",
      "36500000 SAM alignment records processed.\n",
      "36600000 SAM alignment records processed.\n",
      "36700000 SAM alignment records processed.\n",
      "36800000 SAM alignment records processed.\n",
      "36900000 SAM alignment records processed.\n",
      "37000000 SAM alignment records processed.\n",
      "37100000 SAM alignment records processed.\n",
      "37200000 SAM alignment records processed.\n",
      "37300000 SAM alignment records processed.\n",
      "37400000 SAM alignment records processed.\n",
      "37500000 SAM alignment records processed.\n",
      "37600000 SAM alignment records processed.\n",
      "37700000 SAM alignment records processed.\n",
      "37800000 SAM alignment records processed.\n",
      "37900000 SAM alignment records processed.\n",
      "38000000 SAM alignment records processed.\n",
      "38100000 SAM alignment records processed.\n",
      "38200000 SAM alignment records processed.\n",
      "38300000 SAM alignment records processed.\n",
      "38400000 SAM alignment records processed.\n",
      "38500000 SAM alignment records processed.\n",
      "38600000 SAM alignment records processed.\n",
      "38700000 SAM alignment records processed.\n",
      "38800000 SAM alignment records processed.\n",
      "38900000 SAM alignment records processed.\n",
      "39000000 SAM alignment records processed.\n",
      "39100000 SAM alignment records processed.\n",
      "39200000 SAM alignment records processed.\n",
      "39300000 SAM alignment records processed.\n",
      "39400000 SAM alignment records processed.\n",
      "39500000 SAM alignment records processed.\n",
      "39600000 SAM alignment records processed.\n",
      "39700000 SAM alignment records processed.\n",
      "39800000 SAM alignment records processed.\n",
      "39900000 SAM alignment records processed.\n",
      "40000000 SAM alignment records processed.\n",
      "40100000 SAM alignment records processed.\n",
      "40200000 SAM alignment records processed.\n",
      "40300000 SAM alignment records processed.\n",
      "40400000 SAM alignment records processed.\n",
      "40500000 SAM alignment records processed.\n",
      "40600000 SAM alignment records processed.\n",
      "40700000 SAM alignment records processed.\n",
      "40800000 SAM alignment records processed.\n",
      "40900000 SAM alignment records processed.\n",
      "41000000 SAM alignment records processed.\n",
      "41100000 SAM alignment records processed.\n",
      "41200000 SAM alignment records processed.\n",
      "41300000 SAM alignment records processed.\n",
      "41400000 SAM alignment records processed.\n",
      "41500000 SAM alignment records processed.\n",
      "41600000 SAM alignment records processed.\n",
      "41700000 SAM alignment records processed.\n",
      "41800000 SAM alignment records processed.\n",
      "41900000 SAM alignment records processed.\n",
      "42000000 SAM alignment records processed.\n",
      "42100000 SAM alignment records processed.\n",
      "42200000 SAM alignment records processed.\n",
      "42300000 SAM alignment records processed.\n",
      "42400000 SAM alignment records processed.\n",
      "42500000 SAM alignment records processed.\n",
      "42600000 SAM alignment records processed.\n",
      "42700000 SAM alignment records processed.\n",
      "42800000 SAM alignment records processed.\n",
      "42900000 SAM alignment records processed.\n",
      "43000000 SAM alignment records processed.\n",
      "43100000 SAM alignment records processed.\n",
      "43200000 SAM alignment records processed.\n",
      "43300000 SAM alignment records processed.\n",
      "43400000 SAM alignment records processed.\n",
      "43500000 SAM alignment records processed.\n",
      "43600000 SAM alignment records processed.\n",
      "43700000 SAM alignment records processed.\n",
      "43800000 SAM alignment records processed.\n",
      "43900000 SAM alignment records processed.\n",
      "44000000 SAM alignment records processed.\n",
      "44100000 SAM alignment records processed.\n",
      "44200000 SAM alignment records processed.\n",
      "44300000 SAM alignment records processed.\n",
      "44400000 SAM alignment records processed.\n",
      "44500000 SAM alignment records processed.\n",
      "44600000 SAM alignment records processed.\n",
      "44700000 SAM alignment records processed.\n",
      "44800000 SAM alignment records processed.\n",
      "44900000 SAM alignment records processed.\n",
      "45000000 SAM alignment records processed.\n",
      "45100000 SAM alignment records processed.\n",
      "45200000 SAM alignment records processed.\n",
      "45300000 SAM alignment records processed.\n",
      "45400000 SAM alignment records processed.\n",
      "45500000 SAM alignment records processed.\n",
      "45600000 SAM alignment records processed.\n",
      "45700000 SAM alignment records processed.\n",
      "45800000 SAM alignment records processed.\n",
      "45900000 SAM alignment records processed.\n",
      "46000000 SAM alignment records processed.\n",
      "46100000 SAM alignment records processed.\n",
      "46200000 SAM alignment records processed.\n",
      "46300000 SAM alignment records processed.\n",
      "46400000 SAM alignment records processed.\n",
      "46500000 SAM alignment records processed.\n",
      "46600000 SAM alignment records processed.\n",
      "46700000 SAM alignment records processed.\n",
      "46800000 SAM alignment records processed.\n",
      "46900000 SAM alignment records processed.\n",
      "47000000 SAM alignment records processed.\n",
      "47100000 SAM alignment records processed.\n",
      "47200000 SAM alignment records processed.\n",
      "47300000 SAM alignment records processed.\n",
      "47400000 SAM alignment records processed.\n",
      "47500000 SAM alignment records processed.\n",
      "47600000 SAM alignment records processed.\n",
      "47700000 SAM alignment records processed.\n",
      "47800000 SAM alignment records processed.\n",
      "47900000 SAM alignment records processed.\n",
      "48000000 SAM alignment records processed.\n",
      "48100000 SAM alignment records processed.\n",
      "48200000 SAM alignment records processed.\n",
      "48300000 SAM alignment records processed.\n",
      "48400000 SAM alignment records processed.\n",
      "48500000 SAM alignment records processed.\n",
      "48600000 SAM alignment records processed.\n",
      "48700000 SAM alignment records processed.\n",
      "48800000 SAM alignment records processed.\n",
      "48900000 SAM alignment records processed.\n",
      "49000000 SAM alignment records processed.\n",
      "49100000 SAM alignment records processed.\n",
      "49200000 SAM alignment records processed.\n",
      "49300000 SAM alignment records processed.\n",
      "49400000 SAM alignment records processed.\n",
      "49500000 SAM alignment records processed.\n",
      "49600000 SAM alignment records processed.\n",
      "49700000 SAM alignment records processed.\n",
      "49800000 SAM alignment records processed.\n",
      "49900000 SAM alignment records processed.\n",
      "50000000 SAM alignment records processed.\n",
      "50100000 SAM alignment records processed.\n",
      "50200000 SAM alignment records processed.\n",
      "50300000 SAM alignment records processed.\n",
      "50400000 SAM alignment records processed.\n",
      "50500000 SAM alignment records processed.\n",
      "50600000 SAM alignment records processed.\n",
      "50673229 BAM alignments  processed.\n",
      "HTSeq is counting SRR6145436.sorted.bam.\n",
      "100000 GFF lines processed.\n",
      "200000 GFF lines processed.\n",
      "300000 GFF lines processed.\n",
      "400000 GFF lines processed.\n",
      "500000 GFF lines processed.\n",
      "600000 GFF lines processed.\n",
      "700000 GFF lines processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "800000 GFF lines processed.\n",
      "900000 GFF lines processed.\n",
      "1000000 GFF lines processed.\n",
      "1100000 GFF lines processed.\n",
      "1200000 GFF lines processed.\n",
      "1300000 GFF lines processed.\n",
      "1400000 GFF lines processed.\n",
      "1500000 GFF lines processed.\n",
      "1600000 GFF lines processed.\n",
      "1700000 GFF lines processed.\n",
      "1800000 GFF lines processed.\n",
      "1900000 GFF lines processed.\n",
      "2000000 GFF lines processed.\n",
      "2100000 GFF lines processed.\n",
      "2200000 GFF lines processed.\n",
      "2300000 GFF lines processed.\n",
      "2400000 GFF lines processed.\n",
      "2500000 GFF lines processed.\n",
      "2600000 GFF lines processed.\n",
      "2700000 GFF lines processed.\n",
      "2800000 GFF lines processed.\n",
      "2900000 GFF lines processed.\n",
      "2908224 GFF lines processed.\n",
      "100000 SAM alignment records processed.\n",
      "200000 SAM alignment records processed.\n",
      "300000 SAM alignment records processed.\n",
      "400000 SAM alignment records processed.\n",
      "500000 SAM alignment records processed.\n",
      "600000 SAM alignment records processed.\n",
      "700000 SAM alignment records processed.\n",
      "800000 SAM alignment records processed.\n",
      "900000 SAM alignment records processed.\n",
      "1000000 SAM alignment records processed.\n",
      "1100000 SAM alignment records processed.\n",
      "1200000 SAM alignment records processed.\n",
      "1300000 SAM alignment records processed.\n",
      "1400000 SAM alignment records processed.\n",
      "1500000 SAM alignment records processed.\n",
      "1600000 SAM alignment records processed.\n",
      "1700000 SAM alignment records processed.\n",
      "1800000 SAM alignment records processed.\n",
      "1900000 SAM alignment records processed.\n",
      "2000000 SAM alignment records processed.\n",
      "2100000 SAM alignment records processed.\n",
      "2200000 SAM alignment records processed.\n",
      "2300000 SAM alignment records processed.\n",
      "2400000 SAM alignment records processed.\n",
      "2500000 SAM alignment records processed.\n",
      "2600000 SAM alignment records processed.\n",
      "2700000 SAM alignment records processed.\n",
      "2800000 SAM alignment records processed.\n",
      "2900000 SAM alignment records processed.\n",
      "3000000 SAM alignment records processed.\n",
      "3100000 SAM alignment records processed.\n",
      "3200000 SAM alignment records processed.\n",
      "3300000 SAM alignment records processed.\n",
      "3400000 SAM alignment records processed.\n",
      "3500000 SAM alignment records processed.\n",
      "3600000 SAM alignment records processed.\n",
      "3700000 SAM alignment records processed.\n",
      "3800000 SAM alignment records processed.\n",
      "3900000 SAM alignment records processed.\n",
      "4000000 SAM alignment records processed.\n",
      "4100000 SAM alignment records processed.\n",
      "4200000 SAM alignment records processed.\n",
      "4300000 SAM alignment records processed.\n",
      "4400000 SAM alignment records processed.\n",
      "4500000 SAM alignment records processed.\n",
      "4600000 SAM alignment records processed.\n",
      "4700000 SAM alignment records processed.\n",
      "4800000 SAM alignment records processed.\n",
      "4900000 SAM alignment records processed.\n",
      "5000000 SAM alignment records processed.\n",
      "5100000 SAM alignment records processed.\n",
      "5200000 SAM alignment records processed.\n",
      "5300000 SAM alignment records processed.\n",
      "5400000 SAM alignment records processed.\n",
      "5500000 SAM alignment records processed.\n",
      "5600000 SAM alignment records processed.\n",
      "5700000 SAM alignment records processed.\n",
      "5800000 SAM alignment records processed.\n",
      "5900000 SAM alignment records processed.\n",
      "6000000 SAM alignment records processed.\n",
      "6100000 SAM alignment records processed.\n",
      "6200000 SAM alignment records processed.\n",
      "6300000 SAM alignment records processed.\n",
      "6400000 SAM alignment records processed.\n",
      "6500000 SAM alignment records processed.\n",
      "6600000 SAM alignment records processed.\n",
      "6700000 SAM alignment records processed.\n",
      "6800000 SAM alignment records processed.\n",
      "6900000 SAM alignment records processed.\n",
      "7000000 SAM alignment records processed.\n",
      "7100000 SAM alignment records processed.\n",
      "7200000 SAM alignment records processed.\n",
      "7300000 SAM alignment records processed.\n",
      "7400000 SAM alignment records processed.\n",
      "7500000 SAM alignment records processed.\n",
      "7600000 SAM alignment records processed.\n",
      "7700000 SAM alignment records processed.\n",
      "7800000 SAM alignment records processed.\n",
      "7900000 SAM alignment records processed.\n",
      "8000000 SAM alignment records processed.\n",
      "8100000 SAM alignment records processed.\n",
      "8200000 SAM alignment records processed.\n",
      "8300000 SAM alignment records processed.\n",
      "8400000 SAM alignment records processed.\n",
      "8500000 SAM alignment records processed.\n",
      "8600000 SAM alignment records processed.\n",
      "8700000 SAM alignment records processed.\n",
      "8800000 SAM alignment records processed.\n",
      "8900000 SAM alignment records processed.\n",
      "9000000 SAM alignment records processed.\n",
      "9100000 SAM alignment records processed.\n",
      "9200000 SAM alignment records processed.\n",
      "9300000 SAM alignment records processed.\n",
      "9400000 SAM alignment records processed.\n",
      "9500000 SAM alignment records processed.\n",
      "9600000 SAM alignment records processed.\n",
      "9700000 SAM alignment records processed.\n",
      "9800000 SAM alignment records processed.\n",
      "9900000 SAM alignment records processed.\n",
      "10000000 SAM alignment records processed.\n",
      "10100000 SAM alignment records processed.\n",
      "10200000 SAM alignment records processed.\n",
      "10300000 SAM alignment records processed.\n",
      "10400000 SAM alignment records processed.\n",
      "10500000 SAM alignment records processed.\n",
      "10600000 SAM alignment records processed.\n",
      "10700000 SAM alignment records processed.\n",
      "10800000 SAM alignment records processed.\n",
      "10900000 SAM alignment records processed.\n",
      "11000000 SAM alignment records processed.\n",
      "11100000 SAM alignment records processed.\n",
      "11200000 SAM alignment records processed.\n",
      "11300000 SAM alignment records processed.\n",
      "11400000 SAM alignment records processed.\n",
      "11500000 SAM alignment records processed.\n",
      "11600000 SAM alignment records processed.\n",
      "11700000 SAM alignment records processed.\n",
      "11800000 SAM alignment records processed.\n",
      "11900000 SAM alignment records processed.\n",
      "12000000 SAM alignment records processed.\n",
      "12100000 SAM alignment records processed.\n",
      "12200000 SAM alignment records processed.\n",
      "12300000 SAM alignment records processed.\n",
      "12400000 SAM alignment records processed.\n",
      "12500000 SAM alignment records processed.\n",
      "12600000 SAM alignment records processed.\n",
      "12700000 SAM alignment records processed.\n",
      "12800000 SAM alignment records processed.\n",
      "12900000 SAM alignment records processed.\n",
      "13000000 SAM alignment records processed.\n",
      "13100000 SAM alignment records processed.\n",
      "13200000 SAM alignment records processed.\n",
      "13300000 SAM alignment records processed.\n",
      "13400000 SAM alignment records processed.\n",
      "13500000 SAM alignment records processed.\n",
      "13600000 SAM alignment records processed.\n",
      "13700000 SAM alignment records processed.\n",
      "13800000 SAM alignment records processed.\n",
      "13900000 SAM alignment records processed.\n",
      "14000000 SAM alignment records processed.\n",
      "14100000 SAM alignment records processed.\n",
      "14200000 SAM alignment records processed.\n",
      "14300000 SAM alignment records processed.\n",
      "14400000 SAM alignment records processed.\n",
      "14500000 SAM alignment records processed.\n",
      "14600000 SAM alignment records processed.\n",
      "14700000 SAM alignment records processed.\n",
      "14800000 SAM alignment records processed.\n",
      "14900000 SAM alignment records processed.\n",
      "15000000 SAM alignment records processed.\n",
      "15100000 SAM alignment records processed.\n",
      "15200000 SAM alignment records processed.\n",
      "15300000 SAM alignment records processed.\n",
      "15400000 SAM alignment records processed.\n",
      "15500000 SAM alignment records processed.\n",
      "15600000 SAM alignment records processed.\n",
      "15700000 SAM alignment records processed.\n",
      "15800000 SAM alignment records processed.\n",
      "15900000 SAM alignment records processed.\n",
      "16000000 SAM alignment records processed.\n",
      "16100000 SAM alignment records processed.\n",
      "16200000 SAM alignment records processed.\n",
      "16300000 SAM alignment records processed.\n",
      "16400000 SAM alignment records processed.\n",
      "16500000 SAM alignment records processed.\n",
      "16600000 SAM alignment records processed.\n",
      "16700000 SAM alignment records processed.\n",
      "16800000 SAM alignment records processed.\n",
      "16900000 SAM alignment records processed.\n",
      "17000000 SAM alignment records processed.\n",
      "17100000 SAM alignment records processed.\n",
      "17200000 SAM alignment records processed.\n",
      "17300000 SAM alignment records processed.\n",
      "17400000 SAM alignment records processed.\n",
      "17500000 SAM alignment records processed.\n",
      "17600000 SAM alignment records processed.\n",
      "17700000 SAM alignment records processed.\n",
      "17800000 SAM alignment records processed.\n",
      "17900000 SAM alignment records processed.\n",
      "18000000 SAM alignment records processed.\n",
      "18100000 SAM alignment records processed.\n",
      "18200000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18300000 SAM alignment records processed.\n",
      "18400000 SAM alignment records processed.\n",
      "18500000 SAM alignment records processed.\n",
      "18600000 SAM alignment records processed.\n",
      "18700000 SAM alignment records processed.\n",
      "18800000 SAM alignment records processed.\n",
      "18900000 SAM alignment records processed.\n",
      "19000000 SAM alignment records processed.\n",
      "19100000 SAM alignment records processed.\n",
      "19200000 SAM alignment records processed.\n",
      "19300000 SAM alignment records processed.\n",
      "19400000 SAM alignment records processed.\n",
      "19500000 SAM alignment records processed.\n",
      "19600000 SAM alignment records processed.\n",
      "19700000 SAM alignment records processed.\n",
      "19800000 SAM alignment records processed.\n",
      "19900000 SAM alignment records processed.\n",
      "20000000 SAM alignment records processed.\n",
      "20100000 SAM alignment records processed.\n",
      "20200000 SAM alignment records processed.\n",
      "20300000 SAM alignment records processed.\n",
      "20400000 SAM alignment records processed.\n",
      "20500000 SAM alignment records processed.\n",
      "20600000 SAM alignment records processed.\n",
      "20700000 SAM alignment records processed.\n",
      "20800000 SAM alignment records processed.\n",
      "20900000 SAM alignment records processed.\n",
      "21000000 SAM alignment records processed.\n",
      "21100000 SAM alignment records processed.\n",
      "21200000 SAM alignment records processed.\n",
      "21300000 SAM alignment records processed.\n",
      "21400000 SAM alignment records processed.\n",
      "21500000 SAM alignment records processed.\n",
      "21600000 SAM alignment records processed.\n",
      "21700000 SAM alignment records processed.\n",
      "21800000 SAM alignment records processed.\n",
      "21900000 SAM alignment records processed.\n",
      "22000000 SAM alignment records processed.\n",
      "22100000 SAM alignment records processed.\n",
      "22200000 SAM alignment records processed.\n",
      "22300000 SAM alignment records processed.\n",
      "22400000 SAM alignment records processed.\n",
      "22500000 SAM alignment records processed.\n",
      "22600000 SAM alignment records processed.\n",
      "22700000 SAM alignment records processed.\n",
      "22800000 SAM alignment records processed.\n",
      "22900000 SAM alignment records processed.\n",
      "23000000 SAM alignment records processed.\n",
      "23100000 SAM alignment records processed.\n",
      "23200000 SAM alignment records processed.\n",
      "23300000 SAM alignment records processed.\n",
      "23400000 SAM alignment records processed.\n",
      "23500000 SAM alignment records processed.\n",
      "23600000 SAM alignment records processed.\n",
      "23700000 SAM alignment records processed.\n",
      "23800000 SAM alignment records processed.\n",
      "23900000 SAM alignment records processed.\n",
      "24000000 SAM alignment records processed.\n",
      "24100000 SAM alignment records processed.\n",
      "24200000 SAM alignment records processed.\n",
      "24300000 SAM alignment records processed.\n",
      "24400000 SAM alignment records processed.\n",
      "24500000 SAM alignment records processed.\n",
      "24600000 SAM alignment records processed.\n",
      "24700000 SAM alignment records processed.\n",
      "24800000 SAM alignment records processed.\n",
      "24900000 SAM alignment records processed.\n",
      "25000000 SAM alignment records processed.\n",
      "25100000 SAM alignment records processed.\n",
      "25200000 SAM alignment records processed.\n",
      "25300000 SAM alignment records processed.\n",
      "25400000 SAM alignment records processed.\n",
      "25500000 SAM alignment records processed.\n",
      "25600000 SAM alignment records processed.\n",
      "25700000 SAM alignment records processed.\n",
      "25800000 SAM alignment records processed.\n",
      "25900000 SAM alignment records processed.\n",
      "26000000 SAM alignment records processed.\n",
      "26100000 SAM alignment records processed.\n",
      "26200000 SAM alignment records processed.\n",
      "26300000 SAM alignment records processed.\n",
      "26400000 SAM alignment records processed.\n",
      "26500000 SAM alignment records processed.\n",
      "26600000 SAM alignment records processed.\n",
      "26700000 SAM alignment records processed.\n",
      "26800000 SAM alignment records processed.\n",
      "26900000 SAM alignment records processed.\n",
      "27000000 SAM alignment records processed.\n",
      "27100000 SAM alignment records processed.\n",
      "27200000 SAM alignment records processed.\n",
      "27300000 SAM alignment records processed.\n",
      "27400000 SAM alignment records processed.\n",
      "27500000 SAM alignment records processed.\n",
      "27600000 SAM alignment records processed.\n",
      "27700000 SAM alignment records processed.\n",
      "27800000 SAM alignment records processed.\n",
      "27900000 SAM alignment records processed.\n",
      "28000000 SAM alignment records processed.\n",
      "28100000 SAM alignment records processed.\n",
      "28200000 SAM alignment records processed.\n",
      "28300000 SAM alignment records processed.\n",
      "28400000 SAM alignment records processed.\n",
      "28500000 SAM alignment records processed.\n",
      "28600000 SAM alignment records processed.\n",
      "28700000 SAM alignment records processed.\n",
      "28800000 SAM alignment records processed.\n",
      "28900000 SAM alignment records processed.\n",
      "29000000 SAM alignment records processed.\n",
      "29100000 SAM alignment records processed.\n",
      "29200000 SAM alignment records processed.\n",
      "29300000 SAM alignment records processed.\n",
      "29400000 SAM alignment records processed.\n",
      "29500000 SAM alignment records processed.\n",
      "29600000 SAM alignment records processed.\n",
      "29700000 SAM alignment records processed.\n",
      "29800000 SAM alignment records processed.\n",
      "29900000 SAM alignment records processed.\n",
      "30000000 SAM alignment records processed.\n",
      "30100000 SAM alignment records processed.\n",
      "30200000 SAM alignment records processed.\n",
      "30300000 SAM alignment records processed.\n",
      "30400000 SAM alignment records processed.\n",
      "30500000 SAM alignment records processed.\n",
      "30600000 SAM alignment records processed.\n",
      "30700000 SAM alignment records processed.\n",
      "30800000 SAM alignment records processed.\n",
      "30900000 SAM alignment records processed.\n",
      "31000000 SAM alignment records processed.\n",
      "31100000 SAM alignment records processed.\n",
      "31200000 SAM alignment records processed.\n",
      "31300000 SAM alignment records processed.\n",
      "31400000 SAM alignment records processed.\n",
      "31500000 SAM alignment records processed.\n",
      "31600000 SAM alignment records processed.\n",
      "31700000 SAM alignment records processed.\n",
      "31800000 SAM alignment records processed.\n",
      "31900000 SAM alignment records processed.\n",
      "32000000 SAM alignment records processed.\n",
      "32100000 SAM alignment records processed.\n",
      "32200000 SAM alignment records processed.\n",
      "32300000 SAM alignment records processed.\n",
      "32400000 SAM alignment records processed.\n",
      "32500000 SAM alignment records processed.\n",
      "32600000 SAM alignment records processed.\n",
      "32700000 SAM alignment records processed.\n",
      "32800000 SAM alignment records processed.\n",
      "32900000 SAM alignment records processed.\n",
      "33000000 SAM alignment records processed.\n",
      "33100000 SAM alignment records processed.\n",
      "33200000 SAM alignment records processed.\n",
      "33300000 SAM alignment records processed.\n",
      "33400000 SAM alignment records processed.\n",
      "33500000 SAM alignment records processed.\n",
      "33600000 SAM alignment records processed.\n",
      "33700000 SAM alignment records processed.\n",
      "33800000 SAM alignment records processed.\n",
      "33900000 SAM alignment records processed.\n",
      "34000000 SAM alignment records processed.\n",
      "34100000 SAM alignment records processed.\n",
      "34200000 SAM alignment records processed.\n",
      "34300000 SAM alignment records processed.\n",
      "34400000 SAM alignment records processed.\n",
      "34500000 SAM alignment records processed.\n",
      "34600000 SAM alignment records processed.\n",
      "34700000 SAM alignment records processed.\n",
      "34800000 SAM alignment records processed.\n",
      "34900000 SAM alignment records processed.\n",
      "35000000 SAM alignment records processed.\n",
      "35100000 SAM alignment records processed.\n",
      "35200000 SAM alignment records processed.\n",
      "35300000 SAM alignment records processed.\n",
      "35400000 SAM alignment records processed.\n",
      "35500000 SAM alignment records processed.\n",
      "35600000 SAM alignment records processed.\n",
      "35700000 SAM alignment records processed.\n",
      "35800000 SAM alignment records processed.\n",
      "35900000 SAM alignment records processed.\n",
      "36000000 SAM alignment records processed.\n",
      "36100000 SAM alignment records processed.\n",
      "36200000 SAM alignment records processed.\n",
      "36300000 SAM alignment records processed.\n",
      "36400000 SAM alignment records processed.\n",
      "36500000 SAM alignment records processed.\n",
      "36600000 SAM alignment records processed.\n",
      "36700000 SAM alignment records processed.\n",
      "36800000 SAM alignment records processed.\n",
      "36900000 SAM alignment records processed.\n",
      "37000000 SAM alignment records processed.\n",
      "37100000 SAM alignment records processed.\n",
      "37200000 SAM alignment records processed.\n",
      "37300000 SAM alignment records processed.\n",
      "37400000 SAM alignment records processed.\n",
      "37500000 SAM alignment records processed.\n",
      "37600000 SAM alignment records processed.\n",
      "37700000 SAM alignment records processed.\n",
      "37800000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "37900000 SAM alignment records processed.\n",
      "38000000 SAM alignment records processed.\n",
      "38100000 SAM alignment records processed.\n",
      "38200000 SAM alignment records processed.\n",
      "38300000 SAM alignment records processed.\n",
      "38400000 SAM alignment records processed.\n",
      "38500000 SAM alignment records processed.\n",
      "38600000 SAM alignment records processed.\n",
      "38700000 SAM alignment records processed.\n",
      "38800000 SAM alignment records processed.\n",
      "38900000 SAM alignment records processed.\n",
      "39000000 SAM alignment records processed.\n",
      "39100000 SAM alignment records processed.\n",
      "39200000 SAM alignment records processed.\n",
      "39300000 SAM alignment records processed.\n",
      "39400000 SAM alignment records processed.\n",
      "39500000 SAM alignment records processed.\n",
      "39600000 SAM alignment records processed.\n",
      "39700000 SAM alignment records processed.\n",
      "39800000 SAM alignment records processed.\n",
      "39900000 SAM alignment records processed.\n",
      "40000000 SAM alignment records processed.\n",
      "40100000 SAM alignment records processed.\n",
      "40200000 SAM alignment records processed.\n",
      "40300000 SAM alignment records processed.\n",
      "40400000 SAM alignment records processed.\n",
      "40500000 SAM alignment records processed.\n",
      "40600000 SAM alignment records processed.\n",
      "40700000 SAM alignment records processed.\n",
      "40800000 SAM alignment records processed.\n",
      "40900000 SAM alignment records processed.\n",
      "41000000 SAM alignment records processed.\n",
      "41100000 SAM alignment records processed.\n",
      "41200000 SAM alignment records processed.\n",
      "41300000 SAM alignment records processed.\n",
      "41400000 SAM alignment records processed.\n",
      "41500000 SAM alignment records processed.\n",
      "41600000 SAM alignment records processed.\n",
      "41700000 SAM alignment records processed.\n",
      "41800000 SAM alignment records processed.\n",
      "41900000 SAM alignment records processed.\n",
      "42000000 SAM alignment records processed.\n",
      "42100000 SAM alignment records processed.\n",
      "42200000 SAM alignment records processed.\n",
      "42300000 SAM alignment records processed.\n",
      "42400000 SAM alignment records processed.\n",
      "42500000 SAM alignment records processed.\n",
      "42600000 SAM alignment records processed.\n",
      "42700000 SAM alignment records processed.\n",
      "42800000 SAM alignment records processed.\n",
      "42900000 SAM alignment records processed.\n",
      "43000000 SAM alignment records processed.\n",
      "43100000 SAM alignment records processed.\n",
      "43200000 SAM alignment records processed.\n",
      "43300000 SAM alignment records processed.\n",
      "43400000 SAM alignment records processed.\n",
      "43500000 SAM alignment records processed.\n",
      "43600000 SAM alignment records processed.\n",
      "43700000 SAM alignment records processed.\n",
      "43800000 SAM alignment records processed.\n",
      "43900000 SAM alignment records processed.\n",
      "44000000 SAM alignment records processed.\n",
      "44100000 SAM alignment records processed.\n",
      "44200000 SAM alignment records processed.\n",
      "44300000 SAM alignment records processed.\n",
      "44400000 SAM alignment records processed.\n",
      "44500000 SAM alignment records processed.\n",
      "44600000 SAM alignment records processed.\n",
      "44700000 SAM alignment records processed.\n",
      "44800000 SAM alignment records processed.\n",
      "44900000 SAM alignment records processed.\n",
      "45000000 SAM alignment records processed.\n",
      "45100000 SAM alignment records processed.\n",
      "45200000 SAM alignment records processed.\n",
      "45300000 SAM alignment records processed.\n",
      "45400000 SAM alignment records processed.\n",
      "45500000 SAM alignment records processed.\n",
      "45600000 SAM alignment records processed.\n",
      "45700000 SAM alignment records processed.\n",
      "45800000 SAM alignment records processed.\n",
      "45900000 SAM alignment records processed.\n",
      "46000000 SAM alignment records processed.\n",
      "46100000 SAM alignment records processed.\n",
      "46200000 SAM alignment records processed.\n",
      "46300000 SAM alignment records processed.\n",
      "46400000 SAM alignment records processed.\n",
      "46500000 SAM alignment records processed.\n",
      "46600000 SAM alignment records processed.\n",
      "46700000 SAM alignment records processed.\n",
      "46800000 SAM alignment records processed.\n",
      "46900000 SAM alignment records processed.\n",
      "47000000 SAM alignment records processed.\n",
      "47100000 SAM alignment records processed.\n",
      "47200000 SAM alignment records processed.\n",
      "47300000 SAM alignment records processed.\n",
      "47400000 SAM alignment records processed.\n",
      "47500000 SAM alignment records processed.\n",
      "47600000 SAM alignment records processed.\n",
      "47700000 SAM alignment records processed.\n",
      "47800000 SAM alignment records processed.\n",
      "47900000 SAM alignment records processed.\n",
      "48000000 SAM alignment records processed.\n",
      "48100000 SAM alignment records processed.\n",
      "48200000 SAM alignment records processed.\n",
      "48300000 SAM alignment records processed.\n",
      "48400000 SAM alignment records processed.\n",
      "48500000 SAM alignment records processed.\n",
      "48600000 SAM alignment records processed.\n",
      "48649279 BAM alignments  processed.\n",
      "HTSeq is counting SRR6145437.sorted.bam.\n",
      "100000 GFF lines processed.\n",
      "200000 GFF lines processed.\n",
      "300000 GFF lines processed.\n",
      "400000 GFF lines processed.\n",
      "500000 GFF lines processed.\n",
      "600000 GFF lines processed.\n",
      "700000 GFF lines processed.\n",
      "800000 GFF lines processed.\n",
      "900000 GFF lines processed.\n",
      "1000000 GFF lines processed.\n",
      "1100000 GFF lines processed.\n",
      "1200000 GFF lines processed.\n",
      "1300000 GFF lines processed.\n",
      "1400000 GFF lines processed.\n",
      "1500000 GFF lines processed.\n",
      "1600000 GFF lines processed.\n",
      "1700000 GFF lines processed.\n",
      "1800000 GFF lines processed.\n",
      "1900000 GFF lines processed.\n",
      "2000000 GFF lines processed.\n",
      "2100000 GFF lines processed.\n",
      "2200000 GFF lines processed.\n",
      "2300000 GFF lines processed.\n",
      "2400000 GFF lines processed.\n",
      "2500000 GFF lines processed.\n",
      "2600000 GFF lines processed.\n",
      "2700000 GFF lines processed.\n",
      "2800000 GFF lines processed.\n",
      "2900000 GFF lines processed.\n",
      "2908224 GFF lines processed.\n",
      "100000 SAM alignment records processed.\n",
      "200000 SAM alignment records processed.\n",
      "300000 SAM alignment records processed.\n",
      "400000 SAM alignment records processed.\n",
      "500000 SAM alignment records processed.\n",
      "600000 SAM alignment records processed.\n",
      "700000 SAM alignment records processed.\n",
      "800000 SAM alignment records processed.\n",
      "900000 SAM alignment records processed.\n",
      "1000000 SAM alignment records processed.\n",
      "1100000 SAM alignment records processed.\n",
      "1200000 SAM alignment records processed.\n",
      "1300000 SAM alignment records processed.\n",
      "1400000 SAM alignment records processed.\n",
      "1500000 SAM alignment records processed.\n",
      "1600000 SAM alignment records processed.\n",
      "1700000 SAM alignment records processed.\n",
      "1800000 SAM alignment records processed.\n",
      "1900000 SAM alignment records processed.\n",
      "2000000 SAM alignment records processed.\n",
      "2100000 SAM alignment records processed.\n",
      "2200000 SAM alignment records processed.\n",
      "2300000 SAM alignment records processed.\n",
      "2400000 SAM alignment records processed.\n",
      "2500000 SAM alignment records processed.\n",
      "2600000 SAM alignment records processed.\n",
      "2700000 SAM alignment records processed.\n",
      "2800000 SAM alignment records processed.\n",
      "2900000 SAM alignment records processed.\n",
      "3000000 SAM alignment records processed.\n",
      "3100000 SAM alignment records processed.\n",
      "3200000 SAM alignment records processed.\n",
      "3300000 SAM alignment records processed.\n",
      "3400000 SAM alignment records processed.\n",
      "3500000 SAM alignment records processed.\n",
      "3600000 SAM alignment records processed.\n",
      "3700000 SAM alignment records processed.\n",
      "3800000 SAM alignment records processed.\n",
      "3900000 SAM alignment records processed.\n",
      "4000000 SAM alignment records processed.\n",
      "4100000 SAM alignment records processed.\n",
      "4200000 SAM alignment records processed.\n",
      "4300000 SAM alignment records processed.\n",
      "4400000 SAM alignment records processed.\n",
      "4500000 SAM alignment records processed.\n",
      "4600000 SAM alignment records processed.\n",
      "4700000 SAM alignment records processed.\n",
      "4800000 SAM alignment records processed.\n",
      "4900000 SAM alignment records processed.\n",
      "5000000 SAM alignment records processed.\n",
      "5100000 SAM alignment records processed.\n",
      "5200000 SAM alignment records processed.\n",
      "5300000 SAM alignment records processed.\n",
      "5400000 SAM alignment records processed.\n",
      "5500000 SAM alignment records processed.\n",
      "5600000 SAM alignment records processed.\n",
      "5700000 SAM alignment records processed.\n",
      "5800000 SAM alignment records processed.\n",
      "5900000 SAM alignment records processed.\n",
      "6000000 SAM alignment records processed.\n",
      "6100000 SAM alignment records processed.\n",
      "6200000 SAM alignment records processed.\n",
      "6300000 SAM alignment records processed.\n",
      "6400000 SAM alignment records processed.\n",
      "6500000 SAM alignment records processed.\n",
      "6600000 SAM alignment records processed.\n",
      "6700000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6800000 SAM alignment records processed.\n",
      "6900000 SAM alignment records processed.\n",
      "7000000 SAM alignment records processed.\n",
      "7100000 SAM alignment records processed.\n",
      "7200000 SAM alignment records processed.\n",
      "7300000 SAM alignment records processed.\n",
      "7400000 SAM alignment records processed.\n",
      "7500000 SAM alignment records processed.\n",
      "7600000 SAM alignment records processed.\n",
      "7700000 SAM alignment records processed.\n",
      "7800000 SAM alignment records processed.\n",
      "7900000 SAM alignment records processed.\n",
      "8000000 SAM alignment records processed.\n",
      "8100000 SAM alignment records processed.\n",
      "8200000 SAM alignment records processed.\n",
      "8300000 SAM alignment records processed.\n",
      "8400000 SAM alignment records processed.\n",
      "8500000 SAM alignment records processed.\n",
      "8600000 SAM alignment records processed.\n",
      "8700000 SAM alignment records processed.\n",
      "8800000 SAM alignment records processed.\n",
      "8900000 SAM alignment records processed.\n",
      "9000000 SAM alignment records processed.\n",
      "9100000 SAM alignment records processed.\n",
      "9200000 SAM alignment records processed.\n",
      "9300000 SAM alignment records processed.\n",
      "9400000 SAM alignment records processed.\n",
      "9500000 SAM alignment records processed.\n",
      "9600000 SAM alignment records processed.\n",
      "9700000 SAM alignment records processed.\n",
      "9800000 SAM alignment records processed.\n",
      "9900000 SAM alignment records processed.\n",
      "10000000 SAM alignment records processed.\n",
      "10100000 SAM alignment records processed.\n",
      "10200000 SAM alignment records processed.\n",
      "10300000 SAM alignment records processed.\n",
      "10400000 SAM alignment records processed.\n",
      "10500000 SAM alignment records processed.\n",
      "10600000 SAM alignment records processed.\n",
      "10700000 SAM alignment records processed.\n",
      "10800000 SAM alignment records processed.\n",
      "10900000 SAM alignment records processed.\n",
      "11000000 SAM alignment records processed.\n",
      "11100000 SAM alignment records processed.\n",
      "11200000 SAM alignment records processed.\n",
      "11300000 SAM alignment records processed.\n",
      "11400000 SAM alignment records processed.\n",
      "11500000 SAM alignment records processed.\n",
      "11600000 SAM alignment records processed.\n",
      "11700000 SAM alignment records processed.\n",
      "11800000 SAM alignment records processed.\n",
      "11900000 SAM alignment records processed.\n",
      "12000000 SAM alignment records processed.\n",
      "12100000 SAM alignment records processed.\n",
      "12200000 SAM alignment records processed.\n",
      "12300000 SAM alignment records processed.\n",
      "12400000 SAM alignment records processed.\n",
      "12500000 SAM alignment records processed.\n",
      "12600000 SAM alignment records processed.\n",
      "12700000 SAM alignment records processed.\n",
      "12800000 SAM alignment records processed.\n",
      "12900000 SAM alignment records processed.\n",
      "13000000 SAM alignment records processed.\n",
      "13100000 SAM alignment records processed.\n",
      "13200000 SAM alignment records processed.\n",
      "13300000 SAM alignment records processed.\n",
      "13400000 SAM alignment records processed.\n",
      "13500000 SAM alignment records processed.\n",
      "13600000 SAM alignment records processed.\n",
      "13700000 SAM alignment records processed.\n",
      "13800000 SAM alignment records processed.\n",
      "13900000 SAM alignment records processed.\n",
      "14000000 SAM alignment records processed.\n",
      "14100000 SAM alignment records processed.\n",
      "14200000 SAM alignment records processed.\n",
      "14300000 SAM alignment records processed.\n",
      "14400000 SAM alignment records processed.\n",
      "14500000 SAM alignment records processed.\n",
      "14600000 SAM alignment records processed.\n",
      "14700000 SAM alignment records processed.\n",
      "14800000 SAM alignment records processed.\n",
      "14900000 SAM alignment records processed.\n",
      "15000000 SAM alignment records processed.\n",
      "15100000 SAM alignment records processed.\n",
      "15200000 SAM alignment records processed.\n",
      "15300000 SAM alignment records processed.\n",
      "15400000 SAM alignment records processed.\n",
      "15500000 SAM alignment records processed.\n",
      "15600000 SAM alignment records processed.\n",
      "15700000 SAM alignment records processed.\n",
      "15800000 SAM alignment records processed.\n",
      "15900000 SAM alignment records processed.\n",
      "16000000 SAM alignment records processed.\n",
      "16100000 SAM alignment records processed.\n",
      "16200000 SAM alignment records processed.\n",
      "16300000 SAM alignment records processed.\n",
      "16400000 SAM alignment records processed.\n",
      "16500000 SAM alignment records processed.\n",
      "16600000 SAM alignment records processed.\n",
      "16700000 SAM alignment records processed.\n",
      "16800000 SAM alignment records processed.\n",
      "16900000 SAM alignment records processed.\n",
      "17000000 SAM alignment records processed.\n",
      "17100000 SAM alignment records processed.\n",
      "17200000 SAM alignment records processed.\n",
      "17300000 SAM alignment records processed.\n",
      "17400000 SAM alignment records processed.\n",
      "17500000 SAM alignment records processed.\n",
      "17600000 SAM alignment records processed.\n",
      "17700000 SAM alignment records processed.\n",
      "17800000 SAM alignment records processed.\n",
      "17900000 SAM alignment records processed.\n",
      "18000000 SAM alignment records processed.\n",
      "18100000 SAM alignment records processed.\n",
      "18200000 SAM alignment records processed.\n",
      "18300000 SAM alignment records processed.\n",
      "18400000 SAM alignment records processed.\n",
      "18500000 SAM alignment records processed.\n",
      "18600000 SAM alignment records processed.\n",
      "18700000 SAM alignment records processed.\n",
      "18800000 SAM alignment records processed.\n",
      "18900000 SAM alignment records processed.\n",
      "19000000 SAM alignment records processed.\n",
      "19100000 SAM alignment records processed.\n",
      "19200000 SAM alignment records processed.\n",
      "19300000 SAM alignment records processed.\n",
      "19400000 SAM alignment records processed.\n",
      "19500000 SAM alignment records processed.\n",
      "19600000 SAM alignment records processed.\n",
      "19700000 SAM alignment records processed.\n",
      "19800000 SAM alignment records processed.\n",
      "19900000 SAM alignment records processed.\n",
      "20000000 SAM alignment records processed.\n",
      "20100000 SAM alignment records processed.\n",
      "20200000 SAM alignment records processed.\n",
      "20300000 SAM alignment records processed.\n",
      "20400000 SAM alignment records processed.\n",
      "20500000 SAM alignment records processed.\n",
      "20600000 SAM alignment records processed.\n",
      "20700000 SAM alignment records processed.\n",
      "20800000 SAM alignment records processed.\n",
      "20900000 SAM alignment records processed.\n",
      "21000000 SAM alignment records processed.\n",
      "21100000 SAM alignment records processed.\n",
      "21200000 SAM alignment records processed.\n",
      "21300000 SAM alignment records processed.\n",
      "21400000 SAM alignment records processed.\n",
      "21500000 SAM alignment records processed.\n",
      "21600000 SAM alignment records processed.\n",
      "21700000 SAM alignment records processed.\n",
      "21800000 SAM alignment records processed.\n",
      "21900000 SAM alignment records processed.\n",
      "22000000 SAM alignment records processed.\n",
      "22100000 SAM alignment records processed.\n",
      "22200000 SAM alignment records processed.\n",
      "22300000 SAM alignment records processed.\n",
      "22400000 SAM alignment records processed.\n",
      "22500000 SAM alignment records processed.\n",
      "22600000 SAM alignment records processed.\n",
      "22700000 SAM alignment records processed.\n",
      "22800000 SAM alignment records processed.\n",
      "22900000 SAM alignment records processed.\n",
      "23000000 SAM alignment records processed.\n",
      "23100000 SAM alignment records processed.\n",
      "23200000 SAM alignment records processed.\n",
      "23300000 SAM alignment records processed.\n",
      "23400000 SAM alignment records processed.\n",
      "23500000 SAM alignment records processed.\n",
      "23600000 SAM alignment records processed.\n",
      "23700000 SAM alignment records processed.\n",
      "23800000 SAM alignment records processed.\n",
      "23900000 SAM alignment records processed.\n",
      "24000000 SAM alignment records processed.\n",
      "24100000 SAM alignment records processed.\n",
      "24200000 SAM alignment records processed.\n",
      "24300000 SAM alignment records processed.\n",
      "24400000 SAM alignment records processed.\n",
      "24500000 SAM alignment records processed.\n",
      "24600000 SAM alignment records processed.\n",
      "24700000 SAM alignment records processed.\n",
      "24800000 SAM alignment records processed.\n",
      "24900000 SAM alignment records processed.\n",
      "25000000 SAM alignment records processed.\n",
      "25100000 SAM alignment records processed.\n",
      "25200000 SAM alignment records processed.\n",
      "25300000 SAM alignment records processed.\n",
      "25400000 SAM alignment records processed.\n",
      "25500000 SAM alignment records processed.\n",
      "25600000 SAM alignment records processed.\n",
      "25700000 SAM alignment records processed.\n",
      "25800000 SAM alignment records processed.\n",
      "25900000 SAM alignment records processed.\n",
      "26000000 SAM alignment records processed.\n",
      "26100000 SAM alignment records processed.\n",
      "26200000 SAM alignment records processed.\n",
      "26300000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26400000 SAM alignment records processed.\n",
      "26500000 SAM alignment records processed.\n",
      "26600000 SAM alignment records processed.\n",
      "26700000 SAM alignment records processed.\n",
      "26800000 SAM alignment records processed.\n",
      "26900000 SAM alignment records processed.\n",
      "27000000 SAM alignment records processed.\n",
      "27100000 SAM alignment records processed.\n",
      "27200000 SAM alignment records processed.\n",
      "27300000 SAM alignment records processed.\n",
      "27400000 SAM alignment records processed.\n",
      "27500000 SAM alignment records processed.\n",
      "27600000 SAM alignment records processed.\n",
      "27700000 SAM alignment records processed.\n",
      "27800000 SAM alignment records processed.\n",
      "27900000 SAM alignment records processed.\n",
      "28000000 SAM alignment records processed.\n",
      "28100000 SAM alignment records processed.\n",
      "28200000 SAM alignment records processed.\n",
      "28300000 SAM alignment records processed.\n",
      "28400000 SAM alignment records processed.\n",
      "28500000 SAM alignment records processed.\n",
      "28600000 SAM alignment records processed.\n",
      "28700000 SAM alignment records processed.\n",
      "28800000 SAM alignment records processed.\n",
      "28900000 SAM alignment records processed.\n",
      "29000000 SAM alignment records processed.\n",
      "29100000 SAM alignment records processed.\n",
      "29200000 SAM alignment records processed.\n",
      "29300000 SAM alignment records processed.\n",
      "29400000 SAM alignment records processed.\n",
      "29500000 SAM alignment records processed.\n",
      "29600000 SAM alignment records processed.\n",
      "29700000 SAM alignment records processed.\n",
      "29800000 SAM alignment records processed.\n",
      "29900000 SAM alignment records processed.\n",
      "30000000 SAM alignment records processed.\n",
      "30100000 SAM alignment records processed.\n",
      "30200000 SAM alignment records processed.\n",
      "30300000 SAM alignment records processed.\n",
      "30400000 SAM alignment records processed.\n",
      "30500000 SAM alignment records processed.\n",
      "30600000 SAM alignment records processed.\n",
      "30700000 SAM alignment records processed.\n",
      "30800000 SAM alignment records processed.\n",
      "30900000 SAM alignment records processed.\n",
      "31000000 SAM alignment records processed.\n",
      "31100000 SAM alignment records processed.\n",
      "31200000 SAM alignment records processed.\n",
      "31300000 SAM alignment records processed.\n",
      "31400000 SAM alignment records processed.\n",
      "31500000 SAM alignment records processed.\n",
      "31600000 SAM alignment records processed.\n",
      "31700000 SAM alignment records processed.\n",
      "31800000 SAM alignment records processed.\n",
      "31900000 SAM alignment records processed.\n",
      "32000000 SAM alignment records processed.\n",
      "32100000 SAM alignment records processed.\n",
      "32200000 SAM alignment records processed.\n",
      "32300000 SAM alignment records processed.\n",
      "32400000 SAM alignment records processed.\n",
      "32500000 SAM alignment records processed.\n",
      "32600000 SAM alignment records processed.\n",
      "32700000 SAM alignment records processed.\n",
      "32800000 SAM alignment records processed.\n",
      "32900000 SAM alignment records processed.\n",
      "33000000 SAM alignment records processed.\n",
      "33100000 SAM alignment records processed.\n",
      "33200000 SAM alignment records processed.\n",
      "33300000 SAM alignment records processed.\n",
      "33400000 SAM alignment records processed.\n",
      "33500000 SAM alignment records processed.\n",
      "33600000 SAM alignment records processed.\n",
      "33700000 SAM alignment records processed.\n",
      "33800000 SAM alignment records processed.\n",
      "33900000 SAM alignment records processed.\n",
      "34000000 SAM alignment records processed.\n",
      "34100000 SAM alignment records processed.\n",
      "34200000 SAM alignment records processed.\n",
      "34300000 SAM alignment records processed.\n",
      "34400000 SAM alignment records processed.\n",
      "34500000 SAM alignment records processed.\n",
      "34600000 SAM alignment records processed.\n",
      "34700000 SAM alignment records processed.\n",
      "34800000 SAM alignment records processed.\n",
      "34900000 SAM alignment records processed.\n",
      "35000000 SAM alignment records processed.\n",
      "35100000 SAM alignment records processed.\n",
      "35200000 SAM alignment records processed.\n",
      "35300000 SAM alignment records processed.\n",
      "35400000 SAM alignment records processed.\n",
      "35500000 SAM alignment records processed.\n",
      "35600000 SAM alignment records processed.\n",
      "35700000 SAM alignment records processed.\n",
      "35800000 SAM alignment records processed.\n",
      "35900000 SAM alignment records processed.\n",
      "36000000 SAM alignment records processed.\n",
      "36100000 SAM alignment records processed.\n",
      "36200000 SAM alignment records processed.\n",
      "36300000 SAM alignment records processed.\n",
      "36400000 SAM alignment records processed.\n",
      "36500000 SAM alignment records processed.\n",
      "36600000 SAM alignment records processed.\n",
      "36700000 SAM alignment records processed.\n",
      "36800000 SAM alignment records processed.\n",
      "36900000 SAM alignment records processed.\n",
      "37000000 SAM alignment records processed.\n",
      "37100000 SAM alignment records processed.\n",
      "37200000 SAM alignment records processed.\n",
      "37300000 SAM alignment records processed.\n",
      "37400000 SAM alignment records processed.\n",
      "37500000 SAM alignment records processed.\n",
      "37600000 SAM alignment records processed.\n",
      "37700000 SAM alignment records processed.\n",
      "37800000 SAM alignment records processed.\n",
      "37900000 SAM alignment records processed.\n",
      "38000000 SAM alignment records processed.\n",
      "38100000 SAM alignment records processed.\n",
      "38200000 SAM alignment records processed.\n",
      "38300000 SAM alignment records processed.\n",
      "38400000 SAM alignment records processed.\n",
      "38500000 SAM alignment records processed.\n",
      "38600000 SAM alignment records processed.\n",
      "38700000 SAM alignment records processed.\n",
      "38800000 SAM alignment records processed.\n",
      "38900000 SAM alignment records processed.\n",
      "39000000 SAM alignment records processed.\n",
      "39100000 SAM alignment records processed.\n",
      "39200000 SAM alignment records processed.\n",
      "39300000 SAM alignment records processed.\n",
      "39400000 SAM alignment records processed.\n",
      "39500000 SAM alignment records processed.\n",
      "39600000 SAM alignment records processed.\n",
      "39700000 SAM alignment records processed.\n",
      "39800000 SAM alignment records processed.\n",
      "39900000 SAM alignment records processed.\n",
      "40000000 SAM alignment records processed.\n",
      "40100000 SAM alignment records processed.\n",
      "40200000 SAM alignment records processed.\n",
      "40300000 SAM alignment records processed.\n",
      "40400000 SAM alignment records processed.\n",
      "40500000 SAM alignment records processed.\n",
      "40600000 SAM alignment records processed.\n",
      "40700000 SAM alignment records processed.\n",
      "40800000 SAM alignment records processed.\n",
      "40900000 SAM alignment records processed.\n",
      "41000000 SAM alignment records processed.\n",
      "41100000 SAM alignment records processed.\n",
      "41200000 SAM alignment records processed.\n",
      "41300000 SAM alignment records processed.\n",
      "41400000 SAM alignment records processed.\n",
      "41500000 SAM alignment records processed.\n",
      "41600000 SAM alignment records processed.\n",
      "41700000 SAM alignment records processed.\n",
      "41800000 SAM alignment records processed.\n",
      "41900000 SAM alignment records processed.\n",
      "42000000 SAM alignment records processed.\n",
      "42100000 SAM alignment records processed.\n",
      "42200000 SAM alignment records processed.\n",
      "42300000 SAM alignment records processed.\n",
      "42400000 SAM alignment records processed.\n",
      "42500000 SAM alignment records processed.\n",
      "42600000 SAM alignment records processed.\n",
      "42700000 SAM alignment records processed.\n",
      "42800000 SAM alignment records processed.\n",
      "42900000 SAM alignment records processed.\n",
      "43000000 SAM alignment records processed.\n",
      "43100000 SAM alignment records processed.\n",
      "43200000 SAM alignment records processed.\n",
      "43300000 SAM alignment records processed.\n",
      "43400000 SAM alignment records processed.\n",
      "43500000 SAM alignment records processed.\n",
      "43600000 SAM alignment records processed.\n",
      "43700000 SAM alignment records processed.\n",
      "43800000 SAM alignment records processed.\n",
      "43900000 SAM alignment records processed.\n",
      "44000000 SAM alignment records processed.\n",
      "44100000 SAM alignment records processed.\n",
      "44200000 SAM alignment records processed.\n",
      "44300000 SAM alignment records processed.\n",
      "44400000 SAM alignment records processed.\n",
      "44500000 SAM alignment records processed.\n",
      "44600000 SAM alignment records processed.\n",
      "44700000 SAM alignment records processed.\n",
      "44800000 SAM alignment records processed.\n",
      "44900000 SAM alignment records processed.\n",
      "45000000 SAM alignment records processed.\n",
      "45100000 SAM alignment records processed.\n",
      "45200000 SAM alignment records processed.\n",
      "45300000 SAM alignment records processed.\n",
      "45400000 SAM alignment records processed.\n",
      "45500000 SAM alignment records processed.\n",
      "45600000 SAM alignment records processed.\n",
      "45700000 SAM alignment records processed.\n",
      "45800000 SAM alignment records processed.\n",
      "45900000 SAM alignment records processed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "46000000 SAM alignment records processed.\n",
      "46100000 SAM alignment records processed.\n",
      "46200000 SAM alignment records processed.\n",
      "46300000 SAM alignment records processed.\n",
      "46400000 SAM alignment records processed.\n",
      "46500000 SAM alignment records processed.\n",
      "46600000 SAM alignment records processed.\n",
      "46700000 SAM alignment records processed.\n",
      "46800000 SAM alignment records processed.\n",
      "46900000 SAM alignment records processed.\n",
      "47000000 SAM alignment records processed.\n",
      "47100000 SAM alignment records processed.\n",
      "47200000 SAM alignment records processed.\n",
      "47300000 SAM alignment records processed.\n",
      "47400000 SAM alignment records processed.\n",
      "47500000 SAM alignment records processed.\n",
      "47600000 SAM alignment records processed.\n",
      "47700000 SAM alignment records processed.\n",
      "47800000 SAM alignment records processed.\n",
      "47900000 SAM alignment records processed.\n",
      "48000000 SAM alignment records processed.\n",
      "48100000 SAM alignment records processed.\n",
      "48200000 SAM alignment records processed.\n",
      "48300000 SAM alignment records processed.\n",
      "48400000 SAM alignment records processed.\n",
      "48500000 SAM alignment records processed.\n",
      "48600000 SAM alignment records processed.\n",
      "48700000 SAM alignment records processed.\n",
      "48800000 SAM alignment records processed.\n",
      "48900000 SAM alignment records processed.\n",
      "49000000 SAM alignment records processed.\n",
      "49100000 SAM alignment records processed.\n",
      "49200000 SAM alignment records processed.\n",
      "49300000 SAM alignment records processed.\n",
      "49400000 SAM alignment records processed.\n",
      "49500000 SAM alignment records processed.\n",
      "49600000 SAM alignment records processed.\n",
      "49700000 SAM alignment records processed.\n",
      "49800000 SAM alignment records processed.\n",
      "49900000 SAM alignment records processed.\n",
      "50000000 SAM alignment records processed.\n",
      "50100000 SAM alignment records processed.\n",
      "50200000 SAM alignment records processed.\n",
      "50300000 SAM alignment records processed.\n",
      "50400000 SAM alignment records processed.\n",
      "50500000 SAM alignment records processed.\n",
      "50600000 SAM alignment records processed.\n",
      "50700000 SAM alignment records processed.\n",
      "50800000 SAM alignment records processed.\n",
      "50900000 SAM alignment records processed.\n",
      "51000000 SAM alignment records processed.\n",
      "51100000 SAM alignment records processed.\n",
      "51200000 SAM alignment records processed.\n",
      "51300000 SAM alignment records processed.\n",
      "51400000 SAM alignment records processed.\n",
      "51500000 SAM alignment records processed.\n",
      "51600000 SAM alignment records processed.\n",
      "51700000 SAM alignment records processed.\n",
      "51800000 SAM alignment records processed.\n",
      "51900000 SAM alignment records processed.\n",
      "52000000 SAM alignment records processed.\n",
      "52100000 SAM alignment records processed.\n",
      "52200000 SAM alignment records processed.\n",
      "52300000 SAM alignment records processed.\n",
      "52400000 SAM alignment records processed.\n",
      "52500000 SAM alignment records processed.\n",
      "52600000 SAM alignment records processed.\n",
      "52700000 SAM alignment records processed.\n",
      "52800000 SAM alignment records processed.\n",
      "52900000 SAM alignment records processed.\n",
      "53000000 SAM alignment records processed.\n",
      "53100000 SAM alignment records processed.\n",
      "53200000 SAM alignment records processed.\n",
      "53300000 SAM alignment records processed.\n",
      "53400000 SAM alignment records processed.\n",
      "53445187 BAM alignments  processed.\n"
     ]
    }
   ],
   "source": [
    "!./src/htseq.sh -d $WORKDIR -g $GENOMEDIR/genome.gtf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Differential Expression Using DESeq2\n",
    "\n",
    "Now let us use DESeq2 to determine the differential expression.\n",
    "\n",
    "First, make a file containing all the necessary information for running DESeq2.\n",
    "\n",
    "Then, the R-script (DESeq2.r) can be executed. In this R script, the data is also plotted in various figures, which can be found in Rplots.pdf."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Directory where the alignment etc. data is stored\n",
    "workdir = os.environ['WORKDIR']\n",
    "\n",
    "# Location the the file with the metadata\n",
    "metadata = 'metadata.txt'\n",
    "        \n",
    "df = pd.read_csv(metadata, sep=',')\n",
    "\n",
    "\n",
    "# Create a dictionary containing the necessary information\n",
    "file_info = {}\n",
    "\n",
    "# Adding HTSeq data to the dictionary\n",
    "for file in os.listdir(workdir):\n",
    "    split = file.split('.')\n",
    "    if len(split) == 1 and split[0].startswith('SRR'):\n",
    "        condition = df[df['Run'] == split[0]]['study_group'].values[0]\n",
    "        file_info[split[0]] = {}\n",
    "        file_info[split[0]]['HTSeq'] = os.path.join(workdir, split[0], 'Hisat2/' + split[0] + '_HTScount.txt')\n",
    "        file_info[split[0]]['condition'] = condition.replace(', ', '_')\n",
    "        \n",
    "# Write data to a csv file      \n",
    "file_out = 'DESeq2_data_table.csv'\n",
    "with open(file_out, 'w') as f:\n",
    "    f.write('SRR,HTSeq,condition')\n",
    "    \n",
    "    for key in file_info.keys():\n",
    "        f.write(\"\\n\" + key + ',' + file_info[key]['HTSeq'] + ',' + file_info[key]['condition'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "R version 4.0.1 (2020-06-06) -- \"See Things Now\"\n",
      "Copyright (C) 2020 The R Foundation for Statistical Computing\n",
      "Platform: x86_64-pc-linux-gnu (64-bit)\n",
      "\n",
      "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
      "You are welcome to redistribute it under certain conditions.\n",
      "Type 'license()' or 'licence()' for distribution details.\n",
      "\n",
      "  Natural language support but running in an English locale\n",
      "\n",
      "R is a collaborative project with many contributors.\n",
      "Type 'contributors()' for more information and\n",
      "'citation()' on how to cite R or R packages in publications.\n",
      "\n",
      "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
      "'help.start()' for an HTML browser interface to help.\n",
      "Type 'q()' to quit R.\n",
      "\n",
      "> # Script that analyzes the differential expression of the samples with DESeq2\n",
      "> \n",
      "> # Import the library\n",
      "> library(\"DESeq2\")\n",
      "Loading required package: S4Vectors\n",
      "Loading required package: stats4\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "Loading required package: GenomicRanges\n",
      "Loading required package: GenomeInfoDb\n",
      "Loading required package: SummarizedExperiment\n",
      "Loading required package: Biobase\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "Loading required package: DelayedArray\n",
      "Loading required package: matrixStats\n",
      "\n",
      "Attaching package: ‘matrixStats’\n",
      "\n",
      "The following objects are masked from ‘package:Biobase’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "Attaching package: ‘DelayedArray’\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    aperm, apply, rowsum\n",
      "\n",
      "> \n",
      "> directory <- \"\"\n",
      "> \n",
      "> # Prepare data to use for DESeq2\n",
      "> deseq_input <- read.csv(\"/home/mees/NGS_Alzheimer/DESeq2_data_table.csv\")\n",
      "> sampleFiles <- deseq_input$HTSeq\n",
      "> sampleCondition <- deseq_input$condition\n",
      "> sampleNames <- deseq_input$SRR\n",
      "> \n",
      "> sampleTable <- data.frame(sampleName = sampleNames, fileName = sampleFiles, condition = sampleCondition)\n",
      "> sampleTable$condition <- factor(sampleTable$condition)\n",
      "> \n",
      "> # Build the DESeqDataSet\n",
      "> ddsHTSeq <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTable, directory = directory, design= ~ condition)\n",
      "> \n",
      "> # Pre-filtering\n",
      "> keep <- rowSums(counts(ddsHTSeq)) >= 10\n",
      "> dds <- ddsHTSeq[keep,]\n",
      "> \n",
      "> dds <- DESeq(dds)\n",
      "estimating size factors\n",
      "estimating dispersions\n",
      "gene-wise dispersion estimates\n",
      "mean-dispersion relationship\n",
      "final dispersion estimates\n",
      "fitting model and testing\n",
      "> res <- results(dds)\n",
      "> res\n",
      "log2 fold change (MLE): condition Old vs Aged diseased \n",
      "Wald test p-value: condition Old vs Aged diseased \n",
      "DataFrame with 38617 rows and 6 columns\n",
      "                 baseMean log2FoldChange     lfcSE       stat    pvalue\n",
      "                <numeric>      <numeric> <numeric>  <numeric> <numeric>\n",
      "ENSG00000000003  86.30450      -0.197552  0.306487  -0.644570  0.519206\n",
      "ENSG00000000005   1.96599      -0.543290  1.042977  -0.520903  0.602434\n",
      "ENSG00000000419 118.13634       0.127643  0.267939   0.476389  0.633797\n",
      "ENSG00000000457 145.11108       0.214199  0.230552   0.929068  0.352854\n",
      "ENSG00000000460  64.56699      -0.369635  0.297121  -1.244056  0.213479\n",
      "...                   ...            ...       ...        ...       ...\n",
      "ENSG00000288596  89.15276      0.0493721  0.280654  0.1759181  0.860358\n",
      "ENSG00000288597 178.53827     -0.2596842  0.339331 -0.7652838  0.444103\n",
      "ENSG00000288598  32.28253     -0.4345651  0.337105 -1.2891086  0.197360\n",
      "ENSG00000288602   2.69951      0.2123837  0.867013  0.2449601  0.806487\n",
      "ENSG00000288603   2.27991      0.0423341  0.843370  0.0501963  0.959966\n",
      "                     padj\n",
      "                <numeric>\n",
      "ENSG00000000003  0.900325\n",
      "ENSG00000000005        NA\n",
      "ENSG00000000419  0.930977\n",
      "ENSG00000000457  0.857473\n",
      "ENSG00000000460  0.809408\n",
      "...                   ...\n",
      "ENSG00000288596  0.978604\n",
      "ENSG00000288597  0.882758\n",
      "ENSG00000288598  0.801918\n",
      "ENSG00000288602        NA\n",
      "ENSG00000288603        NA\n",
      "> \n",
      "> resultsNames(dds)\n",
      "[1] \"Intercept\"                      \"condition_Old_vs_Aged_diseased\"\n",
      "> \n",
      "> # Shrinkage of effect size, since it is useful for visualization and ranking genes\n",
      "> resLFC <- lfcShrink(dds, coef=\"condition_Old_vs_Aged_diseased\", type=\"apeglm\")\n",
      "using 'apeglm' for LFC shrinkage. If used in published research, please cite:\n",
      "    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for\n",
      "    sequence count data: removing the noise and preserving large differences.\n",
      "    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895\n",
      "> resLFC\n",
      "log2 fold change (MAP): condition Old vs Aged diseased \n",
      "Wald test p-value: condition Old vs Aged diseased \n",
      "DataFrame with 38617 rows and 5 columns\n",
      "                 baseMean log2FoldChange     lfcSE    pvalue      padj\n",
      "                <numeric>      <numeric> <numeric> <numeric> <numeric>\n",
      "ENSG00000000003  86.30450    -0.01170465 0.0755413  0.519206  0.900325\n",
      "ENSG00000000005   1.96599    -0.00288584 0.0764669  0.602434        NA\n",
      "ENSG00000000419 118.13634     0.00978360 0.0744769  0.633797  0.930977\n",
      "ENSG00000000457 145.11108     0.02198607 0.0767993  0.352854  0.857473\n",
      "ENSG00000000460  64.56699    -0.02399979 0.0794223  0.213479  0.809408\n",
      "...                   ...            ...       ...       ...       ...\n",
      "ENSG00000288596  89.15276    0.003472259 0.0739927  0.860358  0.978604\n",
      "ENSG00000288597 178.53827   -0.012719068 0.0762027  0.444103  0.882758\n",
      "ENSG00000288598  32.28253   -0.022146807 0.0792430  0.197360  0.801918\n",
      "ENSG00000288602   2.69951    0.001649518 0.0763196  0.806487        NA\n",
      "ENSG00000288603   2.27991    0.000386014 0.0762774  0.959966        NA\n",
      "> \n",
      "> # Make MA plots\n",
      "> plotMA(res, ylim=c(-3,3) , main=\"Normal data\")\n",
      "> plotMA(resLFC, ylim=c(-3,3), main=\"Shrunken log2 fold changes\")\n",
      "> \n",
      "> # Transform the data for clustering\n",
      "> vsd <- vst(dds, blind=FALSE)\n",
      "> \n",
      "> # Heatmap of the count matrix\n",
      "> library(\"pheatmap\")\n",
      "> select <- order(rowMeans(counts(dds, normalized=TRUE)), decreasing=TRUE)[1:20]\n",
      "> df <- as.data.frame(colData(dds)[c(\"condition\")])\n",
      "> pheatmap(assay(vsd)[select,], annonation_col=df)\n",
      "> \n",
      "> #Heatmap of the sample-to-sample distance\n",
      "> sampleDists <- dist(t(assay(vsd)))\n",
      "> library(\"RColorBrewer\")\n",
      "> sampleDistMatrix <- as.matrix(sampleDists)\n",
      "> rownames(sampleDistMatrix) <- paste(vsd$condtion)\n",
      "> colnames(sampleDistMatrix) <- NULL\n",
      "> colors <- colorRampPalette(rev(brewer.pal(9, \"Blues\")))(255)\n",
      "> pheatmap(sampleDistMatrix, clustering_distance_rows=sampleDists, clustering_distance_cols=sampleDists, col=colors)\n",
      "> \n",
      "> # Principal Component plot of the samples\n",
      "> plotPCA(vsd, intgroup=c(\"condition\"))\n",
      "> \n",
      "> # Dispersion plot\n",
      "> plotDispEsts(dds)\n",
      "> \n",
      "> write.csv(res, file = \"/media/mees/Elements/ngs_data/DESeq/output.csv\")\n",
      "> write.csv(resLFC, file = \"/media/mees/Elements/ngs_data/DESeq/output_LFC.csv\")\n",
      "> \n"
     ]
    }
   ],
   "source": [
    "!mkdir -p $WORKDIR/DESeq\n",
    "!R < ./src/DESeq2.r --no-save\n",
    "!cp -p $WORKDIR/DESeq/output.csv $DIR/data/DESeq_output.csv\n",
    "!cp -p $WORKDIR/DESeq/output_LFC.csv $DIR/data/DESeq_LFC_output.csv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Analyze the Data (in progress)\n",
    "Lastly, analyze the data!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.decomposition import PCA\n",
    "from sklearn.preprocessing import StandardScaler"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['baseMean', 'log2FoldChange', 'lfcSE', 'stat', 'pvalue', 'padj'], dtype='object')"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('data/DESeq_output.csv', index_col=0)\n",
    "df = df.sort_values(by='pvalue', ascending=True)\n",
    "df.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Principal Component Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "pca = PCA(None)\n",
    "pca.fit(X)\n",
    "X = pca.transform(X)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[-2.76480337],\n",
       "       [-2.63289566],\n",
       "       [-1.3580566 ],\n",
       "       ...,\n",
       "       [ 2.68762411],\n",
       "       [ 4.46341567],\n",
       "       [-2.24993793]])"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pca_transformed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[-2.76480337],\n",
       "       [-2.63289566],\n",
       "       [-1.3580566 ],\n",
       "       ...,\n",
       "       [ 2.68762411],\n",
       "       [ 4.46341567],\n",
       "       [-2.24993793]])"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "X"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### MD Plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "def MD_plot(mean, foldChange, y_lim, x_lim, p_val=p_val):\n",
    "    c = [x < 0.1 for x in p_val]\n",
    "    plt.scatter(mean, foldChange,s=2, c=c)\n",
    "    plt.ylim(y_lim)\n",
    "    plt.xlim(x_lim)\n",
    "    plt.ylabel('log fold change')\n",
    "    plt.xlabel('mean of normalized counts')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "MD plot of the data, if the p-value is smaller than 0.1 it is coloured yellow."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAZAAAAEKCAYAAAA8QgPpAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjAsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+17YcXAAAgAElEQVR4nOyddZgVVRvAf2du3+2ku0saBBQUUGwUMTCwWzGwAVtEMTAQLNDPABFBBRVFBQURpEG6Y4ntvj3v98csuyy7CwuyLCvze559bsyJd+7dO++ct44SEUxMTExMTI4WrbIFMDExMTGpmpgKxMTExMTkmDAViImJiYnJMWEqEBMTExOTY8JUICYmJiYmx4SpQExMTExMjolKUyBKKadS6m+l1Eql1Bql1LOVJYuJiYmJydGjKisPRCmlgDARyVVK2YD5wP0isrBSBDIxMTExOSqslTWxGJort+ClreDPzGo0MTExqSJUmgIBUEpZgKVAY2CsiCwqpc3twO0AFiwdmzQIIyy2Eab7xsTExKR8LF26NFVEEo73uJVmwiomhFLRwHTgPhH5p6x2kSpG3njkdm555eUTJ5yJiYlJFUcptVREOh3vcU+K23gRyQTmAOcdqW3In4Gu6xUvlImJiYnJYanMKKyEgpUHSikXcA6w/kj9Zn+RzHm2q5k75c+KFtHExMTE5DBU5gqkBjBHKbUKWAzMFpGZR+rk9WgoTbFp6bYKF9DExMTEpGwqMwprFdD+aPv5PBbOGdyTwc9cUQFSmZiYmJiUl5PCB3I0iA4d+56Gw+WoPBlER7y/IsGdlSaDiYmJSWVTqWG8x0LjNtD7mjMrVwjvTCTrCVDhqGolIo9NTExMTgmq3AqkZefcIzeqaCx1AQFrk8qWxMTExKTSqFIrEE2DQUMiD9smZXcaOem5NDytXoXJoeztoNpKqtjHZ2JiYnJcqVIrEF2H4YOthIKhUo/7fQFubfUg93Z9grULN1aoLErZMMp5mZiYmJyaVCkFArB1tY/Ny0sP4dU0hTPcCYCr4NHExMTEpGKocjYYEajRqFqpx6w2K59sehu/x09kXMRxn1vXdTStyulcExMTkwqhyl0N42sGcNu+pqwaXk63o0KUx4xxP9HPdhX/e3bKcR/bxMTEpCpS5RRIdLwfLe9lCB6x6slxZcOSLVgsGusXbT6h85qYmJicrFQ5E1ZOlgXsncHagL9mLCEqPoKW3ZpV+Lx3j7mJ9r3b0Pm8dhU+l4mJiUlVoMopkP07nKxfPxTUDp6/8nX0kM7U5I8Ijw6r0HndES76XFvJCYwmJiYmJxFVToEoJcRV82Bx1sYV7iSmWhTOsMora2JiYmJyqlLlFEhEbIDEerVQ1ji+TplQ2eKYmJiYnLJUOSd601b5THrmTu7s8Ai7NiRVtjgmJiYmpyxVT4F09vD1+HC2rtzBwhlLK1scExMTk1OWKmfCWvJrFI9//gR//7CMfjefXdnimJiYmJyyVLkVSG6WonO/dtzz5s2Eu+aip16C+FdUtlgmJiYmpxxVToHEVdOLXuR9AMH1iOeryhPIxMTE5BSlyimQuo3z8eR5AVCRz4HrClTYnZUslYmJicmphyqrptTJSKSKka6qD3E145i0a7xZTt3kuCMiLPh2MQl14mjasVFli2NiclxQSi0VkU7He9wq50QHyNifiYiYCsTkuLPsl1W8eM0YRBe+zfwEu9Ne2SKZmJy0VEkFcuaArmZZdZMKoUbDatjsVmo2ro7VXiV/HiYmJ4wqaMLqyyz/ZCxWS2WLY/IfxVzdmvzXqCgTVhW8jRfyczyVLYTJfxhTeZiYlI8qqEBg1PVvVbYIJiYmJqc8VVKBlLUn+qmMeGYi+VPL3KnRxMTE5HhT5byEcdV9vPLriMoW46RCgruRrEcAhbK1BFvLyhbJxMTkFKDSViBKqTpKqTlKqbVKqTVKqfvL0y8vx0LtWtNLvO/z+I67jFUGSyLY2oG1GVjqVrY0JiYmpwiVacIKAkNFpCVwOnCPUuqIt87ePAuo6GLvvXXPB1wUdh0/Tvj1Xwu17NfVfDt2FqFg6F+PdaJQyo4WNwktfjpKC69scUxMTE4RKk2BiMheEVlW8DwHWAfUOnLPECHPfLbP60TIuwSAneuSsNgs7FpftD+IBHci4j0qmULBEE+c9wLjH/qYBd8uPqq+JiYmJqcaJ4UPRClVH2gPLDpS27gaQd4ZmsSsSfU5Z9AX3PNuK576aiir562j8/ntARDvLCTzQbC2QMVPK7ccmkWj28Wd2LJyO007lSxjISJI9jOgp6KiX0UpV7nHNjExMfmvUekKRCkVDnwNPCAi2aUcvx24HcCJC7tDCFlOB/mHnz7fxYYVw/lg1Wv0uLRLUScJAAoIHK0sPDPtkbIb6KngmQxYILAa7F3KbmtiYmLyH6dSw3iVUjYM5fG5iJS6VBCR90Wkk4h0suFg73YXp3XzcvNL16BpGhZLyVNQrotRcd+gYicdX4G1eAh/ENzXgq398R3bxMTEpIpRaaVMlJHu+wmQLiIPlKfPgWq8Nrti+JRHqdeyNnE1Y3G6HRUrrImJiUkV5r9YyqQHcD3QWym1ouDvgiN3E5TSqV5zG7Ua1zCVxwlG/IuR/K8QqTpRaiYmJhVDpflARGQ+hqPiKFHc+tRuGpxW34iyCu1HWesdd/lMSiISQNIHAxpKiwbnOZUtkomJSSVSJUuZjB9Rm+Q91ZD065HU8xDPzMoW6RTBCvYeoMWCtXllC2NiYlLJVHoU1rHQvpeFuFq1ICuEsYgJVrZIpwRKKVTsh5UthomJyUlCFVQgwrNTeqPpi1Gxn0EoCWVrUtlCmZiYmJxyVDkFYrGFsAVegwwN4n8wlYeJiYlJJVHlfCChgJWA3/C9i4qpZGlMTExMTl2qnAIB8ORpgA7+lUgoDfHORsRf2WKdUESCSGAjInpli2JiYnKKUuVMWACaVpD8mP8JkpcNwbWAhrhvRIs8TCmS/xCS9Th4v4ew21ARD1W2OCYmJqcgVWwFYpiu5s9qZby0NgVrEyAEBMD3W6VJduIpUKLmDoQmJiaVRKWVMjkWDpQymbZxNWHhAAoiR4KjF8r7AzjOQFkbHNPYoucDepXZT0MkAMFtYG2MUlXsPsDExOSEUlGlTKqgCUsI+TUO3IErpVCWOAi7/thH1HORlLOBAMTPQlmqHx9RKxClbGBrWtlimJiYnMJUwVtXxeCuzdm3yw1xM1Guy0q0kFAyEtxxFGP6QbwgIRDP8RPVxMTE5D9MFVQg4PNZ2b57GJqtcYljouchqf2Q1AuRwIai9yWEBNYbjyJG9FaB+U5psaj4maj46cVMYHrGEPT9nZHA2oo/KRMTE5MqRhVUIMIDb7bl9P4lVx5bVm5l+Pm3s2KexXhD2Yp6ZT+PpF2GZD+H5IxCUrojuW8XHlfWeijrIQrJvwAkDwoUiOjZiPenAn+JiYmJyalNFVQgimatFkPOs4bvIrgN0TMA+Gr0VP6e7eOjF6tB9Nsoa8ODuh04VQ30ZMBS8FjE12Nm8vpt4/Hm+4wusf9DRT4Lrv4ASObDSOaDSM6LFX2SJiYmJic9VdCJDveea6Vpu8W8Oms8v3wyjX/+juKOtz9gwC2LyUgSBj7QDeXoBRirBgIrwTkALPXAdS1KBcF1Jdg7Fo7p9/oZ/9AnWKwWulzQnjMu64qytQRby6KJbc3BPxeszU7wGZuYmJicfFRJBRLwK9YvcxMIRPPmozXRdWh55gzOu3QNL00KoRI/LgxtlYy7ILAMI4fEgrJURzn7gaNbsTHtTjuDnriM7Wt20b5361Ln1SIeQsLvRSl7BZ/h8UH0fPD9CvbuRqSaiYmJyXGkSiqQGx7dx2ln1sUZP5hrh1tY9ft2ulwyAKyzACcoZ1FjSy0ILAFLHQjtx+Otzp9Tfqd9n9bE1yp+Ub35xWuOOHdVUR4AkvMqeCaBrSMq7rPKFsfExOQ/RpVUIO3OyKXlWf1BfFw34iYYARLcgqRuNBoENxWanlTUyxA5HKVFIqLz2WOfMW3M9zTu0JB3Fr5UiWdxArC1Bo8OtraVLYmJicl/kCroRBcat/EYTvSUc4yMbEC0Brx4Ty+GXNSV7KxaRc31veD5DtGzUEqjWefGiECbM5qj509FTxtYPNw38A961nNIKOlEn9hxQUJJ6FnDEf9yNPcAVLV1p0x9MBMTkxNLFVyBKNYsCqd9Ty/Ju4IsmDSNPte0xOpoyLxv0rFYNTYu3Ubnfu0AkMyhEFgOwQ2oqOfpdUV3eg7shlIKPaU3hHYjnq8RXyzKdhqSMxqCaxDyUVGjKvlcjx7JfQ88U5DASlT8DLPMiYmJSYVRBRUI/PRlNO37JPLyfVmsWfwVWxdnMvSjYTz+6X3s35FChz5tiho7zoLgOrB3L3xLKaMoo4p8AfF8C1oi5L6GYIPI5yDvPbA2R/Ing+sKlLKc4DM8dpTrciSwAty3VLYoJiYm/3GqpAIJ+u1gqUfXvvPYssZJp7OykKxHOfvyN1COAcXaauF3QPgdSGg/IqFiykA5uqMc3ZHgDsTzJdi7obkvRVwXIvvbIFhQWhw4zznRp3jMKHtbVPx3lS2GiYnJKUCVVCDnXbOf7av+JjxKZ8qqNdidAgLimYFynFGivZ7/FWSPAMfZqJhxJY4raz1UwuzC1yI62HtCcGPxPBATExMTk0KqmAIxTE+vDKmHzaaTnhKL12NhwG3JoKpBKAU99RJUzMTCvAeRIAR3Y2SgZx5xBvEvgfTrwd4ZLXFuBZ6LiYmJSdWminlYjeKH7oggbXvk4XQLzdrlkrTNzqv321m/aBkEN0NwNSKCnvs+knwWu1ZNYOKr3UjJLYdTPLQL0IxxjgE99330lPPRAxuPqb+JiYlJVaGKKRCDvdsdtOiUy/QNm2nVOZ/PX6/G7CkxjB3eBMIfAvuZRmhu7msgyYx5pDaTX8/iw8enHXlwZ39U9Duo2C8BjMq9wZ2IhEo0PVDZtxi5YyG0BbKfOx6nalIBiOhI/lTEv7iyRTExqdJUSQWiFHTomQMYe3dcODiDxq09XHn3dpT7SsNRHtxU2P68a6NIqB1J78u2o+d9hJ43CX1fM/T9p6PnfVh8cMkFPRXBhuRPQ3KeQ1LPRbKGFW8W3Izsb4+k9S+uRGztAAWWGhV09hWD6LnoOW8ivoWVLUrF4/sNyX4aSR9cmEdkYmJy9FQxH4iBCORk2KCB8eNv1TnI2J83gaUp4vkZ8c2CsJvB1gnsHel370Oce+svSOa9kPOj4S9BQNIhbwKE3Vo0dvbz4J0JlnpIaAcoB6CV3GgqlGSMEdyGsSe78VGq2PfBvwLs7UuXPbgNfPPBNQClhR33z+aY8XwNeeOQ/M9Q1f7jd+bWZqDFgrUJVfQnYHKUiJ4HegbKWruyRflPUam/HqXUBOAiIFlESq9gWArxNQI0bOkBHAXvFITm6lmQMxrIAP9fgIaKehbd9xeE9nDACY8lHoL7DUXiHlx8cHtn8H4P9i7g3QuOPuAaCMqOiBTmkGDviYp5G7RaKFX0MSrlAEdX9PSbEf9f4LwYLfqVwuOScReEtoOeiop4sPwfVkXj6AH59cDRt7IlqXCUtQ4qcV5li2FyApG0iyG0D2I+Qh1SSNXk2KlsE9bHwHlH26lRmzzu7teU5CQd8HHAlIXsBzKBmIKVA8Zughk3QM4LGCsFQKsOUa+ApEHu64ZNXAQRQXNfiVZ9LYTdBc6LUe5rIOdFIzLLO71QBsl9E8n7DCwJJeQTEWMzKkLgm138oKMPhuLTCk1fEtxZUH5kZbnOX0J7ED0d8S9B8j4rYYbRc99HzxhibIAlfiSUdsQxlbUxWsJPx7Xsiej5SN7HSGDdcRvTxOSYkCDGDWRJX6bJsVOpCkRE/gDSj7bfop+j2bXJwaQx1Qre0Sk6FQEyjBVCwhyUpXrJAYJrULZOhp/C2oqUVT0ZEHcdNzUfgt8XQM+fAqm9jJIg6XeCCjfGVWEFcgch713w/2mYow5BKQXR7xnKImp8sQu8irgfCELeBwRy5xAKhZC8ccZc2U8d+FzQc15Dz3rSUAAFEWV6zhj0wEYkpZ9RByz9RiRnJHh/OOgzDUHuq4bi8v1h7MKYcgbiW3S0H/O/J/9zY/fHjDtO/NwmJgdhbFldep6YybFzRAWilHIrpUYopT4oeN1EKXVRxYtWOP/tSqklSqklAXzFjnXolVXwzAFRbwEHtrDVILgaZYlH2buw7K+ePHpFQzasiITwEajYCShrbbSEX0HySd2Tg8/jZ//2VDx7H4Xs4UWT2DtB9NtgqYnkvo+u+4AgRDwDrgHgOLtUuTVnL1T0O5DzDLK/o+H7AENGawtS9rq5ovYErmtwNwE1AKzNIew2o4meDHnvg+cbCKwwQotzX8ef/j5k3An4DUeQ42zQ4sBWVLpFKQtEPArOfoilToGvRgPJ+VffwzFh7wwqGpz9KmwKCSUhue8hof0VNodJ1UdpkcV3KP0PInomeuYD6HlfnLA5y+MDmQgsBQ4YDpOAr4CZFSXUwYjI+8D7AJEqpljMbHjUgZc+yHoMKLjTt/fGk7WBT0ZcQMuu8Ux/z8LavyOYOi7IsHP6og6OkArtpXl7D8980ZKImtcRETECAhrGqsYCUaNR+nYktBfYA2mDkNA/4L4bLepIW9uGILS74Ok+sDYwVidxk8neuZWA/xmyUrLxBZvjOLj8iJYI7ptBTyuI6rLw28yevHJHBudenc5Dr2kQOwHN3q7UWbWCoAA97TqQfLCdjnKeeN+GsrdDVavYqC7JGm6YCwNrUDFvVehcJiYnNd6fwfuT8Rh25L2NjgflUSCNROQqpdQgABHJV4We5MrltO65hc/zsj1s/ieM1j1asnNNCuOesLF6oZsZE/MYNWUrU8clcPV9KajQ/uIhtuEPQP4HdDq3KVpYE0R/D0k5FyQDtHqQehGiRUDkcygtFskscHz7fkOCAxHPlyjXAJS1QQn5lLJD3DTENxfxzgBrA9CikZRzaFQrl5dnvY07ujaRsRGH9FOoyMeKvbd7Z2806zds39QEFfUgqgzlAQV5DnnjDEVkqYsKP3JhRQmsBclH2Tsdse1JhfN8CKyt0FUOGGZLyRwKBFDRY0D8SMadoKeA41xUxEOU9rMQEaOYp6UuSguvUBlNTnGcfcH3G9i7nrApy6NA/EopFwVp4EqpRnCILakS0DQdiwX27rCTUNPP8OsbsGG5mzbd/KyYBxCB3aFz6a0ptOmaR5uueYAVsbZm4YwlVG+QSIPWdVHkI3oK5IxG90xFRY8GKTg9favxGLIYd9PWxkjMe5D7kVGgMedl8P2EBFahYj8pJp94ZiJ6Psp9BWTeYZSNV9Go8NtBzwCgdfdYlK1+uc732uGX06RDQ1p2b4ZyRZU4LsGdoO9D2btAcL2R0IiOip+Fsh5+DtHTkbQrjRdxU1G25uWS6WRAc18J7isrfqLQLvD9BCijSoGeAYGlQAjyPzBksNYp2c/7jZFDZG2Civ+24uWsYhi+OUE5Tq9sUao8SotFxYw/oXOWx4n+NDALqKOU+hz4FXj0eEyulJoE/AU0U0rtVkqVuwa5rms8eU19burRnGHXNiQ2MYiuK4IBD5pFwx2hce3QfaxZHMaktxIBDZxXsPjHlbxw1evc22UogaSmSHAHWFoDurEPSP7X4LoJb77i8zcSWbGgEUS+jLI2NmS2d4LQasi4EaxtQasJrsuLySahvUjWw5DzDAT/gbA7wdoa5eqPCMb2upaaYKlb7s/KZrfR7dwtROk9S9g4RUJI2iVI+i2Ibx6iZ4GlCdj7gqUcce/KBVpB5JoWU3zs4GbjMzrVsdSH8Ach/G7DX2XvapTMt54GrkHG1sml4sAIwHCWcfzURYI7kYybkYxbjBsgkxKI6OjZr6Bnv2YUeT3JOOIKRERmK6WWAadjxMHdLyKpx2NyERn0b/qvXBCO6LB6YRgzt68mO8NC8m47z9zUmD4DkwkL11m3JIzNq90MeqwX+P4iMeIXUHWp18SDpmHYDMkDNFDxkD8BsPDL1Bp89lo8DreT77IvQdd1vHk+nGoK6OmABeXohIq4FT3nLfTUS1HRb6GsdREVaygWPQXx/YkWfie4r0TPHgX5XwBewAKhnaAVr/ZrhPYKSmmInoWk3wKWRMMh75sP+MA/v5iNUwKbQEJAEFGRkH41oKEihxbLUSkLpVyQMLdgXisiXiNxUs9EUi8zGiX+jtJi/83XVaVRSqHCD44m01CRDx+5n+sCsJ0GlsSKE66CEP9SCKxEbO1RyoayFU/VOhCGfswWbS3GSOg88NykJMENkD/ReO7uDwU3sicL5YnC6gDUAwq8yNRVSjVS5bkyHXeK/6MG/Yb4IjDi+gakJNlZ+Wc06fsVX4+rRnhUiPOudzL0/XPA3p2kLXuIjMng2625vP3jRpQC1EF5HJLKgYKN7S54khqN4jl3cEv07FEMv/h5Lo25gYUzfjfksJ6GZNyKnvkA5H9smI18cwsEWw/6bsAHuWMKZPQXKCev0d91NVhbFDsfET+Seg6S3Bk961kk434IrgXfryBZqMjnUBHDUZGH1NnyTjXm0mqhbG3A1t74QVobAaDnT0dPv+mwd3lKWQqURwhJORdJPhMJ7gZlKbh7tpfaT895DT35TCSwqsyxy4MENyOeHwprjon4EP9yI2T6MOi5HxhbEIu/oJ8ggXWInnvYfoeVRc9H/IuPOHd5Udbahj+snHjyvOzeuOe4zP1vkIzbkZxXIP0aJO1qJLir6JgEkNR+SHKnggCTo0dpEaiEP4w/LeLIHU5FrI3BeRE4LzFWwScZ5VEC7wIdgFUYV/DWwBogSil1l4j8XIHyHYKU+q4eUiyZE0koACM+3MG0D6JJ32/j6/dq8c6sv4G/2bKgGvdf1AKrVefGx/dw1qUWouN0sJ8Glr4FWl6B6ybQ91C75jQm/D4XsEG+n/1bT8disZCSdiG4G4B3LkF/LmsXLKHZWS/jsCxGbJ3A+yMEMggFYdVf4TRtl09Y5LeGIogYbtTocg9CK22fEfHAgR+j53PjUasH4fehDtyhhV1XoptyD0aCSRDajGQNQ4s7JIwvd7RR3yvrSVTcZyWnDaWAcqC0SEA36oEBSnNBwnxAlV12xfOtEXbsW2DcaZeCBLcCFpS1XunH9Wwk9UKM8jIjUe6BSOYj4PsF3INRkY+X0S/DyHnBAs4+Rja991sk60mwNj7mjbUkc4gR2eW+BRU51Ph80FGWakfsezwY0m0Yu9bv5snxSZzRvyYq9svCu3wRHSTvmC644l9u/P+5BpRrZYrzfPD+DuQDGhwcBCDeghBxBYcGphwFJ0k8zkmLUjbDL3uSUh4Fsge4RUTWACilWgLPYfhBpgEnUIEcDmHDyjB++TqanEzjtJwuD8EAWKyQtMULCAG/xvinarHyz3CenrADfN9RaKe2dTAuHKGNFCYnqihQFkbPvp+tay2076lD5gggyIcv1Oa7ibF06DuPjP1eEms8ytMfbgVrKz59rRpTxibS5vQ8Xp7yqDG+tTUq9jOU5jYkloBxAba1QtlaoLQoiJuMhDIh8x7AC/oOvMHTcB1cRuUQlLUuuPsjmQ9BaCciz6GUkRMjoVSwdQPfDAj8jei5xaKBdN9CyBgMaEji32haJMT/AHo66iAlJ3o2kn4DaNGomA8KL0AqZqyhPNzXlv6tBHcjqf2NFwm/ogpMOSJiBCFItrEaO3BzcKAMvhZufAeHu1CqaGPe0F5j1QUFFQiksBLBMaHFAiGwxBkBBil9jTETfgYUkvsWynkRW9ZV55Onp3DF0Is5rWdFbDwWMiLMDqq1Jhk3gX8RRI9DOUvmIElgHZL5ALguRQu/q+h9EST9ekBHKTe4jpzKpUW9AFEUru4OXkUpLQLiJoGeediIwFOVwsoWWmUX+6hYyqNAmh5QHgAislYp1VxEtlbe3YOitNVIZGyAd4cVOY1XLwrniUEN6TMgkzcfq42mCW16NuCf+dtpd2YuIvDC7fXYt9POyC+2Egis5JevYunSN5GGHQeQsj+SbYvfJTPFQrP2NzJlVHdGXZfNsPEO2nYPEh3nQw9B0O9l8/JtbF7uYsWfDtr1WEl89VroIUV8rWpADpBiONR9cxDbaRDcyO4NW4mLGIMzLAyJGgWBlajwu9BsTvSIxyHnGWZNiueNoQ9zwS2tefCDZ0qcs4gY4aShXeC+AaxtUMpWkGD3oZGRrmcY+ST2M0qGkob2FTzRCwpGRhrZ+4dk8EtgrWGPRTfGKyjhYpjM2lAmymGYwbDAwWYcfW+RbTd4wASiQItCvHPAfQMq/B7Dl1TW0EqhIp8q/qajD1hbgvhKKMvyoqJGQcQjKEsCoqdj5BcFkcBGCPwNnq8Q/wImj7qYhTOWkJ2azZt/HiknqPy89deLpO5Oo3a9f8BSp/hqIbQPsBirvtLwzTVqreV/DgcpEKUU4jgTAiuPepfNssxvqowV56mOruvc1fFR9mzex7hlo6ndpGpV5j4ayqNA1iilxgGTC15fBaxVSjkozNw7URxQWHLIe0LDVh62r3cVvmuxCqGgInmXne0bHOghw9S1fdU2Zm5bxZ7tdr58J4H53xshsTd2b44334IeUnz/eRRtz7Iy+38/orR6KBTR8UGy0lMJBRVPXt2QD+dt4Kr7UjjnqgxcdR7n2vqv4PPqxCQYd/4X3ZBEgy6GDTkzvQbR8U7QYhDlgLRL+f1bG6PuqUPtRo344K96kHknW/5xMu+HP7j0jmhimr6LKDu7t7yHxQq7NpRhZ5Y88M8FrIZzP/sBdNlr3Ml7pwNuQDfKseR/hh5Yg4p+GaVFA+CT88jOTiOhbgO0Mkw0EtwJGbcbY0WNQJVS/6tkn21I+s1g74JKmIdhBjvoYq7VAFsXCCyCwELje7R1Alszw4QEUG0F4pmK5I5FRT2PcpxpjB1KK1gF9EM5uhefWE+D4BpAg9A20A6j3MpAKa1IQWqxiBZlrMhCG8F5MeJbAO6ruGJoR7JScxj0xIAjjqlnvwze71DR76LsbQ/b1hXmpE6zWkDJyC4V+4Wx1fJBsf6i5xrx/44zwD0IJBflOKtEX62U7ZwPhwS3QigF5Ti+eQV6KHpKmQ4AACAASURBVAVy3wH3VWi2lkhoH3img/MiVGmh0FWMYCDErvVJKBVi74aZ1G5yW2WLVGGoEhsiHdrAyAG5GzhQROZPDL+IF3CLyLF7K4+SSBUrXVXvUo9pVh09WLRcvP3pJP7+NZLOvbP58PlaoOCMCzM5/Zxs2pyexwu312fTShendctl1V/haBZjX/VQSNH/didLfsknaasFEOwOofv5WfS9Ip3nbm6ALnDBtWlcPSSZuGo6RI1lw/zx2K0badD+EvDNYfmC+jx5pRc9BOHRIaamTEVTPiS5I6Dz0YvVmTI2AVCERbl4/ZsVjLyjHjs2Orj01jTueuNGyPuQgDeZJfM60KrnuUTVGliq7Vu8vyChPeD7G/y/grUVhN8LOaMg7E6U82wkuSdG0UmFinwW5b4agMFN7iV5ZyqjZg2n3dmlF0QW/0ok3fC9qMSFJfwhoVCIcQ9+DMCdr3RC8/0A3m+MZEwcaNVXlz5uaJ+RI2HvavhfnOdCKA1JG2CEOEsO6NlAPjj6o8UYtmA95y3IewdUHDjPRrlvQNmaFY3r+YFlv20lstrpNO7Y8V/b2cW/AvEtQIUNPuZkQD25F+j7UeEPHhLNdRRy6OmGecrWHu2gSs561nCjHL/9dLTYicc0dgl5Axsh7VJAoWI+PK4VbPXU/kZyJU606qvQ0+8E/29g61TSf1dF2bzwI5I3vUfXvjlYalR+MVGl1FIROe5ZwuUJ4/UArxX8HcoJUx5H4mDl0aZbLvO/j2b98jBWLYigZadc1iwO58/vo5g3w7jzrtPIh1JQra4P50o3ETFB7n95N7UaWomMyaZF2yiCAaHrOTlERIf4eXIsoYDGp2u7c23zBXw3MZ6dmxMY+W0vti94lqGXJKBpjZm86heS069m+NV/oIeMC5fD6QR9P5LzLIZvBc661M+094VgQOHJ9fHoFe3oeYkNb346Z1yYCTkvgaMfNv0XuvVZCixDsjcYkVXOvsUy35Wzr1Fn1NGfL0bsIiIyiUtufRkV9TLK3tbYCyF6LGQP45X7wtiwch6jfupDtXoJ+D1+lFIEfIdZTFrrGHumOM4uVB669zfIHYuKfJKdGxOYMc5whV0w8EvqN92OEeCnIPr1ModVluqo2I8KX4voYLGiqq1AfH9C5l0Yi1wFtqLwReU8H/H9ChIAz1QktBd10IVz3YpGPDXwY3SZw1dbOhFe64myz61w7iD45iJaJBCFCq0BrCjXJUYS6SF2fhEf6BnokoDFajni+CpmnLHFgOvqI7YtE9988C8G/yIk/H5jpQRG8ILnK7AdfmVzVGQ9CRhRaBLaw3E1Vts6FWbnGxMUXEYKEmwPcPDKqqqFkDfqfCENm8wCW+fKFqVCOaICUUr1AJ7BCOUtbC8ilVCZTCjyf5TuBwHY8o+T/BwrDpeOz6MRCBg/NF0v+hns2uIABbO/jEMEPHl2vp0QzxPjdvLYlQ3ZuNJNz4syOefKTFYvCuONR2ohuqJesznoupEUtn1DkAvj5/DwGEMZOd0hrJZc9u/wEgwYc/UZmM59L1tR+RPAPwcAn1cx/ulI3BEh6jfLY9+uMJJ3hwj60vjf30lFJ+Lqb5hhgusBKwQ2I95pkD8BlfgXaXszWDnnH7pd0oBQ2jDu7Olh/y47qFp067eOBBmKhN8B2c8C4Yhk8tvXrRFJ5ueJU7l2aJDxy4eTvnMO9Q+zG4tkPlwQXLAb7AURIbnjjb3nc8dQt8XHXHCbUWurTtsAeL8kz9ebMPt0yHoUsf1czOxllLtfZChDLcZwxFrikYxbwP8XKnosytEbol5A8j6F4NpihfCUzcjqFt+fSM4oVNiNxeSNqxmLxQZxcQHs9syyT6wAPX0w+JdjXDB1wIoUKHpsbUuNHpO0q3jxljzmfx/DCzOfoPN5pW8gViRzy6P2PZTA0QdcV4KtTZHywMjG11UMhLYjEigMoCgLkRB4Zxgba1lqIjmvgr0HmuugnRVcF0POakCVXtH6X6BFjUAiH6Ow+KlroOGbcV1WXM6cUYUrK3UcVlaiZ4KebQSdVDDKUh0V91WFz1PZlMcH8hHwIEZBxZOgmP7hlQdAfo5xWjUb+Ni+3sXeHSWdgLGJfvJyrPg8xg9RKeHvXyO5qUcz7A4dBBb/FomIkzqNvGgWCOmwY6MTRDF28Qvc09nY5nbGx7FMWluNrNyu/PTlp3Ts9RlgmFTiawSw2/eAZGJEe/kYO6wmqxYYpqjVC8N58YutrJgfzqW3pBphu/oOwALWFqi4KUjWU+D9DlRBXoKejZ7an6cvbcDm5UkMuH0/516VQereZqAUHfvEEVs9HPR9kP10wRmno5RG4zZWNq0MsfyX2az6LZ+aDX5hyKh1kAGSuKx0E00oDdDBv6boPWc/yF0J/sVoWoghY29l8qjpTHlH8HnuZNLIaVx5Tww3PpaK9dDdHD3TkewRhqNeq26UBIl+qyCEWQM9xTA7uS4D5yUkbdxCvCsVe1YXcA1AKwjrVY4eKMeMEuJWq5fAtLRP0PRVqDJ2hjyAiA7+Zcb5FUY72cFa18h/OTSYQASjGnI+W9e40CyKHet2l6lAvPk+lv68kna9WxMW6T6sLEeUU9+LinzKqLhc7JgPsu7F+J+pD85zDj+Yd4ZhOlQOVPgDxurFOwMOUiALf21JKDCeM/rXL7XO27/lYMf8L1Oj2bPpSa4dMbB4YtpxXFkZ+U3nGaud2P+h7B1KaaMDAWNTuKMd3/cn4l+ECrvj5NpptIIpjwLJEpEfK1ySclGaE71sOvTMYcd6F3aHoDRBDlqBpCcXVyoiGiBkphTdvXnyNL7/1M2ebXZCBSsKpYTmnTxoFjthkYq8bCE3y4I1YiB3tBhLwBdNnSbOgrYKXTuXC+uu4oLrlrJifn3iatWlQfMUQLBYNS6+MZWOvfLp2CsXrG0g7EbIeowp78azc8f7DH72LhKj/Ma5a/XZl9ySMOdCIiLX0bytsHGJncZtPGiaUL+5h05n53DzEysxTEgCWCDyWcj/HwQ3c/eYq/l05Go69PQx8dn1rJgPt43QcIfrSPZzqOhX2Ld9D0+e/xzN2+3m4bd0I8ILQC9aHSlnPyT/Y/zBRsweP4eY6tF8/PSXKBTtexk+pSlj41i3vA6vzzfu+PbvSCEvO58GTROAkKE8QrsAMcwkcV9AYEMxB/FvXyxg9E3v0LR9OGO+zQbfbAL+oaz6fS0tTm+KO6IocAJg2piZfPzUlwwZdxt9r+1Z+L7f62fe14to07MFiXXiDUWgp4IWj4qbBMHtiKNvoc9FK+OiKVnDjMTNiOcY9XM9NiyDbhc1MkKyQ3uMKCjXZQU5NfDu/RP4+X+/0/WCDjw7/dgrAEnOy8Z36BqIinr+kKN2cF5g5HiUcbGV4E4IrALnecbKQzkMU5Kzj7GfzEHbEuzdup/nBr4GCt5bPpp6FRGhXIA338erN72LZtVo2rkx3S8pMvlo7isR18DC1ZaI/6gSMkuglJF1XEb+tKRfDYF/IPaLow5Nlsz7QPJBS4Cw649dxipGeRTIHKXUaIycj8IiiiKyrMKkKhelrUKKv/f1eMNskrbPXkrbQyn9uMMlzPs+uqiVrli32M1dHR6lddc8/lkURtI2J09c/BUBn6FkvB6Nm5/cS2aalbz0+WiWCP78MYqcDCtJ25LZtjYcm8NLwBdkw+rm+LW+vHTjXGyOLPrfMoLqte189EI1YDW/TbqHCUvbUi0ePn1xBZPfjiM8KpZJq9106pXK799UIyO9BTM/d7HlnzTS99vo1i+LP76rxcAHOxPf9An0kI0Fs2qzZ/Muel9zJi//dCl+byaZyU8hgZXkZllwh2NkvAP//PwIe7b4SNrsZOjrqyjyQXuQwFqUrSU7NwrZae+zefk2xg+dQFScTrdzcxDVkIdG/8Znr1dnxsdxZKYYq4+JT03mixe+RmmK0b8+TZuuEyD/U9Azjc9edMPOfYizVg/pgEKXOFTEE2DvxsRhk5j25g906NOGkT8MK9Z+0Q/L8Ob7WDJrBUkb9/LVazN4/NP72LJyO5NGTqdui1q8v/I1JPctY1Mw9w1okU8i1lbgmQbZLwE+JG5KsTyYQkKbMCLe9lGt4dUk1pyDZPZEbG1BfBBcz94tW5gytha7N+6hQZu6hAIh6rYoq1aWQWZKFjaH7TCrlNAhjwf3zeaz51txxmU307536SVTJP16Y2Un2Sj3NahqRT9fFTe5WNuY6tHUaV6LUDBIfO24w8pdXnTPz5D1ILhvLLbrpcNlp99NZ7FzfRKte5Qs4nlAeeh5H0LOK0jYvWgRQ456fqUsEP8zSDaqrLplwe0Y0Xt7gKPMbXENMn4/p9iGVeVRIAduBw/24AtQejjUCaEsE9ah4b2Hvn/g7uNwyuSAiQxempxE+zNz+XlqE5J3e0u00VU9rrpvDdPfT2DtwtTCfqGAYvoHCUREB9m91UnDVvlsXmVcGFp0DmPd4qLYgx1r9/DThOUsmGXkO/zxXTT9BqXjChM8eaCLEMz9hdWbXUx+O55QEEQPsXj2Xr58pzmZqX7mfluTB8bfwTfvPUp2hpUX76hPSpId3RZGnWZfMP2dFezetBcEfpwwnwfG3U5c+Et8P8GDNz+BnyYnMm3XxQiCnnIeZ5wfIPlRP43a1EZpkWTl9sChfsXpDiHBrXg8Edzd6TEEuPetm3FHuujSO4mHXt8L9rqEPDqX35HCmVdcRL22hl17yitGJVrRBafbYVSw9f+JsZ+9BqWYFABadm/K8MkP0rFfW1SYsbJLqLMZPRgioU4cevotENpjJGha4nh4wj3Mn7aI3tecwbCLXsLv8THppen0u+lsRIQW3Zqie+cW1BXTSN65j9++m0afKywkRDyN4bS3ccCBDJCdnkNaUjoN2tRDxbwH/qXg6FXwr5BlnIOeCc6LIbSLN4ZksfIPI/LGGeZkZt5nOFylm0X0nDHsXDWTe85JwBnm4oud40ptqyKeMIp2WpuUODZj3E989+5PLPhmMZN2vVfqPNjaGA7pUvofIH1fBim702nWqREfrCotZuZfEFiKYQY19ofxe/1kJmeRWDeBoR/eXY7+GwGrEcJ8jBgRjGUnpqq4r40cGvvRKwFDKR6/7aCrCuWJwip9y71KQx3yvCxlUNr7h/edGBgKRrMI7z0by/b1tYyyDQf1bdYhjwat63PPO8Ox51/PptW5LJsbWXg8fb9hBssoMIdtXx8B6FjtsGllNkVLaEV+joWfp8QXnpeI4AoTRky6iuGXTkEPCbf0iGfIKz4SakcSHuXhqYlBbujUAM0SpO91XRh0/35q1P2Exm3dJG0J0OGcDiz/bRs9zv6KJ6+tS8BX5OfZu3kfj/d7gRGfROP3eRFRxNWsjbiuY9eSu6lVeys6Hbh06FCcUR1Z9ftanrxgJBHRjfls8Xos3nlYcx8mJrEdmalW2p3dmulpHyOBdfw26Tc+GrGSuMTOrF/qJyL6LyYsnEW+60vOv7k386ctotulXYiMi0CF34VkbMbYx95eaJoQEdYt2kRs9WjckS5ub/swouuMXfw8DVobkViX3XcBvQe6CedB8OcBNqMWl3cL8YltueSefjzQYzj7t6fQoG19Ni7ZgtVuYcK6MdzV4VHWzZvBu7O3okXcy7t3ZPPXzEms+bMFz0+MAks9iByJshWZsO5s/wgZ+7N4eurDnH5Rx+I+Bmd/lKUeWOujtBh277ucNQsfQmmK+q3rcO3wy8tUHgB4viboy0YkjoAvgK6X/v+plAVsLUo9dublp7Pgm4WcO8iFBDYUC2k+gBbzDlJQzUAC65HcMaiwW1D2IpPRHW0fJi/bw9NTh9L1wo5ly3wMqPAhRl2ngjv0Id2Hse2fnTz9RW+6XXZZYYWCMvtHPm2Y2+zdD9tOfIsMU6vz0mKBBmWx6o+1JNSJo0aDakYOyn8gD+VQ9mzZd+RGx0i5CiIqpS4EWgGFNalF5Lmye1QkUsbzY6E0n4rxXA8ptq9zlegBsGFZGBuWpTDrf/fT92o7W1Y7SpGlKFos6NeJTowkM/lg5WHMpZQirroXo1ihonm7fJbPC8Onb0HXi8ZZPi+MO55J4dkbnYy+x4vVFoHSrPS/xcPDF2zGm6+BgoDPxuxPVwLC1PGJNGrpZf1yF3UaB0jeY8HvVZx+YXtOv/Jhrlz9CZNfmcWuDUnc3ekxtq3OoN/1ZzHvGw966G16DOjC7E9+R7NqZGdYuLF7M4ZNCFC3vpVGrXVi6vSkRsNqfP7i10wZ/Q352cYqzZNjfBa5WYpfv/Ty5bsPk5vp46yrujHrw1/Y8Pcm3v2jcYHyAFzXoGwtWf7bah7v9wKI4HA7+GLnOJxuC36PF1fgPsBwxe3bkcKrgz+n35V2+lzhQoXfa5TQz3kNlBOfewGblm1Fs2hc/uCF5KTmcN5Nvdm/I5VQMMSuLS6CoTo43JfT66pNrF24mbOu6oOW+By6ZzZk3oNEDkM5egDgcDsQEewuO79P/Ysxd7zH4Kev5LIhFxjO/oMc9am709A0hbJZeWfhS9idh7fZq+i3adh5HuOX9cMdGYerYJUlIiycsYSoxChant70sGM0aF2Xd/+oSdLqT/jzi5V0v/67EiU0QqEQ6xdtplG7+th9440N0fR0VNyUwjYRseHkZOQx5dXv2LF2N1c+0v+w8x4NSgsD9xWFr/1eP5oSfOkTkYwfUfHfHLn/IZuGrZ63Dk+uly7nG5+/iBfJuBHQUFpc0SqxDBbPWs4zA0ZjsVn4JuOTCik7EgwEsdpKv8xuXr6N3Rv30POKbmiaZpSM8XwH9g7HdfvdiqwYUp4w3vEY6cxnAx8CA4G/K0yiI3LoRb88q4qyKG8/ddBj8SiwXyaHH3LsoHGVolbjGiRt2lugPIq3i6upkVg7kdiEFCwWIRSC9SuMCI7t65cWzu50WzjzonymvFsfyGL1X+EYysdPdk4HsjNWEQoqNAu4wh2EQoI3z4tfupCXu4YOvbK5+GYXI29X2Gw69z03BU2uoVmX1oj8iOiw7Z+d6CGdXycXmdfS96RjsVnoOaAJa+avIiXJxqI5zdletyULfvgJmE3Hc07j4xEHbOiC1QavfZPDsrn5THs/gfFPVwdyUQossgCb00anc9tCYAFGgb56oEWj577H1lWGbVrXhejEKNyRbj7feD2B1If4/tOabHnlDe5562ZevPp11i/KZM2i2vQZ3BvlvgaCm5G86qxZ0ZEpY99iyLu34Qp3cdZV3bnqgZpI7hgIf4ARU4YSXysWZx1jhdHnmur0uebMoq86/xOjKGX+pEIFMn7ZK+Sk5xJfK46Xb3ib3Iw85n75J5cNuaD4f5OezmkdZ/LMV5cSX7drCeWRl5XHZy98TYe+p9G5n2FjP5BjUveQxcXTl73CX98tAQ0m73qfuBqHL3euHD25/+IFePOsDLX+SZ9rjXPKy84nJz2XWRN/Y/Kob+jQ5zRe/PYWRNJRYcUTGseveJWfJs5h7JCPWP3HOgYOvbjCajm9/ddI9qz/lkb1ngVr+a3h6xZtYvRNYzn3hl588tSXKE3x9sKXaNS2PuAAe2fDl3EYU90BYqpFI7oQXzO2zIusN9/Hd2Nn0frMFkdU5Ify9n0f8d27s3j043s553pDmem6zv4dKSTWjWdIj2EggsVm5cwBXSH/MyRnNGjVUYlzjmquw1GjYcUVAS3Pf0d3ERkMZIjIsxh7ox/dJ1khHF1E1r/jEMVQOP+hfhZV/E9g77b91Glas1i3PgMz+W63jXMG92fHhmy+/zSBUOjQsYrm8ubrjLyjFuv+zip8z2YXrnusBpHu6YSCRt8WpzfDYrNSo2EiCKTt2cuuTcK+3bXoPmgMb/wQy/2jk7iuYwOGXTyW2s1q0vm89ihN0aR9A6w2C7qu8+w3j/LO3y/w9LRHOXNAV6xqG4+/u51BD8HlQwfR9ZIBuCNs2J3gz0vmtF4tsTk1WnbK4/1FzWlwxpdc/sQ4PHlF5yQi/PR5AF++n6/HfM/bT7aAyFFGqGbeGLzJb/DH5K+o3TSBd/4exYR1Y/Dl+5g73cLOfa/y0YuRzP1yAU+e/yJNOhh3Z6075xXU5wJlbYyWOIeJI8NYOGMJi75fxtlX9zBMNnnvGTb4vA84/aKONGpXBz3zAfSMuwsLBRZ+PZEjwDUIFTG08D2Hy0F8LcOZfPvowUQnRrJh8WZWzyueYSx5n0P+RDp2focGbUrmjvz08Vy+fmMmIwcZJf4zkrPY9k/pJfb3b08xnuhgsR75Z6rsXWjUri0iFmo1qV74md/S8gFuan4/Oem5SEgnIjYMZWvDvF/v5KNntuHzFG0uanfYOPvqHpwxoCs3vzQIn8dfYh7RszhS9YpD2bFuNzPG/YQ3v2iusKgwmnS9BlXtH7TokeUea87nk9i1PomfJsymWv0EwqPDiKtpKFelFFrs/9AS/0BZyq6hdoDG7RvwddpEPlj9eqECSd6ZQtreooTGHz74hY+e+JzhF73Em3e9z9q/NpRb1i3Lt2GxWNi2uug7HnnNmwxudC8P936GFl2aYLPbigIsbG0AO9g7IaG0cs9TmZTHhHUgiD9fKVUTSANOkupgR1qNHO3q5Ejty1ZaSoHDbcGbF+LglYoe1Nl10N4OiXXi+XUqrFkWS17WL+RnFeVIKA0ObDrWb1Aa82ZGE/Rb8ZfYQFiRWNdJrwvmcP9FjTnwNbbtWYv1izaRnZoDwJ7NKVx0o4/eNz2HZq1Hs7M/YNP6z1BqBst+3c4dbR/mobE16DdgGx36JjDxlb7M/t9c9q97CXdwE998ewtzpywAERq1jCArqw13d3yMF79/gumb9pCXtY+BLT8FNHRdZ++u2tRp94IhohZN5/M78vtXK9BDxmcRnWAhY3+IoD/IzPf+YNs/zUjetglXeDOGvpHChmV5iOQTHu3GarPywWOf8c3bP9KkQ0OsNitBf5Bq9RO5751buerhXiQk/m1s2FRAbmYenc9rx/7tyfS/tyinQUU8iOSGFZUQCe0G7yzjewpuAlurora2ZqioZ8v8D4hJjDKKOGqKt+7+gD1b9jHqpxG0ObMFytkX8f4A7stL7dvl/PbMmjiHHv07o+s6t7Z6EE+uh5E/DKP1Gc355dM/aNKxIY3a1ueRiXczpPtwYqpHER5dvryCV355Cl3XsViK8kREFxTQuF0Dbn35Oi68oy+hUIgXrx4DCqISorjioYsL24dHh/Hge3dwfcN7+OLF6Uxc/2bh6kfP+wxyngfXFaioFw4rixxUPXr4RS+RvDOVvOx8rn6seLLgkZIewVhFrV+0ibZnteKK2+YiXp2zrm5Fq97PlOtzORwHTIZg3Ozd2uohNIvi8+3jiIyLoO1ZrYitGUNETDgz35vNqj/W8dGaN8o19jPTH2Hl3LV0u7jIn3Rgn5dNS7cyM/fzYu2VvTNUW4qk9EJSekHcV6gy/F4nC+VZgcxUSkUDo4FlwHZgUkUKVT5KWxUcrEiOhDrk8dCxSjsmlLUKEQFvXgiH237QsUPbQ/JOYzPHfVuTyUk7YC4SImOC2B0FX4dSzJ4ST36OBb/P6B8W4ypsC0LSZi/3nNeYvGxDeWgWBZ4v6T0gC6UZMp3WLY/elztp2rkRqUlpiAh9b7yCDue2IxQKEQqG+OSF3cydHsWnI/9h7YKVePN8vP1IiPsvasiEEXMLxd++wcmPH69l/44Uvhg5nf3pt+CMakfTjnWIrhZJ9/5dsDmsTH/r+8JzXbdoV0EYrhAR42biqht44NXdxCT6QcGaBRvISHWQtNVFfJP7uPvFdB54pz21Ghv3Jy27N0PTFG3PboUzzIHFZmHH2l2MvnksifWboIVdV1jiYse63VxV8zYmDptEyu40tq82clc8uR5CtECLebtwR72gXosFv19LRt7NhZt6pSb9n72zDo/i+v7/a1bixBNCCEnw4BLc3a3FCsWtuDsUdyhWXItDcXeKB7cIwRIkhLjr2vz+mGSTkIQESguf76/v5+EhO3vnzp3Z3XvuPed93iecm8fvEhUq7fJEbRCX952nX+lRPPrLK8NnuP7REjqMbs1r73eoktT43HqOVqvlyDp/bt+ajcy0H1nBqZgjGx4toefMzgiCgKGJAaIIBsYGHP79NMsHbmBcyqRYpEIhBi3ridJQie+dl1n29zEEQchgPD74BbP8xhw2ei1l4/gdbJq4i3YWvZjV4TfsnG0RdSK3T9zP1E9yoorkRBUatYak+HTsQ+17pEqa7zOdkx7bpu+jmcFPnN0quWGqNK+AsZkRJatLwX1/zzdsnrw7w0r/U5jZfglTWs1n2/Q/sSkymkELClKyTq9cnfs5kMlkCEJKzCDlp1+4nCt73q5n9MaBuJZ2puPY1tmer1ap+bXNAqa0nIcqWY2lnQV1O1bP4Mqcd3oyjbrVYc7x7OR1dCmkHSRa+HeOHMUUMzSWUjSNRFGMzrHxPwBJTLFhyqs0um0asqLxfm6sJKtdzMd9f3w8sxEytTQhPuqjDOwskdnoKQzkaFTaT7TJnq5saaclOlxJrfZVyWOp5OzWm+Qv6sC7Z4H8NKEdMoWMXXMOfnQ70gtzW0NiwlTIlQJadcb7rf1jWa4deoLCQIFOq6NweVfW3F2IKOpQxXvSznYBGpUGmVxGqZpudJnYjogPQSzptxm5QmSL90xMLaz464/xVG3hyr5VzuR1tsWlpDNKYyWX99xArdIw9Pc+mFtnplqGBUZwetMFts+Q5CFGrv+FlinyKapkNc/uvGBi0zlotTocC+dlxqHxqJJUDK82GXsXO7b6rpA0v1Rq9i8+xvZZ+3EtVYB1DxZz98xDJreQ3Ch5XezQapIxNg7F1FzE94FkgMdsHEShsmluqct/3tCv4vcHbeLZ3VdMb7cQrUaHS0knVt1ZINGVP4GkhGQSYhJIiE2if9nRaJI11GxXhRmHxuFx/B6rR2wh+HUorX5pzIi1Az7Z18fwRuN7GQAAIABJREFUuuHL+EYzMbUwYV/gRuZ2Wc6tE/fRqjXYO9vyy5KerB+7jf6LulP7x2qZzn/r+x6tWpPBFSeKKlDdAWX5T4pKjms0kydXfGjWpwGj1mcWjhxUaTwvH/jTon9DRq0fmOO9LOm7hnN//MXAZb3wuu7Lk8ve9FvYjWa9vyyTQJWs5sH5J5Sp7YapRcbdXURQJHKFHAtb88/u963vewaUHQ0IrLm3MMP35XMgakNAF42gzDmOk1t8MzHFlIvXAFxT2wuCgCiK27/2YD4PWe0y0ge5IfPEmxsjkt3knP7cj49nRu6MR9bnZzQen4aBoQ63Kq48ufYWEIgKVSCTQeibMLxvSKyjsMAIRFHk7un7vHryNotHIN1bbLiK1KQ+5xIWvH0ai3EeIwRB4MYxL/JYm5G/aD78n7zBvbFUC0IT9Tu9y1xCo5JWWTqtDs+rPiTGJlK7fTUQBbRqgVF1V1Kufmku71VzYlsCb3wuALDq9nxMLUy4sPMqoijy157r/DylPT+ObIm5jWRIIoOj2LfwCKVqFscmvzXhgREYGEmuj91zD7J95n4MTQyYdWwi9gVsUqTQ4daJ+yBIk4JOpyPgWSCDK03AzMosQ3JfXFRC6neasMAIZAJE6gyYs/MDO3+vy9Nbr5jUbA77Ajfqn1i9TjVxKuaIqYUJlnYWFK1YEHtnWz74hfDGJ4CHFz2p3jrz7zU2Mo5DK05StUVF3KoUxcjEkNfeT0AEYzMjxv0xhNfe75jV8Td0Oh1Ne9Xn56lpLjGNWoPHsXu4VS2K3SeS/MTkyyBqEUXYMWs/Lfs3YvKuEXgcv0cx90IYmRlRp2ONbIPzzm5ZSMkLBmBYC3/PN3hdv0HT3vWzZJlN3jWCu2ceUevHrGXgW/RrxL6FR6jXuWa240+PMZsGMWBxdx7/5c21g7dAhONrz32xAdk4YSfH156lQoMyzD+dMRHV2uHLa7MXKO5IvwXd0Gq1FCzz5XpbgtwecqA1fy/IDQtrB1AYeERaGqwIfGMDkh2y20F8/DorTa3cxkA+RnrD8nmUOSsHS4b93pdVwzYTEZSd8J/Up0wu6OMJqdfUaASCAzKKIut0Yga3R2qc5eWjN/r+5Eo5WnV6QyUgiqBQCsgVIpP+aM7QmgdJjE2SXC06kbioeJ7decGB0C36XYJWZ0Bk2EdfIwHevwiidC03zKxMiY9OIDo0hmLuBbl55DYadVqSnpGpIfmL5uOnST9w/eAt3j0L5MCyE+xbfJSVN+cSH53Apkm7eHbnJUd+P82Itf3xvvkcz6s+PLzoyfntV6R7jEnkwG/HuHf2MbXbV2Pc1sEsH7geUYSpe0dL5YgDIhCBiBTXSbn6pVCr1CgMFLTo35Cnt1/QuEdd8hfJh04TRvmW7pi5RjG5+VxK1y7B+MYzqd2+Oq0HNgGkuEIqrB2s2OS9nJbGXQFYMWiD3oBc2X8Tr2u+9J7bhUMrTrJr9kHOb7/CTr81JMQmMr3dIgRB8pmbmpsg5hexzW+N0lDJ0FV9M+xkDv9+ms2TduFY2IEtPsuz/LaI2jBKlVzGpmtG3LzyCxsnHcLQ2IBjMTsoWrEQf0zbhyiKXNh5lb/2Xmf7i1VMa7eIhNgE5p2cjLFZ1vT1VExuMY+IoCjUyRp+HNky0/tWeS1p0rNetue3HthE/wxzA0EQMLfOQ9m6JXGrWhRNsoaxmwflfGI2yF/UAa1aqycbfC7e+Lzj1okHtOjfkDxWaTsxQRDoMDp7F9c/Ae+bz1jcZzXtR7b6rGf6tZCbHUgloKT4udSLb4rPiYFk9ffn3uqX86wjg6K4uPsakSFZewVLVCtKp3Ftmdt1OZpkTbp3pGvqtBATlpBt/6XrlCDAN5CodP13meBO1dY/cGDZcW6ffIBGpUFMSWCr9UM1wgLD8LlnTgG3/Lz2eouBsQEatRatWosgE3h+3498Be155xtI5eYDsHN6zAe/NNaIIAhotVpOb7pAXGQ8AFqdlvjoBHrO6MTGCTuRKWTMOjwel5JS4lbXST/w5LI3ZhYmxEbGI8gFFnRfSZB/CKqkNKn5i7uu4XXdN8t7vXfuMQDXDt7CwjYPibFJCIKAmaWkAuDeuCx953Zh3Rhp7bN1yl7WjtyGKjEZuULO4ovTKV2rBI8vezOhyVaKVLjMqtsLOBT+B/sWHWXTxJ28fOhP64FNiAqNZtWwzbg3LkfzvpJb1cBQSZOe9bi46yo12lUm5F0Y+xYe4eTGC4g6kYJlXajW0p3z26/QrI+0epYrZBiaGJIUl0ReF0l6x9TChEk7h+NS0imTG8zZLT+iVkfhcp9wj8iswbARDoXCqWbehFObn1GpmUQb3jlrP+e3X8E2vzWOhRxo3q8B4YER3D3zEEEQ8Pd6lyNdtUqLilzd74Fbtdy7WLQaLUOrTiL8QyRr7i3E1vHz5dktbM35/WbuGVvZod2Q5jTpUQ9jM6OcG2eB2Z2X8dYngKiQKH5Z0vNvjye3iI2Mw/fOSyo2LKMvIfDXnuu8f/6Bo6tOfxMDkpuCUvuB4aIoZlMS799DxhjIP4ms4ivfDjKFDJ1Gl/37+p1JZkzcOYzFvdfodxuGpkqs7K2o26k6h1ecQqvRok3p28TcGI1Ko5+wh6/ux5FVp2nWtyEbxqZtOA1NDFAlqVEo5YzdMoR8hfLyW981hLwLp+OY1myf8SdyhRynYvl44xOArZM1jkXy4X39qXQtAdoMacbgZb1QJ6uZ1m4ReazMuH74drb3WevHqrz2fsfEHcMZ33imflflVCwfoe/DSY6XKKdGZoYkxSVTpUVFRq4bQGxEXAZftCiKHFl1GqWBgrWj/kAEXEsVICE2kXFbB1OquhuX991gYc9VWNqZ66VBgl6H0L/MaAD+eP47Nw7fYfXwzSgNlZyIz8imScWKwRs5sU6qkyLIBOwL2DL35CS90UxFUkIyGpVGz7Y6uvo060Zvo1A5F1bfWZip308lp+WE4TUm8/TWC0pUK8rKlMk4MT6JthY9EASBBWenUqFBzlUc/b3e4u/5ljw2ZiTHJ1Prh+yrFga+CuLaoVtsmbwbuULOwnPTKFM7a3bR0dVn8Lz+lOGr+2UZC/sesGvuAY78fobJu0fk6ll9LYysNRXfuy/pOqU9PaZJSZnhHyLZv+Qo9bvUpnilwtme+6/HQARBOI40k+ZBKmF7h4xiim2+9mC+H3w/xgP4pPEAsjUeMrmMK/s99MZDJpeRHK8myD+EfQuPZmqfEJOoZ3ABPLnmw9un7zmz+SKNutXmwq5ryBUykpNUIEqusnyF8lKialHCP0SSGJfIq8ev6TW7M/6e7zAwVpKcqGLc1iHYO9vSq9gwQNqh9JjWkYAUscOHF6SKhe5NyvLa+x3h7z9i5whw8+hdCpYpQPFKhVl9ewF9S41Cp9XR8pfGbJqwE4DCFVyZsmcUf+25Tot+DbG0t0BppOTm0buUru3G5km7MLfJQ5+5XYmNiKOAW34EQaBgGWe6FPiFcQ1nsfbeQup2qoF1Pivyp9SyVqvU/DFtL8kJycgUcsIDI5ArZLiUKkCzvln74UVRRKGUp5ApEhB1IsFvQrl39nEmA2JkYgjpdhp5rPOg04l694hOp8uQ0PclxuPFAz92zNxPiWrFeOMTQJshaTRnAyMlhcq6EPwmNNdJZ6PrTCMxPglRq0OmkLPixhyKuWc9gc3utJRXj19T+8dq1O1UHXsXW1YN20zjHnUpXrlIhrZrR/2BKIpUalxOv0tLj8S4RI6uPouzW362Td9HufqlGLys92c8ib+Pn6d04OcpHf7VawLkLWiPj8czHFzTauvY5LNi4G+9/vWxpOJT38Ql/9oo/sNXQSp7y6WkE298AtBpdXgcvad/X6LUgomFMQnRiSiNlKiT1JStV5Inl32wtLeg7dBm1OlQDUt7C4JfhxIeGEWLfg24ffIBiKBV66jfpRb2zrZ0n96Rc39cZlbHJXQa35Ytk3bjcewelvbmPL7srXebvXr8mrJ1SrLJaxl+nm8pVMYZE3NjujoPRBRF5Eo5BoZKHlzwTAnUOyCTy3EoaM/d0w8lY6XV8erRG8bUn86kXSOYe3IyMWExvHjoT4mqxfC64YupuQkFijnSY3onEuIS6VZwMBFBUZJ7LsUuCoJAg661GVFzCjqtji1PVxATHovSUIk6WY2hiSGCIFC2jqTEGxEUyerhW7l28BaiKOVVWNpbsHLIJgSZQKkamRVkQZqwj687h6gTqfVjVR5e8sSxsANxUXEZciSyQoMutShXrxSWduZMaDKLx1d8mLB9GHVTJC9yC1WyGgNDiWywc/YBPI7fo2ydkhyNyhi+lMvlrHuwONf9gkSxfnzZCxsXOyk/x8Uu27bVW1ci8FUQTXrWpWpLd5YP3MDJDed5ctWHDY8zijb2W/gznlefUr1N1ovlY2vOsWXKbkzyGJMYlyTJ8PzLBuRjxEbGsX7MNsrWK0WTHvX+setM3D6M4av6ZmKOfUtka0BEUbwCIAhCQeCDKErk5JQa6f9cbvx/+HIIAp3Gt6Vu5+oMcZ+YbbNU90+1lu64Ny7L8oEbkCtkjNk0SBILBJYOWMfpTRcZ+FtPEmOTuPLnTRDA3tmWO6cekJyowqWEE2e2XiIsIIKjq87oDdLJ9RLDSq6Qo9NqiUwhBzgVc8QpJStfp9Nh72LH+xcfUBoqKV23JHdPPsDE3Jjg16FotToCngcik8v0hg/gyRUfTqw7R/tRrdgyZTcn1p3HxtEKCztz6nWqQUxELIMrTUjL5E7HmTAwUtJ6UFMs7S3QaUVEHfjees787r9j7WDJuperSIxL0ruIYsJjGVVnGoEvgzA0NUQmCLg3LotVXgtK1XQjISYBp2wCsQXc8pO/iAMmeUyYsmckQf4h9C01Cr/Hb6jZripFKhTM8rxUyOQyds09yIOU3dn8n1fw/N4rflnc45PnpeL4urOsHLyJjmNaM2BxDzpPaEd8dALdfv06K+e5J3IuE6xWSd+Fio3L0mNGJ/3xxj3q4nnNh/ajWmU6p8Oo1hStWIh3vu+xqJWZSluhYWls8llRp0N1jMwMP1te5J/AtQO3OL/jKpf23PhHDYggCN+V8YDcBdH3A+klMLUpx/5vF/v9H4QmWUOQXzDmVrnzHWs1Go6vP5fyt47pPy5i8LLetB3SjBuHbgOwc+4BKjUpjyiCVV4LdvmvZUrLeTy+4o1LKSe6TPyBme2XEP4+AqVhxqziqq0qIiBkoKGm4tFf3lRtWZGDS0+gSlTx5C8v5Ao5VvYWdB7XDuM8RpzaeAGNWkvbwU1Z1Hs1pubG5CvsQNWWFfnZdRDJCSqs8lrw08QfaDe0OcOqTWLlkE36axgYG5DH2hRRB+1HtaRK84q4lpLcRytuzsX/yRuM8xgjCJK439WDHiwfsAGZXMaWp8vxuu5LkL8k+Ni8b0OGLE9b6S69nH22OsDze6/44BeMTqsjOVGFQ0F7arSthDpZk2NtEIDJLebi9+g1giBgYKxEo9J8xJr7NF57v0OukOPn+RZRFIkKiWbIit76vA5RlPTScmJc/R38tecG68ZsQ2Gg4ETcTv3xUjWKs9k7awZZwPNAJjaZjVarY+S6AbTo1yjD+8XcC2cvWf+NULWVO+X+vEn5Bp+oC/1/FLkJoj8SRbH8R8cei6L4t+tMCoLQDFiBVBRikyiKCz7V/t8Lov/vQmGoYNmVWZxYf16fCZwd3BuXpWord9aM2KrPgyhRtSgGxgZ4ezzTs75kMhkyuYBcqaBB11o06VGX0rVKkJSQzPR2C3l4yQtRJ1K7YzWu7ZfqPcjkqVXkRH5Z0oOE2EQ6j2urzxtoZ9WTxNhEXEoVIDwokmGr+yOXyShT2w1LO4sM40yITeTZ3ZeUrVMSuUJOTEQsnfL1R6vWYmRqyO636wDo7TaC6NAYDE0M6TSuDfmL5GNRr98B2OG3BvsCtvo+f6kwlrc+AfRb2I2Kjcpi7WDJ8fXn2PbrPgDsXWxZcmk6G8fvxOPYvRR5dmeKuRdixNoBOSqchr0PZ0jlidg62bDSY26GDHGQclRmtl9Mi/6NGLYqc+b6vK7LJcHGES35YVhz4mMScS1dgMjgaHbOOsCDi0+YeXg8BUtnzDd4cOEJM35cTOOe9ShfrxSla7nh7fGc+V2lCftY7A7kcjnLB67n5IYLjN44UM8i02q0LBu4HgGBkesHZBrz5+Lds/eMbzSLsnVLMmnniFydExsZx8+ug1Lo44YZDM/nYO3oPwh4Fsjk3SO+u1X7t8C3TCQMFQShjSiKx1IG0hYI+7sXFqTCzquBxkAAcFcQhGOiKPr83b7/LyN1os8OmmQNw2tMQZYLCef755+g1ejY9vJ3ehaRAtxhHyKJDokGUURpJK18XUo50XJgEy7uuMrpTRe5vPc6CgMFsZHxesazQ0F7mvduwI1Dd/TyJRZ2FkQFR7Fu9DYQ4MyWS1RvXYmhK/vSon8jbh69i7/XW+QKOaZ5jPUKtanYPe8g26btQ2GoRJWkomS1Yiy7NhtDYwNkMgEtktLojza9USjlTDs4lseXvWnaqx4FS7swrd1CRBHK1C6RwXiAlHHu7/kW+wK2+kn458ntCX4dypnNlwh5E8Yv5cdxJHIbfUtKLii/x695+cCPfgu66RlTGrWGPxcfZdecg9T8oQr1Otck4HkgP45omSHx8GP43nmBKIqZBBlTMWnXCEZvGpSBxhsTEUvv4sNJik8GAR5e8MxkQDyvPyU5UcWji550m9qePiVGSsZcANdSzvoYSsDzD8iVcr02E8DbpwFc2H4FnU7asbQb1hz3xl++TixQPP9n7xbyWJmx/tES5nReppdp/1wkxidxaPlJ5Ao5T64+zTKh8z98HeRmB1IY2AWkylsGAN1FUXz1ty4sCNWBGaIoNk15PQlAFMX52Z3z3w6ELBIAcw+FoSJDLokgExizcRBJCcmsGrYZgDzWpiy6MJ13voGUru1GUlwSjoUdaGvVQ0+VzY5W7Fg0H8H+wXpacGpKTWocQxAEBAH2h2zm0V9e2OSz4trB2wS9DmHCtqEYmxkTFhjOsgEbqN2+KquHb5Emy3SwdrQiIjASuUKGVqPDwNgAVYpqrCATGLqyDwXc8hP8Joyo4Ci2Tt3D4BV9aJvCOvK64cu8rstpPagJbQY1zXJ1+mubBdw6cR8jU0OOx+5Eq9WSFJ/MrjkHyOtiR9shzfVt9y89zqYJO9FpdVjltSAyWCIOTNg+jEbd6mTqOzkxmd3zDlGorAvJCSrK1Sv1yQB0esRFxfOz6yDUyRpaDWxMr1k/ZaoJnxifxIUdV6nQsAwJMQmMrDUVENj28vcMsuWxkXF4XfelUtNyKA0k16NOp2PD2O14Xn/K83t+OBS0Z8er1bka27dEbGQct08+oGrLinrm2pFVp3jnG8gvS3rkWJPlWyA+Op7X3gGUrF7sH63XkYpvtgNJMRTVBEEwS3kdl8MpuUV+4F261wGklc/VQxCEAcAAACOyqxf9/w+0ai3N+tTnzJas3VOCIOBSyonXXmmP1sTcmISYRErXKE611u6c33GVN97vMLU0wzq/Fcv6r9W3VSdr2LvwMLdPPmTAou60HtgEnU6HmYUZyfERIMDYzYNY3GsNoijiXCI/kcHRxEbEUaFBKS7siKB07SKoElU4uNrT4OfaLOm9mpjwOAQZdBzbhu6FhpAQIwXye8/rgo2jFeGBkdjkF1jWfz13Tj/kzqkHIGTccdm72BLyRtr8ajU6FEo5lZuVp0AxR+6de4y/11uiw2NZ3XQOMrmMmYfHc0YtuaRuHLmDhW0ePI7fIzQgnHN/XKbLxB+zfIYzj4zn5rG7lKkl5SrI5XJMzU0YsKgHqmQ1o+pOQ5OsZtHF6VJinyjiVqUIDX6uzZoRWwFwKeWUZd9X999i38IjgMDp5D2fNXmYWZqy2XsZR1afoXy90pjkMSYyOIrhNaZgbm3G77fnY2xqROuBTVCr1HR1HoQoikzaNRy7/BllT/JYmWVamctkMgYu7cVr73csG7COlgMa82/h/csPrB+7nRb9GumJHDnhyVUfTMyN2T33EDeP3qHWj1WZulfK1Wk3tEUOZ385EuOT8LruS7m6Jb/YOE1oMpuXj14z8LeetBvaPOcTvlPkmlD+FQ3HZ0EUxQ3ABpB2IN9iDN8TBJnA2a1/ZbsTMTIz5MeRLVnab53+WEJMIoXLuzJqw0AmNZ9D4EspMBwdEs28n5YTFxWPIBOwdrAkPDCSK/s8ADi4/Dh+T95QvU0lhq/px7yuK0hOSCb0XTglaxTD+8YzQt6G8dvVWSzougJEqfhSeqbVxKZziA6LRWEgR6cVEWQZWVVbJ+9BJpexYdwODIwNaDO4KXfPPCKPjRkxYbEYmCgpWa0YDy95oUpWY5PPSq/i2ntuF9oMboaRiSE//9oBf8+3FHUvyJMrPrx/EaRnSHnffMbcLsvQaUW2PluBoYkhNdpkzwGRyWTUapd1YlxYQDg+Hs8QBIEPr4Kp2qIipxJ3o1Aq8Dh+D4WBArlclkHm5PDKUxz+/RQTtw+jXP1SuJZyplTN4l+08rx37gn7lxzj0LKTnErcje/dlwT5hxDkH8KNI3elwkRItOfkhGQEmQwH18/TVXItVYAVN+Z+9tg+F1qNFt87LylSwZVja87iceweb30CcmVAXjzwk6pXIkoyOIdu41b164kPfgq/9VnD9cN3aNKzLqM3fpmkSh4rM3RaXQYplP9FfFk669fBeyB9RpVTyrH/8AmkSo5o1dosM9ATY5O4dfJBpvPyWJuR18VObzxSXUCpxkMmk1GyRnGuHZCC4EojJVq1jhPrzumzqVPh7/kW7xtSYZ2k+GTGN5hBfHQi754Fcm77FVbfnk/Y+wimtl5A6ZqShLdGo6XrpB/pMulHbhy6zbtngSCAobEBhsaGxITHokpUcXTVaeaemszO2fvxCYtFlajG87ovNvmtadKzHn3mdOHE+nPcOHyHvfMPs3PWATZ5LyM6NIYCxR1RKBQsvjCdO6cf0sttBNVaujN8bX9MLUyxdrDEzsmGHtMlSum57ZeJCo7+rMp7joUdGLdlCKoktV4wLzWxr1ord+Ycn4hDQfsMxuH4urN88Avmyn4PBi3txbqHmXMufDyesXv+Ybr/2iFTcl16lKtbkvxFHChbV6phUrlZBewK2BAdFpNBwM/Q2JDNPtLi4OM4SW4R8jaUPNZm/xhTa+vUPRxYeoKqLSvSd/7PvPZ6S6tcynFY2lugNFRgamHKT+Pb0W1qh8+unvja+x0Tm86hUpOyjN0yJNfn2bvYotNqyfuZhjk95pyYRFRoTI6VJr93fEsDchcompJn8h74Cej6DcfzP4fsMtA9jt4FpMRCQSZDnaTm0SUvpraaj5GpIUnxyWlxCqBs3ZKUq1eanTPT6mP/9tcMDi4/SZB/SKb+r/wp7VDsnG3I62wnrcjlAgaGSkSdiMJAgZ/nWwSZwMtHrwFQKpV0Ht8OIxND3JuUI9AvWBL+FWHr85V0su+LVqsDBPK62KE0UEo1ThCkuhQpuSh5rEzpOKYNtdtXo6vzwJTx3GTL5N3kK5SXLU9XpFT6e4NcLuOt73tMzY2p2a4yxasU0U/2UaHRLOmzBrlcRvHKRShXr1Sm+8wOWcU2QHK3fRx0XjlkI1EhMTTuXpefJrTLts+tv+7lUQqbbe6JSWg1WjZP3o2ZpQldJv2oN0j5CuVli88K/XkKhZzdb9ZlmZxo52TzScXeT+HxZW8mNp2NjaM1O/3XfFEfOcHMylRyj1qa4uyWn4XnpuX6XDsnGw6GbkGQCV/MFvO5+YzIkCgu/+mBdT4rOk9oh6l5zm7y/gu789PEH/6W1IpcIf+fNx7waSmTrB3EKRBF8dDfubAoihpBEIYCZ5FovFtEUfT+O33+X0X6oHVR90L4eb5Bm07y3dDEgOSEtPKjeazNSE5QodVoUKcTIvT3eptBmDAVj//y5vFfGR/92lF/sPTqbO6eekhC7EfS9CnB8aiQGGYdmYAqUUVsZDyla7mhSlJjZW/BjyPsyetsy/YZf5IQk4hcIdMHfIes6EONtpWZ2HQOqkQVszv9hqWDJeHvI2jauz7ObvmZe3ISgS+DGFp9MqoEFfExCcTHJLBj5n46jmmDha05ax8sRpWoIjRAKpZlaGLAjA5LuH38HoqU1anCQM7JDec5tfEiJzdeoGmv+shkMsxt8lCjbWXC30fgWDhjXuyq4Zvxe/yG6QfHflFdiPS4dvA28dEJFKlQEKu8ltm2656ibdR1svSz873zkkMrTiJqdTTqVgd7508H2r92IFarkYQz0ysnp0fI21Au77tJw251vngi7Dy+HQ261sY2/+cLK8KXSbqkR6MedUlKSGbbtH3sXXgESzuLLNWFP0aqOvB/+AQLSxCErSl/2iMlEl5KeV0fuCmKYuY00n8Y/z+xsAyNDbKsSQ3oM76zg8JQgValIcNHm4XIcB5rM2Ij0kJbCqUcTUpcxTa/NUuvzqJPiRFoVFocC+elgFt+Qt6G0XFMK5YP2oSo06FWaajfuSaTd4/U97P11z28ffoet6pF2DRhF25VijBmy2Bc02lAPbz4hPGNZ2can4m5cQapja1T97B7/iGMTA1xb1yOOh2q0aBL7Uz3HBoQjuc1Hxb2WIVOq8PASCntaESRKi3c8Th2F6t8lvz5Po1aq9Pp6Fl0GOGBkcw/PYVTmy5QoWEZfuu3FrlczpQ9I6WaJukQFxWPsZmRXg111bDNeBy/x5wTk7J0FT279wrv6760/KURhsafLjCVHsmJycz/eQWmlmaM2TTws90zXwNvfd9jldciSz/9hKazeXjBk9odqvHrvtGf3ffcLsu4f/4Jiy5MyxAv+hbYMmU31w/fYcahcVnWQfm/gH+dhSWKYu+UC59DknP/kPI6H/DH1x7If8g4gWdnPABsHC0J8gvN9v2CpQvw4r4j7110AAAgAElEQVS//vXkPSNY0mcNqsSMRqdBl5ocX3sO55ISa6uAW36KVy6MTC6nx4yOgIDSQImhsSELzv2Kg6vk2/d78obO49oSHRbDiXXn8PF4Rgf7PozdMoTyDUqze+4h5Ep5ys5DjgiMqDGFou6F+HlKeyo0KMOmSbsQZGBibkK+Qnl59fg1JmbGTNqVMeHM3sUWmUyGRqVl6Mo+2ObP2iVjkscIf6+3lK7lRqFyrnQa2wYfj+eYmBtjYKjkne97/eo+FaJOJDosBgS4evAWl/Zc5+axe4zdPJjX3u+o2rIiIBmaNz4BRAZLtUEKly/I6jtSzuvF3ddIiEnkyRWfLA1I8UqFP6mSmh0MjQ2ZcWj8Z5/3NfGpybRO+2q8uO9HzR+qfFHf9849JjE2ief3/LI1IKIo4u/5FqfijnpNr38CfeZ2pc/c/7znX4Lc5IE8FUWxRLrXMsA7/bF/C/9XdyAymQxRFClY1hm/x28QBDLsHlJpuOkhyAS9SOAvi7uzfuwOANyblsPn5jMS45KQCQLG5sZ0n9aBdaOlVX2lZuUIfBlE+fplGLy8F53y9ScxLiklOC/J2C+6MIU984/hffMZE7YNJehNKBvH7cDI1JBxW4fw+9DNRIfFULl5ee6eeZRSpl2kWKXCFKtUGGMzIz74BTNsdT88rz7F+4Yvh1ac0o/drUoRRFHk+X0/Jm4fRp2O1fUsmvSMocjgKDrnH4CoEzExN2bX67X6BL6PsXPOAbZP/xNbJ2t2v1mXZZtUJMYncXnvDSo0lKS4X/sEsLD7SlRJanrO7ESnsW0ztN88eRf7lxzDrVoxnt15iYWtOXsDpAQ5rxu+eN94RtuhzbItYRsTEctbnwBK1XRj78Ij7F9yjIk7hn9xotzHUKvUBDwLxLW087+SU/A5SE5M5p1vIIXLu2YY24sHfry470eTXvWydUXtW3SErVP34t64LHNPTv63hvx/Ev/UDiQ3++KLgiCcFQShlyAIvYCTwIWvPZCvh++J6ZvzWNqPbolDIXtEUcTvsVQxsFb7ahl0dT42HiCtng1NDBm4pKeekQNw/6y0skstaZIUl6w3HgD3zjwm8FUwdTvV4NXjNyTEJOqZXalj9r7hycOLnqgSVdw981ASQxQkxtW6sdsllV0RarSugqWtOdXbVKLr1A48u/uKUxsvcHjlaW6ffEBibBK121ejqHthKjRKq5vge+cllvaW7A1YT4nqxZjSaj6qJHUmumkeazOKVy6CY+G8bPFZztPbL/i17ULePXtPcmIyDy56okqSdmqVm1XAztmGpr3qZ/usVclqbh69y9ape1g+cAMzfliMg6s9to5WJCcko9NoaZxODE+VrJZyUFImPntnWxadn8bqu2mKO6VrutF5fNtP1j8fW38G4xrO5NTGC1w94EFcZDx3Tmdmyn0pZndayiD38exdcPir9fm18GubhQytOomDy05kOF60YiFa9G/0yTiGTC4trFLdhf/h+0NuEgmHpgTUUx3PG0RR/P6+qRmQVe3yf7r/3F8z/Q7j0PJTGOcxyhCj8Hv0mnyFsxI8zljoquPY1kQERXJhx9UsryOxtNKC7daO1kQERiAIAi8f+mWIf0jXF5DJBep2bsihlZeIDY/D++Zz1j9aTIHijlzZ78Fr73f69sUqFeLPD5J44cVdV6VaIoKQkvMhJfrdO/eYpf3XIep09F3Qja1TdqPT6eg8vi3WDlbsnH2ABxee8NrzLU161tNLa9jkt8bIxICVN2eQnCgypt50XqeQAOzyW5MQm8jlP2/SqFsdxm4eTPFKhdnlLyVErhm1Fa/rvsw8PB47JxuCXocwtdV8Ap4HIgIubk4olHK9kS5SviCzj03E2MwIK3tJh+vumYdMaTWf6q0rMf3gWBp0qYWzW/7PmszC3ofjffM5eWwkzr+5TR6m7BmFx9G7NO2TvaED8Lr+lIU9V9F+VKscE81kMkH6ZnyDOElOUBoqEUVRX8P+c9BhdGuqtnTPRHL4D2kIC4zg/LYr1O9S87Pzfb4GckVjSGFc/S3W1b+D9PXCs5rI00/y6dt8icHJzbkfGxjpde+5XdkyZbdEY9WJJEQn4ljEgdo/VmHfomN88Asm6E0IgkzA0t6C2Mg4jIxVxEUJGbq+sv8mgS+C0WrSjIQgE0CEfEXyEvgiSH88r4sdblWL0qxPfaKDH7Oo9w50OkHfV6exbfhz8THqtK+CU2EYv3Uo09stpHilwigNlLTo14jtM/4kIjCSSk3LUb5+6QyS5PkKOyCTCRQs7cyCc1Oluhl2Fmg1OgyNDchf1IEmPetwZvMFZDIZEcGSxHuTXvV49fg19X6qScjbUH4pPxaNWotOq2PMCg1NOj4nOGQNrx69RtSJlKhWjDZDmnF++xV0Gh2WeTMKL4qiyNFVZxAEeHTJi8Y96rJswDre+ATo25SpU4INTzLWofiYfvv+ZRByuYw3PgHIZLJP5lKklmuNCIpk9d20cq2Tms/j3bP3dPu1A1P3jtYbp45jc67FdmW/B0H+IZxYfy5HAzJl7ygCXwbhXCLr7Pd/C/5eb7G0t9DfJ8DMw+MIfhOKY+HPrz8uCML/2aA2SFTpRT1X0WFMa34Y/mWZ82tGSrVqHl32YuHZX7/yCHPGp2i8sWTtgxEAURTFv8dv/Gr4mGqU+n9ujEJ6I5KTQcitkfnYSEmoWDcG77s2JCdqUEVfwNjMSHI1AaVrudF3/s/YOFoR9CYMOycbfDye8+qRP+1HtqTD6JpsHL2Eg6v8ADC3MycmLIZ3TwMxt5ajTtaQlKhABOp3rknhCgUpXM6F4+vOcfPIXZxL5CfgxQeuHfCg/aiWTGl5Ap0ubazm1qbcP/+EI1HbME5ugxi6haqNN3EqeU8Gjv3w1f25vO8GfeZ15e7phzy+7E35+tIqvmS1YlRuVoG7Zx5xZOVpes7sDEh5C4cj/tD3MW3/WAZXnsD8rsspXqkwDq72TD8wFpBiHnKlAo1ai0wh4/2rGERRxsphh1AaKek5sxMdRrUGoN+Cn3n8lxf7lxyjRutKlKxeHK1Gy4wfF+Ps5oh70/LU7iAxqOp1rsnz+3406FKLvvN/zqQfJeqiQBeOoEgLdrce1AR7Z1uKuRfK4fOGxLgk/D3fIJPLCPIL1huQou4FeePzjkJlXTJMqrlB18k/YmhiSL1ONXJsqzRQZqpy+G/j8WVvJjWfi5GpIQdCNutZYwqlgvxF8n3TsSXEJvJb3zUUcMtPr1k/fdOxpMeVAx6EBIRxZsulLzYgdTpU48kVH+p1zvl78k/gUyys75zonNUu42N797FR+Jx22Rmm9P9n7NPYTEdinPTDkStlaNVpooIPrpgDalr3DMPM5D2Nu7XjxtGX2DpJGdbjGs5Ep9UhV8pZfGE6dTtWZ0TNqfwxbR/Fi86lz7hozCxqsXNhDDGhMfqr5nPVsfTwM1RCB8Jj+pOcqGJ4jSkA7PRfw4yD49CoNRxYepyCpZ2JDIlBkMkQdDqciiRh56jh0XUZ8THv+OAXTCFHFSAQERTN5ulrqduxOlVbStISNdpWpkbbylw7eIs1I7eiE0VOJe7WGxljMyNEUcTINPt4QF5XOwoUc0RppMTawZK4qHgmNJlN2PsIes7sxJ5364iNjOO15zsqNrBEFAPxubUaQSZkWMUKgkDQ61DkchlBr0MpWb04oQHh3Dn9EICR63/RxyWa922olyxPxeoRW7h1/B6zjk3Exb4L6KLBagOCofRDlMvlmeROtFotiGRyY5lZmjJp90ieejyneJW0LPLxW4cyZuOgL/LhW+W1pN/8nz/7vG8FA2MDSPns0ws2fg9SHY/+8uLGkbtoNbfoPKEdxqZG33pIAHT/tQPm1mbU6VD9i/uo16km9TrV/Iqj+jzkyoUlCEI50mIgV0VRfPLPDSk3SO+qyqldbvvKCTm3kYyH1E6r1mKTz5AS1ctz/dAdfZsT2+0QRZAp7jN4eW82jN3OikEb0Yk6RJ2IXJSxbMA63j0LxLGwA0kJ8RxYZ0Wd1gKnt0Wh1cgyjLlqmzYo7fqhNKyDmZMRC7qtRKvWYmFnTh4ribEU+i6cnbMPYm5jSsW6WkSdjiot3Jl9pAfo4rl6OJTkRBWFy7mCeBx0oZxZ8JjzO65w/8xl9jz5gGBzBEEuJbMVqVgQa0dripR3RS6Xo9PpWNp/HYnxSax9sAilgSJTHe9UmJqbsNFzqf714ys+vHzgh04nsmnCTlr0a4SpuQkOLqn+3MIsverAh1dBmTSSfr89j3e+gVRuJsnAO7ja06RXPc5u+YuNE3ay/NqcbD+rCzuukhCbyJMr3rh0VErPVMjeT58Qm0jv4sNRJ2vY/HR5ph3FH7/u5YNfMNb5LDOwuP5XA8AJsVLyZ25zV0pULcrO12swMTdBEAS2z/yTnbP202l8u1wbwviYBIxMDL/6M6vYqCwNfq5FgeL5vxvjAdIi4XvaEX0JcjQggiCMAPqTFgPZJQjCBlEUf/9HR5YjstpFfMqwfMrgfOyeyt7w2DoqiYlUY5s3idLV4rh81B5VYmZpc5lMJPxDslRLPKW/vM4GBL+VcjFcSxXIIKjnUrIAb33eYWRmhNJQgSBAjxmd8Ly4meObzblz0RxRJ4AAJSvrePNMS9cxVnSa2g1VkoqDC05QvEpRIoKjkCvlNOvTAL8nb5nUbA6la7uh0+mIDo3GyvoDOp0dTsUcEOSOaEUtlZo6YmIuBzESQWYNMgvqdjLh9qk7NGx7HXRRoA2AFAOSr2Betj1fyc2jdwkNCEcQ4Ny2y8jkMoxMDLl64BYtBzRi+Or+2T7HVFRsWIYfRrTkyRVv2o9unWWbElWLUiKdUF5ifBJyhZx8BfOSr2DGAGupGm6c334lkwz8x5h9bAJeN3xp2rsBglEdEGMQ5PlQq9TsX3IM5xJO1PohTVAxITaR2EiJdBAbEZfJgBStWJDAFx9wLfVlulPfE0LehdG35CgMjJTs9F+day0sa4e0jPTgN6HIFHKC32Sfr5QeT2+/YHSdX8lfzJFN6RYYXwNGJoaM3zr0q/b5HyTkZgfSF6gqimI8gCAICwEP4BsakPQTfaphENMdywq53Y2k9i/9LZORIV4QFqiiViuBvpP9cXRVodXacnF/RgNkYSsQHSb9XbaOM/fP+1Gmtj3eN8OQyWXU+qEKv/45Bs/rT5EpZGgTVQSnaE5pVVr8nkjByCD/YILf58XUIpZ4qcwETkUssHPS4HMnjitHDOk0FS7tvs4f0/ahUCr488MGfG69oHxdB06s2UZifBI+Hs+Ry2XUaV+J3tMe06R3cS4dMubZ3ZcsHbCON94BLD4cSin3YLDZi6AsjVMxR1beXIiouiMZEGXGYk+HVpxky5Q9OBZ2YLP3MvrO60pMWCwJ8YkIgoAqOXOmvCjqEKMngC4awWoFgmCMXCFn4G89c/HZgKiLIej5aQZUPo6RmRE7/VZnWiE37VWPwuVccCzy6aBt6VolKF2rBMmJyfjeDaRYpULIgVvH77N95n50Wh2nEnfz2vsd/p5vadC1FsuuzUGj0mQZ2J28ayQTtg/721X8vgdIWmlakhJ0qJM1GH+BF2r46n7U61yTMrVzly4WHRqDIJcRHhiRpa7Xf/g+kRsDIpCeCyr9/R19urnZbaS9NjTWkpyY3Y88vTGSblGnS2+cpDbXT4CZZWVGLU2i/1RP4qMduXUujVOgStTQa0IwL7zMeXgtEHMbM+wcDQAd5eqVYPRGSQRw04Sd+kC6oYkhiXFJku6UABEfItm3+CjxUZKOkom5EU+u+vDBP5JuI9+jMGhAm+G9AIlVlK+QPeXql8bUwpTKTcujC+9C8/aPUSqa8tLHjdObLvL09htk1tu4uHwvu+ce5MKOqyTGJyGTC0SHpmS+i4mIugjQfkBQlkIwqIIqScWFTRcpVcsNlxSmj0vJAohaHUUqSAlincdLQoFajZYWfRtlUIbVQxcCSccBGah9wCB3dR9SIUZPJD7QA62mCAnR0uT2sQFJjJNEF8vVL52pwmFWmNN5GXfPPKLzhHb0nv0TRSoWwsLWnIKlnVEoFYyuMw21SoMgCNmKKKbi2sFb+Nx8Tu85P/2jtcb/aTi75WfV7fkYGBtgbvNloVBDY8NcPf9UVGvlzqJzv2LvYvef8fgfQm4MyFbgtiAIqbkf7YDN/9yQcoNPiTxlxcBKez+j8ZDayhUiVRrFcOusBaIISkM56mRdpnYgIMgEbJ1sqN/mKXfPxlKhjorpWwJYMro896+YEBcRhVYr0KRzJAXLV+XW2UDUci1dB1+lfFUlr1+a0c6qF92mdaD33C7M6bQM9yZlKVzelY3jpfrPSqWOZr1dsXWuwdYpe/H3fM3O16u4tKETMRE66rRV06D/QASFNEnnL5KPP55l3BCGhpQiOuApLfrWIDF0NwULyajYVtKrqtbKnfM7rtCsbwOa9KxHkH8IpapbI4iRCEo3dCG1iYuMwtjxN5R5mnB01Rk2TdqFnZONXpm1aouKnErak8lfLVfIKVoxG+aSLC/kGYcqPoJQf3sKFM+6WbZQlKRQyQssO18DU/vOWWaln99+hf1LjnF83bkMmlrZwTiPMaJOh4m5NOE/ueJNdGg0ntelUrNl6pTE+4Yvhcq65NjXgu6/I+pECpVzpVnvT+d55Bai9j1iRB9QlkVmmVkG/p9Cbu73a6N0rX9d3OI//E1km3mUIrOOKIpLgd5ARMq/3qIoLv93hpcbfIph9fHuIWvY5FXxy/QgBBkIMjC3MUWmkNFpfAt6THFAaSAgyGS0G96cURv6opAFM7GzLVO7F2TeIBfu3p3H1aMi9TrVYMOjaay78wMyy1ksGxmNVqOlaouK2BUfRNOfIogKCkCukBHkF0L5eqXpNq0DcVEJeF3z1Y9HrRIwVLzCraIU3NVqdDy6dJuOv+6l72/rUDre1xuPrKBRaxhQ7Tkj25bE664lxkbPaN3DB6fCkkF1q1KUXf5r+XlKe+ycbChTuwQyRV4EpRsAvg8s6FSmOAMqHyAuOp5Hf3khU8io2DAvuqCy6KJnAJ8fIBYEAZlpXyZ3iqVf6bGc3Zp1VcXsIMszFCGvN8VrjdIXrfoY7k3K4VzSiTaDm+aqzwnbh7Lt5So6peRmlKrphp2TLfVTaJFzT0ziSOS2XE2oHce0xq1KESo1Kas/dmHHFTo69OXi7mu5Gk8mqB6C9h0kHUcUM8fa/sP/39g97yAja00l5G3uYk1fG5/agRwA3AVBuCiKYkPg62kv/C3kNlAuItnHtHYyuUjL7uG89jXE81ba1jzkvSEPrplQqGQSWo1AtzHezP2lAAeXnUKb4spv0b8hvyzpyeaJW/jwRkjpVuDORXPUml1oNTIeXzzJoCk7SDLeQ+/ivxIVLNFtrx++zfP7tux4UIcRS17SoFc/yjWQ3CGbJu5ClaTCtVR6Hr/AofUG+D72QKEEjRoW9drBgwtvGLtlcI7+Q0EmYGppQlRINKZWRREsFgAyBIVrrp5wbEJlZHIvokKTuXn0LvfPP0anExmxzBbi1aDyyFU/2UGrkeqja7W5nxB977xgUa/V/DCsOa0HZW8cnIrmY+OT3Adh5XJ5hgxep6L52OEn1QFPjE9i1bDNuJR0yqSPlR4f/IKxdbKm77zMbKO/9t0kKjSGK3/epGHXzCrCOcKoSYo70Q1Jhu4//FMQdbEgGCII318N9eywZ/5hkhNVeBy/T9shzf71639Kzv0hsB8YBCz7+P2Uncm/CklMsUEOrSQjYmSqRacV0GkFNOq0KVcmFxi7Qs6lQ2BgFMbNU1KNhoJuSSy50JPkqMtM7+rPSy8TRB3IFZC3QDIz91YAnR+/1EpCp02t1Q0IUsCvTO2ijFl0grxOWrpXq0bou5gMo7LKa8K+dzMzTeJntlzi3rlH/LKkB9N/WMyL+34Z3m/YrTYPLnoS+SEKmVxkw/1uuJTNvjBRKlRJKpISkrOtWxD2PhwDo6x93LqgEjy9b4xdsdEYWrVhQfffKVGtIN1+bYuQdAoMqn5yB5QTkhKS+eAXjGupAnp/d0x4LMZ5jFAaZE2lXTv6Dw4tP4lTcUe2Pl2RZZuvjeuHbzOn8zK0Gi0n4ndmSWk9+8dfLO2/jrJ1StJtWgfC30dQv0st/X2FvA3lws5rNOpeB/sCtv/KuP/D50NUv0AM/wFk1gh2lxCEb1lrL/e4etCDhxc86TOv6ydzbv51OXekCoHtUtp8R0mFWe0+Pt55QFK8AhARBGjWJZyrJyxIiFWg04rEhL+jdBXYviiNqfP6mRG9y52mWQ9DXjwxQSYXWbA/AbUqjsr1g8G0AaqIB1SoVQDP22Zo1ALdf+3Is7svuH3yEQ261Cdf2ZaIuljUyftRGMiZvGcULo5zCfIPoHj5JMSwNmB/BUGWRnds1qcBzfpIRvF3j3mMbzwL7xvPKF6lCGEB4TTr3YBRy2UsH3QUhVKLo+NZpI8la4iiyJMrPjgVd9QX+vmY1fLW9z2DKo5DaahkT8D6zNx48xmUqPkAwbwNgsyceUdbIUZ0gYg/EWyPfMZnlcX4tEEYGmozSIN4XnvKuIYzcSqWj01emdYqAHQa1xadTqT+T/9e0lSFhmWo1tod11IFss2HUCWpEASBpPgkxjeahUwuwzKvJRVTlH7tne0yychnB1Htgxg9CUy6IjPp/NXu4z/kAmIsIIAYA/zvuArrtK9OnfZfnoj4d/GpTPRnwEJBEJ6Ionj6XxxTDshqx5T1sSoNo3EuqkKuFHFzj+PBZWm3cWKHFe9fGaUIGkrGRyYHrToOI2UACObIFWBkEkLRBqfweuzLX38+p+voEcw7pmV080c8vRONKMKc41PQarT6eIAAbPKqQVRINC4lC6CL70x+l3XcOmuIraOconX9gZcIBhmznF888OP64dv4eDzDxNyYBeemYmwiTey60OGMWxEAijIIlhM++XQu7b7Okr5rsHawZNfrtVw94MHcLstpN6w5g5b2kp6MxEtGFEXEhANcPwW7Fz5h8Io+lK7pJk1e6ScwXTAgA+3fK1kvasMRQ5siiiL37v+GjVMRRJ3I1l/3IAgQHRab7bk2+awYsrz337r+58LU3IQZB8d9sk2rX5pQolox8hXOy+Rmcwl+E0qB4lnHZ3KCmHgcNE8hfkPG5/8f/nEIBhXBZg8Ilv9TLqxvjdyo8X5HxiMVOWWhS3ESn3tm+NyD+Fg5zkXTJNFLuscT8NIopT64RF9t1TOU83ttOLTejK4jIkhO1DG6XWFcSy0hMS6JwJdBIDZmxNrB2Dsvx8fjJjaO0gr/42Cyha05FrbmeBy/x9mtgdT8YQ7LB61Hp9Nx2LcnBkYgWu9AZpBGcxzfeBYJMVIOhU6nQ6lUcGrTBfYuPMKYdaMoU8UfjH+ApJOIhvURlFlTmKwcLNFpdfoyoc/uvkIQwOt6WpDepWQBNvssx0DmgZF2MvsWuvLigQnH1pyhdE23zJ0aNkaw2gjZuK1EtSdiRA8wqI3MamX2H4sgqfU+vmHMrO5bEEUoWaM4nlefUqqmG9P2f35lu28NQRAoUr4goi6S5efLg1FDKRnzS/oy7YkoxiEYZ51Q+R/+WQjK0jk3ygFxUfHM/3kFziWd+GVxj68wqu8b/6NRuZyEDaXjdVol06pHOE6FkhmzthpulS2QyyHsgwH5XA3QqLQ06VmJBh1tqNM6iqREGXHRMuJj5RSvkIyoA1MLE5r3bYAgwNltl4kIimTizuFsfb6CIP8Qnlz10V912/R9NDf6iYs7tgGSUuaNI3fwvPYUuVKOc4n8KIysAC1EdEIXv1V/bsWGZTDJY0CfeV1QJamZ0GQ2x9ac5cOrYC79+Z4EbV/m/TSHQ0u3I0alVe3T6TJutys2LMOhsC0svToLgO7TO1K+QRlCA8J5+TCtSqGDqz1Wju4gs2XQgrz/r737jo+iWhs4/nu2JCSBJCSh9w4R6VIUFBSkCiqIBal6UWxXvXZ9xYYFOxcLdkWvF0VQFAVBReAq0qv03iEBkpCe3fP+MQPZQBJCyLIseb6fz8Ls7MzZZ2Yne3ZmznkO3QZfdnxc7pP2pggS2gFxFpAZNftvMJmQ9Zf16eTswJt4Q57tAxD7+nKlCz/B6XJRtX4Vrn+oH40uqs/wZ2/I05P5RCbzD0zqJxhT8FC+gWSSRmOSn8QkPVHsMsRZGUfUs0hI8Ub5M8aLN/FGvAcuxXj2nXoFVeJWzFnDktkrmfzq92SmF54N4XwQHHeKCpR7FlI2MofUFFeekfwuuzqBFpccYfij+4AtdL02knWLqlGzYSbuMkfZtyOES6+czEVdkli1IBavsSqe7z+J4c533qVpzxDC3POZP11wlwmxvkhFcDgczJ88n0kvfceMj37j633WmBjLf12G15PDml8/4/IbO3Lri4P4YcLPDLi/D/dNuM06uzh8P2T+ADggZ8PxWA/uWElaSjqJ238CAxuXbOG1uc/wyxfz6H9vb/78fjHzvzvM3KlVufrezgB899YMxt/zIUOfGsjN/5f75R8Rlds/okx4KJtXbCPpYDIr5qzJk4JdXNWRir8T3xXiu57BxxB2LYIb3BewYclmxg5+lj6Dt9N3xNsQkfeykzhiqNYwhm+PfHp8f17UvfCR+YwxmMO3Aga8KVD2rnOvs1lIW8icbf0fKCYdspdjHVtbwHn6KdTVmWnTvTm9bu1KzfhqRc4jFsyKkgsrvzuAScAqY8yBkg+pKPJewhKn4Wiyi7JROUTF5XBoXwg5OQJkH18+KzOHmV/GElclmz9mRDHg/p48MeEDQkKt6+6NWh7hou492f73XgaPHsiuzWVY8v1YNi5ZzS/fxNC8UzkefH0BUeVmknMULmo3hhn1LqTzTd2O36R+7D+3s2jycDpdlQqO8rS8vCJv3fMRj/V6nvdXvWbdrBa7eXFoV6Rcbv7+tOQcRKBOvItHJt5NjcbVqNO05kM3J5UAACAASURBVPHxotv1bkX7vhcR36EhzkirSen6xZtwOh2s/WtToXvr2e8eZuXvf9Nr5JnUEgUTcUO4dZjMmTSR7etSmfphHfreW/A9i/wSLRZcvmDK9ICMnyD1HXDVhnPsMo8j4maIuDmgMYgjAsp/bN2zCgncjdXSLDQslHveujXQYZw1Rc2F1QE41uurM7AEqCMizxhjJvoptkLkvf/Rb/hBEveGMG96NEeTXHTqc4TWnYVX7m1E2dhqjJ1Wl/H/nMqmVdbIf8YLk16eRZebvsIdeYA9f39N5cY3sX7RW6QlpwEw+uqx7Nqwh8iYcnhyoN4FacRUhDXzl/LgVVtp1akqH/xhuKv7Eia/+j3vLBlL1Srf0OPGfeBqiThiOLRvJ0ePpGIOpZJ6JJWwiDJI1FiIuA1cDY+3609NSuXQfjeh4UK7/k9TvmLsSVscGVPupBu6d745gjbdmtOmR96UEWkp6cz46FdaX9mcWk2q07htAxq3bcDZMOD+PmRlZNP5+otxhOVzP6WYHNGv4U3YATlrwFGeuZP/ZPvfu7jhkasLbPpbGklo+0CHoEqRolQgLqCJMWY/gIhUAj4D2gFzgQBUIL6Ebz+oSOvOuf0uGrcK4e0nypCVkUPC3u0sm3WYedNjMMYQVwUO7feSuFd4tMfrtOhSgynjdnFxv9+I79CQZb+sIulgMuViy+IOFZIPu3A6Dbe+0Iev3mjJt+9uBhxsWlMRT9nX2bbmTkSEvRuXUrVqNOAFt3XWUPuCGjw77RHcoW7iqlmVgogb3LlfrF6vl5HN/0XK4VREhLTkLMoXcWTKiMhwLs+nc9pXL3/Hf56fQrX6lfl4XSE3tf0gpnJ57ho3wi9lS+x/waSQk1OW564fhMPloHbTmnS6tt2pV1ZKlbiiVCA1jlUetgP2vEMicpbvaOY32p/1/5I5uckMF/1iyMpwIAJ9hx+kU89VbHvkCSaNnU7Cnhx7fWHT8n1sX7sbwcHeLft5Y/5zZGVk0T9uBA6ng3snDGfRtxNo2u4gxtmAr9/8leSEFLrc2JGhTw/k1y9+wpPtoVGbKrRq9RAcDYUKfyGO3FTfvkOlLpqxjEUzlzNk9MDjeZyyM7M5uOsQAFXqVTo+eltmeiYbl26lSbsGp50ypEWXpkx7eyaXXBPA6/F+IOICKY87BLoNuYwtq7ZzwcUNAx2WUqVWUSqQOSLyA1avdIAB9rwI4IjfIjsNNRums2ODlQwvIsrLbU/v5qW749m+Ppulv0fiKNuDYc/cQNbRNUwZv5nQMKHPbV35+tVZZGc6qFJb2LpqBx888DQNWlotjbweL217tqfH8B5AFuPvnkjKoaPUb1mbu8aNoFxMGPM/m4DDWQmHHEIkB/AijogCb/A+P+hNUpPSqFSzAtfe0waTeD1uZ3Ve+XU0q+at49p7ex9fdsyNb7Dwp2UMuK8Pt754etfWW3RpypSEjwtdxpvyBqR/C9HjwRV/WvckzgUPfnxnoENQPt57aCIr5qxh9DcPaI/7UqQo3xp3YmXkbWE/PgXuNMakGmNKJuXoacu9B9JzUCIJe62OPxe2T+Htmdup2+oqhj1eBUFIPuxm0AXbWDjtfa67dTpRsTmkpxrCI2Oo3qiqlT32ovZ4vV7qN5xD63bv0Oyyhgx6YgCxVcrj9Xo5vD+N7KwcHA6hSVs3ZT09MYk30P+uirz09Xaem1QZcIIr3rpEVYD+9/ambrNatOvTGnI2gzcBspdyYcc6DHq8f54e4eXKl8Xr8VIuxk9DgqZ/gzdnL7e3foV+UUPYtXGvf95HlQpTx/3IpmVbWTorwIOVqrOqwFxYeRay7nu0xfrmXhio1leRUt60k9yxrcPLeXju8y3865oGGC/UapTBG99vY83CCJpfcoS0FCeD2jQhJ8uBOISyUR7ufS2UedNrMPK1h4iMLUvK4VTiqsaQk5WAI+lmcMSwbfdzvDjkbXoM68zKuX/zx7TF/POdf9CgZQXqVL4ep8tuSRX9LhwZZQUT81/EXXgF4ssYAxlTwVEFCT25xYzX6yVh66/EVdqDRNxY5HKLymStIDN5LtfW+AMR4dlpj9Cqa7NTr1hCPB4PGamZRESGn7X3VP4zb8pfrF2wnsGjB55Tw8Yqi79yYZ3yDEREBgILsS5dDcQaG2TAmbypiFwnImtExCsixdqomIrZTF2/hgsuSmfEY3twuQ1tr0jmhTsaMXpYDT58rgrRcV4uv87OB+U1uEKiiW+5ikfHTSeuwkZCw0KJq2r1GnaFxJHqmsz2PS8wf8oitq7aztevfU9SQgoOp4OUQ6k0aH0hzrAO4KgMkc9bHb7cLSCkPeJugoibIweTeP/hiaxbuLHA2Ncv2sQN1Uby7hPJ+VYe9j4iLuIeOPoiZMwszi4qlIQ0p0zc3fx7wQs8+fW/aGnnbjpbHrziGa6NHc6SWSvO6vsq/+h0bTtGjh2ilUcpU5R7II8DFx076xCRCsBsrHTvxbUauBaYcHqr5d5bOHTAzVV1m/LPsbsYeEcWA+9YBcD40a3wzN5FpdrlcVSew8gXPiSuwkTqtenNlHeSuKlVIx4en8Dl/6hxUul3tHmYhN2HuG/CbfS7swcdr2lHwzb12LJiG/EXN0LEgcR+kjei2C/zPP/6lWl89co05k9dyKcb8h/1d8WcNRw+kMTvX/3JqFeHHZ9vPPvBexBxN7X7PlwJWcvB7b8zg3rNa1OveW2/lV+QAzsO4nA6SNxzuETLXTp7JcmJKVw28OJzr7OhUueZolQgjhMuWSVyhilQjDFrgWL8gee93JaV4WDp3HJ0HZCAxwNOZzgN211Fjbnf0KSzNWxsuYglDH1oH//qv4w1f2YjDgdH0m8DR27COys9hpPQsBCM11C+cjR3jbuFrIwsnh/0JhGR4cRfXLTh8zpe2575UxfSt5AxK64adSXicNC8c7xPDF5MQm8wGVB+AhJ6CY5oa9wuT46HSS9MoXKtCvk22w20fdsOsGruWi4b2IGQMkVLRPfm/8awY+0uWnQ58/xDxyQnpvBorzE4HA4q1IjjgiJ+Zkqp4ilKBTJDRGYCx35qXw/86L+Q8hKRkcBIgDLkvV5evX4mtz+1h/GPVeWHT2N4+NO+fPPGdHauP8jszxdwwSXNkKjnIONnjCzH6drKNXdUpd/1T2OS1yFRT2Nytllp1p2VGb/wO5ITU4+3Itm0bCsLvl+CJ8dDj1sut3qBOwtvUtu4bS0+XtoUcZ083OoxYWXDuO5fJ/akFnBEgScdJO+N86W/rGLiU1/hyfHSrnerPKlKTmSM96wPPPR47+fZvXEvB3YmMOjx/kVaJ7ZK+ePp5ktKeGQY9ZvX5sjBZKrWq1SiZSulTnbKbxpjzIPAe0Az+/GeMabwnOKAiMwWkdX5PAoe2i3/93/PGNPGGNPGjfXr1uH00LJTMi9ObUtkjIeNK0NxOJ1sWQ33vZrIVcMSuP7uHRiTDhmzIeQSXpr1NO8ue4Vbn47F4RB++ng1j3a7gf0r+gIe8OwhNMyRpwliw1Yh9BkRgjPEwf2XPsmYa6/Bm/pp4QFnzIajb2KOjMIYz0kvj7npDfpXGMHmFdtO3F9I3E9IxflISPM8rzVqU496Letw6XUdCC/kprM35d+Y/Y3xpn5ReIwlrGnHJogIDVsXMBb6WeJyu3hr0Ut8se0dyleK9tv7THnjB4Y1uqfQ+1zKf4wnEZO1iKI0AFL+VaRWWH57c5E5wAPGmMVFWd63FZbDCROWjqZmjU85lFiL1Ytb0aF7Gu6Mu/DkGCY8257wqFiG3jcDcdfDEfcDYF2uMmmTGFDjG44mObnl8X1cd38niBgCaROBECTySUQceJOeZs/ar7mlY2O8XqHFJSm8NEVwVMg/w73xHsZkb4akB8G7B0K7n5Te/JqYoaQfzeTOcSOoc2FNGretj8tdMjktvYdGQtY8CL0UiRwDnu3gbqX3AkrY0IZ3s2fTPgY+2Jd/vDQ40OGUOt6DXa1hfqOeRcKKNlhXaXfWRyQUkRQKGXjcGBOZz2t+lvtFWKdxOlWjB0LoOMqHP0WHjj/gDhkF2XXZsCKKHz7Ownh30+O6MCIqtqFM2E5CwqtbzWHDejFqzH/4349hXD50JBJ1PeRsxqRPtQqPGAKuukj4jVRtsJEXp3dl15YouvbfiET3zjcyYzyYg93BpEH4IKsyyj65TfxLs55kw1+zWPHzq7x1TyS9RnbnnvEllHwt8lFImA+Z8zGJ/cB7BIl6CcL6lEz5JcRk/g8ACT17owuWpEcm3sO8bxZw3QN9Ax1K6eSoYA1u5tAOi4FW2IiEfhvGVkSuAf4NVACmi8hyY0zBd51zozo+1eOmRDIzDK6UV/hnnyg2rw7jje+fptGVy/jstWdxONfScUB7Ul0vcmujR4iucBfvzG9LStYIqjeoQrdRP9JtlE/JrkYQdgOIG5y1rTjdDZHYz2nZHQpPOH5sw5xWiKFXIu4L82091bB1Peo3/I4tS7Lweg3logu+n3G6xFER44i0xuZwxIL3EBRzcCN/MTk7MIdHWk/ipp80RnwwaNKuAU3anZ3klOpkEjMRzFHE4b/LlKpoAjIeiDFmKjC1uOuP/nArL95Vm0/HVmHiihrs2rQLccC+nWWo78lh5ZxVIB4u6ZNFekoGCKQmObm7y2oO7r6Pp799mLY9rSohKzObxD2HqFKnEhI1+oQ4rVRfhXXiMyYD0n+EkLZI3M9gUhBn4UOaSsRt3DWuItc/fgEV67QodNnTIY4IqDAXa0xnV5H/yIxnLyblFSSsHxJ6aYnFky9HDDgrWimRHSdnHVbqVKycaFp5nAuCckCpKe/H4cmBnGwHg5vv5IZ79lOzUSbtrjiKw+zh5Wll2bl2LRd3/RtnmPD63EeIjjnMQz2mYkjIU9ZDXZ9m7YKNPPzZ3Vx+Y8fj8433kHVJCidUmIU48j8hM0ffs8aocNbFUWE6cOoTN3FEIBE3U8kP95zzjOdcxD8yk/YFZHyPyVqCVJxT8kH5EEdZpMJvp16whJjs9ValHlLil3+VKvWCsgJZtaAcTrehbnw6m1aF8/ELValUI5uLel2EZG8kvvUB4i++DtK+gtQ/adA4DLw7eWteK5KO9qRq3VCy05czdvh0tqzcAWJlxc3Dm2KN8AZgjlJQxSAhzTCpDgi5yL8b7UcSdjUmazGEDQx0KCXKeJMxif0BgdgvS2TMa6VUrqCsQAA82cKRRDe9Bifx838jSdzvwuPujzPtOWuoWGcdCB8MGT+AhEL6NMLxEF6zLebAQA7vhd8nx+NwuLj//du5ckjnPOWLqxbE/gdwIs4qBcYhoZ2Rymv8u7F+Jq761lgb5xspY10mM8l+vxfk8Xh4/8GJOENc3PL8TUGX3Vip4gjaCgSgSet07hmbSpdrDNGxBwhhJhL5GCbtcyTiFsRVH8rdifEesX5hu5si5GAQ4qpkc/szh9i9dyDrFyynUfwsarW4GXHn9g4XP6YQUf4nEgIVfgU8eS/t+cHWlTv4dvxPGAM9hnehRqNqfn0/pc4FQVmBXNb3CBdeXI2rhqwGr4Om7cP590NxJCTu5PH/1iUs6oU8y4sjGhM+ENKnQNjVSOxUDA6ueaQqj/V+jcUzV7BvUzLP/XcjEvt1Ae+qgpGIEzi9AbmKo3bTGlwx6FJcIU6q1q/s9/dT6lwQlBXIqgVleXDcAnC1AM9WkhJ28+PnTXG4drP2r020yi+zbPKz4N2DcdXBUe6B4z1K+t7Rg31bd9JnWDqEDzmr26HOHy63Swe5UqVOUFYghw64yMoQ3CHLAIiOhVvHxJN4oA4Xdmqc/0rlHoD0yUhY3lxNHa5qQ4ertIWOUkqdrqC803dxzyQiIr32MwFnfS4f/A8q167I0cOpGM9+vIfvwJs25fg6EtoRiX4DcdUJTNBKKXWeCcoKpGqtLGvC1QJivoKIEbz5j8d45/5PePOO962WV5m/QMrzgJ187WBn6+FJKKTkgq2at5Z+UUN4/fbTHMJEKaXOU0F5CWv/LjdIRYj4BxwaBOTQtks0K+ZVp12vVlDmAshcCGWODX+bY/V8PjZdDOv+2khGWiaLZywviU1QSqmgF9BsvKfrWDbeCtWy+HzROlKTnWzbWJf4VjsQdx2IuAtH2JXHlzc5myBrGYT1A481Jpa4qhfrvbMyspj12e807dSEWk2KV8b5ypv+M6R/jpT7P8StOaKUOtec9Wy857JaDTMBePiGC9m6xsHIV17gmrtPzjhrDo2whog1aUjE0DN6z5AyIfQe2e2MyjhvHX0VPFsxaV8iUU8GOhql1FkSlPdANq0OY8eWeKIqNcfj8RIVG5X/giEXW73QNYWFX0nkExDaFYnQZtBKlSZBeQnL6YL7XvfQ9bbJJCUkE1O5aEOjWiME5iAS6t9AlVLqHOKvS1hBeAbi4crrD9G+Z32cLudpVB5eTEJvzP42mGwdilQppc5UEFYgws+TYnjnsX1402fnu4QxmZiczSfMtcY9B8B7wL8hKqVUKRCUFYgnR6jdcAskP4b3yIN4074B7LOMjJmYxJswCX3wpn2Vu5a4kbjvkPIfBO1QqkopdS4JwlZYwhX9DzHwjlRwXwYZ30PGDAjvDxnTMEmPY43Ih0/fD3tNVx3QnuhKKVUigqwCsVIgtu+eBKRBxBDWLTnCigXV6XdvOmWcBjDgboNEPgmu+gGNVimlzmdBVoFYLcY2rQrj0r7lkZALGT3YTVLCRlzul+g/7CtwNcUROzHAcSql1PkvCO+BwObVYVDuQURCuGJQJ+KqxtCic3XAC46IQIenlFKlQtD1A7kk9FI+mr+FyvGPIeE35nnd5OwCZyVE3AGKUCmlzj3aD8SWk+Vm84arIGzASa+Jq3rAKw+v18vBXYkEU8WslFLFEXQVCED5qBmcmFV3+W+ruSZmGO89+FlggrK9PnICg2qPYtLYbwMah1JK+VvQVSAhYdk0ae3BmLz3/1f8vobU5DT+mLY4QJFZkhNTcDgdpCQeDWgcSinlb0HWCgtq1s8AMhBSMKY8IlbT3oEP9CUqNpJW3ZoFNL7H/vNPNi7ZQpP2DQMah1JK+VvQnYFc0jsFwoZh0qZi9jfBm/oJAGFlw7j67p7UbFwtoPGFhoXStGMTnC5nQONQSil/C0gFIiIvi8g6EVkpIlNFJLqo62ZnOsC7E7KXAwJZS/0YqVJKqYIE6gxkFtDUGNMM2AA8WtQVF/8WAc6aSNSzSNQYJOoZvwWplFKqYAGpQIwxPxtjjjWjWgAUeYzYPdvcSNnbEUc0EnYt4ijyyYtSSqkSdC7cAxkB/FTUhV1uIT01lCMHk/wYklJKqVPxWyssEZkNVM7npceNMd/ZyzyO1aHji0LKGQmMBChDGOWisxlcdyRpydk0bteA+967PeA3zpVSqjQKWCoTERkG3AZcYYxJK8o6x4a0DSvrJj01BwxcfXdP7nxzhF9jVUqpYOavVCYB6QciIj2Ah4DLilp5HOMK8fLR2vH8MW0xf01fSt87uue7nMn8C5yxiKZ0V0opvwhUR8LxQCgwy+4IuMAYc3tRVoyI9LJ70z76jupO31EFVB5ZizGHb7WeVFqISFiJBK2UUipXQCoQY0yxTwuSEtx8OWYSzS97xi7Lg8gJnfYccYCAsyKgmXmVUsofzoVWWKdFxDD06SsB8CY9hdkfjzd9et5lXLWRSouRuJmIBF22FqWUCgpBV4HUjU+nSYeO1pOc1YATctadtJxIyElnJt6jE/Dua443/fuzEKlSSp3fgq4C2bMthEP7DgMg0e8gUS8jZe8q2sqZc4FMyPrDfwEqpVQpEXQVSHqqkwU/WPmvxFkBCeuNSGiR1pXoV5HI/0PKPeLPEJVSqlQIuhsEUXGGSwe0L9a64qwM4YNKOCKllCqdgu4MpMvVaZSNjgh0GEopVeoFXQUipJOZtpP01IxAh6KUUqVa0FUg65eX4ea6jzGw8q3s334w0OEopVSpFXT3QLatK0tOjgdjIDXptLKgKKWUKkFBdwbSqV8ob8+rwxvzn6Fus1olVm5SQjLj7/mQpbNXlliZSil1Pgu6CiT18H5qVJtCg2bZJVruj+//wrS3ZvLi4HElWq5SSp2vgq4Cqd8sB8IGgKtBiZZ78dUXUbd5LQb866oSLVcppc5XARsPpDgipbzpFtuFj9d/SrmYshhjcDiCrg5USqmzyl/jgQTdt2/yYQd7t+zn9lYP0i9qCLs27Al0SEopVSoFVQUiAk6ng8S9h9i5bjdej5d927Qpr1JKBUJQNeM1xvonukIU/17wAgd3JtK6W7NAh6WUUqVSUFUgAHe93pz4Do0AqNe8dmCDUUqpUiyoLmEBrJ73JzlZetlKKaUCLegqkF8nl2HZjG8DHYZSSpV6QVeBVK0D9dtfHegwlFKq1Au6CqRe8wok7j4a6DCUUqrUC6oKRIC5UxN4aei/j89L2HOI37/+k+yskk1topRSqnBBVYEYIK5KNn1G5o5I+ETvF3hh0JtMGvtd4AJTSqlSKKia8ToExs2oSOwFAwA4fCCJAzsO4snxUCu+eoCjU0qp0iWozkC8Bga1OMI/L34cgA2LN5N+NAMRoc2VzQMcnVJKlS5BdQYCIA6Opy9p0705w8fcRJW6lQgrGxbgyJRSqnQJqmy8zRqHmY+f7UD1tpOoVKtCoMNRSqmgoNl4gYR9bpq03kPFqgmBDkUppUq9gFQgIvKsiKwUkeUi8rOIVC3KeqnJTlYvrAKuOv4O8ZSOHkllxke/cvhAUqBDUUqpgAjUGcjLxphmxpgWwA/Ak0VZKSTUS9PLuiFSxr/RFcFb937M67dNYOzQ8YEORSmlAiIgFYgxJtnnaQRWF49TyspwcGDbYf8EdZou7NgYcQjNO8cHOhSllAqIgN1EF5ExwBAgCehijMk3xa6IjARG2k+bAqvPToTnvDhAbwZZdF/k0n2RS/dFrkbGmHIlXajfKhARmQ1Uzuelx40x3/ks9yhQxhgzughlLvZHS4JgpPsil+6LXLovcum+yOWvfeG3fiDGmK5FXPQL4EfglBWIUkqpc0egWmE18HnaD1gXiDiUUkoVX6B6or8oIo0AL7AduL2I673nv5CCju6LXLovcum+yKX7Ipdf9kVQ9URXSil17giqnuhKKaXOHVqBKKWUKpagqEBEpIeIrBeRTSLySKDj8QcRqSEiv4nI3yKyRkT+ac+PEZFZIrLR/r+8PV9EZJy9T1aKSCufsobay28UkaGB2qYzJSJOEVkmIj/Yz+uIyF/2Nk8SkRB7fqj9fJP9em2fMh61568Xke6B2ZIzIyLRIjJZRNaJyFoR6VBajwsRuc/++1gtIl+KSJnSclyIyEcickBEVvvMK7HjQERai8gqe51xIiKnDMoYc04/ACewGagLhAArgPhAx+WH7awCtLKnywEbgHhgLPCIPf8R4CV7uhfwE9ZIv+2Bv+z5McAW+//y9nT5QG9fMffJ/cB/gB/s518BN9jT7wKj7Ok7gHft6RuASfZ0vH28hAJ17OPIGejtKsZ++BS41Z4OAaJL43EBVAO2AmE+x8Ow0nJcAJcCrYDVPvNK7DgAFtrLir1uz1PGFOidUoSd1gGY6fP8UeDRQMd1Frb7O6AbsB6oYs+rAqy3pycAN/osv95+/UZggs/8PMsFywOoDvwCXI6VL02wehW7TjwugJlAB3vaZS8nJx4rvssFywOIsr805YT5pe64sCuQnfaXn8s+LrqXpuMCqH1CBVIix4H92jqf+XmWK+gRDJewjh00x+yy55237FPtlsBfQCVjzF77pX1AJXu6oP1yvuyvN4CHsJp6A8QCR4wxOfZz3+06vs3260n28ufDvqgDHAQ+ti/nfSAiEZTC48IYsxt4BdgB7MX6nJdQOo+LY0rqOKhmT584v1DBUIGUKiJSFvgGuNfkTTqJsX4anPftrkWkD3DAGLMk0LGcA1xYly3eMca0BFKxLlUcV4qOi/JYHY/rAFWxErH2CGhQ55BAHAfBUIHsBmr4PK9uzzvviIgbq/L4whgzxZ69X0Sq2K9XAQ7Y8wvaL+fD/roE6Csi24D/Yl3GehOIFpFjnV99t+v4NtuvRwGJnB/7Yhewyxjzl/18MlaFUhqPi67AVmPMQWNMNjAF61gpjcfFMSV1HOy2p0+cX6hgqEAWAQ3slhYhWDfDpgU4phJnt3j4EFhrjHnN56VpwLGWEkOx7o0cmz/Ebm3RHkiyT2VnAleKSHn7F9uV9rygYYx51BhT3RhTG+vz/tUYMwj4DRhgL3bivji2jwbYyxt7/g12a5w6QAOsG4VBwxizD9gpVuYGgCuAvymFxwXWpav2IhJu/70c2xel7rjwUSLHgf1asoi0t/ftEJ+yChbom0JFvHHUC6tV0masbL4Bj8kP29gR6/RzJbDcfvTCumb7C7ARmA3E2MsL8Ja9T1YBbXzKGgFssh/DA71tZ7hfOpPbCqsu1h/6JuBrINSeX8Z+vsl+va7P+o/b+2g9RWhVci4+gBbAYvvY+Bar9UypPC6Ap7Fy560GJmK1pCoVxwXwJda9n2ysM9NbSvI4ANrY+3UzMJ4TGm7k99BUJkoppYolGC5hKaWUOgdpBaKUUqpYtAJRSilVLFqBKKWUKhatQJRSShWLViDqvCIijUVkuZ32o16g4/ElIp+IyAB7+gMRiT/D8mr7ZmY9m0TksUC8rzq3aAWizjdXA5ONMS2NMZtLqlCfns4lwhhzqzHm75Is8yzTCkRpBaKKx/71u87+Vb1BRL4Qka4i8j97nIG29nIR9jgGC+2zgn4+688TkaX242J7fmcRmSO54198kd+4BCLSQkQW2GMdTLV71vYC7gVGichv+axzVETGiMgKe91KPrH8apf1i4jUtOd/IiLvishfwFgReUpEPrXj3i4i14rIWHsMhRl2KhpE5EkRWSTWmBXvFRD/HBFpIyJ97TOm5WKNTbHVfr21iPwu3MUc8QAAA8FJREFUIktEZKZPuorWdvwrgDsL+XwetuNaISIvFrTPfGOxp+PESiGDiAwTkSn2tm0UkbH2/BeBMDvmL+zPeLr9XqtF5PpTH0HqvBDo3pX6CM4HVlrpHOBCrB8iS4CPsHrA9gO+tZd7HrjZno7GyigQAYQDZez5DYDF9nRnrKyp1e1y/wQ65vP+K4HL7OlngDfs6aeABwqI2QBX2dNjgSfs6e+Bofb0CJ/YP8FKGe70KXs+4AaaA2nYvZiBqcDV9nSMz3tO9HnPT4AB9vQcfHoH2/O+wqoU3MAfQAV7/vXARz7bfak9/TI+qb19yulprx/uG08h++x4LEAcsM2eHoY1XkQUVq/u7UAN+7WjPu/XH3jf53lUoI9PfZydh56BqDOx1RizyhjjBdYAvxjrG2QVVgUDVq6dR0RkOdYXVRmgJtaX5Psisgor3YTv/YCFxphddrnLfcoCQESigGhjzO/2rE+xBts5lSysCgGsCu9YuR2wBq4C6wu/o886XxtjPD7PfzJWIr9VWIOdzbDn+25zF7FGwFuFlQjyglMFJiIPAenGmLeARkBTYJa9354AqotINNZ2z/WJNT9dgY+NMWkAxphDZ7DPfjHGJBljMrDyTtXKZ5lVQDcReUlEOhljkopQrjoPlOh1XVXqZPpMe32ee8k9tgTob4xZ77uiiDwF7Mf6Je8AMgoo10PJHafZdgV3OuWmnvA8E8AY4xUR3/K8gEtEygBvY/2i32lvZ5nC3kBEugLXkfuFLsAaY0yHE5aLLkK8xZFD7uXsE2M95WdhjNkg1pCpvYDnROQXY8wzfolUnVP0DET520zg7mP3AUSkpT0/Cthrn2UMxvo1XyT2L9zDItLJnjUY+L2QVU7lD6ysvwCDgHlnUNaxL+AEscZ2GVDYwiJSCyvp3XXGmHR79nqggoh0sJdxi8gFxpgjwBEROXaGNKiAYmcBw0Uk3F4/5hT7bBvQ2p4uNF4f2T73fKoCacaYz7Euq7UqdE113tAzEOVvz2KNLrhSRBxYw7P2wfqV/o2IDMG6DHTiL/1TGQq8a39JbgGGn0GMd2ON+Pcg1uh/xS7LGHNERN7Hymq6D2s4gsIMw8qo+q1dx+4xxvQSq7nvOPvSkwtrH66xY/tIRAzwcwExzBCRFsBiEckCfsRqNVXQPnsF+EpERgLTi7ip72F9pkuBz4CXRcSLlSl2VBHLUEFOs/EqpZQqFr2EpZRSqli0AlFKKVUsWoEopZQqFq1AlFJKFYtWIEoppYpFKxCllFLFohWIUkqpYvl/u+EgvkN6KM8AAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "mean = df['baseMean'].to_numpy()\n",
    "foldChange = df['log2FoldChange'].to_numpy()\n",
    "p_val = df['pvalue'].to_numpy()\n",
    "MD_plot(mean, foldChange, (-3,3), (0, 10000), p_val=p_val)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Cutting Adapters (testing)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 1.15 with Python 3.6.9\n",
      "Command line parameters: -a NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACGAGATTCCATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAA -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACGAGATTCCATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAT -o trimmed/SRR6145423.sra.fastq SRR6145423.sra.fastq\n",
      "Running on 1 core\n",
      "Trimming 3 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1383.31 s (37 us/read; 1.63 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              37,552,625\n",
      "Reads with adapters:                37,332,522 (99.4%)\n",
      "Reads written (passing filters):    37,552,625 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,785,289,949 bp\n",
      "Total written (filtered):      7,860,791 bp (0.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN; Type: regular 3'; Length: 35; Trimmed: 36966506 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-35 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 0.0%\n",
      "  G: 0.0%\n",
      "  T: 0.0%\n",
      "  none/other: 100.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t6\t586759.8\t0\t6\n",
      "4\t1\t146689.9\t0\t1\n",
      "5\t1\t36672.5\t0\t1\n",
      "6\t2\t9168.1\t0\t2\n",
      "10\t1\t35.8\t1\t0 1\n",
      "11\t1\t9.0\t1\t0 1\n",
      "14\t1\t0.1\t1\t0 1\n",
      "18\t1\t0.0\t1\t0 1\n",
      "19\t1\t0.0\t1\t0 1\n",
      "26\t1\t0.0\t2\t0 0 1\n",
      "27\t1\t0.0\t2\t0 0 1\n",
      "28\t2\t0.0\t2\t0 0 2\n",
      "29\t1\t0.0\t2\t0 1\n",
      "30\t6\t0.0\t3\t0 1 1 4\n",
      "31\t3\t0.0\t3\t0 0 2 1\n",
      "32\t13\t0.0\t3\t0 0 7 6\n",
      "33\t6\t0.0\t3\t3 0 0 3\n",
      "34\t119\t0.0\t3\t68 25 11 15\n",
      "35\t1766\t0.0\t3\t1740 23 1 2\n",
      "36\t1918\t0.0\t3\t1903 13 2\n",
      "37\t1911\t0.0\t3\t1894 8 6 3\n",
      "38\t2088\t0.0\t3\t2071 9 6 2\n",
      "39\t2248\t0.0\t3\t2218 20 7 3\n",
      "40\t2558\t0.0\t3\t2520 18 13 7\n",
      "41\t2892\t0.0\t3\t2871 18 2 1\n",
      "42\t3186\t0.0\t3\t3166 10 10\n",
      "43\t3375\t0.0\t3\t3357 11 7\n",
      "44\t3762\t0.0\t3\t3740 11 9 2\n",
      "45\t3819\t0.0\t3\t3779 18 15 7\n",
      "46\t3878\t0.0\t3\t3862 11 4 1\n",
      "47\t4076\t0.0\t3\t4071 4 1\n",
      "48\t4491\t0.0\t3\t4487 4\n",
      "49\t4929\t0.0\t3\t4926 3\n",
      "50\t5268\t0.0\t3\t5267 1\n",
      "51\t5830\t0.0\t3\t5829 1\n",
      "52\t6649\t0.0\t3\t6645 3 1\n",
      "53\t7287\t0.0\t3\t7284 1 0 2\n",
      "54\t8017\t0.0\t3\t8017\n",
      "55\t8618\t0.0\t3\t8617 0 1\n",
      "56\t9322\t0.0\t3\t9320 2\n",
      "57\t9885\t0.0\t3\t9882 1 1 1\n",
      "58\t10903\t0.0\t3\t10901 1 1\n",
      "59\t12398\t0.0\t3\t12393 2 3\n",
      "60\t13294\t0.0\t3\t13275 5 13 1\n",
      "61\t14749\t0.0\t3\t14748 1\n",
      "62\t16303\t0.0\t3\t16301 2\n",
      "63\t18013\t0.0\t3\t18008 5\n",
      "64\t20453\t0.0\t3\t20446 6 0 1\n",
      "65\t22173\t0.0\t3\t22169 3 1\n",
      "66\t23564\t0.0\t3\t23563 1\n",
      "67\t25306\t0.0\t3\t25306\n",
      "68\t28613\t0.0\t3\t28613\n",
      "69\t34355\t0.0\t3\t34355\n",
      "70\t65260\t0.0\t3\t65259 0 1\n",
      "71\t171613\t0.0\t3\t171606 5 1 1\n",
      "72\t712748\t0.0\t3\t712721 20 4 3\n",
      "73\t2622245\t0.0\t3\t2622181 40 14 10\n",
      "74\t8493368\t0.0\t3\t8493275 73 9 11\n",
      "75\t24553208\t0.0\t3\t24552919 211 50 28\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GATCGGAAGAGCACACGTCTGAACTCCAGTCACGAGATTCCATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAA; Type: regular 3'; Length: 74; Trimmed: 283380 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-69 bp: 6; 70-74 bp: 7\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.8%\n",
      "  C: 0.0%\n",
      "  G: 0.0%\n",
      "  T: 0.0%\n",
      "  none/other: 99.2%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "36\t1\t0.0\t3\t0 0 0 0 1\n",
      "38\t15\t0.0\t3\t0 11 4\n",
      "39\t15\t0.0\t3\t0 10 3 1 1\n",
      "40\t49\t0.0\t4\t0 30 4 3 12\n",
      "41\t22\t0.0\t4\t0 1 3 5 13\n",
      "42\t21\t0.0\t4\t0 15 4 1 1\n",
      "43\t15\t0.0\t4\t0 5 1 2 7\n",
      "44\t22\t0.0\t4\t0 11 4 1 6\n",
      "45\t23\t0.0\t4\t0 14 4 2 3\n",
      "46\t15\t0.0\t4\t0 9 3 1 2\n",
      "47\t25\t0.0\t4\t0 9 6 8 2\n",
      "48\t13\t0.0\t4\t0 8 1 1 3\n",
      "49\t27\t0.0\t4\t0 15 8 1 3\n",
      "50\t26\t0.0\t5\t0 16 4 2 2 2\n",
      "51\t40\t0.0\t5\t0 20 9 3 3 5\n",
      "52\t31\t0.0\t5\t0 18 7 3 1 2\n",
      "53\t31\t0.0\t5\t0 17 6 4 2 2\n",
      "54\t40\t0.0\t5\t0 22 10 4 3 1\n",
      "55\t41\t0.0\t5\t0 22 15 1 1 2\n",
      "56\t43\t0.0\t5\t0 18 17 2 3 3\n",
      "57\t44\t0.0\t5\t0 23 9 9 0 1 2\n",
      "58\t57\t0.0\t5\t0 25 20 5 3 4\n",
      "59\t64\t0.0\t5\t0 38 14 6 4 0 2\n",
      "60\t80\t0.0\t6\t0 43 26 5 3 2 1\n",
      "61\t131\t0.0\t6\t0 72 20 11 10 11 7\n",
      "62\t100\t0.0\t6\t0 53 22 8 10 6 1\n",
      "63\t86\t0.0\t6\t0 47 22 11 3 1 2\n",
      "64\t204\t0.0\t6\t0 107 52 29 10 1 5\n",
      "65\t123\t0.0\t6\t0 62 30 16 5 6 4\n",
      "66\t153\t0.0\t6\t0 88 32 17 8 5 3\n",
      "67\t131\t0.0\t6\t0 78 32 12 5 3 1\n",
      "68\t141\t0.0\t6\t0 86 39 7 7 1 1\n",
      "69\t230\t0.0\t6\t0 134 47 26 11 2 6 4\n",
      "70\t211\t0.0\t7\t2 129 39 17 14 7 2 1\n",
      "71\t423\t0.0\t7\t141 124 70 41 21 14 10 2\n",
      "72\t2727\t0.0\t7\t1245 661 384 189 97 86 36 29\n",
      "73\t39273\t0.0\t7\t25027 8609 3023 1304 669 304 210 127\n",
      "74\t230817\t0.0\t7\t153146 44970 16477 7698 4077 2303 1361 785\n",
      "75\t7870\t0.0\t7\t1920 2281 1356 1095 574 317 195 132\n",
      "\n",
      "=== Adapter 3 ===\n",
      "\n",
      "Sequence: GATCGGAAGAGCACACGTCTGAACTCCAGTCACGAGATTCCATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAT; Type: regular 3'; Length: 75; Trimmed: 82636 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-69 bp: 6; 70-75 bp: 7\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 0.0%\n",
      "  G: 0.0%\n",
      "  T: 0.0%\n",
      "  none/other: 100.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "69\t1\t0.0\t6\t0 0 0 0 0 0 0 1\n",
      "70\t1\t0.0\t7\t0 0 0 0 0 1\n",
      "71\t5\t0.0\t7\t0 0 0 0 2 1 1 1\n",
      "72\t94\t0.0\t7\t0 0 0 29 26 16 12 11\n",
      "73\t1579\t0.0\t7\t0 0 782 405 188 105 68 31\n",
      "74\t9377\t0.0\t7\t0 4536 2218 1226 641 389 234 133\n",
      "75\t71579\t0.0\t7\t40576 16528 6981 3487 1869 1094 653 391\n",
      "\n",
      "This is cutadapt 1.15 with Python 3.6.9\n",
      "Command line parameters: -a NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACGAGATTCCATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAA -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACGAGATTCCATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAT -o trimmed/SRR6145433.sra.fastq SRR6145433.sra.fastq\n",
      "Running on 1 core\n",
      "Trimming 3 adapters with at most 10.0% errors in single-end mode ...\n",
      "Finished in 1468.99 s (36 us/read; 1.65 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              40,299,174\n",
      "Reads with adapters:                40,099,302 (99.5%)\n",
      "Reads written (passing filters):    40,299,174 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,938,374,823 bp\n",
      "Total written (filtered):      7,130,919 bp (0.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN; Type: regular 3'; Length: 35; Trimmed: 40083274 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-35 bp: 3\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 0.0%\n",
      "  G: 0.0%\n",
      "  T: 0.0%\n",
      "  none/other: 100.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2\t629674.6\t0\t2\n",
      "4\t2\t157418.6\t0\t2\n",
      "5\t1\t39354.7\t0\t1\n",
      "19\t1\t0.0\t1\t0 1\n",
      "32\t1\t0.0\t3\t0 0 0 1\n",
      "33\t11\t0.0\t3\t5 2 1 3\n",
      "34\t71\t0.0\t3\t53 12 3 3\n",
      "35\t13127\t0.0\t3\t13098 28 1\n",
      "36\t14698\t0.0\t3\t14687 9 1 1\n",
      "37\t15515\t0.0\t3\t15507 7 1\n",
      "38\t17286\t0.0\t3\t17276 8 1 1\n",
      "39\t19525\t0.0\t3\t19505 15 4 1\n",
      "40\t22940\t0.0\t3\t22925 11 3 1\n",
      "41\t26615\t0.0\t3\t26603 9 2 1\n",
      "42\t28227\t0.0\t3\t28215 7 4 1\n",
      "43\t32130\t0.0\t3\t32118 10 2\n",
      "44\t34112\t0.0\t3\t34104 5 2 1\n",
      "45\t36902\t0.0\t3\t36891 8 2 1\n",
      "46\t39893\t0.0\t3\t39889 3 1\n",
      "47\t44210\t0.0\t3\t44203 6 0 1\n",
      "48\t49266\t0.0\t3\t49253 10 1 2\n",
      "49\t54484\t0.0\t3\t54477 5 1 1\n",
      "50\t61619\t0.0\t3\t61614 4 1\n",
      "51\t69405\t0.0\t3\t69404 1\n",
      "52\t78455\t0.0\t3\t78455\n",
      "53\t85820\t0.0\t3\t85817 2 1\n",
      "54\t96918\t0.0\t3\t96907 8 2 1\n",
      "55\t103733\t0.0\t3\t103725 5 1 2\n",
      "56\t112491\t0.0\t3\t112481 8 1 1\n",
      "57\t121967\t0.0\t3\t121956 8 2 1\n",
      "58\t136199\t0.0\t3\t136195 2 2\n",
      "59\t148483\t0.0\t3\t148472 8 3\n",
      "60\t161694\t0.0\t3\t161687 3 4\n",
      "61\t169879\t0.0\t3\t169875 2 2\n",
      "62\t186624\t0.0\t3\t186616 6 1 1\n",
      "63\t201396\t0.0\t3\t201387 8 1\n",
      "64\t221578\t0.0\t3\t221555 23\n",
      "65\t235687\t0.0\t3\t235685 2\n",
      "66\t239396\t0.0\t3\t239395 1\n",
      "67\t255498\t0.0\t3\t255497 1\n",
      "68\t273952\t0.0\t3\t273947 4 0 1\n",
      "69\t295764\t0.0\t3\t295758 5 0 1\n",
      "70\t330764\t0.0\t3\t330760 4\n",
      "71\t455904\t0.0\t3\t455894 6 2 2\n",
      "72\t1000995\t0.0\t3\t1000985 9 1\n",
      "73\t2696958\t0.0\t3\t2696924 27 2 5\n",
      "74\t7912250\t0.0\t3\t7912200 41 9\n",
      "75\t23980826\t0.0\t3\t23980660 136 20 10\n",
      "\n",
      "=== Adapter 2 ===\n",
      "\n",
      "Sequence: GATCGGAAGAGCACACGTCTGAACTCCAGTCACGAGATTCCATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAA; Type: regular 3'; Length: 74; Trimmed: 13001 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-69 bp: 6; 70-74 bp: 7\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 6.3%\n",
      "  C: 0.0%\n",
      "  G: 0.0%\n",
      "  T: 0.0%\n",
      "  none/other: 93.7%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "36\t675\t0.0\t3\t0 0 0 0 675\n",
      "38\t142\t0.0\t3\t0 0 0 0 142\n",
      "39\t1\t0.0\t3\t0 0 0 0 1\n",
      "58\t1\t0.0\t5\t0 0 0 0 0 0 1\n",
      "71\t8\t0.0\t7\t0 0 0 0 0 0 0 8\n",
      "72\t74\t0.0\t7\t0 0 0 0 0 0 1 73\n",
      "73\t1477\t0.0\t7\t0 0 0 0 0 0 2 1475\n",
      "74\t10513\t0.0\t7\t0 0 0 0 0 0 7 10506\n",
      "75\t110\t0.0\t7\t0 0 0 0 0 7 2 101\n",
      "\n",
      "=== Adapter 3 ===\n",
      "\n",
      "Sequence: GATCGGAAGAGCACACGTCTGAACTCCAGTCACGAGATTCCATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAT; Type: regular 3'; Length: 75; Trimmed: 3027 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20-29 bp: 2; 30-39 bp: 3; 40-49 bp: 4; 50-59 bp: 5; 60-69 bp: 6; 70-75 bp: 7\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 0.0%\n",
      "  C: 0.0%\n",
      "  G: 0.0%\n",
      "  T: 0.0%\n",
      "  none/other: 100.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "75\t3027\t0.0\t7\t0 0 0 0 0 2 6 3019\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#!./src/trim.sh -d $WORKDIR"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "FastQC analysis after trimming the adapters."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started analysis of SRR6145423.sra.fastq\n",
      "Approx 5% complete for SRR6145423.sra.fastq\n",
      "Approx 10% complete for SRR6145423.sra.fastq\n",
      "Approx 15% complete for SRR6145423.sra.fastq\n",
      "Approx 20% complete for SRR6145423.sra.fastq\n",
      "Approx 25% complete for SRR6145423.sra.fastq\n",
      "Approx 30% complete for SRR6145423.sra.fastq\n",
      "Approx 35% complete for SRR6145423.sra.fastq\n",
      "Approx 40% complete for SRR6145423.sra.fastq\n",
      "Approx 45% complete for SRR6145423.sra.fastq\n",
      "Approx 50% complete for SRR6145423.sra.fastq\n",
      "Approx 55% complete for SRR6145423.sra.fastq\n",
      "Approx 60% complete for SRR6145423.sra.fastq\n",
      "Approx 65% complete for SRR6145423.sra.fastq\n",
      "Approx 70% complete for SRR6145423.sra.fastq\n",
      "Approx 75% complete for SRR6145423.sra.fastq\n",
      "Approx 80% complete for SRR6145423.sra.fastq\n",
      "Approx 85% complete for SRR6145423.sra.fastq\n",
      "Approx 90% complete for SRR6145423.sra.fastq\n",
      "Approx 95% complete for SRR6145423.sra.fastq\n",
      "Analysis complete for SRR6145423.sra.fastq\n",
      "Started analysis of SRR6145433.sra.fastq\n",
      "Approx 5% complete for SRR6145433.sra.fastq\n",
      "Approx 10% complete for SRR6145433.sra.fastq\n",
      "Approx 15% complete for SRR6145433.sra.fastq\n",
      "Approx 20% complete for SRR6145433.sra.fastq\n",
      "Approx 25% complete for SRR6145433.sra.fastq\n",
      "Approx 30% complete for SRR6145433.sra.fastq\n",
      "Approx 35% complete for SRR6145433.sra.fastq\n",
      "Approx 40% complete for SRR6145433.sra.fastq\n",
      "Approx 45% complete for SRR6145433.sra.fastq\n",
      "Approx 50% complete for SRR6145433.sra.fastq\n",
      "Approx 55% complete for SRR6145433.sra.fastq\n",
      "Approx 60% complete for SRR6145433.sra.fastq\n",
      "Approx 65% complete for SRR6145433.sra.fastq\n",
      "Approx 70% complete for SRR6145433.sra.fastq\n",
      "Approx 75% complete for SRR6145433.sra.fastq\n",
      "Approx 80% complete for SRR6145433.sra.fastq\n",
      "Approx 85% complete for SRR6145433.sra.fastq\n",
      "Approx 90% complete for SRR6145433.sra.fastq\n",
      "Approx 95% complete for SRR6145433.sra.fastq\n",
      "Analysis complete for SRR6145433.sra.fastq\n"
     ]
    }
   ],
   "source": [
    "#!./src/fastqc.sh -d $WORKDIR/trimmed/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then align with trimming."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Hisat2 is aligning SRR6145423.sra.fastq\n",
      "--read-lengths arg must be at least 20\n",
      "HISAT2 version 2.2.0 by Daehwan Kim (infphilo@gmail.com, www.ccb.jhu.edu/people/infphilo)\n",
      "Usage: \n",
      "  hisat2 [options]* -x <ht2-idx> {-1 <m1> -2 <m2> | -U <r> | --sra-acc <SRA accession number>} [-S <sam>]\n",
      "\n",
      "  <ht2-idx>  Index filename prefix (minus trailing .X.ht2).\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <SRA accession number>        Comma-separated list of SRA accession numbers, e.g. --sra-acc SRR353653,SRR353654.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "  --sra-acc          SRA accession ID\n",
      "\n",
      " Presets:                 Same as:\n",
      "   --fast                 --no-repeat-index\n",
      "   --sensitive            --bowtie2-dp 1 -k 30 --score-min L,0,-0.5\n",
      "   --very-sensitive       --bowtie2-dp 2 -k 50 --score-min L,0,-1\n",
      "\n",
      " Alignment:\n",
      "  --bowtie2-dp <int> use Bowtie2's dynamic programming alignment algorithm (0) - 0: no dynamic programming, 1: conditional dynamic programming, and 2: unconditional dynamic programming (slowest)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-repeat-index  do not use repeat index\n",
      "\n",
      " Spliced Alignment:\n",
      "  --pen-cansplice <int>              penalty for a canonical splice site (0)\n",
      "  --pen-noncansplice <int>           penalty for a non-canonical splice site (12)\n",
      "  --pen-canintronlen <func>          penalty for long introns (G,-8,1) with canonical splice sites\n",
      "  --pen-noncanintronlen <func>       penalty for long introns (G,-8,1) with noncanonical splice sites\n",
      "  --min-intronlen <int>              minimum intron length (20)\n",
      "  --max-intronlen <int>              maximum intron length (500000)\n",
      "  --known-splicesite-infile <path>   provide a list of known splice sites\n",
      "  --novel-splicesite-outfile <path>  report a list of splice sites\n",
      "  --novel-splicesite-infile <path>   provide a list of novel splice sites\n",
      "  --no-temp-splicesite               disable the use of splice sites found\n",
      "  --no-spliced-alignment             disable spliced alignment\n",
      "  --rna-strandness <string>          specify strand-specific information (unstranded)\n",
      "  --tmo                              reports only those alignments within known transcriptome\n",
      "  --dta                              reports alignments tailored for transcript assemblers\n",
      "  --dta-cufflinks                    reports alignments tailored specifically for cufflinks\n",
      "  --avoid-pseudogene                 tries to avoid aligning reads to pseudogenes (experimental option)\n",
      "  --no-templatelen-adjustment        disables template length adjustment for RNA-seq reads\n",
      "\n",
      " Scoring:\n",
      "  --mp <int>,<int>   max and min penalties for mismatch; lower qual = lower penalty <6,2>\n",
      "  --sp <int>,<int>   max and min penalties for soft-clipping; lower qual = lower penalty <2,1>\n",
      "  --no-softclip      no soft-clipping\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (L,0.0,-0.2)\n",
      "\n",
      " Reporting:\n",
      "  -k <int>           It searches for at most <int> distinct, primary alignments for each read. Primary alignments mean \n",
      "                     alignments whose alignment score is equal to or higher than any other alignments. The search terminates \n",
      "                     when it cannot find more distinct valid alignments, or when it finds <int>, whichever happens first. \n",
      "                     The alignment score for a paired-end alignment equals the sum of the alignment scores of \n",
      "                     the individual mates. Each reported read or pair alignment beyond the first has the SAM ‘secondary’ bit \n",
      "                     (which equals 256) set in its FLAGS field. For reads that have more than <int> distinct, \n",
      "                     valid alignments, hisat2 does not guarantee that the <int> alignments reported are the best possible \n",
      "                     in terms of alignment score. Default: 5 (linear index) or 10 (graph index).\n",
      "                     Note: HISAT2 is not designed with large values for -k in mind, and when aligning reads to long, \n",
      "                     repetitive genomes, large -k could make alignment much slower.\n",
      "  --max-seeds <int>  HISAT2, like other aligners, uses seed-and-extend approaches. HISAT2 tries to extend seeds to \n",
      "                     full-length alignments. In HISAT2, --max-seeds is used to control the maximum number of seeds that \n",
      "                     will be extended. For DNA-read alignment (--no-spliced-alignment), HISAT2 extends up to these many seeds\n",
      "                     and skips the rest of the seeds. For RNA-read alignment, HISAT2 skips extending seeds and reports \n",
      "                     no alignments if the number of seeds is larger than the number specified with the option, \n",
      "                     to be compatible with previous versions of HISAT2. Large values for --max-seeds may improve alignment \n",
      "                     sensitivity, but HISAT2 is not designed with large values for --max-seeds in mind, and when aligning \n",
      "                     reads to long, repetitive genomes, large --max-seeds could make alignment much slower. \n",
      "                     The default value is the maximum of 5 and the value that comes with -k times 2.\n",
      "  -a/--all           HISAT2 reports all alignments it can find. Using the option is equivalent to using both --max-seeds \n",
      "                     and -k with the maximum value that a 64-bit signed integer can represent (9,223,372,036,854,775,807).\n",
      "  --repeat           report alignments to repeat sequences directly\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0), only valid with --no-spliced-alignment\n",
      "  -X/--maxins <int>  maximum fragment length (500), only valid with --no-spliced-alignment\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>           write unpaired reads that didn't align to <path>\n",
      "  --al <path>           write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>      write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>      write pairs that aligned concordantly at least once to <path>\n",
      "  (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "  --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --summary-file <path> print alignment summary to this file.\n",
      "  --new-summary         print alignment summary in a new style, which is more machine-friendly.\n",
      "  --quiet               print nothing to stderr except serious errors\n",
      "  --met-file <path>     send metrics to file at <path> (off)\n",
      "  --met-stderr          send metrics to stderr (off)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  --met <int>           report internal counters & metrics every <int> secs (1)\n",
      "  --no-head             suppress header lines, i.e. lines starting with @\n",
      "  --no-sq               suppress @SQ header lines\n",
      "  --rg-id <text>        set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>           add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                        Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq        put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "\n",
      " Performance:\n",
      "  -o/--offrate <int> override offrate of index; must be >= index's offrate\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'hisat2's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --remove-chrname   remove 'chr' from reference names in alignment\n",
      "  --add-chrname      add 'chr' to reference names in alignment \n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "Error: Encountered internal HISAT2 exception (#1)\n",
      "Command: /home/mees/NGS_Alzheimer/hisat2-2.2.0/hisat2-align-s --wrapper basic-0 -p 4 -x /media/mees/Elements/ngs_data/genomes/ref_genome//ref_genome -S SRR6145423.sra_out.sam --read-lengths 0,35,6,21 -U SRR6145423.sra.fastq \n",
      "(ERR): hisat2-align exited with value 1\n",
      "Hisat2 is aligning SRR6145433.sra.fastq\n",
      "--read-lengths arg must be at least 20\n",
      "HISAT2 version 2.2.0 by Daehwan Kim (infphilo@gmail.com, www.ccb.jhu.edu/people/infphilo)\n",
      "Usage: \n",
      "  hisat2 [options]* -x <ht2-idx> {-1 <m1> -2 <m2> | -U <r> | --sra-acc <SRA accession number>} [-S <sam>]\n",
      "\n",
      "  <ht2-idx>  Index filename prefix (minus trailing .X.ht2).\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <r>        Files with unpaired reads.\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\n",
      "  <SRA accession number>        Comma-separated list of SRA accession numbers, e.g. --sra-acc SRR353653,SRR353654.\n",
      "  <sam>      File for SAM output (default: stdout)\n",
      "\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\n",
      "\n",
      "Options (defaults in parentheses):\n",
      "\n",
      " Input:\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\n",
      "  --qseq             query input files are in Illumina's qseq format\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\n",
      "  -r                 query input files are raw one-sequence-per-line\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\n",
      "  --phred33          qualities are Phred+33 (default)\n",
      "  --phred64          qualities are Phred+64\n",
      "  --int-quals        qualities encoded as space-delimited integers\n",
      "  --sra-acc          SRA accession ID\n",
      "\n",
      " Presets:                 Same as:\n",
      "   --fast                 --no-repeat-index\n",
      "   --sensitive            --bowtie2-dp 1 -k 30 --score-min L,0,-0.5\n",
      "   --very-sensitive       --bowtie2-dp 2 -k 50 --score-min L,0,-1\n",
      "\n",
      " Alignment:\n",
      "  --bowtie2-dp <int> use Bowtie2's dynamic programming alignment algorithm (0) - 0: no dynamic programming, 1: conditional dynamic programming, and 2: unconditional dynamic programming (slowest)\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\n",
      "  --nofw             do not align forward (original) version of read (off)\n",
      "  --norc             do not align reverse-complement version of read (off)\n",
      "  --no-repeat-index  do not use repeat index\n",
      "\n",
      " Spliced Alignment:\n",
      "  --pen-cansplice <int>              penalty for a canonical splice site (0)\n",
      "  --pen-noncansplice <int>           penalty for a non-canonical splice site (12)\n",
      "  --pen-canintronlen <func>          penalty for long introns (G,-8,1) with canonical splice sites\n",
      "  --pen-noncanintronlen <func>       penalty for long introns (G,-8,1) with noncanonical splice sites\n",
      "  --min-intronlen <int>              minimum intron length (20)\n",
      "  --max-intronlen <int>              maximum intron length (500000)\n",
      "  --known-splicesite-infile <path>   provide a list of known splice sites\n",
      "  --novel-splicesite-outfile <path>  report a list of splice sites\n",
      "  --novel-splicesite-infile <path>   provide a list of novel splice sites\n",
      "  --no-temp-splicesite               disable the use of splice sites found\n",
      "  --no-spliced-alignment             disable spliced alignment\n",
      "  --rna-strandness <string>          specify strand-specific information (unstranded)\n",
      "  --tmo                              reports only those alignments within known transcriptome\n",
      "  --dta                              reports alignments tailored for transcript assemblers\n",
      "  --dta-cufflinks                    reports alignments tailored specifically for cufflinks\n",
      "  --avoid-pseudogene                 tries to avoid aligning reads to pseudogenes (experimental option)\n",
      "  --no-templatelen-adjustment        disables template length adjustment for RNA-seq reads\n",
      "\n",
      " Scoring:\n",
      "  --mp <int>,<int>   max and min penalties for mismatch; lower qual = lower penalty <6,2>\n",
      "  --sp <int>,<int>   max and min penalties for soft-clipping; lower qual = lower penalty <2,1>\n",
      "  --no-softclip      no soft-clipping\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\n",
      "                     (L,0.0,-0.2)\n",
      "\n",
      " Reporting:\n",
      "  -k <int>           It searches for at most <int> distinct, primary alignments for each read. Primary alignments mean \n",
      "                     alignments whose alignment score is equal to or higher than any other alignments. The search terminates \n",
      "                     when it cannot find more distinct valid alignments, or when it finds <int>, whichever happens first. \n",
      "                     The alignment score for a paired-end alignment equals the sum of the alignment scores of \n",
      "                     the individual mates. Each reported read or pair alignment beyond the first has the SAM ‘secondary’ bit \n",
      "                     (which equals 256) set in its FLAGS field. For reads that have more than <int> distinct, \n",
      "                     valid alignments, hisat2 does not guarantee that the <int> alignments reported are the best possible \n",
      "                     in terms of alignment score. Default: 5 (linear index) or 10 (graph index).\n",
      "                     Note: HISAT2 is not designed with large values for -k in mind, and when aligning reads to long, \n",
      "                     repetitive genomes, large -k could make alignment much slower.\n",
      "  --max-seeds <int>  HISAT2, like other aligners, uses seed-and-extend approaches. HISAT2 tries to extend seeds to \n",
      "                     full-length alignments. In HISAT2, --max-seeds is used to control the maximum number of seeds that \n",
      "                     will be extended. For DNA-read alignment (--no-spliced-alignment), HISAT2 extends up to these many seeds\n",
      "                     and skips the rest of the seeds. For RNA-read alignment, HISAT2 skips extending seeds and reports \n",
      "                     no alignments if the number of seeds is larger than the number specified with the option, \n",
      "                     to be compatible with previous versions of HISAT2. Large values for --max-seeds may improve alignment \n",
      "                     sensitivity, but HISAT2 is not designed with large values for --max-seeds in mind, and when aligning \n",
      "                     reads to long, repetitive genomes, large --max-seeds could make alignment much slower. \n",
      "                     The default value is the maximum of 5 and the value that comes with -k times 2.\n",
      "  -a/--all           HISAT2 reports all alignments it can find. Using the option is equivalent to using both --max-seeds \n",
      "                     and -k with the maximum value that a 64-bit signed integer can represent (9,223,372,036,854,775,807).\n",
      "  --repeat           report alignments to repeat sequences directly\n",
      "\n",
      " Paired-end:\n",
      "  -I/--minins <int>  minimum fragment length (0), only valid with --no-spliced-alignment\n",
      "  -X/--maxins <int>  maximum fragment length (500), only valid with --no-spliced-alignment\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\n",
      "  --no-discordant    suppress discordant alignments for paired reads\n",
      "\n",
      " Output:\n",
      "  -t/--time          print wall-clock time taken by search phases\n",
      "  --un <path>           write unpaired reads that didn't align to <path>\n",
      "  --al <path>           write unpaired reads that aligned at least once to <path>\n",
      "  --un-conc <path>      write pairs that didn't align concordantly to <path>\n",
      "  --al-conc <path>      write pairs that aligned concordantly at least once to <path>\n",
      "  (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\n",
      "  --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\n",
      "  --summary-file <path> print alignment summary to this file.\n",
      "  --new-summary         print alignment summary in a new style, which is more machine-friendly.\n",
      "  --quiet               print nothing to stderr except serious errors\n",
      "  --met-file <path>     send metrics to file at <path> (off)\n",
      "  --met-stderr          send metrics to stderr (off)\n",
      "  --met <int>           report internal counters & metrics every <int> secs (1)\n",
      "  --no-head             suppress header lines, i.e. lines starting with @\n",
      "  --no-sq               suppress @SQ header lines\n",
      "  --rg-id <text>        set read group id, reflected in @RG line and RG:Z: opt field\n",
      "  --rg <text>           add <text> (\"lab:value\") to @RG line of SAM header.\n",
      "                        Note: @RG line only printed when --rg-id is set.\n",
      "  --omit-sec-seq        put '*' in SEQ and QUAL fields for secondary alignments.\n",
      "\n",
      " Performance:\n",
      "  -o/--offrate <int> override offrate of index; must be >= index's offrate\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\n",
      "  --reorder          force SAM output order to match order of input reads\n",
      "  --mm               use memory-mapped I/O for index; many 'hisat2's can share\n",
      "\n",
      " Other:\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\n",
      "  --seed <int>       seed for random number generator (0)\n",
      "  --non-deterministic seed rand. gen. arbitrarily instead of using read attributes\n",
      "  --remove-chrname   remove 'chr' from reference names in alignment\n",
      "  --add-chrname      add 'chr' to reference names in alignment \n",
      "  --version          print version information and quit\n",
      "  -h/--help          print this usage message\n",
      "Error: Encountered internal HISAT2 exception (#1)\n",
      "Command: /home/mees/NGS_Alzheimer/hisat2-2.2.0/hisat2-align-s --wrapper basic-0 -p 4 -x /media/mees/Elements/ngs_data/genomes/ref_genome//ref_genome -S SRR6145433.sra_out.sam --read-lengths 0,35,6 -U SRR6145433.sra.fastq \n",
      "(ERR): hisat2-align exited with value 1\n"
     ]
    }
   ],
   "source": [
    "#!./src/align.sh -g $REF_GENOME -f $WORKDIR/trimmed/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
